%% Generated by Sphinx.
\def\sphinxdocclass{jupyterBook}
\documentclass[letterpaper,12pt,english]{jupyterBook}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}
%% turn off hyperref patch of \index as sphinx.xdy xindy module takes care of
%% suitable \hyperpage mark-up, working around hyperref-xindy incompatibility
\PassOptionsToPackage{hyperindex=false}{hyperref}
%% memoir class requires extra handling
\makeatletter\@ifclassloaded{memoir}
{\ifdefined\memhyperindexfalse\memhyperindexfalse\fi}{}\makeatother

\PassOptionsToPackage{warn}{textcomp}

\catcode`^^^^00a0\active\protected\def^^^^00a0{\leavevmode\nobreak\ }
\usepackage{cmap}
\usepackage{fontspec}
\defaultfontfeatures[\rmfamily,\sffamily,\ttfamily]{}
\usepackage{amsmath,amssymb,amstext}
\usepackage{polyglossia}
\setmainlanguage{english}



\setmainfont{FreeSerif}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Italic,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldItalic
]
\setsansfont{FreeSans}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]
\setmonofont{FreeMono}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]



\usepackage[Bjarne]{fncychap}
\usepackage[,numfigreset=1,mathnumfig]{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}

\let\cleardoublepage\clearpage



        % Start of preamble defined in sphinx-jupyterbook-latex %
         \usepackage[Latin,Greek]{ucharclasses}
        \usepackage{unicode-math}
        % fixing title of the toc
        \addto\captionsenglish{\renewcommand{\contentsname}{Contents}}
        \hypersetup{
            pdfencoding=auto,
            psdextra
        }
        % End of preamble defined in sphinx-jupyterbook-latex %
        

\title{From raw data to kinetic parameters: an EnzymeML-based workflow for reproducible enzyme kinetics}
\date{Jan 30, 2023}
\release{}
\author{Max Häußler}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle

\pagestyle{plain}
\setcounter{tocdepth}{3}
\sphinxtableofcontents


\sphinxstepscope


\chapter*{Declaration of authorship}
\label{\detokenize{declaration:declaration-of-authorship}}\label{\detokenize{declaration::doc}}
\sphinxAtStartPar
I hereby declare that the master thesis submitted is my own work. It was made unaided by third parties and only using the quoted sources and tools. All sources used are labelled as references. The thesis, in equal or similar form, was not previously presented to another examination board, and the electronic copy matches the other copies.

\sphinxAtStartPar
Max Häußler

\sphinxstepscope

\chapter*{Acknowledgements}
\label{\detokenize{acknowledgements:acknowledgements}}\label{\detokenize{acknowledgements::doc}}
\sphinxAtStartPar
First, I would like to thank Prof. Dr. Jürgen Pleiss for the opportunity to work on this project and the support throughout the work of this thesis.
Furthermore, I would like to thank Associate Professor Marilize Le Roes\sphinxhyphen{}Hill together with Prof. Dr. Jürgen Pleiss for organizing and enabling my research visit to South Africa, which fueled the collaboration on the projects of this thesis.

\sphinxAtStartPar
My special thanks go to my project partners Alaric Prins, Marwa Mohamed, Chantal Daub, and Paulo Durão who provided the experimental data for the projects of this thesis as well as the fruitful and pleasant project discussions.

\sphinxAtStartPar
I would like to thank Jan Range and Torsten Gieß for their support as well as the whole group for inspiring and motivating seminar discussions.

\sphinxstepscope
\pagestyle{plain}

\chapter*{Abstract}
%\label{\detokenize{abstract:abstract}}\label{\detokenize{abstract::doc}}
\sphinxAtStartPar
In this work, an EnzymeML\sphinxhyphen{}based workflow for kinetic parameter estimation was developed, enabling a continuous data flow from raw data to kinetic parameters.
For this purpose, two Python modules were developed and published on PyPI: CaliPytion enables precise calculation of concentrations based on analytical absorption data, whereas EnzymePynetics facilitates kinetic parameter estimation of single substrate enzyme reactions. Furthermore, EnzymePynetics is suited for inhibition studies.
Moreover, the workflow was implemented in a Jupyter Notebook environment, enabling conclusive documentation of each data processing step. Hence, the workflow is transparent, additionally the application of the workflow is reproducible. FAIR data principles were implemented on multiple levels of this thesis. Consequently, this thesis was conceptualized as a Jupyter Book, making all data analysis and methodologies of this work Findable, Accessible, Interoperable and Reusable to the most possible degree.


\section*{Abstract (deutsch)}
\label{\detokenize{abstract:abstract-deutsch}}
\sphinxAtStartPar
In dieser Arbeit wurde ein EnzymeML basierter Workflow zur Bestimmung kinetischer Parameter entwickelt, welcher einen kontinuierlichen Datenfluss von Rohdaten zu kinetischen Parametern ermöglicht.
Zu diesem Zweck wurden zwei Python\sphinxhyphen{}Module entwickelt und auf PyPI veröffentlicht: CaliPytion, welches die präzise Berechnung von Konzentrationen auf der Grundlage analytischer Absorptionsdaten ermöglicht und EnzymePynetics, welches die Schätzung kinetischer Parameter von Enzymreaktionen mit einem Substrat ermöglicht. Außerdem ist EnzymePynetics für Inhibitionsstudien geeignet.
Darüber hinaus wurde der Workflow in einer Jupyter\sphinxhyphen{}Notebook\sphinxhyphen{}Umgebung implementiert, was eine schlüssige Dokumentation der einzelnen Datenverarbeitungsschritte ermöglicht. Somit ist der Workflow transparent, während die Anwendung des Workflows reproduzierbar ist. Die FAIR\sphinxhyphen{}Prinzipien wurden in dieser Arbeit auf mehreren Ebenen umgesetzt. Daher wurde diese Arbeit als Jupyter Book konzipiert, um alle Datenanalysen und Methoden dieser Arbeit so weit wie möglich den FAIR\sphinxhyphen{}Kriterien Auffindbarkeit, Zugänglichkeit, Interoperabilität und Wiederverwendbarkeit entsprechend zu machen.

\sphinxstepscope

\pagestyle{normal}
\chapter{Introduction}
\label{\detokenize{introduction:introduction}}\label{\detokenize{introduction::doc}}
\sphinxAtStartPar
Reproducible research is the cornerstone of scientific advances and builds the core principle of the scientific method itself. Within the last decade, an increasing number of reports raised attention on unreproducible results across various scientific disciplines (\hyperlink{cite.references:id33}{Begley and Ioannidis, 2015}). Reportedly, more than 70 \% of researchers were not able to reproduce the results of another researcher, whereas more than 50 \% were not able to reproduce their own findings (\hyperlink{cite.references:id32}{Baker, 2016}).
Since reproducibility is a non\sphinxhyphen{}standardized term, it often serves as a vague umbrella term for mingling (i) repeatable analysis based on given information by the same scientist, (ii) reproducible analysis based on given information by another scientist, or (iii) replicable analysis with new information by another scientist.
Therefore, reproducibility is not an absolute term but rather relative, depending on the scope of the notion (\hyperlink{cite.references:id31}{Goodman \sphinxstyleemphasis{et al.}, 2016}). Goodman \sphinxstyleemphasis{et al.} (\hyperlink{cite.references:id31}{2016}) proposed to combine the word reproducibility with its relative scope, hence subdividing reproducibility in “methods reproducibility”, “results reproducibility” and “inferential reproducibility”. According to Goodman \sphinxstyleemphasis{et al.} (\hyperlink{cite.references:id31}{2016}), “methods reproducibility refers to the provision of enough detail about study procedures and data so the same procedures could, in theory or in actuality, be exactly repeated”. Results reproducibility requires methodological reproducibility, so that the same results are obtained from an independent study. Inferential reproducibility is given, if qualitatively similar conclusions are drawn based on the results of independent studies or reanalysis of a given study. Therefore, inferential reproducibility constitutes the highest level of reproducibility. Possible reasons for non\sphinxhyphen{}reproducible experiments are broadly discussed and range from study design, to publication bias and malpractice (\hyperlink{cite.references:id34}{Munafò \sphinxstyleemphasis{et al.}, 2017}), (\hyperlink{cite.references:id21}{Miyakawa, 2020}) (\hyperlink{cite.references:id32}{Baker, 2016}). Once published, the results of non\sphinxhyphen{}reproducible studies are harmful for science, since the reported results along the generated data assets are likely incorrect.

\sphinxAtStartPar
Low reproducibility also affects the field of enzymology, where varying experimental standards are described as the leading cause (\hyperlink{cite.references:id42}{Bell \sphinxstyleemphasis{et al.}, 2021}). With the aim to increase reproducibility on biocatalytic experiments, the STRENDA commission has defined standards for reporting enzymology data (STRENDA) (\hyperlink{cite.references:id11}{Tipton \sphinxstyleemphasis{et al.}, 2014}). The \sphinxhref{https://www.beilstein-institut.de/en/projects/strenda/guidelines/}{guidelines} recommend reporting on identity information of the enzyme, enzyme preparation steps, storage conditions, assay conditions and method, as well as enzyme activity with the corresponding kinetic parameters and units. Furthermore, the precision of measurements and preferably the raw data itself should be shared. Hence, the STRENDA guidelines pave the way for reproducible and therefore reusable functional enzymology data. Besides the reproducibility of data, availability of data assets must be given in order to enable reuse of extant data. This is especially important to enable the application of big data technologies, like machine learning which rely on large amounts of high quality data. In the field of biocatalysis, machine learning bares the potential to investigate how the structures of an enzyme is related to its catalytic properties. Thereby, valuable conclusions for enzyme engineering might be drawn.
One prerequisite for applying big data technologies and therefore use their potential in biocatalysis is the availability of data. With the scope to enhance data reusability, FAIR guiding principles for scientific data management were established (\hyperlink{cite.references:id40}{Wilkinson \sphinxstyleemphasis{et al.}, 2016}). The acronym FAIR denotes four foundational principles for contemporary data management in a Findable, Accessible, Interoperable and Reusable fashion. Special emphasis is placed on machines being able to find and use data, due to their fundamental importance in modern science. Thereby, (meta)data should be findable by a globally unique identifier. Data should be accessible by a standardized and open protocol. Interoperability is achieved, if data contains qualitative references to other (meta)data and uses conclusive vocabulary. For reusability, the data needs to be extensively describes with accurate and relevant attributes (\hyperlink{cite.references:id40}{Wilkinson \sphinxstyleemphasis{et al.}, 2016}). Broad application of the stated principles will promote the reuse of data and ultimately amplify the knowledge which can be deduced from each published dataset.

\sphinxAtStartPar
To enable data exchange in biocatalysis the standardized exchange format EnzymeML was developed, abiding by FAIR data principles as well as STRENDA guidelines (\hyperlink{cite.references:id4}{Pleiss, 2021}). Within an EnzymeML document, information on reaction conditions, obtained measurement data, as well as modeling results are stored. As such, reaction conditions contain information on the pH value, temperature and the reaction vessel. Each species within a reaction (e.g. substrate, product or inhibition species) is uniquely labeled with an identifier, which allows unequivocal referencing of species. Thereby, proteins are labeled with their UniProt ID, and reactants are labeled with their respective SMILES or InChI code. A special focus of the EnzymeML format is to store the modeling results, information on kinetic parameters, and model equations alongside the measurement data on which the results are based. In summary, data and meta data of biocatalytic experiments can be stored and shared between scientists and databases compliant with contemporary data management practices.

\sphinxAtStartPar
In this work, a computational workflow based on EnzymeML for kinetic parameters estimation of enzyme reactions was developed with the following requirements: The workflow should enable comprehensive data analysis. Should be based on raw data. And yield reproducible kinetic parameter estimates. Furthermore, the workflow should be iteratively developed by applied it to on ongoing research projects of EnzymeML project partners. Therefore, two software packages should be developed, which facilitate the parameter estimation process. Finally, the estimated parameters should be written back to the EnzymeML file, unifying measurement data with information on the applied estimation model as well as the resulting kinetic parameters.

\sphinxstepscope


\chapter{Methods}
\label{\detokenize{methods:methods}}\label{\detokenize{methods::doc}}

\section{EnzymeML and PyEnzyme}
\label{\detokenize{methods:enzymeml-and-pyenzyme}}
\sphinxAtStartPar
EnzymeML documents were read, edited, and written via the Python application programming interface PyEnzyme (\hyperlink{cite.references:id12}{Range \sphinxstyleemphasis{et al.}, 2021}), providing the possibility to integrate data from EnzymeML documents in Python\sphinxhyphen{}based data analysis workflows. Thereby, EnzymeML documents were used as a data source.


\subsection{Creation of EnzymeML documents}
\label{\detokenize{methods:creation-of-enzymeml-documents}}
\sphinxAtStartPar
\sphinxcode{\sphinxupquote{EnzymeMLDocuments}} were created using the EnzymeML Excel template, in combination with the \sphinxcode{\sphinxupquote{.fromTemplate()}} method of PyEnzyme. Within the spreadsheet, data and metadata of an experiment were filled in the respective sheets. Alternatively, only metadata was entered to the spreadsheet, whereas measurement data was parsed by a custom Python function from the output file of the analytical device to the measurement data section of the EnzymeML document.


\subsection{Saving of modeling results}
\label{\detokenize{methods:saving-of-modeling-results}}
\sphinxAtStartPar
Modeling results from parameter estimation were written back to the EnzymeML document, which contained the measurement data by using PyEnzyme.


\section{CaliPytion}
\label{\detokenize{methods:calipytion}}
\sphinxAtStartPar
CaliPytion was developed to provide an easy way to find the best model by which the relationship between analytical signal to analyte concentration of photometric measurements can be determined. The resulting \sphinxcode{\sphinxupquote{StandardCurve}} is created based on a provided analyte standard. Besides a linear calibration equation, non\sphinxhyphen{}linear calibration models were implemented. All implemented calibration equations \eqref{equation:methods:linear}, \eqref{equation:methods:quadratic} \eqref{equation:methods:3rd_polynominal}, \eqref{equation:methods:poly_e} \eqref{equation:methods:fractional} are listed below, whereas \(A\) denotes the absorption, \(x\) the concentration, and \(a, b, c\) the respective parameters of each equation.
During the fitting process, all calibration equations are fitted against the analyte standard by non\sphinxhyphen{}linear least\sphinxhyphen{}square minimization. Therefore, the Python library lmfit (\hyperlink{cite.references:id35}{Newville \sphinxstyleemphasis{et al.}, 2016}) was utilized. The best fitting model is then selected based on the lowest Akaike information criterion (AIC). Furthermore the fit of the models can be visually assessed by calling the \sphinxcode{\sphinxupquote{.visualize()}} method of the \sphinxcode{\sphinxupquote{StandardCurve}} class. After the calibration curves are fitted, concentration of unknown samples are determined by calculating the root of the parameterized calibration model.
\begin{equation}\label{equation:methods:linear}
\begin{split}A = ax\end{split}
\end{equation}\begin{equation}\label{equation:methods:quadratic}
\begin{split}A = ax^2 + bx\end{split}
\end{equation}\begin{equation}\label{equation:methods:3rd_polynominal}
\begin{split}A = ax^3 + bx^2 + cx\end{split}
\end{equation}\begin{equation}\label{equation:methods:poly_e}
\begin{split}A = ae^{\frac{b}{x}}\end{split}
\end{equation}\begin{equation}\label{equation:methods:fractional}
\begin{split}A = \frac{ax}{b+x}\end{split}
\end{equation}

\subsection{Data model}
\label{\detokenize{methods:data-model}}\label{\detokenize{methods:calipytion-datamodel}}
\sphinxAtStartPar
The \sphinxcode{\sphinxupquote{StandardCurve}} functionality of CaliPythion is based on a \sphinxhref{https://github.com/FAIRChemistry/CaliPytion/blob/main/specifications/CalibrationModel.md}{data model}, structuring data and metadata of calibration measurements. Thereby, calibration conditions like temperature, pH, and information about the analytical device can be provided. Furthermore, the name as well as an ID can be specified for the analyzed substance. All of this information is stored in the \sphinxcode{\sphinxupquote{Calibration}} root object. Additionally, the root object can contain a \sphinxcode{\sphinxupquote{Standard}} and a \sphinxcode{\sphinxupquote{Spectrum}}. A \sphinxcode{\sphinxupquote{Standard}} contains measurements of multiple predefined concentrations and the wavelength at which the measurement was conducted. Additionally, a \sphinxcode{\sphinxupquote{Spectrum}} can be defined by providing measurement data as well as the respective wavelengths at which the data was measured. The data model was generated using \sphinxhref{https://github.com/JR-1991/software-driven-rdm}{sdRDM}, a Python tool allowing the creation and versioning of data models.


\subsection{Initialization}
\label{\detokenize{methods:initialization}}
\sphinxAtStartPar
A \sphinxcode{\sphinxupquote{StandardCurve}} can be initialized directly from an Excel file by calling \sphinxcode{\sphinxupquote{StandardCurve.from\_Excel()}}
Therefore, the first column must contain concentration data, whereas all following columns contain the respective measurement data. The first row is not considered and should be used to label the data.
Alternatively a \sphinxcode{\sphinxupquote{StandardCurve}} can be initialized by providing a \sphinxcode{\sphinxupquote{Calibration}} object, containing the calibration data.

\sphinxAtStartPar
After \sphinxcode{\sphinxupquote{StandardCurve}} initialization, concentrations can be calculated by calling the \sphinxcode{\sphinxupquote{.get\_concentration()}} method. Alternatively, a \sphinxcode{\sphinxupquote{StandardCurve}} can be directly applied to an \sphinxcode{\sphinxupquote{EnzymeMLDocument}} by calling the \sphinxcode{\sphinxupquote{.apply\_to\_EnzymeML()}} method.


\subsection{Code availability}
\label{\detokenize{methods:code-availability}}
\sphinxAtStartPar
CaliPytion was published on the Python packaging index (\sphinxhref{https://pypi.org/project/CaliPytion/}{PyPI}) whereas the source code is accessible on \sphinxhref{https://github.com/FAIRChemistry/CaliPytion}{GitHub}.


\section{EnzymePynetics}
\label{\detokenize{methods:enzymepynetics}}\label{\detokenize{methods:method-enzymepynetics}}
\sphinxAtStartPar
EnzymePynetics is a Python package for kinetic parameter estimation of single\sphinxhyphen{}substrate enzyme reactions. The \sphinxcode{\sphinxupquote{ParameterEstimator}} of EnzymePynetics estimates the turnover number and Michaelis\sphinxhyphen{}Menten constant by fitting time\sphinxhyphen{}course measurement data of enzyme reactions to different Michaelis\sphinxhyphen{}Menten models. Thereby, the residuals between measurement data and integrated Michaelis\sphinxhyphen{}Menten rate equations are minimized through a non\sphinxhyphen{}linear least\sphinxhyphen{}square algorithm utilizing the lmfit library. Additionally, the inhibition constant \(K_{i}\) can be assessed for potential substrate or product inhibition. Furthermore, \(K_{i}\) of an enzyme inhibitor apart from substrate or product can be determined for inhibitor studies.


\subsection{Data model}
\label{\detokenize{methods:enzymepynetics-datamodel}}\label{\detokenize{methods:id3}}
\sphinxAtStartPar
EnzymePynetics is based on a data model, resembling the experimental design of an enzyme kinetics assay. Thereby, all relevant data and metadata of a kinetic experiment are structured in the \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}} base object. Therein, reaction temperature with its respective unit, pH value, and the name of the substance from which the measurement data originates are described. An \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}} contains one or multiple \sphinxcode{\sphinxupquote{Measurements}}, which contain the measurement data. Each measurement stores information of the experimental conditions of an enzyme reaction. Hence, each \sphinxcode{\sphinxupquote{Measurement}} contains information on the initial substrate concentration, enzyme concentration, and inhibitor concentration, along with the respective concentration units. Each \sphinxcode{\sphinxupquote{Measurement}} contains measured data, which itself consist of one or multiple replicates of the respective experimental condition. The data model was generated using \sphinxhref{https://github.com/JR-1991/software-driven-rdm}{sdRDM}.

\sphinxAtStartPar
Extensive documentation of the data model is accessible in the \sphinxhref{https://github.com/haeussma/EnzymePynetics/blob/main/specifications/EnzymeKinetics.md}{specifications} of the software package.


\subsection{Kinetic models}
\label{\detokenize{methods:kinetic-models}}
\sphinxAtStartPar
Besides the irreversible Michaelis\sphinxhyphen{}Menten rate equation \eqref{equation:methods:irreversible_mm} inhibition models for competitive \eqref{equation:methods:competitive_inhibition}, uncompetitive \eqref{equation:methods:uncompetitive_inhibition}, non\sphinxhyphen{}competitive \eqref{equation:methods:noncompetitive_inhibition} and partially competitive inhibition \eqref{equation:methods:partially_inhibition} were implemented, according to (\hyperlink{cite.references:id5}{Bisswanger, 2017}). Thereby, \(S\), \(E\), and \(I\) respectively denote the concentration of substrate, enzyme, and inhibitor. In terms of kinetic parameters, \(k_{cat}\) denotes the turnover number, \(K_{m}\) the Michaelis\sphinxhyphen{}Menten constant of the substrate, whereas \(K_{ic}\) and \(K_{iu}\) respectively describe the competitive and uncompetitive inhibition constant.
\begin{equation}\label{equation:methods:irreversible_mm}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} + S}\end{split}
\end{equation}\begin{equation}\label{equation:methods:competitive_inhibition}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{ic}}) + S}\end{split}
\end{equation}\begin{equation}\label{equation:methods:uncompetitive_inhibition}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{iu}}) + S}\end{split}
\end{equation}\begin{equation}\label{equation:methods:noncompetitive_inhibition}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{ic}}) + (1+\frac{I}{K_{iu}}) + S}\end{split}
\end{equation}\begin{equation}\label{equation:methods:partially_inhibition}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * \frac{(1+\frac{I}{K_{ic}})}{(1+\frac{I}{K_{iu}})} + S}\end{split}
\end{equation}
\sphinxAtStartPar
By default, \(E\) is assumed to be constant throughout the reaction. If required, the kinetic models can be extended by the parameter \(k_{inact}\), which describes the time\sphinxhyphen{}dependent inactivation rate of the enzyme. The decrease in active enzyme concentration is modeled as describe in equation \eqref{equation:methods:enzyme_inactivation}:
\begin{equation}\label{equation:methods:enzyme_inactivation}
\begin{split}\frac{dE}{dt} = -k_{inact} * E\end{split}
\end{equation}
\sphinxAtStartPar
Each kinetic model therefore consists of a set of ordinary differential equations, describing the change in concentration of substrate, product and enzyme.


\subsection{Run parameter estimation}
\label{\detokenize{methods:run-parameter-estimation}}
\sphinxAtStartPar
\sphinxstylestrong{Initialization}

The \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}} object solely serves as a data container, whereas
the \sphinxcode{\sphinxupquote{ParameterEstimator}} harbors the functionalities for parameter estimation. Data can be provided as an \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}} object. Alternatively, an \sphinxcode{\sphinxupquote{EnzymeMLDocument}} document can be provided as the data source via PyEnzyme by calling the \sphinxcode{\sphinxupquote{.from\_EnzymeML()}} method. Within the method, the ID of the measured species needs to be specified according to the nomenclature of the \sphinxcode{\sphinxupquote{EnzymeMLDocument}}. If the document contains information on an inhibitor, the respective id needs to be provided as well, if its inhibition constant should be estimated.

\sphinxAtStartPar
Initially, missing data for the fitting process is calculated based on the assumption of mass conservation. As such, missing product concentration \(P\) is calculated based on the given substrate measurement data \(S\) and the specified initial substrate concentration \(S_{0}\) for each measurement \(t\) \eqref{equation:methods:missing_product}. If product data is provided, missing substrate concentrations are calculated accordingly \eqref{equation:methods:missing_substrate}.
\begin{equation}\label{equation:methods:missing_product}
\begin{split}P_{t} = S_{0} - S_{t}\end{split}
\end{equation}\begin{equation}\label{equation:methods:missing_substrate}
\begin{split}S_{t} = S_{0} - P_{t}\end{split}
\end{equation}
\sphinxAtStartPar
\sphinxstylestrong{Fitting of models}

After the \sphinxcode{\sphinxupquote{ParameterEstimator}} is initialized, all kinetic models are initialized by calling the \sphinxcode{\sphinxupquote{.fit\_models()}} method. Furthermore, it can be specified if enzyme inactivation should be considered for modeling. Depending on whether an inhibitor was specified, the models are initialized accordingly. If no inhibitor was specified, product and substrate inhibition models are initialized besides the irreversible Michaelis model. If inhibitor data was provided, the inhibition models are initialized with concentration data of the inhibitor. Thereafter, kinetic parameters of each model are initialized with estimates based on the highest reaction rate in the dataset. For minimization, the parameter space is limited to ± 1000\sphinxhyphen{}fold of the initial parameter estimates. After all models are fitted, an overview table is printed which lists all kinetic parameters of each model together with the respective 1σ standard deviation in percent. The table is sorted by ascending AIC.

\sphinxAtStartPar
\sphinxstylestrong{Visualization}

After model fitting, the modeling results can be visualized by calling the \sphinxcode{\sphinxupquote{.visualize()}} method, which was implemented utilizing Matplotlib (\hyperlink{cite.references:id36}{Caswell \sphinxstyleemphasis{et al.}, 2020}).
Thereby, the measurement data is visualized together with the fitted kinetic model. If the experiment was carried out in replicates, mean values of the measurement data with the corresponding standard deviations are shown. If the experiment data contained reactions with different inhibitor concentrations, individual inhibitor concentrations are denoted by individual markers. By default, the best fitting model according to its AIC is visualized. Different models can be visualized by passing the name of the respective model to the function call. Besides the figure, a detailed statistical fit report is printed, if not specified otherwise.

\sphinxAtStartPar
\sphinxstylestrong{Data units}

So far, EnzymePynetics neither converts nor validates units of the measurement data. Hence, all data needs to be provided using the same molar or mass unit. Only the concentration unit of the inhibitor may differ from all other data units.


\subsection{Code availability}
\label{\detokenize{methods:id6}}
\sphinxAtStartPar
EnzymePynetics was published on the Python packaging index (\sphinxhref{https://pypi.org/project/EnzymePynetics/}{PyPI}) whereas the source code is available on \sphinxhref{https://github.com/haeussma/EnzymePynetics}{GitHub}.


\section{Model comparison}
\label{\detokenize{methods:model-comparison}}
\sphinxAtStartPar
The kinetic model, which best describes the experimental data, was selected based on AIC, standard deviation of parameter estimates, and visual fit between measurement data and fitted model. AIC serves as a statistical metric for information loss, which allows to relatively compare different models with differing number of parameters for a given data set. Hence, AIC can be applied for model selection (\hyperlink{cite.references:id16}{Arnold, 2010}), (\hyperlink{cite.references:id17}{Akaike, 1998}). Thereby, models with a lower AIC indicate less information loss.
Since AIC is based on the chi\sphinxhyphen{}square statistic, it does not consider the standard deviation of the estimated parameters of a model. Therefore, standard deviation was additionally considered in model selection. Models with low standard deviation were therefore preferred over models with high standard deviation. Furthermore, fit quality was assessed visually by confirming that the model describes the progress curve of the experimental data.


\section{Jupyter Notebook}
\label{\detokenize{methods:jupyter-notebook}}
\sphinxAtStartPar
All developed tools were deployed in Jupyter Notebooks (\hyperlink{cite.references:id37}{Kluyver \sphinxstyleemphasis{et al.}, 2016}) for data analysis within a Python environment. A Jupyter Notebook is a digital document containing code cells and text cells. Thereby, executable code with its corresponding output can be supplemented with narrative text, figures, and equations.
Besides its documentation capabilities, notebooks can be uploaded to platforms like GitHub or be deployed with Binder (\hyperlink{cite.references:id20}{Ragan\sphinxhyphen{}Kelley \sphinxstyleemphasis{et al.}, 2018}).
GitHub is a cloud\sphinxhyphen{}based Git repository, which enables versioning as well as collaborative working on projects involving any kind of code. Besides GitHub, sharing of Jupiter Notebooks is possible with the open\sphinxhyphen{}source project Binder, which allows hosting of notebooks. Thus, Jupyter Notebooks can be shared in an executable form, which is independent from the operating system and the location of the user. Only an internet connection and a browser is required.
As a result, Jupyter Notebooks, which are executable through Binder, form an user\sphinxhyphen{}friendly environment for scientific data analysis. Due to the extensive documentation capabilities of Jupyter Notebooks, the analysis workflow is comprehensible and accessible for programmers as well as for non\sphinxhyphen{}programmers.


\section{Jupyter Book}
\label{\detokenize{methods:jupyter-book}}
\sphinxAtStartPar
This written thesis, except for the cover page, was entirely conceptualized as a Jupyter Book, which is an open\sphinxhyphen{}source tool that allows to incorporate computational material in publication\sphinxhyphen{}quality books (\hyperlink{cite.references:id38}{Community, 2020}). Thereby, Jupyter Notebooks can natively be integrated in Jupyter Books, thus making all advantages of the notebook also available in the book.
Jupyter Books allow text formatting with Markdown, which is a simple markdown language. Furthermore, Jupyter Book supports cross\sphinxhyphen{}referencing, citations, and numbering of equations.
Together with all data, the final and interactive book was deployed on GitHub Pages whereas the print version was generated by exporting the JupyterBook to LaTeX. The LaTeX version was then adjusted for printing.

All data of this JupyterBook as well as the experimental data is stored on the data repository of the University of Stuttgart (DaRUS) and can be found under the following DOI:  \sphinxhref{https://doi.org/10.18419/darus-3337}{https://doi.org/10.18419/darus-3337}. In order to host the JupyterBook on the internet, the data from DaRUS is mirrored to GitHub. The JupyterBook can be found under the following URL:  \sphinxhref{https://haeussma.github.io/masterthesis/welcome.html}{https://haeussma.github.io/masterthesis/welcome.html}.


\sphinxstepscope


\chapter{Results}
\label{\detokenize{results:results}}\label{\detokenize{results::doc}}

\section{Workflow for enzyme kinetics parameter estimation}
\label{\detokenize{results:workflow-for-enzyme-kinetics-parameter-estimation}}
\sphinxAtStartPar
In this work, a Python\sphinxhyphen{}based analysis workflow for kinetic parameter estimation of enzyme reactions was developed. The workflow itself consists of the steps (i) data acquisition, (ii) raw data preparation, (iii) concentration calculation, (iv) quality control, (v) modeling, and (vi) saving of results. For precise concentration calculations as well as kinetic parameter estimation the two Python modules CaliPytion and EnzymePynetics were respectively developed over the course of this work.
The two packages were designed to seamlessly interact with EnzymeML documents which were modified using PyEnzyme and thus allowing a continuous data stream from analytical raw data to kinetic parameters. Starting from raw data of photometric measurements, which are stored in an EnzymeML document, concentration values are calculated by CaliPytion, whereupon kinetic parameters are determined by EnzymePynetics. Finally, the modeling results are written back to the EnzymeML document via PyEnzyme (Fig. 1).

\sphinxAtStartPar
\sphinxincludegraphics{{concept_workflow}.png}
\sphinxstyleemphasis{Fig. 1: Components of the kinetic parameter estimation workflow for enzyme reaction data.}

\sphinxAtStartPar
A key aspect of the workflow design was to enable reproducible analysis. Therefore, documentation of raw data treatment prior to modeling is vital. As a result, Jupyter Notebooks were used as a platform, since they allow to write text and code within the same document.


\subsection{CaliPytion}
\label{\detokenize{results:calipytion}}
\sphinxAtStartPar
CaliPytion was developed to facilitate robust conversion of measurement data to concentration data. Based on an analyte standard, a calibration curve is created. Since linear and non\sphinxhyphen{}linear relations between analyte concentration and analytic signal may occur, (\hyperlink{cite.references:id14}{Hsu and Chen, 2010}), (\hyperlink{cite.references:id15}{Martin \sphinxstyleemphasis{et al.}, 2017}), linear, polynomial, exponential, and rational calibration equations were implemented. After fitting, the best model is preselected based on Akaike information criterion (\hyperlink{cite.references:id17}{Akaike, 1998}), offering a metric to compare models with different number of parameters. CaliPytion was designed to work with PyEnzyme. Hence, a \sphinxcode{\sphinxupquote{StandardCurve}} of CaliPytion can natively by applied to an \sphinxcode{\sphinxupquote{EnzymeMLDocument}} of PyEnzyme. As a result, absorption data of an EnzymeML document can be converted into concentration data based on the previously generated standard curve.


\subsection{EnzymePynetics}
\label{\detokenize{results:enzymepynetics}}
\sphinxAtStartPar
EnzymePynetics was developed to enable easy applicable parameter estimation of single substrate enzyme reactions directly from EnzymeML documents. At its core, the \sphinxcode{\sphinxupquote{ParameterEstimator}} of EnzymePynetics fits experimental data to different Michaels\sphinxhyphen{}Menten type models. Besides the irreversible Michaelis\sphinxhyphen{}Menten model, competitive, non\sphinxhyphen{}competitive, uncompetitive, and partially competitive inhibition models were implemented. Hence, the inhibition constant of potentially inhibiting substrate or product can be estimated. Furthermore, the inhibition constant of an applied inhibitor can be quantified. Since, the experimental data is fitted against multiple kinetic models, the best fitting model is suggested based on AIC. Hence, a proposal of the enzyme’s kinetic mechanism is given based on the best fitting model.
For quality control, visualization of experimental data as well as the fitted kinetic model was a priority during development. Thereby, detection of systematic deviation between model and measurement data is possible. Furthermore, the \sphinxcode{\sphinxupquote{ParameterEstimator}} allows to subset the measurement data by time as well as initial substrate concentration without deleting data. Hence, identified systematic deviations (e.g. lag\sphinxhyphen{}phases or measurements with incorrect enzyme concentrations) can be excluded from parameter estimation. Like the \sphinxcode{\sphinxupquote{StandardCurve}} of CaliPytion, the \sphinxcode{\sphinxupquote{ParameterEstimator}} of EnzymePynetics natively supports \sphinxcode{\sphinxupquote{EnzymeMLDocument}}s as a data source.


\section{Scenario\sphinxhyphen{}driven workflow development}
\label{\detokenize{results:scenario-driven-workflow-development}}
\sphinxAtStartPar
The development of the workflow was driven by different research scenarios of EnzymeML project partners. The goal of all projects was to reliably determine kinetic parameters of enzyme reactions.
The collaboration consisted of multiple rounds of lab experiments (preformed by the experimental partners) and kinetic modeling of experimental data. Each round included (i) wet lab experiments, (ii) kinetic modeling, (iii) design of follow\sphinxhyphen{}up experiments, and (iv) discussion of results with the respective project partners. Hence, a short feedback loop between lab experiments and modeling\sphinxhyphen{}based experimental suggestions was established. In parallel, the Python modules CaliPytion and EnzymePynetics were developed. Thereby, individual requirements of each research scenario fostered the implementation of various functionalities, resulting in a feature\sphinxhyphen{}rich yet generic workflow.


\section{Data analysis with Jupyter Notebooks}
\label{\detokenize{results:data-analysis-with-jupyter-notebooks}}
\sphinxAtStartPar
All data analysis was carried out in Jupyter Notebooks by applying the developed workflow for kinetic parameter estimation. Within the next chapters, the latest results of all four experimental scenarios are shown. Thereby, the applicability of the workflow is demonstrated while analyzing and discussing the results of the respective scenario.
Since the analysis was conducted in Jupyter Notebooks, each of the following chapter is a executed Jupyter Notebook. Hence all figures and tables for data visualization were generated at runtime of the analysis.
Thereby, the applicability of Jupyter Notebooks for reproducible and comprehensive computational workflows in a scientific environment is demonstrated.
Every scenario consists of a short description of the project’s background and methodology carried out by the project partners in the laboratory. Additionally, data preparation, kinetic modeling steps, and the results are shown was well as project specific results are discussed.
All notebooks can be launched interactively by clicking on the rocket icon in the upper section of each scenario page. As a result, all analysis can be interactively repeated. Therefore, it is highly recommended to \sphinxhref{https://haeussma.github.io/masterthesis/welcome.html}{read this work in its conceptualized form}.

\sphinxstepscope

\newpage
\section{Scenario A: Chymotrypsin inhibiton by an \sphinxstyleemphasis{in silico} designed albumin fusion protein}
\label{\detokenize{scenarios/chymotrypsin_inhibition:scenario-a-br-chymotrypsin-inhibiton-by-an-in-silico-designed-albumin-fusion-protein}}\label{\detokenize{scenarios/chymotrypsin_inhibition::doc}}
\sphinxAtStartPar
Data provided by Marwa Mohamed (Institute of Cell Biology and Immunology, University of Stuttgart, Germany)


\subsection{Project background}
\label{\detokenize{scenarios/chymotrypsin_inhibition:project-background}}
\sphinxAtStartPar
In this scenario, the binding of an \sphinxstyleemphasis{in silico} designed protein to an enzyme was assessed by determining the inhibitory constant \(K_{i}\). Thereby, the efficiency of a newly developed approach to computationally design protein binders to deliberately chosen targets was demonstrated.
This was done by comparing \(K_{i}\) of the designed human serum albumin variant (HSA(M3)) to the wild\sphinxhyphen{}type (HSA(WT)) by respectively applying the proteins to chymotrypsin enzyme reactions. Both designed proteins were individually dimerized into fusion proteins through a huFc.


\subsubsection{Experimental design}
\label{\detokenize{scenarios/chymotrypsin_inhibition:experimental-design}}
\sphinxAtStartPar
Enzyme activity was monitored by measuring the product formation of p\sphinxhyphen{}nitroanilin (p\sphinxhyphen{}NA) photometrically at 410 nm for 30 min at 30°C. Therefore, Succinyl\sphinxhyphen{}gly\sphinxhyphen{}gly\sphinxhyphen{}phe\sphinxhyphen{}p\sphinxhyphen{}nitroanilide (SGGPpNA) was applied as substrate in a concentration range of 0.25 \sphinxhyphen{} 2 mM. For concentration calculations, a p\sphinxhyphen{}NA standard was prepared in the range of 0 \sphinxhyphen{} 0.3 mM in duplicates. \(K_{i}\) of HSA(WT)\sphinxhyphen{}huFc and HSA(M3)\sphinxhyphen{}huFc on chymotrypsin were investigated in independent experiments. Each experiment consisted of enzyme reactions with and without the respective HSA variant. Each enzyme reaction contained 0.2 µM of enzyme and 26.88 µM of the respective HSA variant, if inhibitor was applied. All enzyme reactions were prepared in duplicates.


\subsubsection{Data management}
\label{\detokenize{scenarios/chymotrypsin_inhibition:data-management}}
\sphinxAtStartPar
Experimental data and meta data was filled in EnzymeML Excel templates for each of the two inhibition experiments respectively. Calibration data was stored as Excel files. Measurement data was already blanked.


\subsection{Data preparation}
\label{\detokenize{scenarios/chymotrypsin_inhibition:data-preparation}}

\subsubsection{Imports}
\label{\detokenize{scenarios/chymotrypsin_inhibition:imports}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{import} \PYG{n+nn}{pyenzyme} \PYG{k}{as} \PYG{n+nn}{pe}
\PYG{k+kn}{import} \PYG{n+nn}{copy}
\PYG{k+kn}{import} \PYG{n+nn}{string}
\PYG{k+kn}{from} \PYG{n+nn}{IPython}\PYG{n+nn}{.}\PYG{n+nn}{display} \PYG{k+kn}{import} \PYG{n}{display}
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{parameterestimator} \PYG{k+kn}{import} \PYG{n}{ParameterEstimator}
\PYG{k+kn}{from} \PYG{n+nn}{CaliPytion} \PYG{o}{.}\PYG{n}{tools}\PYG{o}{.}\PYG{n}{standardcurve} \PYG{k+kn}{import} \PYG{n+nn}{StandardCurve}

\PYG{k+kn}{import} \PYG{n+nn}{warnings}
\PYG{n}{warnings}\PYG{o}{.}\PYG{n}{filterwarnings}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ignore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Concentration calculation}
\label{\detokenize{scenarios/chymotrypsin_inhibition:concentration-calculation}}
\sphinxAtStartPar
Product standard data was imported directly from an Excel file. Then, a standard curve was created.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{product\PYGZus{}standard} \PYG{o}{=} \PYG{n}{StandardCurve}\PYG{o}{.}\PYG{n}{from\PYGZus{}excel}\PYG{p}{(}
    \PYG{n}{path}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/chymotrypsin\PYGZus{}inhibition/pNA\PYGZhy{}standard.xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{wavelength}\PYG{o}{=}\PYG{l+m+mi}{410}\PYG{p}{,}
    \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{csv}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{concentration\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{temperature}\PYG{o}{=}\PYG{l+m+mi}{30}\PYG{p}{,}
    \PYG{n}{temperature\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    
\PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Calibration data was automatically blanked.
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fa8d19a4bb0\PYGZgt{}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_3_2}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 2: Fitted quadratic calibration model to standard data of p\sphinxhyphen{}NA.}

\sphinxAtStartPar
Based on the Akaike information criterion (AIC), the relation between concentration and absorption is best described by a quadratic function. The calibration measurements and the fitted calibration model are shown in Fig. 2.


\subsubsection{Experimental data}
\label{\detokenize{scenarios/chymotrypsin_inhibition:experimental-data}}
\sphinxAtStartPar
The EnzymeML documents of each experiment were loaded and the standard curve was applied to the absorption data to yield concentrations.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load data from }
\PYG{n}{chymo\PYGZus{}HSAwt} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/chymotrypsin\PYGZus{}inhibition/chymo\PYGZus{}HSAwt.xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{chymo\PYGZus{}HSAM3} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/chymotrypsin\PYGZus{}inhibition/chymo\PYGZus{}HSA(M3).xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Apply standard curve to \PYGZsq{}EnzymeMLDocument\PYGZsq{}}
\PYG{n}{chymo\PYGZus{}HSAwt} \PYG{o}{=} \PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAwt}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{chymo\PYGZus{}HSAM3} \PYG{o}{=} \PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAM3}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Comparability of the experiments}
\label{\detokenize{scenarios/chymotrypsin_inhibition:comparability-of-the-experiments}}
\sphinxAtStartPar
Since experimental data with HSA(WT)\sphinxhyphen{}huFc and HSA(M3)\sphinxhyphen{}huFc originate from independent experiments, the control reactions without the respective inhibitor were compared by performing a parameter estimation (Fig. 3). Thereby, catalytic efficiency \(\frac{k_{cat}}{K_{m}}\) was used to assess comparability between the data sets, since \(k_{cat}\) and \(K_{m}\) were highly correlated (\sphinxstyleemphasis{r2} > 0.98). High correlations between parameters indicate that the parameters cannot be determined independently with certainty. In this case, the highest initial substrate concentration is presumably too low, compared to the true \(K_{m}\) of the enzyme under the given experimental conditions. However, higher substrate concentration were not applied for multiple reasons. On the one hand dimethyl sulfoxide (DMSO) was used as a co\sphinxhyphen{}solvent of the substrate, which inhibits enzyme activity (\hyperlink{cite.references:id9}{Busby \sphinxstyleemphasis{et al.}, 1999}). Hence, higher initial substrate concentrations would have led to higher enzyme inhibition, which would have distorted the assessment of \(K_{i}\).
On the other hand, high substrate viscosity denied the application of higher concentrations without sacrificing pipetting precision.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Create copies of the data sets and delete measurements with inhibitor.}
\PYG{n}{wt\PYGZus{}control} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAwt}\PYG{p}{)}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m4}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m5}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m6}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m7}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{n}{m3\PYGZus{}control} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAM3}\PYG{p}{)}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m4}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m5}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m6}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m7}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{c+c1}{\PYGZsh{} Estimate kinetic parameters of the control reactions of the HSA wild\PYGZhy{}type data set.}
\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{wt\PYGZus{}control}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Kinetic parameters of HSA(wt) chymotrypsin control reactions:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYGZbs{}
    \PYG{o}{.}\PYG{n}{style}\PYG{o}{.}\PYG{n}{set\PYGZus{}table\PYGZus{}attributes}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{style=}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{font\PYGZhy{}size: 12px}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}


\PYG{c+c1}{\PYGZsh{} Estimate kinetic parameters of the control reactions of the HSA(M3) data set.}
\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{m3\PYGZus{}control}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{Kinetic parameters of HSA(M3) chymotrypsin control reactions:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYGZbs{}
    \PYG{o}{.}\PYG{n}{style}\PYG{o}{.}\PYG{n}{set\PYGZus{}table\PYGZus{}attributes}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{style=}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{font\PYGZhy{}size: 12px}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}


\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{e}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control} \PYG{p}{,}\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control}\PYG{p}{]}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin control reactions HSA(wt)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin control reactions HSA(M3)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{doc}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{n}{title}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{4\PYGZhy{}nitroanilin [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time after reaction start [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xticks}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{,} \PYG{l+m+mi}{20}\PYG{p}{]}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{n}{e}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{ax}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
            \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{l+m+mi}{4}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial SGGPpNA  [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.15}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters of HSA(wt) chymotrypsin control reactions:
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fa8d65fc820\PYGZgt{}
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters of HSA(M3) chymotrypsin control reactions:
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fa8d601f160\PYGZgt{}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_8_4}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 3: Measurement data and fitted irreversible Michaelis\sphinxhyphen{}Menten model for chymotrypsin reactions without HSA inhibitor.}

\sphinxAtStartPar
The two datasets were fitted against all kinetic models, which are displayed in the tables above.
Each dataset is best described by the irreversible Michaelis\sphinxhyphen{}Menten model in terms of AIC and standard deviation on the estimated parameters. Models with product or substrate inhibition resulted in large uncertainties above 80\% on the parameter estimates. Therefore, irreversible Michaelis\sphinxhyphen{}Menten model was utilized for comparison of parameters. \(\frac{k_{cat}}{K_{m}}\) was estimated to be 25.353 min\sphinxhyphen{}1mM\sphinxhyphen{}1 ± 8.57\% for the control reaction of the HSA(WT)\sphinxhyphen{}huFc data set and 24.776 min\sphinxhyphen{}1mM\sphinxhyphen{}1 ± 4.35\% for the HSA(M3)\sphinxhyphen{}huFc data set. As a result, the two experiments showed to be comparable, since the catalytic efficiency differs less than 3 \% between the two data sets.


\subsection{Determination and comparison of \protect\(K_{i}\protect\)}
\label{\detokenize{scenarios/chymotrypsin_inhibition:determination-and-comparison-of-k-i}}
\sphinxAtStartPar
Experimental data of chymotrypsin inhibition by HSA(M3)\sphinxhyphen{}huFc contained negative absorption values for the first measurement point, presumably sourcing from an incorrect blank measurement. Therefore, only measurement data from the second data point (minute 5 and onward) was considered for parameter estimation.
Parameter estimates for all applied kinetic models are displayed in the output below and visualized in Fig. 4.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Parameter estimation for HSA(wt) data set}
\PYG{n}{kinetics\PYGZus{}HSAwt} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAwt}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{inhibitor\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}HSAwt}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{start\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(WT)\PYGZhy{}huFc:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}HSAwt}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYGZbs{}
    \PYG{o}{.}\PYG{n}{style}\PYG{o}{.}\PYG{n}{set\PYGZus{}table\PYGZus{}attributes}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{style=}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{font\PYGZhy{}size: 12px}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Parameter estimation for HSA(M3) data set}
\PYG{n}{kinetics\PYGZus{}HSAM3} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAM3}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{inhibitor\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{start\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(M3)\PYGZhy{}huFc:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYGZbs{}
    \PYG{o}{.}\PYG{n}{style}\PYG{o}{.}\PYG{n}{set\PYGZus{}table\PYGZus{}attributes}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{style=}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{font\PYGZhy{}size: 12px}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize experimental data and fitted models}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{e}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{n}{kinetics\PYGZus{}HSAwt} \PYG{p}{,}\PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{p}{]}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin inhibition by HSA(WT)\PYGZhy{}huFc}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin inhibition by HSA(M3)\PYGZhy{}huFc}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{doc}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{n}{title}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{4\PYGZhy{}nitroanilin [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time after reaction start [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xticks}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{,} \PYG{l+m+mi}{20}\PYG{p}{]}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{n}{e}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{ax}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
            \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial SGGPpNA  [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.2}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(WT)\PYGZhy{}huFc:
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fa8d73792b0\PYGZgt{}
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(M3)\PYGZhy{}huFc:
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fa8d71daf70\PYGZgt{}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_10_4}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 4: Measurement data and fitted product inhibition model for chymotrypsin reactions with respective HSA inhibitior.}

\sphinxAtStartPar
Both reaction systems are best described by the competitive inhibition model, which is indicated by the lowest AIC and standard deviation on the estimated parameters. Thereby, a \(K_{i}\) of 0.460 mM ± 27.24\% was estimated for HSA(WT)\sphinxhyphen{}huFc and 0.059 mM ± 8.51\% for HSA(M3)\sphinxhyphen{}huFc. This resembles a roughly 7\sphinxhyphen{}fold increase in affinity of HSA(M3) to the enzyme compared to the HSA(wt).
Since the competitive inhibition model describes the data the best, HSA(M3)\sphinxhyphen{}huFc presumably interacts with the enzyme in the active site region.


\subsection{Save modeling results}
\label{\detokenize{scenarios/chymotrypsin_inhibition:save-modeling-results}}
\sphinxAtStartPar
Lastly, the modeling results are written to the EnzymeML documents and the files are exported.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{for} \PYG{n}{result}\PYG{p}{,} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{n}{kinetics\PYGZus{}HSAwt}\PYG{p}{,} \PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{p}{]}\PYG{p}{,} \PYG{p}{[}\PYG{n}{chymo\PYGZus{}HSAwt}\PYG{p}{,} \PYG{n}{chymo\PYGZus{}HSAM3}\PYG{p}{]}\PYG{p}{)}\PYG{p}{:}

    \PYG{c+c1}{\PYGZsh{} Write modeling results to kinetic parameters}
    \PYG{n}{k\PYGZus{}cat} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{1 / }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{time\PYGZus{}unit}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

    \PYG{n}{K\PYGZus{}m} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}m}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{data\PYGZus{}conc\PYGZus{}unit}\PYG{p}{)}

    \PYG{n}{K\PYGZus{}ic} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{data\PYGZus{}conc\PYGZus{}unit}\PYG{p}{)}

    \PYG{c+c1}{\PYGZsh{} Define kinetic model}
    \PYG{n}{model} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{KineticModel}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{competitive inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{equation}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZhy{}k\PYGZus{}cat * p0 * s0 / (K\PYGZus{}m*(1+(s2 / K\PYGZus{}ic))+s0)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{parameters}\PYG{o}{=}\PYG{p}{[}\PYG{n}{k\PYGZus{}cat}\PYG{p}{,} \PYG{n}{K\PYGZus{}m}\PYG{p}{,} \PYG{n}{K\PYGZus{}ic}\PYG{p}{]}\PYG{p}{)}

    \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{model} \PYG{o}{=} \PYG{n}{model}

    \PYG{c+c1}{\PYGZsh{} Export EnzymeML documents}
    \PYG{n}{export} \PYG{o}{=} \PYG{k+kc}{False}
    \PYG{k}{if} \PYG{n}{export}\PYG{p}{:}
        \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{toFile}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope

\newpage
\section{Scenario B: α\sphinxhyphen{}glucosidase inhibition by fucoidan}
\label{\detokenize{scenarios/glycosidase_inhibition:scenario-b-br-glucosidase-inhibition-by-fucoidan}}\label{\detokenize{scenarios/glycosidase_inhibition::doc}}
\sphinxAtStartPar
Data provided by Chantal Daub (Biochemistry, Rhodes University, Makhanda, South Africa)


\subsection{Project background}
\label{\detokenize{scenarios/glycosidase_inhibition:project-background}}
\sphinxAtStartPar
In this scenario, the inhibitory properties of fucoidan on \(\alpha\)\sphinxhyphen{}glucosidase from \sphinxstyleemphasis{Saccharomyces cerevisiae} were analyzed. Fucoidan is a sulfated polysaccharide found in various brown algae. The polysaccharide is investigated as a potential active compound in the fields of anti\sphinxhyphen{}cancer, anti\sphinxhyphen{}inflammation, and anti\sphinxhyphen{}coagulate research, among others (\hyperlink{cite.references:id7}{Li \sphinxstyleemphasis{et al.}, 2008}). Furthermore, fucoidan showed to be a promising drug for diabetes mellitus treatment, since it effectively inhibits \(\alpha\)\sphinxhyphen{}glucosidase. Recently, Daub \sphinxstyleemphasis{et al.} (\hyperlink{cite.references:id6}{2020}) showed that \(\alpha\)\sphinxhyphen{}glucosidase inhibition by fucoidan from \sphinxstyleemphasis{E. maxima} exhibited significantly stronger inhibition, compared to the established diabetes drug acarbose. In the following analysis, \(\alpha\)\sphinxhyphen{}glucosidase was exposed to fucoidan from \sphinxstyleemphasis{Ecklonia maxima}, \sphinxstyleemphasis{Ecklonia radiata}, \sphinxstyleemphasis{Fucus vesiculosus}, and \sphinxstyleemphasis{Schimmelmannia elegans} to test their respective capability of \(\alpha\)\sphinxhyphen{}glucosidase inhibiton by assessing their respective \(K_{i}\) values.


\subsubsection{Experimental design}
\label{\detokenize{scenarios/glycosidase_inhibition:experimental-design}}
\sphinxAtStartPar
\(\alpha\)\sphinxhyphen{}glucosidase reactions, catalyzing the hydrolysis of p\sphinxhyphen{}nitrophenyl glucopyranoside (p\sphinxhyphen{}NPG) to p\sphinxhyphen{}nitrophenol (p\sphinxhyphen{}NP) were conducted with and without fucoidan from each seaweed species as well as acarbose. Fucoidan was applied in two different concentrations. p\sphinxhyphen{}NPG was applied in a range from 0.1 mM \sphinxhyphen{} 5 mM, to enzyme reactions containing 9.19 µM \(\alpha\)\sphinxhyphen{}glucosidase.
Product formation was recorded photometrically at 405 nm and 37°C for 20 min. Product concentrations were calculated utilizing a photometric p\sphinxhyphen{}NP standard. Additionally, control reactions without enzyme were prepared to subtract the absorption contribution of the respective inhibitor, buffer, enzyme and substrate from each measurement.


\subsection{Data preparation}
\label{\detokenize{scenarios/glycosidase_inhibition:data-preparation}}

\subsubsection{Imports and parser function}
\label{\detokenize{scenarios/glycosidase_inhibition:imports-and-parser-function}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{typing} \PYG{k+kn}{import} \PYG{n}{Dict}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{colors} \PYG{k}{as} \PYG{n+nn}{mcolors}
\PYG{k+kn}{import} \PYG{n+nn}{re}
\PYG{k+kn}{import} \PYG{n+nn}{os}
\PYG{k+kn}{import} \PYG{n+nn}{pyenzyme} \PYG{k}{as} \PYG{n+nn}{pe}
\PYG{k+kn}{from} \PYG{n+nn}{IPython}\PYG{n+nn}{.}\PYG{n+nn}{display} \PYG{k+kn}{import} \PYG{n}{display}
\PYG{k+kn}{from} \PYG{n+nn}{CaliPytion}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{standardcurve} \PYG{k+kn}{import} \PYG{n}{StandardCurve}
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{parameterestimator} \PYG{k+kn}{import} \PYG{n}{ParameterEstimator}


\PYG{k+kn}{import} \PYG{n+nn}{warnings}
\PYG{n}{warnings}\PYG{o}{.}\PYG{n}{filterwarnings}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ignore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{colors} \PYG{o}{=} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{mcolors}\PYG{o}{.}\PYG{n}{TABLEAU\PYGZus{}COLORS}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Parser}
\PYG{k}{def} \PYG{n+nf}{measurement\PYGZus{}data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{template\PYGZus{}path}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{measurement\PYGZus{}data}\PYG{p}{:} \PYG{n}{np}\PYG{o}{.}\PYG{n}{ndarray}\PYG{p}{,}
    \PYG{n}{species\PYGZus{}id}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{time}\PYG{p}{:} \PYG{n}{np}\PYG{o}{.}\PYG{n}{ndarray}\PYG{p}{,}
    \PYG{n}{data\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{time\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}
    \PYG{p}{)} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZgt{}} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{:}

    \PYG{n}{enzmldoc}\PYG{p}{:} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{n}{template\PYGZus{}path}\PYG{p}{)}

    \PYG{k}{for} \PYG{n}{IDs}\PYG{p}{,} \PYG{n}{concentration} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{keys}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{n}{measurement\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{for} \PYG{n}{counter}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{)}\PYG{p}{:}
            
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getMeasurement}\PYG{p}{(}\PYG{n}{IDs}\PYG{p}{)}\PYG{o}{.}\PYG{n}{addReplicates}\PYG{p}{(}\PYG{n}{pe}\PYG{o}{.}\PYG{n}{Replicate}\PYG{p}{(}
                \PYG{n+nb}{id}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Measurement}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{counter}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
                \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{n}{species\PYGZus{}id}\PYG{p}{,}
                \PYG{n}{data}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{replicate}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{n}{data\PYGZus{}unit}\PYG{p}{,}
                \PYG{n}{time}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{n}{time\PYGZus{}unit}\PYG{p}{)}\PYG{p}{,} \PYG{n}{enzmldoc}\PYG{p}{)}

    \PYG{k}{return} \PYG{n}{enzmldoc}

\PYG{c+c1}{\PYGZsh{} Ignore hidden files in file stystem}
\PYG{k}{def} \PYG{n+nf}{listdir\PYGZus{}nohidden}\PYG{p}{(}\PYG{n}{path}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{for} \PYG{n}{f} \PYG{o+ow}{in} \PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{path}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{f}\PYG{o}{.}\PYG{n}{startswith}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{.}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{:}
            \PYG{k}{yield} \PYG{n}{f}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
Measurement data was provided as an Excel file, whereas metadata was filled in EnzymeML Excel templates for each fucoidan seaweed species and acarbose respectively. In preliminary experiments, p\sphinxhyphen{}NPG showed to slightly absorb at the product detection wavelength. Therefore, the absorbance contribution of substrate at the product detection wavelength was subtracted as well as the contributions of enzyme, buffer and inhibitor. Then, the blanked absorbance data was written to the EnzymeML documents by a parser function.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{dataset\PYGZus{}path} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/glucosidase\PYGZus{}inhibition/experimental\PYGZus{}data\PYGZus{}real.xlsx}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{template\PYGZus{}directory} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/glucosidase\PYGZus{}inhibition/EnzymeML\PYGZus{}templates}\PYG{l+s+s2}{\PYGZdq{}}   

\PYG{n}{enzml\PYGZus{}docs} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{c+c1}{\PYGZsh{} Load experimental data from Excel}
\PYG{n}{excel\PYGZus{}sheets} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{pd}\PYG{o}{.}\PYG{n}{ExcelFile}\PYG{p}{(}\PYG{n}{dataset\PYGZus{}path}\PYG{p}{)}\PYG{o}{.}\PYG{n}{sheet\PYGZus{}names}\PYG{p}{)}
\PYG{n}{inhibitors} \PYG{o}{=} \PYG{n}{excel\PYGZus{}sheets}\PYG{p}{[}\PYG{p}{:}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{]}
\PYG{n}{substrate\PYGZus{}controls} \PYG{o}{=} \PYG{n}{excel\PYGZus{}sheets}\PYG{p}{[}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{]}
\PYG{n}{initial\PYGZus{}substrates} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mf}{2.5}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{]} \PYG{c+c1}{\PYGZsh{} mM}

\PYG{c+c1}{\PYGZsh{} Blank data}
\PYG{c+c1}{\PYGZsh{}\PYGZsh{} Absorption contribution from substrate}
\PYG{n}{substrate\PYGZus{}absorption\PYGZus{}data} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{dataset\PYGZus{}path}\PYG{p}{,} \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{n}{substrate\PYGZus{}controls}\PYG{p}{)}\PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{buffer\PYGZus{}enzyme\PYGZus{}absorption} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{substrate\PYGZus{}absorption\PYGZus{}data}\PYG{o}{.}\PYG{n}{iloc}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{substrate\PYGZus{}absorptions} \PYG{o}{=} \PYG{n}{substrate\PYGZus{}absorption\PYGZus{}data}\PYG{o}{.}\PYG{n}{subtract}\PYG{p}{(}\PYG{n}{buffer\PYGZus{}enzyme\PYGZus{}absorption}\PYG{p}{)}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Buffer+ Enzyme}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{substrate\PYGZus{}absorptions} \PYG{o}{=} \PYG{n}{substrate\PYGZus{}absorptions}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{21}\PYG{p}{)}
\PYG{n}{substrate\PYGZus{}absorptions} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{substrate\PYGZus{}absorptions}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{substrate\PYGZus{}absorptions} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{substrate\PYGZus{}absorptions}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{mapper\PYGZus{}substrate\PYGZus{}enzyme\PYGZus{}absorption} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{initial\PYGZus{}substrates}\PYG{p}{,} \PYG{n}{substrate\PYGZus{}absorptions}\PYG{p}{)}\PYG{p}{)}

\PYG{k}{for} \PYG{n}{inhibitor}\PYG{p}{,} \PYG{n}{template} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{inhibitors}\PYG{p}{)}\PYG{p}{,} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{listdir\PYGZus{}nohidden}\PYG{p}{(}\PYG{n}{template\PYGZus{}directory}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{dataset\PYGZus{}path}\PYG{p}{,} \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{n}{inhibitor}\PYG{p}{)}\PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{time} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{index}\PYG{o}{.}\PYG{n}{values}
    \PYG{n}{inhibitor\PYGZus{}controls} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{iloc}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{p}{:}\PYG{l+m+mi}{4}\PYG{p}{]}
    \PYG{n}{inhibitor\PYGZus{}concs} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{unique}\PYG{p}{(}\PYG{p}{[}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{conc}\PYG{o}{.}\PYG{n}{split}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ }\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{[}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{conc} \PYG{o+ow}{in} \PYG{n}{inhibitor\PYGZus{}controls}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{]}\PYG{p}{)}
    \PYG{n}{inhibitor\PYGZus{}absorptions} \PYG{o}{=} \PYG{n}{inhibitor\PYGZus{}controls}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{21}\PYG{p}{)}
    \PYG{n}{inhibitor\PYGZus{}absorptions} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{inhibitor\PYGZus{}absorptions}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{)}
    \PYG{n}{inhibitor\PYGZus{}absorptions} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{inhibitor\PYGZus{}absorptions}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{)}
    \PYG{n}{mapper\PYGZus{}inhibitor\PYGZus{}absorption} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{inhibitor\PYGZus{}concs}\PYG{p}{,} \PYG{n}{inhibitor\PYGZus{}absorptions}\PYG{p}{)}\PYG{p}{)}
    \PYG{n}{mapper\PYGZus{}inhibitor\PYGZus{}absorption}\PYG{p}{[}\PYG{l+m+mf}{0.0}\PYG{p}{]} \PYG{o}{=} \PYG{n}{buffer\PYGZus{}enzyme\PYGZus{}absorption}
    \PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{iloc}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{:}\PYG{p}{]}
    \PYG{k}{for} \PYG{n}{column} \PYG{o+ow}{in} \PYG{n}{df}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{:}
        \PYG{n}{init\PYGZus{}substrate} \PYG{o}{=} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{column}\PYG{o}{.}\PYG{n}{split}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ }\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{[}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{inhibitor\PYGZus{}conc} \PYG{o}{=} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{column}\PYG{o}{.}\PYG{n}{split}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ }\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{df}\PYG{p}{[}\PYG{n}{column}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{column}\PYG{p}{]} \PYG{o}{\PYGZhy{}} \PYG{n}{mapper\PYGZus{}substrate\PYGZus{}enzyme\PYGZus{}absorption}\PYG{p}{[}\PYG{n}{init\PYGZus{}substrate}\PYG{p}{]} 
        \PYG{n}{df}\PYG{p}{[}\PYG{n}{column}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{column}\PYG{p}{]} \PYG{o}{\PYGZhy{}} \PYG{n}{mapper\PYGZus{}inhibitor\PYGZus{}absorption}\PYG{p}{[}\PYG{n}{inhibitor\PYGZus{}conc}\PYG{p}{]}
    
    \PYG{n}{data} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{21}\PYG{p}{)}
    \PYG{n}{data} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{moveaxis}\PYG{p}{(}\PYG{n}{data}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{21}\PYG{p}{)}

    \PYG{c+c1}{\PYGZsh{} Parse measurement data to EnzymeML documents}
    \PYG{n}{enzml\PYGZus{}docs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{measurement\PYGZus{}data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
        \PYG{n}{template\PYGZus{}path}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{template\PYGZus{}directory}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{template}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{measurement\PYGZus{}data}\PYG{o}{=}\PYG{n}{data}\PYG{p}{,}
        \PYG{n}{time}\PYG{o}{=}\PYG{n}{time}\PYG{p}{,}
        \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{min}\PYG{l+s+s2}{\PYGZdq{}}
    \PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Data quality}
\label{\detokenize{scenarios/glycosidase_inhibition:data-quality}}
\sphinxAtStartPar
In the next cell, the blanked absorption data of each EnzymeML document is visualized with the \sphinxcode{\sphinxupquote{.visualize()}}\sphinxhyphen{}method of \sphinxcode{\sphinxupquote{PyEnzyme}} for quality control.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_output_scroll}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{for} \PYG{n}{doc} \PYG{o+ow}{in} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{:}
    \PYG{n+nb}{print}\PYG{p}{(}\PYG{n}{doc}\PYG{o}{.}\PYG{n}{name}\PYG{p}{)}
    \PYG{n}{doc}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
a\PYGZhy{}glucosidase inhibition by acarbose
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_5_1}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
a\PYGZhy{}glucosidase inhibition by fucoidan from E. maxima
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_5_3}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
a\PYGZhy{}glucosidase inhibition by fucoidan from E. radiata
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_5_5}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
a\PYGZhy{}glucosidase inhibition by fucoidan from F. vesiculosus
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_5_7}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
a\PYGZhy{}glucosidase inhibition by fucoidan from S. elegans
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_5_9}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The technical output of the cell above visualizes the blanked product absorbance data of each dataset. Therein, the individual measurements are labeled from m0 \sphinxhyphen{} m17, which represent individual experimental conditions: measurements m0 \sphinxhyphen{} m5 are from reactions without inhibitor, m6 \sphinxhyphen{} m11 reactions with the lower inhibitor concentration, and m12 \sphinxhyphen{} m17 originate from reactions with the higher inhibitor concentration. Each subplot contains the data of two experimental repeats. In most reactions a decrease followed by an increase of reaction rate is visible around minute 5. Since each dataset originates from a continuous photometric measurement carried out on a single MTP, the observed behavior likely sources from an analytical device malfunction. Additionally, one repeat of measurement ‘m8’ from the acarbose dataset shows low absorption, which is inconsistent with comparable measurements. Therefore, ‘m8’ was excluded from the dataset.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{del} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m8}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Concentration calculation}
\label{\detokenize{scenarios/glycosidase_inhibition:concentration-calculation}}
\sphinxAtStartPar
Standard data of p\sphinxhyphen{}NP was loaded from an Excel file, and a standard curve was created. Then, the standard curve was applied to the EnzymeML documents.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{path\PYGZus{}calibration\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/glucosidase\PYGZus{}inhibition/p\PYGZhy{}NP\PYGZus{}standard.xlsx}\PYG{l+s+s2}{\PYGZdq{}}

\PYG{n}{product\PYGZus{}standard} \PYG{o}{=} \PYG{n}{StandardCurve}\PYG{o}{.}\PYG{n}{from\PYGZus{}excel}\PYG{p}{(}
    \PYG{n}{path}\PYG{o}{=}\PYG{n}{path\PYGZus{}calibration\PYGZus{}data}\PYG{p}{,}
    \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{csv}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{wavelength}\PYG{o}{=}\PYG{l+m+mi}{405}\PYG{p}{,} 
    \PYG{n}{concentration\PYGZus{}unit} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{cutoff\PYGZus{}absorption}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{)}

\PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Quadratic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Apply calibration curves to absorption EnzymeML documents}
\PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{:}
    \PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Quadratic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Calibration data was automatically blanked.
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fd0ac227520\PYGZgt{}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_9_2}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 5: Fitted quadratic calibration model to standard data of p\sphinxhyphen{}NP (s1).}

\sphinxAtStartPar
Based on AIC, 3rd\sphinxhyphen{}degree polynomial, quadratic and rational calibration models describe the relation between absorption to concentration ration of p\sphinxhyphen{}NP equally. Since the 3rd\sphinxhyphen{}degree polynomial model has one additional parameter compared to the quadratic model, the calibration model with less parameters is preferred, since the additional parameter did not give additional information. The fit between quadratic calibration model and calibration standard is visualized in Fig. 5.


\subsection{Parameter estimation}
\label{\detokenize{scenarios/glycosidase_inhibition:parameter-estimation}}
\sphinxAtStartPar
Comparison between different inhibition models and thus determining the kinetic mechanism was not possible, as no model was consistently favored over the other based on AIC and standard deviations of the parameter estimates.

\sphinxAtStartPar
Thus, different inhibition models cannot be compared, since information provided by the initial slopes is insufficient to distinguish kinetic inhibition mechanisms by modeling. In consequence, kinetic parameters were estimated based on initial rates, assuming competitive inhibition, since this kinetic mechanism has been proposed (\hyperlink{cite.references:id26}{Shan \sphinxstyleemphasis{et al.}, 2016}). Additionally, the faulty measurement with an initial substrate concentration of 0.5 mM from the acarbose data set was excluded from parameter estimation.
Based on an initial parameter estimation run with subsequent visual analysis, severals measurements from the \sphinxstyleemphasis{E. maxima} dataset were excluded. Thereby, measurements from both applied inhibitor concentrations and an initial substrate concentration of 2.5 mM and 5 mM showed similar activity compared to reactions without inhibitor. In contrast, the reactions with inhibitor present showed inhibition, according to the difference in slope.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Discard measurement with 2.5 mM and 5 mM and lower inhibition concentration from E. maxima dataset}
\PYG{k}{del} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m10}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m11}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m16}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m17}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Run parameter estimation for all data sets}
\PYG{n}{results}\PYG{o}{=}\PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{:}
    \PYG{n}{result} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{o}{=}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{inhibitor\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{result}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{l+m+mi}{3}\PYG{p}{)}
    \PYG{n}{results}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Display results for competitive inhibition model}
\PYG{n}{labels} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Acarbose}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Fucoidan (E. maxima)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Fucoidan (E. radiata)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Fucoidan (F. vesiculosus)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Fucoidan (S. elegans)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{concat}\PYG{p}{(}\PYG{p}{[}\PYG{n}{result}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{competitive inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{to\PYGZus{}frame}\PYG{p}{(}\PYG{p}{)}\PYG{o}{.}\PYG{n}{T} \PYG{k}{for} \PYG{n}{result} \PYG{o+ow}{in} \PYG{n}{results}\PYG{p}{]}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{keys}\PYG{o}{=}\PYG{n}{labels}\PYG{p}{)}\PYGZbs{}
        \PYG{o}{.}\PYG{n}{reset\PYGZus{}index}\PYG{p}{(}\PYG{p}{)}\PYGZbs{}
        \PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{level\PYGZus{}1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{AIC}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Ki uncompetitive [g / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYGZbs{}
        \PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{level\PYGZus{}0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYGZbs{}
        \PYG{o}{.}\PYG{n}{rename\PYGZus{}axis}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Inhibitor}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{axis} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Kinetic parameters based on competitive inhibition model:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{df}\PYG{o}{.}\PYG{n}{style}\PYG{o}{.}\PYG{n}{set\PYGZus{}table\PYGZus{}attributes}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{style=}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{font\PYGZhy{}size: 12px}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters based on competitive inhibition model:
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fd0ac230700\PYGZgt{}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The output above displays estimated kinetic parameters for all data sets, assuming competitive inhibition between enzyme and inhibitor. As a result \(k_{cat}\) was estimated between 2.317 min\sphinxhyphen{}1 ± 2.14\% and 1.343 min\sphinxhyphen{}1 ± 2.75\%, whereas \(K_{m}\) was estimated between 0.415 mM ± 7.37\% and 0.180 mM ± 14.57\%.
Since all measurements were conducted under equal conditions, \(k_{cat}\) and \(K_{m}\) should be similar across experiments. Deviations in parameter values might therefore result from pipetting or deviations in reaction temperature.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Get K\PYGZus{}i parameter results}
\PYG{k}{def} \PYG{n+nf}{get\PYGZus{}parameter}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{model\PYGZus{}name}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,} \PYG{n}{parameter}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{return} \PYG{n}{result}\PYG{o}{.}\PYG{n}{models}\PYG{p}{[}\PYG{n}{model\PYGZus{}name}\PYG{p}{]}\PYG{o}{.}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{n}{parameter}\PYG{p}{]}

\PYG{n}{ki} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki\PYGZus{}stderr} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{result} \PYG{o+ow}{in} \PYG{n}{results}\PYG{p}{:}
    \PYG{n}{ki}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{get\PYGZus{}parameter}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{competitive inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{ki\PYGZus{}stderr}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{get\PYGZus{}parameter}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{competitive inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}
\PYG{n}{ki} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{ki}\PYG{p}{)}
\PYG{n}{ki\PYGZus{}stderr} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{ki\PYGZus{}stderr}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize}
\PYG{n}{kis} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{,}\PYG{n}{ki}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{f}\PYG{p}{,} \PYG{p}{(}\PYG{n}{a0}\PYG{p}{,} \PYG{n}{a1}\PYG{p}{)} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{gridspec\PYGZus{}kw}\PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{width\PYGZus{}ratios}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mf}{1.4}\PYG{p}{]}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{a0}\PYG{o}{.}\PYG{n}{bar}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{[}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{]}\PYG{p}{,} \PYG{n}{ki}\PYG{p}{[}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{]}\PYG{p}{,} \PYG{n}{yerr}\PYG{o}{=}\PYG{n}{ki\PYGZus{}stderr}\PYG{p}{[}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{]}\PYG{p}{,} \PYG{n}{align}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{center}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{capsize}\PYG{o}{=}\PYG{l+m+mi}{3}\PYG{p}{)}
\PYG{n}{a0}\PYG{o}{.}\PYG{n}{set\PYGZus{}xticklabels}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{[}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{]}\PYG{p}{,} \PYG{n}{rotation} \PYG{o}{=} \PYG{l+m+mi}{90}\PYG{p}{)}
\PYG{n}{a1}\PYG{o}{.}\PYG{n}{bar}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{ki}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{yerr}\PYG{o}{=}\PYG{n}{ki\PYGZus{}stderr}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{align}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{center}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{capsize}\PYG{o}{=}\PYG{l+m+mi}{3}\PYG{p}{)}
\PYG{n}{a1}\PYG{o}{.}\PYG{n}{set\PYGZus{}xticklabels}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{rotation} \PYG{o}{=} \PYG{l+m+mi}{90}\PYG{p}{)}
\PYG{n}{a0}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZdl{}K\PYGZus{}}\PYG{l+s+si}{\PYGZob{}i\PYGZcb{}}\PYG{l+s+s2}{\PYGZdl{} [\PYGZdl{}mg\PYGZdl{} \PYGZdl{}ml\PYGZca{}}\PYG{l+s+s2}{\PYGZob{}}\PYG{l+s+s2}{\PYGZhy{}1\PYGZcb{}\PYGZdl{}]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{glycosidase_inhibition_14_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 6: Estimated inhibitory constants for \(\alpha\)\sphinxhyphen{}glucosidase inhibition by fucoidan from different algae species and acarbose}

\sphinxAtStartPar
Fucoidan from \sphinxstyleemphasis{E. radiata}, \sphinxstyleemphasis{F. vesiculosus}, and \sphinxstyleemphasis{S. elegans} all showed a lower \(K_{i}\) (28, 3, and 16 µg ml\sphinxhyphen{}1, respectively) compared to the acarbose reference (112 µg ml\sphinxhyphen{}1). This resembles a 37\sphinxhyphen{}fold higher binding affinity of fucoidan from \sphinxstyleemphasis{F. vesiculosus} to \(\alpha\)\sphinxhyphen{}glucosidase compared to acarbose. In contrast to the previous study, fucoidan from \sphinxstyleemphasis{E. maxima} showed a more than 7\sphinxhyphen{}fold higher \(K_{i}\) (858 µg ml\sphinxhyphen{}1) compared to acarbose. This likely originates from inconsistencies during experimental preparation. Additionally, 4 of 10 measurements with applied inhibitor were excluded from parameter estimation, since they showed higher activity compared to the respective reactions without inhibitor. Therefore, these measurement should be repeated. Estimated inhibition constants for each tested inhibitor are visualized in Fig. 6.


\subsection{Project discussion}
\label{\detokenize{scenarios/glycosidase_inhibition:project-discussion}}
\sphinxAtStartPar
In general, the estimated parameters are not reliable on a quantitative level, since the malfunction of the analytical device undermines the validity of the whole dataset. Therefore, it cannot be ruled out, that some of the calculated concentrations deviate from the real concentrations. Furthermore, all estimated parameters are only estimated based on three time points of measure. In addition each experimental condition was applied as duplicate repeats, which additionally showed deviations.
In conclusion, the estimated parameters qualitatively confirm that fucoidan from different host organisms, inhibit \(\alpha\)\sphinxhyphen{}glucosidase more strongly, compared to acarbose.


\subsection{Save modeling results}
\label{\detokenize{scenarios/glycosidase_inhibition:save-modeling-results}}
\sphinxAtStartPar
Lastly, the modeling results are written to the EnzymeML documents and the files are exported.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{for} \PYG{n}{result}\PYG{p}{,} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{results}\PYG{p}{,} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{)}\PYG{p}{:}

    \PYG{c+c1}{\PYGZsh{} Write modeling results to kinetic parameters}
    \PYG{n}{k\PYGZus{}cat} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{1 / }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{time\PYGZus{}unit}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

    \PYG{n}{K\PYGZus{}m} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}m}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{data\PYGZus{}conc\PYGZus{}unit}\PYG{p}{)}

    \PYG{n}{K\PYGZus{}ic} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ic}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{data\PYGZus{}conc\PYGZus{}unit}\PYG{p}{)}

    \PYG{c+c1}{\PYGZsh{} Define kinetic model}
    \PYG{n}{model} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{KineticModel}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{competitive inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{equation}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZhy{}k\PYGZus{}cat * p0 * s0 / (K\PYGZus{}m*(1+(s2 / K\PYGZus{}ic))+s0)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{parameters}\PYG{o}{=}\PYG{p}{[}\PYG{n}{k\PYGZus{}cat}\PYG{p}{,} \PYG{n}{K\PYGZus{}m}\PYG{p}{,} \PYG{n}{K\PYGZus{}ic}\PYG{p}{]}\PYG{p}{)}

    \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{model} \PYG{o}{=} \PYG{n}{model}

    \PYG{c+c1}{\PYGZsh{} Export EnzymeML documents}
    \PYG{n}{export} \PYG{o}{=} \PYG{k+kc}{False}
    \PYG{k}{if} \PYG{n}{export}\PYG{p}{:}
        \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{toFile}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope

\newpage
\section{Scenario C: SLAC characterization}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:scenario-c-br-slac-characterization}}\label{\detokenize{scenarios/SLAC_kinetic_characterization::doc}}
\sphinxAtStartPar
Data provided by Alaric Prins (Applied Microbial and Health Biotechnology Institute, Cape Peninsula University of Technology, Capetown, South Africa)


\subsection{Project background}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:project-background}}
\sphinxAtStartPar
The small laccase from \sphinxstyleemphasis{Streptomyces coelicolor} (SLAC) is a member of the muticopper oxidase family. Due to their potential to oxidize various electron\sphinxhyphen{}rich substrates, laccases are of interest for industrial application for lignin degradation (\hyperlink{cite.references:id13}{Widsten and Kandelbauer, 2008}). SLAC is currently used in ongoing biocatalysis and synergy projects. Recently, the wild\sphinxhyphen{}type SLAC was synthesised (previously cloned by restriction cloning) and codon\sphinxhyphen{}optimized for expression in \sphinxstyleemphasis{E. coli}. Therefore, the catalytic properties of the codon\sphinxhyphen{}optimized SLAC were investigated in a broad pH and temperature range, for rapid and accurate determination of biochemical data, which is useful for downstream applications.


\subsubsection{Experimental design}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:experimental-design}}
\sphinxAtStartPar
In this scenario the enzymatic oxidation of 2,2’\sphinxhyphen{}Azino\sphinxhyphen{}bis(3\sphinxhyphen{}ethylbenzothiazoline\sphinxhyphen{}6\sphinxhyphen{}sulfonic acid) (ABTS) to its radical form ABTS+· was studied in the pH range of pH 3 \sphinxhyphen{} pH 5.5 and temperature range of 25°C \sphinxhyphen{} 45°C. In total 30 kinetic enzyme assays in a substrate range between 0 \sphinxhyphen{} 200 µM of ABTS were conducted. Additionally, for each enzyme reaction with a given initial substrate concentration, a control reaction without enzyme was prepared. For each pH \sphinxhyphen{} temperature condition an individual ABTS standard curve and absorption spectrum was recorded to account for varying ABTS absorption properties due to reaction conditions. Each enzyme reaction was followed photometrically at two wavelengths for 15 min, measuring substrate depletion and product accumulation simultaneously. Preliminary experiments confirmed, that ABTS absorbs at 340 nm, whereas the ABTS+· absorbs at 420 nm, in contrast to the substrate. Furthermore, cross absorbance of product at the substrate detection wavelength and vice versa was ruled out.


\subsubsection{Data management}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:data-management}}
\sphinxAtStartPar
Overall, the dataset consists of more than 100 000 individual absorbance reads. Thus, data preparation was automated by custom parser functions, which were tailored to the output of the used spectrophotometer. Information on the involved reactants, and the enzyme was filled in an EnzymeML Excel spreadsheet, which served as a meta data container. All other information was parsed from the output of the spectrophotometer. Fig. 7 illustrates the schematic data flow of this project.

\sphinxAtStartPar
\sphinxincludegraphics{{workflow_SLAC}.png}
\sphinxstyleemphasis{Fig. 7: Schematic data pipeline of the SLAC project.}


\subsection{Data preparation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:data-preparation}}

\subsubsection{Imports}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:imports}}

\subsubsection{Experimental data}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:experimental-data}}
\sphinxAtStartPar
Data from SLAC reactions was loaded from the output files of the photometer and written to individual EnzymeML documents. Then, information of the control reactions was used to subtract the absorption contribution from enzyme and buffer from the substrate and product signal.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Specify the location of the data sets}
\PYG{n}{directory\PYGZus{}measurement\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/TimeCourseData}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{directory\PYGZus{}standard\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/StandardData}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{directory\PYGZus{}spectrum\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/SpectrumData}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{path\PYGZus{}EnzymeML\PYGZus{}templates} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/EnzymeML\PYGZus{}templates}\PYG{l+s+s2}{\PYGZdq{}}

\PYG{c+c1}{\PYGZsh{} Define IDs for species, listed in the EnzymeML Excel template}
\PYG{n}{substrate\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{product\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{substrate\PYGZus{}control\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{product\PYGZus{}control\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{species\PYGZus{}ids} \PYG{o}{=} \PYG{p}{[}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{,} \PYG{n}{product\PYGZus{}id}\PYG{p}{,} \PYG{n}{substrate\PYGZus{}control\PYGZus{}id}\PYG{p}{,} \PYG{n}{product\PYGZus{}control\PYGZus{}id}\PYG{p}{]}


\PYG{c+c1}{\PYGZsh{} Parse measurement data from photometer output}
\PYG{n}{raw\PYGZus{}data\PYGZus{}dict} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
\PYG{k}{for} \PYG{n}{path} \PYG{o+ow}{in} \PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{directory\PYGZus{}measurement\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{data} \PYG{o}{=} \PYG{n}{read\PYGZus{}measurement\PYGZus{}data}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{directory\PYGZus{}measurement\PYGZus{}data}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{path}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{pH} \PYG{o}{=} \PYG{n}{data}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
    \PYG{n}{temp} \PYG{o}{=} \PYG{n}{data}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
    \PYG{n}{raw\PYGZus{}data\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{ }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{temp}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{data}

\PYG{n}{EnzymeML\PYGZus{}template\PYGZus{}dict} \PYG{o}{=} \PYG{p}{\PYGZob{}}
    \PYG{l+m+mf}{3.0}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH3.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{3.5}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH3\PYGZus{}5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{4.0}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH4.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{4.5}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH4\PYGZus{}5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{5.0}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{5.5}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH5\PYGZus{}5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
\PYG{p}{\PYGZcb{}}

\PYG{c+c1}{\PYGZsh{} Write absorption data to EnzymeMLDocuments}
\PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{:} \PYG{n}{List}\PYG{p}{[}\PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{name}\PYG{p}{,} \PYG{n}{data} \PYG{o+ow}{in} \PYG{n}{raw\PYGZus{}data\PYGZus{}dict}\PYG{o}{.}\PYG{n}{items}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{pH} \PYG{o}{=} \PYG{n}{data}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
    \PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{measurement\PYGZus{}data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
        \PYG{n}{template\PYGZus{}path}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{path\PYGZus{}EnzymeML\PYGZus{}templates}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{EnzymeML\PYGZus{}template\PYGZus{}dict}\PYG{p}{[}\PYG{n}{pH}\PYG{p}{]}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{measurement\PYGZus{}data}\PYG{o}{=}\PYG{n}{data}\PYG{p}{,}
        \PYG{n}{species\PYGZus{}ids}\PYG{o}{=}\PYG{n}{species\PYGZus{}ids}\PYG{p}{,}
        \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{umole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Sort documents by ascending pH and temperature}
\PYG{n}{absortion\PYGZus{}enzymemldocs} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{,} \PYG{n}{key}\PYG{o}{=}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{ph}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Blanc measurement data}
\PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{:}
    \PYG{n}{blanc\PYGZus{}measurement} \PYG{o}{=} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}
    \PYG{n}{blanc} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{p}{[}\PYG{n}{repeat}\PYG{o}{.}\PYG{n}{data} \PYG{k}{for} \PYG{n}{repeat} \PYG{o+ow}{in} \PYG{n}{blanc\PYGZus{}measurement}\PYG{p}{]}\PYG{p}{)}

    \PYG{k}{for} \PYG{n+nb}{id}\PYG{p}{,} \PYG{n}{measurement} \PYG{o+ow}{in} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{items}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}
        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}

        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}
        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}
    
    \PYG{c+c1}{\PYGZsh{} Delete control measurement \PYGZsq{}m0\PYGZsq{}}
    \PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Concentration calculation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:concentration-calculation}}
\sphinxAtStartPar
Calibration data was loaded and converted into individual instances of the calibration data model. Some meta data of the calibration needed to be provided to the custom \sphinxcode{\sphinxupquote{read\_calibration\_data}} function, since the output of the used spectrophotometer only contained a minimum of information. Thereafter, a standard curve was created for each calibration data set. Thereby, only absorption values below 3.2 were considered, since higher absorption values could not be converted into concentration accurately due to the detection limit of the photometer.
Lastly, the fit of each standard curves was visualized.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load calibration raw data}
\PYG{n}{calibration\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{standard\PYGZus{}directory} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{sort}\PYG{p}{(}\PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{directory\PYGZus{}standard\PYGZus{}data}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{spectrum\PYGZus{}directory} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{sort}\PYG{p}{(}\PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{directory\PYGZus{}spectrum\PYGZus{}data}\PYG{p}{)}\PYG{p}{)}

\PYG{k}{for} \PYG{n}{standard}\PYG{p}{,} \PYG{n}{spectrum} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{standard\PYGZus{}directory}\PYG{p}{,} \PYG{n}{spectrum\PYGZus{}directory}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{standard} \PYG{o}{=} \PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{directory\PYGZus{}standard\PYGZus{}data}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{standard}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}
    \PYG{n}{spectrum} \PYG{o}{=} \PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{directory\PYGZus{}spectrum\PYGZus{}data}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{spectrum}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}
    \PYG{n}{result} \PYG{o}{=} \PYG{n}{read\PYGZus{}calibration\PYGZus{}data}\PYG{p}{(}
        \PYG{n}{path\PYGZus{}standard}\PYG{o}{=}\PYG{n}{standard}\PYG{p}{,}
        \PYG{n}{path\PYGZus{}spectrum}\PYG{o}{=}\PYG{n}{spectrum}\PYG{p}{,}
        \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{,}
        \PYG{n}{wavelengths}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mi}{340}\PYG{p}{,} \PYG{l+m+mi}{420}\PYG{p}{]}\PYG{p}{,}
        \PYG{n}{concentrations}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{15}\PYG{p}{,}\PYG{l+m+mi}{25}\PYG{p}{,}\PYG{l+m+mi}{50}\PYG{p}{,}\PYG{l+m+mi}{75}\PYG{p}{,}\PYG{l+m+mi}{100}\PYG{p}{,}\PYG{l+m+mi}{125}\PYG{p}{,}\PYG{l+m+mi}{150}\PYG{p}{,}\PYG{l+m+mi}{175}\PYG{p}{,}\PYG{l+m+mi}{200}\PYG{p}{]}\PYG{p}{,}
        \PYG{n}{concentration\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{umole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{device\PYGZus{}manufacturer}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Molecular Devices}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{device\PYGZus{}model}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{SpectraMax i3x}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{spectrum\PYGZus{}reactant\PYGZus{}concentration}\PYG{o}{=}\PYG{l+m+mi}{100}
    \PYG{p}{)}
    \PYG{k}{for} \PYG{n}{pH} \PYG{o+ow}{in} \PYG{n}{result}\PYG{o}{.}\PYG{n}{keys}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{p}{[}\PYG{n}{pH}\PYG{p}{]}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Sort calibration data by ascending pH and temperature}
\PYG{n}{calibration\PYGZus{}data} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{calibration\PYGZus{}data}\PYG{p}{,} \PYG{n}{key} \PYG{o}{=} \PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Generate standard curves for ABTS calibration data}
\PYG{n}{standard\PYGZus{}curves}\PYG{p}{:} \PYG{n}{List}\PYG{p}{[}\PYG{n}{StandardCurve}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{calibration} \PYG{o+ow}{in} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{calibration\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{standard\PYGZus{}curves}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{StandardCurve}\PYG{p}{(}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{=}\PYG{n}{calibration}\PYG{p}{,} \PYG{n}{wavelength}\PYG{o}{=}\PYG{l+m+mi}{340}\PYG{p}{,} \PYG{n}{cutoff\PYGZus{}absorption}\PYG{o}{=}\PYG{l+m+mf}{3.2}\PYG{p}{,} \PYG{n}{show\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Sort standard curves by ascending pH and temperature.}
\PYG{n}{standard\PYGZus{}curves} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{standard\PYGZus{}curves}\PYG{p}{,} \PYG{n}{key} \PYG{o}{=} \PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize all fitted standard curves }
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{13}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{standard}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{standard\PYGZus{}curves}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{30}\PYG{p}{]}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{i}\PYG{o}{\PYGZpc{}}\PYG{k}{5}:
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{standard}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{standard}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{, }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{standard}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{°C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mi}{25}\PYG{p}{,}\PYG{l+m+mi}{26}\PYG{p}{,}\PYG{l+m+mi}{27}\PYG{p}{,}\PYG{l+m+mi}{28}\PYG{p}{,}\PYG{l+m+mi}{29}\PYG{p}{]}\PYG{p}{:}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_6_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 8: ABTS standard curves for different experimental condition.}

\sphinxAtStartPar
Since the absorption characteristics of ABTS change with pH, the calibration range differs between pH values, due to the upper absorption limit (Fig. 8). In consequence, the upper calibration limit for reactions at pH 3, pH 3.5, and pH 5.5 is at 150 µM of ABTS, whereas for all other pH values the upper limit is at 175 uM. This might source from the protonation state of ABTS, since the sulfonate groups of ABTS are deprotonated for less acidic pH values. Therefore, the absorption properties of ABTS might decrease. Calibration curve data at pH 5.5 showed larger variation between the repeats. In this case, pipetting of one of the three repeats differs from the other two, which should be considered for kinetic parameter estimation. In contrast to pH, the temperature during calibration affected the calibration curve only marginally.

\sphinxAtStartPar
The generated standard curves were used to convert the absorption measurement data into concentration data. Thereby, the respective concentration values were only calculated, if the measured absorption was within the respective calibration bounds to avoid extrapolation.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_output_scroll}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Calculate concentrations by applying standard curves to \PYGZsq{}EnzymeMLDocuments\PYGZsq{}.}
\PYG{n}{concentration\PYGZus{}enzymemldocs} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{standard\PYGZus{}curve}\PYG{p}{,} \PYG{n}{abso\PYGZus{}enzmldoc} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{standard\PYGZus{}curves}\PYG{p}{,} \PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{)}\PYG{p}{:}

    \PYG{c+c1}{\PYGZsh{} Validate, that measurement data is paired with respective standard curve}
    \PYG{k}{assert} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH} \PYG{o}{==} \PYG{n}{abso\PYGZus{}enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{ph}
    \PYG{k}{assert} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature} \PYG{o}{==} \PYG{n}{abso\PYGZus{}enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{temperature}

    \PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{abso\PYGZus{}enzmldoc}\PYG{p}{)}
    \PYG{n}{conc\PYGZus{}enzmldoc} \PYG{o}{=} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ommit\PYGZus{}nan\PYGZus{}measurements}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
    \PYG{n}{conc\PYGZus{}enzmldoc} \PYG{o}{=} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ommit\PYGZus{}nan\PYGZus{}measurements}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}

    \PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{conc\PYGZus{}enzmldoc}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Quality control through mass balance analysis}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:quality-control-through-mass-balance-analysis}}
\sphinxAtStartPar
Since product and substrate of the SLAC reaction were simultaneously recorded, mass balance analysis was conducted as a control of quality. By assuming mass conservation, the following concentration balance can be established:
\begin{equation}\label{equation:scenarios/SLAC_kinetic_characterization:mass_balance}
\begin{split}0 = S_{(t)} + P_{(t)} - S_{0}\end{split}
\end{equation}
\sphinxAtStartPar
Where, \(S_{0}\) denotes the initial substrate concentration, whereas \(S_{(t)}\) and \(P_{(t)}\) describe the substrate and product concentration for each time point \(t\). Since \(S_{t}\) and \(S_{0}\) were individually measured, each enzyme reaction with a given initial substrate concentration had a control reaction with identical substrate concentration. In contrast to the substrate, no calibration standard was available for the product. Therefore, an additional parameter \(k\) was introduced to the mass balance equation, assuming linear relationship between the product concentration and its signal:
\begin{equation}\label{equation:scenarios/SLAC_kinetic_characterization:mass_balance_with_k}
\begin{split}0 = S_{(t)} + P_{(t)}k - S_{0}\end{split}
\end{equation}
\sphinxAtStartPar
\(k\) was determined for each dataset individually by a minimization algorithm. The minimization objective was to find the optimal \(k\), which minimizes all slopes of an experiment. If the slopes are zero, mass conservation is given over the observed time\sphinxhyphen{}course. Mass balances of all measurements are visualized in Fig. 9.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Defenition of parameter \PYGZsq{}k\PYGZsq{}}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{Parameters}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{params}\PYG{o}{.}\PYG{n}{add}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{value}\PYG{o}{=}\PYG{l+m+mi}{30}\PYG{p}{,} \PYG{n+nb}{min}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n+nb}{max}\PYG{o}{=}\PYG{l+m+mi}{200}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Target function for the minimizer}
\PYG{k}{def} \PYG{n+nf}{residual}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{x}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{k}\PYG{o}{=}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

    \PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,}  \PYG{n}{control} \PYG{o}{=} \PYG{n}{x}
    \PYG{n}{slopes} \PYG{o}{=} \PYG{n}{substrate}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{o}{*} \PYG{l+m+mf}{0.0}
    \PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{p}\PYG{p}{,} \PYG{n}{c}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,} \PYG{n}{control}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{sub} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{prod} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{p}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{cont} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{c}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}

        \PYG{n}{model} \PYG{o}{=} \PYG{n}{sub} \PYG{o}{+} \PYG{n}{prod}\PYG{o}{*}\PYG{n}{k} \PYG{o}{\PYGZhy{}} \PYG{n}{cont}

        \PYG{n}{slopes}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]} \PYG{o}{=} \PYG{n}{linregress}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{model}\PYG{p}{)}\PYG{p}{)}\PYG{p}{,} \PYG{n}{model}\PYG{p}{)}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}

    \PYG{k}{return} \PYG{n}{slopes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}


\PYG{n}{f} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.5}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{e}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{substrate} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{product} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{initial\PYGZus{}substrate} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{control} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{k}{for} \PYG{n}{measurement} \PYG{o+ow}{in} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{initial\PYGZus{}substrate}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{)}\PYG{o}{.}\PYG{n}{init\PYGZus{}conc}\PYG{p}{)}
        \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
            \PYG{n}{substrate}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}
        \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{n}{product\PYGZus{}id}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
            \PYG{n}{product}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}
        \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
            \PYG{n}{control}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}

    \PYG{n}{time} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{time}\PYG{p}{)}

    \PYG{n}{product} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{product}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{p}{(}\PYG{n}{product}\PYG{p}{)}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{3}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{11}\PYG{p}{)}
    \PYG{n}{substrate} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{)}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{3}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{11}\PYG{p}{)}
    \PYG{n}{control} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{control}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{p}{(}\PYG{n}{control}\PYG{p}{)}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{3}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{11}\PYG{p}{)}
    \PYG{n}{initial\PYGZus{}substrate} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{initial\PYGZus{}substrate}\PYG{p}{)}

    \PYG{n}{x} \PYG{o}{=} \PYG{p}{(}\PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,} \PYG{n}{control}\PYG{p}{)}

    \PYG{n}{result} \PYG{o}{=} \PYG{n}{minimize}\PYG{p}{(}\PYG{n}{residual}\PYG{p}{,} \PYG{n}{params}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{(}\PYG{n}{x}\PYG{p}{,}\PYG{p}{)}\PYG{p}{)}

    \PYG{n}{factor} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
    \PYG{n}{f}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{factor}\PYG{p}{)}

    \PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{p}\PYG{p}{,} \PYG{n}{c}\PYG{p}{,} \PYG{n}{l}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,}  \PYG{n}{control}\PYG{p}{,} \PYG{n}{initial\PYGZus{}substrate}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{init} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{c}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{sub} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{prod} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{p}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{cont} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{c}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{balance} \PYG{o}{=} \PYG{p}{(}\PYG{n}{sub} \PYG{o}{+} \PYG{n}{prod}\PYG{o}{*}\PYG{n}{factor} \PYG{o}{\PYGZhy{}} \PYG{n}{cont}\PYG{p}{)}

        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{doc}\PYG{o}{.}\PYG{n}{getMeasurement}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{time}\PYG{p}{,} \PYG{n}{balance}\PYG{p}{,} \PYG{n}{label} \PYG{o}{=} \PYG{n}{l}\PYG{p}{)}
        \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{i}\PYG{o}{\PYGZpc{}}\PYG{k}{5}:
            \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mass balance [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{k}{if} \PYG{n}{i} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mi}{24}\PYG{p}{,} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{l+m+mi}{26}\PYG{p}{,} \PYG{l+m+mi}{27}\PYG{p}{,} \PYG{l+m+mi}{28}\PYG{p}{,} \PYG{l+m+mi}{29}\PYG{p}{]}\PYG{p}{:}
            \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{pH} \PYG{o}{=} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{ph}
        \PYG{n}{temp} \PYG{o}{=} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{temperature}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{, }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{temp}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{°C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{c+c1}{\PYGZsh{}, factor: \PYGZob{}factor:.0f\PYGZcb{}\PYGZdq{})}
    \PYG{k}{if} \PYG{n}{e} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
        \PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{)}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.03}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_10_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 9: Quality control of each experiment through modeled mass balance.}

\sphinxAtStartPar
For all experiments, except for pH 4, 45°C and pH 4.5, 35°C, the product and substrate is, qualitatively, in balance over the reaction time\sphinxhyphen{}course (Fig. 9). Mass balances, differing from 0 µM originate from differing substrate concentration in the enzyme reaction and the substrate control. In the reactions at pH 4, 45°C and pH 4.5, 35°C the mass balance slopes are not linear. This indicates issues with the measurement. Therefore, the respective measurements likely result in wrong parameter estimations.


\subsection{Kinetic parameter estimation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:kinetic-parameter-estimation}}

\subsubsection{Choice of kinetic model}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:choice-of-kinetic-model}}
\sphinxAtStartPar
Model selection is vital for parameter estimation. Thus, different settings for the parameter estimator were tested. Firstly, product inhibition models were excluded, since in some of the experiments the measured initial substrate concentration was higher than the specified one, which is used to calculate product concentrations. This resulted in calculated product concentrations with negative values. Secondly, substrate inhibition models were excluded, since the model was not able to describe the observed reaction kinetics. This was evident from a higher AIC as well as more than 100 \% standard deviation on the estimated parameters. Lastly, the irreversible Michaelis\sphinxhyphen{}Menten model with and without time\sphinxhyphen{}dependent enzyme inactivation was compared, since enzyme inactivation was observed in previous experiments. Estimated parameters and the fitted models are shown for the experimental data at pH 3 and 25˚C under two scenarios \sphinxhyphen{} taking enzyme inactivation into consideration and, conversely, without enzyme inactivation (Fig. 10).

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{n}{ax} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{ax}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
            \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
        \PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Estimated parameters without time\PYGZhy{}dependent enzyme inactivation:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{without enzyme inactivation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Estimated parameters with time\PYGZhy{}dependent enzyme inactivation:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{with enzyme inactivation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{)}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.09}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Estimated parameters without time\PYGZhy{}dependent enzyme inactivation:
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                               AIC    Km [umole / l]  \PYGZbs{}
irreversible Michaelis Menten \PYGZhy{}162  35.473 +/\PYGZhy{} 5.52\PYGZpc{}   

                              kcat / Km [1/s * 1/umole / l]       kcat [1/s]  
irreversible Michaelis Menten               0.009 +/\PYGZhy{} 5.89\PYGZpc{}  0.308 +/\PYGZhy{} 2.07\PYGZpc{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Estimated parameters with time\PYGZhy{}dependent enzyme inactivation:
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                               AIC    Km [umole / l]  \PYGZbs{}
irreversible Michaelis Menten \PYGZhy{}476  35.137 +/\PYGZhy{} 3.00\PYGZpc{}   

                              kcat / Km [1/s * 1/umole / l]       kcat [1/s]  \PYGZbs{}
irreversible Michaelis Menten               0.013 +/\PYGZhy{} 3.58\PYGZpc{}  0.460 +/\PYGZhy{} 1.95\PYGZpc{}   

                              ki time\PYGZhy{}dep enzyme\PYGZhy{}inactiv. [1/s]  
irreversible Michaelis Menten                   0.001 +/\PYGZhy{} 4.47\PYGZpc{}  
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_13_4}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 10: Fitted irreversible Michaelis\sphinxhyphen{}Menten model to experimental data of SLAC reaction at pH 3 and 25°C with and without considering time\sphinxhyphen{}dependent enzyme inactivation.}

\sphinxAtStartPar
The model with enzyme inactivation describes the data better (Fig. 10B). This is characterized through a lower AIC, as well as lower standard deviations on the estimated parameters.
In absolute terms, the estimated \(K_{m}\) of both models is approximately identical, whereas the \(k_{cat}\) estimate is approximately 33\% lower for the model without enzyme inactivation. Hence, the model without enzyme inactivation underestimates the turnover number of the enzyme. As a result, irreversible Michaelis\sphinxhyphen{}Menten model with time\sphinxhyphen{}dependent enzyme inactivation was selected for the parameter estimation for all data sets.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Run parameter estimator for all datasets, utilizing multi\PYGZhy{}processing.}
\PYG{k}{def} \PYG{n+nf}{run\PYGZus{}ParameterEstimator}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{:} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{)}\PYG{p}{:}    
    \PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
    \PYG{k}{return} \PYG{n}{kinetics}

\PYG{n}{results} \PYG{o}{=} \PYG{n}{Parallel}\PYG{p}{(}\PYG{n}{n\PYGZus{}jobs}\PYG{o}{=}\PYG{l+m+mi}{8}\PYG{p}{)}\PYG{p}{(}\PYG{n}{delayed}\PYG{p}{(}\PYG{n}{run\PYGZus{}ParameterEstimator}\PYG{p}{)}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{)} \PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{)}
\PYG{n}{results} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{results}\PYG{p}{,} \PYG{n}{key}\PYG{o}{=}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize all fitted models}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.5}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{results}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{ph} \PYG{o}{=} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}
    \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{i}\PYG{o}{\PYGZpc{}}\PYG{k}{5}:
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{doc}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title} \PYG{o}{=} \PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{doc}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{, }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{doc}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{temperature}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{°C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mi}{25}\PYG{p}{,}\PYG{l+m+mi}{26}\PYG{p}{,}\PYG{l+m+mi}{27}\PYG{p}{,}\PYG{l+m+mi}{28}\PYG{p}{,}\PYG{l+m+mi}{29}\PYG{p}{]}\PYG{p}{:}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
        \PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{)}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.04}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_15_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 11: Experimental data and fitted irreversible Michaelis\sphinxhyphen{}Menten model with time\sphinxhyphen{}dependent enzyme inactivation of SLAC reactions under various conditions.}

\sphinxAtStartPar
Experimental data as well as the fitted model are visualized in Fig 11. Based on the reaction slopes, no catalytic activity was observed for reactions at pH 5 or higher.
As in the mass balance analysis, the reactions at pH 4, 45°C and pH 4.5, 35°C visually differed from other experiments with identical pH. In both cases, all applied substrate concentrations were higher than intended in the design of the experiment. Furthermore, the resulting parameter estimates have a high uncertainty. Therefore, the respective measurements were excluded from further analysis. Additionally, the results from pH 5.5, 35°C were excluded, since the uncertainty of the parameters could not be estimated. All calculated kinetic parameters are listed in the table below.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}
\begin{sphinxuseclass}{tag_output_croll}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Extract kinetic parameters of all datasets}
\PYG{n}{kcat} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{pH} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{temperature} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{corr\PYGZus{}kcat\PYGZus{}km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{result} \PYG{o+ow}{in} \PYG{n}{results}\PYG{p}{:}
    \PYG{n}{params} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}parameter\PYGZus{}dict}\PYG{p}{(}\PYG{p}{)}

    \PYG{n}{kcat}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{kcat\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{Km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{Km\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{ki}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{ki\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}  

    \PYG{n}{correlation} \PYG{o}{=} \PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{correl}
    \PYG{k}{if} \PYG{n}{correlation} \PYG{o}{==} \PYG{k+kc}{None}\PYG{p}{:}
        \PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n+nb}{float}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{nan}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{correlation}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}

    \PYG{n}{pH}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{)}
    \PYG{n}{temperature}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Organize kinetic and experimantal parameters in a \PYGZsq{}DataFrame\PYGZsq{}}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{o}{.}\PYG{n}{from\PYGZus{}dict}\PYG{p}{(}\PYG{p}{\PYGZob{}}
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pH}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{pH}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature [C]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{temperature}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat [1/s]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km [uM]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki}\PYG{p}{,}
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki\PYGZus{}std}\PYG{p}{,}
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{correlation kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{/} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr} \PYG{o}{=}\PYG{p}{(}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{o}{+}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mf}{0.5} \PYG{o}{*} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr}

\PYG{c+c1}{\PYGZsh{} Exclued results from failed experiments}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{25}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} reached parameter boundaries \PYGZhy{}\PYGZhy{}\PYGZgt{} inactive}
\PYG{c+c1}{\PYGZsh{}df = df.drop(index=12) \PYGZsh{}Km really high stddev}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{17}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{}kcat outlier, mass balance outlier}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{14}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{}kcat outlier, mass balance outlier}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{zeros}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{zeros}\PYG{p}{(}\PYG{l+m+mi}{27}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Calculate halflife of enzyme}
\PYG{k}{def} \PYG{n+nf}{calulate\PYGZus{}halflife}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{return} \PYG{n}{np}\PYG{o}{.}\PYG{n}{log}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{/}\PYG{n}{x}

\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{apply}\PYG{p}{(}\PYG{n}{calulate\PYGZus{}halflife}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{60}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{/} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{*} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{c+c1}{\PYGZsh{} Delete inactivation rates}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
     pH  temperature [C]  kcat [1/s]  kcat stderr     Km [uM]  Km stderr  \PYGZbs{}
0   3.0             25.0    0.459962     0.008966   35.137357   1.053968   
1   3.0             30.0    0.470366     0.009855   32.114083   1.037088   
2   3.0             35.0    0.546029     0.016598   37.454496   1.765891   
3   3.0             40.0    0.712488     0.021754   46.441229   2.155285   
4   3.0             45.0    0.815724     0.022451   47.967746   2.047138   
5   3.5             25.0    0.420480     0.045642   89.752447  13.848584   
6   3.5             30.0    0.399649     0.007455   61.224383   1.674235   
7   3.5             35.0    0.246773     0.009247   28.237534   1.572755   
8   3.5             40.0    0.650031     0.012397   79.432405   2.189303   
9   3.5             45.0    0.978983     0.026556  103.784356   4.040850   
10  4.0             25.0    0.115133     0.008752   65.852866   2.476592   
11  4.0             30.0    0.192374     0.020265   95.314777   5.569713   
12  4.0             35.0    0.282991     0.029279  100.647793  11.576143   
13  4.0             40.0    0.225429     0.012766   97.966631   1.677116   
15  4.5             25.0    0.368730     0.025801  110.902372   0.830267   
16  4.5             30.0    0.486402     0.036084  148.193267   3.440830   
18  4.5             40.0    0.549232     0.021351  178.481651   2.544896   
19  4.5             45.0    0.235716     0.013338   64.718239   0.037679   
20  5.0             25.0    0.008085     0.006871   15.067578  20.447918   
21  5.0             30.0    0.040665     0.005659   50.772187   3.784251   
22  5.0             35.0    0.061677     0.006158   68.874565   0.267258   
23  5.0             40.0    0.131196     0.017131   91.314407   2.756719   
24  5.0             45.0    0.129399     0.014577   96.735544   2.562607   
26  5.5             30.0    0.014374     0.007402   96.125085   0.393705   
27  5.5             35.0    0.124370     0.023968   81.996103  22.257271   
28  5.5             40.0    0.055892     0.016036   48.740460  19.833207   
29  5.5             45.0    0.075249     0.015052  115.390172  27.541849   

    correlation kcat/Km  kcat/Km [1/s * uM]  kcat/Km stderr  zeros  \PYGZbs{}
0              0.535141            0.013090        0.000468    0.0   
1              0.522963            0.014647        0.000564    0.0   
2              0.558865            0.014578        0.000818    0.0   
3              0.608947            0.015342        0.000852    0.0   
4              0.623270            0.017006        0.000864    0.0   
5              0.746575            0.004685        0.000884    0.0   
6              0.666682            0.006528        0.000216    0.0   
7              0.554406            0.008739        0.000587    0.0   
8              0.731830            0.008183        0.000274    0.0   
9              0.774126            0.009433        0.000448    0.0   
10             0.247717            0.001748        0.000148    0.0   
11            \PYGZhy{}0.718208            0.002018        0.000243    0.0   
12             0.608286            0.002812        0.000435    0.0   
13             0.168743            0.002301        0.000136    0.0   
15            \PYGZhy{}0.736267            0.003325        0.000234    0.0   
16             0.816213            0.003282        0.000255    0.0   
18             0.084106            0.003077        0.000127    0.0   
19            \PYGZhy{}0.113239            0.003642        0.000206    0.0   
20             0.276523            0.000537        0.000859    0.0   
21             0.009319            0.000801        0.000126    0.0   
22             0.299400            0.000895        0.000089    0.0   
23            \PYGZhy{}0.426692            0.001437        0.000193    0.0   
24            \PYGZhy{}0.696719            0.001338        0.000155    0.0   
26            \PYGZhy{}0.200038            0.000150        0.000077    0.0   
27             0.771045            0.001517        0.000505    0.0   
28             0.552798            0.001147        0.000571    0.0   
29             0.604743            0.000652        0.000203    0.0   

    half life [min]  half life std  
0          9.139562       0.408568  
1          9.522170       0.476531  
2         10.304462       0.776312  
3          9.127765       0.600883  
4          9.087343       0.530573  
5          8.824513       1.680327  
6         14.545045       0.804521  
7         18.269277       2.766875  
8         11.026573       0.453021  
9          9.110325       0.425387  
10       111.702355     222.064200  
11        23.819987      10.874634  
12        14.929497       4.974473  
13        29.441552      12.288914  
15        22.536023       6.247470  
16        19.046651       4.109261  
18        34.946634      11.931005  
19       108.053681     160.970528  
20       115.522134    2593.086797  
21       115.512956     453.016208  
22       115.519891     286.705892  
23         8.399863       2.519856  
24        11.438272       2.986966  
26       115.523248    1471.972292  
27         5.841192       1.400781  
28         6.132460       2.957303  
29        10.834329       5.343971  
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Parameter estimates for \protect\(k_{cat}\protect\) and \protect\(K_{m}\protect\)}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:parameter-estimates-for-k-cat-and-k-m}}
\sphinxAtStartPar
The parameter estimates for \(k_{cat}\) and \(K_{m}\) showed correlation. For experiments at pH 3 and pH 3.5, the parameters were correlated between \sphinxstyleemphasis{r2} = 0.5 \sphinxhyphen{} 0.85. This indicates, that the highest initial substrate concentration, which was applied for parameter estimation was not sufficiently high. Ideally, the highest initial substrate concentration applied for a kinetic experiment should be 10\sphinxhyphen{}fold higher than \(K_{m}\) to allow independent estimates for \(k_{cat}\) and \(K_{m}\) (\hyperlink{cite.references:id11}{Tipton \sphinxstyleemphasis{et al.}, 2014}). In this scenario, only reactions with an initial substrate concentration up to 175 µM were used for parameter estimation, due to the limited photometric detection range of ABTS. Hence, ABTS was only applied 1.5\sphinxhyphen{}fold to 6\sphinxhyphen{}fold of the estimated \(K_{m}\). For reactions at pH 4 and above, positive as well as negative correlations were observed. This might result from wrong parameter estimates, due to minimal enzyme activity at the given reaction conditions.

\sphinxAtStartPar
Catalytic efficiency for all experimental conditions is visualized in Fig. 12B, whereas the pH is color\sphinxhyphen{}coded. For reactions with identical pH values, \(k_{cat}\) and \(K_{m}\) both increase with temperature. This might be attributed to the high correlation between the parameters. Thus, the individual change of \(k_{cat}\) or \(K_{m}\) through pH or temperature cannot be assessed with certainty. Therefore, the catalytic efficiency \(\frac{k_{cat}}{K_{m}}\), was a better measure to assess the enzyme activity under different reaction conditions.

\sphinxAtStartPar
For reactions at pH 5 and above, almost no catalytic activity was observed. Hence, data of these experimental conditions is excluded from further analysis.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Visualize estimated parameters}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,} \PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{n}{ax} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{ax}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
            \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i}\PYG{o}{==}\PYG{l+m+mi}{1}\PYG{p}{:}
        \PYG{n}{plot}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{xdata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ydata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{xerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{colors}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{)}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{n}{plot}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{xdata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ydata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{xerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{zeros}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{colors}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature [C]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{)}
    
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_19_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 12: A: Catalytic efficiency of SLAC under different pH and temperature conditions. B: Kinetic parameters \(k_{cat}\) and \(K_{m}\) estimates for SLAC reactions at different pH values.}

\sphinxAtStartPar
\(\frac{k_{cat}}{K_{m}}\) in relation to reaction pH is visualized in Fig. 12A. The highest catalytic efficiency was observed at pH 3 and 45°C. Increasing the pH by 0.5 reduced \(\frac{k_{cat}}{K_{m}}\) approximately by half. For higher pH values, \(\frac{k_{cat}}{K_{m}}\) is even further reduced. The catalytic efficiency is therefore highly sensitive to the pH. This might source from changes in the protonation state of either substrate or enzyme, hindering formation of the enzyme\sphinxhyphen{}substrate complex.
In terms of temperature, higher \(\frac{k_{cat}}{K_{m}}\) was achieved at higher temperatures. SLAC might even be more active above 45°C and under pH 3, since in previous studies SLAC was the most active at 95°C (\hyperlink{cite.references:id30}{Prins \sphinxstyleemphasis{et al.}, 2015}).


\subsubsection{Time\sphinxhyphen{}dependent enzyme inactivation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:time-dependent-enzyme-inactivation}}
\sphinxAtStartPar
The half life of SLAC was calculated with equation \eqref{equation:scenarios/SLAC_kinetic_characterization:enzyme_halflife} (\hyperlink{cite.references:id8}{Buxbaum, 2015}).
\begin{equation}\label{equation:scenarios/SLAC_kinetic_characterization:enzyme_halflife}
\begin{split}t_{\frac{1}{2}} = \frac{ln(2)}{k_{inact}}\end{split}
\end{equation}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Exclued inactive enzyme reactions above pH 4.5 and discard measurements with more than 100 \PYGZpc{} standard deviation on enzyme inactivation parameter}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{19}\PYG{p}{]}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} more than 100 \PYGZpc{} standard deviation on enzyme inactivation parameter}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{19}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} more than 100 \PYGZpc{} standard deviation on enzyme inactivation parameter}

\PYG{c+c1}{\PYGZsh{} Plot half life over catalytic efficiency}
\PYG{n}{plot}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{xdata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ydata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{xerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{colors}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_21_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The enzyme’s calculated half life was between 8 \sphinxhyphen{} 18 min for reactions at pH 3 and pH 3.5, whereas enzyme reactions at higher pH values showed a half life between 15 \sphinxhyphen{} 35 min. Generally, reactions with higher catalytic efficiency showed a shorter half life compared to reactions with lower catalytic efficiency.
In order to check for corrections between \(k_{inact}\), \(k_{cat}\), \(K_{m}\), \(\frac{k_{cat}}{K_{m}}\), as well as the reaction temperature and pH, a correlation analysis was conducted across all datasets.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Calculate correlations}
\PYG{n}{df\PYGZus{}corr} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{correlation kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{zeros}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{rho} \PYG{o}{=} \PYG{n}{df\PYGZus{}corr}\PYG{o}{.}\PYG{n}{corr}\PYG{p}{(}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize correlation heatmap}
\PYG{c+c1}{\PYGZsh{}sns.heatmap(rho, }
\PYG{c+c1}{\PYGZsh{}        xticklabels=rho.columns,}
\PYG{c+c1}{\PYGZsh{}        yticklabels=rho.columns,}
\PYG{c+c1}{\PYGZsh{}        cmap = sns.color\PYGZus{}palette(\PYGZdq{}vlag\PYGZdq{}, as\PYGZus{}cmap=True))}
\PYG{c+c1}{\PYGZsh{}plt.show()}

\PYG{c+c1}{\PYGZsh{} Calculate confidence intercal for correlations}
\PYG{n}{pval} \PYG{o}{=} \PYG{n}{df\PYGZus{}corr}\PYG{o}{.}\PYG{n}{corr}\PYG{p}{(}\PYG{n}{method}\PYG{o}{=}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{,} \PYG{n}{y}\PYG{p}{:} \PYG{n}{pearsonr}\PYG{p}{(}\PYG{n}{x}\PYG{p}{,} \PYG{n}{y}\PYG{p}{)}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)} \PYG{o}{\PYGZhy{}} \PYG{n}{np}\PYG{o}{.}\PYG{n}{eye}\PYG{p}{(}\PYG{o}{*}\PYG{n}{rho}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{)}
\PYG{n}{p} \PYG{o}{=} \PYG{n}{pval}\PYG{o}{.}\PYG{n}{applymap}\PYG{p}{(}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{.}\PYG{n}{join}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{*}\PYG{l+s+s1}{\PYGZsq{}} \PYG{k}{for} \PYG{n}{t} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mf}{.05}\PYG{p}{]} \PYG{k}{if} \PYG{n}{x}\PYG{o}{\PYGZlt{}}\PYG{o}{=}\PYG{n}{t}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{rho}\PYG{o}{.}\PYG{n}{round}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{.}\PYG{n}{astype}\PYG{p}{(}\PYG{n+nb}{str}\PYG{p}{)} \PYG{o}{+} \PYG{n}{p}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
                        pH temperature [C] kcat [1/s] Km [uM]  \PYGZbs{}
pH                    1.0*           \PYGZhy{}0.15      \PYGZhy{}0.39   0.88*   
temperature [C]      \PYGZhy{}0.15            1.0*       0.6*    0.07   
kcat [1/s]           \PYGZhy{}0.39            0.6*       1.0*   \PYGZhy{}0.03   
Km [uM]              0.88*            0.07      \PYGZhy{}0.03    1.0*   
kcat/Km [1/s * uM]  \PYGZhy{}0.89*            0.27      0.58*  \PYGZhy{}0.75*   
half life [min]       0.8*            0.03      \PYGZhy{}0.49   0.68*   

                   kcat/Km [1/s * uM] half life [min]  
pH                             \PYGZhy{}0.89*            0.8*  
temperature [C]                  0.27            0.03  
kcat [1/s]                      0.58*           \PYGZhy{}0.49  
Km [uM]                        \PYGZhy{}0.75*           0.68*  
kcat/Km [1/s * uM]               1.0*          \PYGZhy{}0.71*  
half life [min]                \PYGZhy{}0.71*            1.0*  
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The above table shows correlation between kinetic parameters and experimental conditions. Thereby, significant (p < 0.05) correlations are labeled with ‘*’. The half life of SLAC is significantly correlated to \(\frac{k_{cat}}{K_{m}}\) as well as to pH. Due to correlation between \(k_{cat}\) and \(K_{m}\), the enzyme’s half life is also correlated to \(K_{m}\). Multiple reasons are possible this observation. On the one hand, the observed correlation might be of technical origin. As such, the system of ordinary differential equations, describing the change in substrate and enzyme concentration, are not an accurate model for the reaction system. Therefore, high cross\sphinxhyphen{}correlation occurs, since individual observations cannot be attributed to individual parameters. On the other hand, enzyme activity and enzyme inactivation might be causally related. In this case, the formed ABTS radical might inactivate the enzyme by potential suicide inhibition, or the enzyme deteriorates through catalysis.


\subsection{Project discussion}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:project-discussion}}

\subsubsection{Data management}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:id5}}
\sphinxAtStartPar
The established data pipeline allows scalable data preparation and analysis of enzyme kinetics experiments. In future experiments this workflow could be used to further expand the experimental parameter space. As a result, different enzymes with different substrates in different buffers at different temperatures and pH values could be analyzed.


\subsubsection{Correlation between parameters}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:correlation-between-parameters}}
\sphinxAtStartPar
The highest catalytic efficiency of SLAC was observed at pH 3 at 45°C and therefore on the edge of the investigated parameter space. Optimal reaction conditions might therefore be at an even lower pH and higher temperature. \(k_{cat}\) and \(K_{m}\) were correlated, which is likely the result of a too low initial substrate concentration in relation to the true \(K_{m}\) of the enzyme at a given experimental condition. Hence, in adjoining experiments the product signal could be used for parameter estimation, since the concentration to absorbance ratio is lower compared to the one of the product. However, a reliable method for product quantification would need to be established first.


\subsubsection{Enzyme inactivation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:enzyme-inactivation}}
\sphinxAtStartPar
SLAC showed a short half life of approximately 10 min at reaction conditions with the highest catalytic efficiency, which was correlated to the catalytic efficiency. Further experiments are required to investigate the observed inactivation. This might be done by comparing the half life of SLAC with different substrates as well as higher enzyme concentrations. Enzyme inactivation is further discussed in the next chapter.


\subsection{Save modeling results}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:save-modeling-results}}
\sphinxAtStartPar
Lastly, the modeling results are written to the EnzymeML documents and the files are exported.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{for} \PYG{n}{result}\PYG{p}{,} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{results}\PYG{p}{,} \PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{)}\PYG{p}{:}

    \PYG{c+c1}{\PYGZsh{} Write modeling results to kinetic parameters}
    \PYG{n}{k\PYGZus{}cat} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{1 / }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{time\PYGZus{}unit}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

    \PYG{n}{K\PYGZus{}m} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}m}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{value}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
        \PYG{n}{unit}\PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{data\PYGZus{}conc\PYGZus{}unit}\PYG{p}{)}

    \PYG{c+c1}{\PYGZsh{} Define kinetic model}
    \PYG{n}{model} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{KineticModel}\PYG{p}{(}
        \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis\PYGZhy{}Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{equation}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZhy{}k\PYGZus{}cat * p0 * s0 / (K\PYGZus{}m + s0)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{parameters}\PYG{o}{=}\PYG{p}{[}\PYG{n}{k\PYGZus{}cat}\PYG{p}{,} \PYG{n}{K\PYGZus{}m}\PYG{p}{]}\PYG{p}{)}

    \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{model} \PYG{o}{=} \PYG{n}{model}

    \PYG{n}{add\PYGZus{}enzyme\PYGZus{}inactivation\PYGZus{}model}\PYG{p}{(}
        \PYG{n}{result}\PYG{o}{=}\PYG{n}{result}\PYG{p}{,}
        \PYG{n}{enzmldoc}\PYG{o}{=}\PYG{n}{enzmldoc}\PYG{p}{)}
    


    \PYG{c+c1}{\PYGZsh{} Export EnzymeML documents}
    \PYG{n}{export} \PYG{o}{=} \PYG{k+kc}{False}
    \PYG{k}{if} \PYG{n}{export}\PYG{p}{:}
        \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{toFile}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope


\newpage
\section{Scenario D: Time\sphinxhyphen{}dependent enzyme inactivation}
\label{\detokenize{scenarios/enzyme_inactivation:scenario-d-br-time-dependent-enzyme-inactivation}}\label{\detokenize{scenarios/enzyme_inactivation::doc}}
\sphinxAtStartPar
Data provided by Paulo Durão (Microbial \& Enzyme Technology, Instituto de Tecnologia Química e Biológica, Oeiras, Portugal)


\subsection{Project background}
\label{\detokenize{scenarios/enzyme_inactivation:project-background}}
\sphinxAtStartPar
All investigated enzyme reactions without inhibitor applied to the reaction showed progress curve behavior, which was not explainable by irreversible Michaelis\sphinxhyphen{}Menten kinetics. All experiments had in common, that enzyme reactions were carried out in 96\sphinxhyphen{}well polystyrene micro titer plates (MTP), whereas the change in substrate or product absorption was monitored photometrically. One hypothesis for the suspected enzyme inactivation was hydrophobic interaction between the enzyme and the MTP surface. Thereby, hydrophobic regions of the enzyme’s surface might interact with the polystyrene reaction vessel, potentially preventing substrate access to the active site of the enzyme. Another hypothesis for the observed gradual activity decrease was inactivation which is linked to the catalytic activity, as observed in the previous scenario.

\sphinxAtStartPar
In order to test the hypothesis, an experiment was designed which detaches enzyme inactivation from enzyme activity. Therefore, enzyme was incubated in MTP wells prior to reaction start for different periods. Thereby, the enzyme activity should decrease depending on prior incubation time. If inactivation is independent from catalysis, the calculated half life from the adsorption experiment should match the half life of an enzyme kinetics experiment which was conducted in parallel and also considered an inactivation term.


\subsubsection{Experimental design}
\label{\detokenize{scenarios/enzyme_inactivation:experimental-design}}
\sphinxAtStartPar
\sphinxstylestrong{Determination of enzyme inactivation through adsorption}

The inactivation experiment was conducted by incubating CotA laccase from \sphinxstyleemphasis{Bacillus subtilis} in individual MTP wells up to 1 h prior to reaction start. Then, enzyme reactions were started in 10 min intervals by transferring 2 µL of incubated enzyme to individual MTP wells. Each proceeding enzyme reaction contained 256 nM CotA, 1 mM ABTS and was buffered in acetate buffer at pH 4. Product formation was followed photometrically at 420 nm and 25°C for 5 min, whereas concentrations were calculated assuming an extinction coefficient of ε = 36000 M\sphinxhyphen{}1cm\sphinxhyphen{}1 for the ABTS radical product. Each experiment was performed as technical triplicates.

\sphinxAtStartPar
\sphinxstylestrong{Enzyme kinetics experiment}

Enzymatic oxidation of ABTS to its radical form was followed photometrically at 420 nm at 25°C for 70 min.
Thereby, ABTS was applied in a range from 0.01 mM \sphinxhyphen{} 2 mM. Each proceeding enzyme reaction contained 256 nM CotA, and was buffered in acetate buffer at pH 4. Each experimental condition was repeated as technical quadruplicates. Concentrations were calculated via the provided extinction coefficient of the ABTS radical.


\subsubsection{Experimental data}
\label{\detokenize{scenarios/enzyme_inactivation:experimental-data}}
\sphinxAtStartPar
Measurement data was provided as an Excel file, containing time\sphinxhyphen{}course absorption data. Meta data was written into the EnzymeML Excel template. Then, the experimental data was written to an EnzymeML document by a parser function.


\subsection{Adsorption of enzyme to micro titer plate surface}
\label{\detokenize{scenarios/enzyme_inactivation:adsorption-of-enzyme-to-micro-titer-plate-surface}}

\subsubsection{Imports}
\label{\detokenize{scenarios/enzyme_inactivation:imports}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{from} \PYG{n+nn}{scipy}\PYG{n+nn}{.}\PYG{n+nn}{stats} \PYG{k+kn}{import} \PYG{n}{linregress}
\PYG{k+kn}{import} \PYG{n+nn}{pyenzyme} \PYG{k}{as} \PYG{n+nn}{pe}
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{parameterestimator} \PYG{k+kn}{import} \PYG{n}{ParameterEstimator}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{cm} \PYG{k}{as} \PYG{n+nn}{cm}
\PYG{k+kn}{from} \PYG{n+nn}{IPython}\PYG{n+nn}{.}\PYG{n+nn}{display} \PYG{k+kn}{import} \PYG{n}{display}
\PYG{k+kn}{import} \PYG{n+nn}{string}
\PYG{k+kn}{from} \PYG{n+nn}{parser\PYGZus{}functions} \PYG{k+kn}{import} \PYG{n}{add\PYGZus{}enzyme\PYGZus{}inactivation\PYGZus{}model}

\PYG{k+kn}{import} \PYG{n+nn}{warnings}
\PYG{n}{warnings}\PYG{o}{.}\PYG{n}{filterwarnings}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ignore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Calculation of inactivation rate}
\label{\detokenize{scenarios/enzyme_inactivation:calculation-of-inactivation-rate}}
\sphinxAtStartPar
Experimental data was loaded into a Pandas DataFrame and the slopes of each incubation condition were calculated through linear regression. The resulting initial rates were then used to calculate the rate by which the enzyme activity decreases, when subjected to the reaction vessel.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load excel}
\PYG{n}{path} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{../../data/enzyme\PYGZus{}inactivation/Slide 2 \PYGZhy{} Activity effect of incubating CotA in MTP.xlsx}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{path}\PYG{p}{,} \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{time (min)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} replace values of \PYGZsq{}0**\PYGZsq{} with nan\PYGZhy{}values, since the measurement is incorrect}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{0**}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{nan}

\PYG{c+c1}{\PYGZsh{} Get data from Excel file}
\PYG{n}{columns} \PYG{o}{=} \PYG{p}{[}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{df}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{)} \PYG{k}{if} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)}\PYG{o}{.}\PYG{n}{endswith}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{]}
\PYG{n}{time} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{index}\PYG{o}{.}\PYG{n}{values}
\PYG{n}{absorption} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{7}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{22}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Calculate concentrations}
\PYG{n}{extinction\PYGZus{}coefficient} \PYG{o}{=} \PYG{l+m+mi}{36} \PYG{c+c1}{\PYGZsh{} (1/mM * 1/cm)}
\PYG{n}{optical\PYGZus{}length} \PYG{o}{=} \PYG{l+m+mf}{0.65} \PYG{c+c1}{\PYGZsh{} cm}

\PYG{k}{def} \PYG{n+nf}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{abso}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{return} \PYG{n}{abso} \PYG{o}{/} \PYG{p}{(}\PYG{n}{extinction\PYGZus{}coefficient}\PYG{o}{*}\PYG{n}{optical\PYGZus{}length}\PYG{p}{)}

\PYG{n}{concentration} \PYG{o}{=} \PYG{n}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{absorption}\PYG{p}{)}
\PYG{n}{concentration\PYGZus{}mean} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{nanmean}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{concentration\PYGZus{}std} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{nanstd}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Linear regression}
\PYG{n}{slopes} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{intercepts} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{stderrs} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{time\PYGZus{}set} \PYG{o+ow}{in} \PYG{n}{concentration}\PYG{p}{:}
    \PYG{n}{mask} \PYG{o}{=} \PYG{o}{\PYGZti{}}\PYG{n}{np}\PYG{o}{.}\PYG{n}{isnan}\PYG{p}{(}\PYG{n}{time\PYGZus{}set}\PYG{p}{)}
    \PYG{n}{time\PYGZus{}regression} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{tile}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n}{mask}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{)}\PYG{p}{[}\PYG{n}{mask}\PYG{p}{]}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{data\PYGZus{}regression} \PYG{o}{=} \PYG{n}{time\PYGZus{}set}\PYG{p}{[}\PYG{n}{mask}\PYG{p}{]}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{slope}\PYG{p}{,}\PYG{n}{intercept}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{stderr} \PYG{o}{=} \PYG{p}{(}\PYG{n}{linregress}\PYG{p}{(}\PYG{n}{time\PYGZus{}regression}\PYG{p}{,} \PYG{n}{data\PYGZus{}regression}\PYG{p}{)}\PYG{p}{)}
    \PYG{n}{slopes}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{slope}\PYG{p}{)}
    \PYG{n}{intercepts}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{intercept}\PYG{p}{)}
    \PYG{n}{stderrs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{stderr}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Regression of the slopes between 10 and 60 min}
\PYG{n}{slope\PYGZus{}of\PYGZus{}slopes}\PYG{p}{,} \PYG{n}{intercept\PYGZus{}of\PYGZus{}slopes}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{\PYGZus{}} \PYG{o}{=} \PYG{n}{linregress}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{,} \PYG{n}{slopes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Calculate time at which enzyme activity is reduced by half}
\PYG{n}{half\PYGZus{}activity} \PYG{o}{=} \PYG{p}{(}\PYG{p}{(}\PYG{n}{slopes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{2} \PYG{o}{\PYGZhy{}} \PYG{n}{slopes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)} \PYG{o}{/} \PYG{n}{slope\PYGZus{}of\PYGZus{}slopes}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Plot results}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Calculated time after enzyme activity is halved based on linear regression: }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{half\PYGZus{}activity}\PYG{l+s+si}{:}\PYG{l+s+s2}{.2f}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{ min}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{colors} \PYG{o}{=} \PYG{n}{cm}\PYG{o}{.}\PYG{n}{tab10}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{linspace}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{absorption}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}

\PYG{k}{for} \PYG{n}{mean}\PYG{p}{,} \PYG{n}{std}\PYG{p}{,} \PYG{n}{label}\PYG{p}{,} \PYG{n}{slope}\PYG{p}{,} \PYG{n}{intercept}\PYG{p}{,} \PYG{n}{color} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}mean}\PYG{p}{,} \PYG{n}{concentration\PYGZus{}std}\PYG{p}{,} \PYG{n}{columns}\PYG{p}{,} \PYG{n}{slopes}\PYG{p}{,} \PYG{n}{intercepts}\PYG{p}{,} \PYG{n}{colors}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{n}{mean}\PYG{p}{,} \PYG{n}{std}\PYG{p}{,} \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{n}{label}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=} \PYG{l+m+mf}{0.4}\PYG{p}{,} \PYG{n}{color}\PYG{o}{=}\PYG{n}{color}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{o}{*}\PYG{n}{slope}\PYG{o}{+}\PYG{n}{intercept}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZhy{}\PYGZhy{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{color}\PYG{o}{=}\PYG{n}{color}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{CotA incubation}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{in MTP\PYGZhy{}well [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ABTS radical [mM]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{time [min]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Experimental data with calculated reaction rate by linear regression}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
            \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}


\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{,} \PYG{n}{slopes}\PYG{p}{,} \PYG{n}{stderrs}\PYG{p}{,} \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial rates within the first 5 min}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{CotA incubation time}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{in MTP\PYGZhy{}well [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{initial rate [mM min\PYGZdl{}\PYGZca{}}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZhy{}1\PYGZcb{}\PYGZdl{}]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}\PYG{p}{,} \PYG{n}{slope\PYGZus{}of\PYGZus{}slopes}\PYG{o}{*}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}\PYG{o}{+}\PYG{n}{intercept\PYGZus{}of\PYGZus{}slopes}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZhy{}\PYGZhy{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{color}\PYG{o}{=}\PYG{n}{colors}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{regression}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Initial rates of CotA reaction}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
        \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Calculated time after enzyme activity is halved based on linear regression: 105.09 min
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{enzyme_inactivation_3_1}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 13: Experimental data and regression results of CotA reaction with different enzyme incubation times.}

\sphinxAtStartPar
Enzyme reactions without prior incubation showed the fastest reaction rate, whereas all reactions with prior incubation showed an approximately halved reaction rate (Fig. 13A). Thereby, reactions with longer incubation time showed gradually reduced reaction rates with increased incubation time. Enzyme reactions without incubation showed large deviations across all replicates. This presumably resulted from inhomogeneous mixing or deviating enzyme concentration due to the small volume of enzyme solution, which was used to start the reactions. Since the true reaction rate of reactions without prior incubation was highly uncertain and showed mixing issues, the respective data was not used to estimate the enzyme’s inactivation rate.
Based on linear regression of the initial reaction rates, excluding the reaction without prior incubation, the enzyme activity is approximately halved after 105 min (Fig. 13B). This assumes validity of extrapolation.


\subsection{Enzyme kinetics experiment}
\label{\detokenize{scenarios/enzyme_inactivation:enzyme-kinetics-experiment}}

\subsubsection{Experimental data}
\label{\detokenize{scenarios/enzyme_inactivation:id1}}
\sphinxAtStartPar
Experimental data was provided as an Excel file, whereas meta data of the experiment was filled into an EnzymeML Excel spreadsheet. Measurement data was written to the EnzymeML document by a parser function.
Kinetic parameters were estimated with and without considering enzyme inactivation.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load experimental data from excel file}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/enzyme\PYGZus{}inactivation/Repetition CotA ABTS kinetics higher volumes 2nd time.xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{csv}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Time(min)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{data} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{8}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{,}\PYG{l+m+mi}{72}\PYG{p}{)}
\PYG{n}{time} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{index}\PYG{o}{.}\PYG{n}{values}

\PYG{c+c1}{\PYGZsh{} Calculate concentrations}
\PYG{n}{extinction\PYGZus{}coefficient} \PYG{o}{=} \PYG{l+m+mi}{36} \PYG{c+c1}{\PYGZsh{} (1/mM * 1/cm)}
\PYG{n}{optical\PYGZus{}length} \PYG{o}{=} \PYG{l+m+mf}{0.65} \PYG{c+c1}{\PYGZsh{} cm}

\PYG{k}{def} \PYG{n+nf}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{abso}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{return} \PYG{n}{abso} \PYG{o}{/} \PYG{p}{(}\PYG{n}{extinction\PYGZus{}coefficient}\PYG{o}{*}\PYG{n}{optical\PYGZus{}length}\PYG{p}{)}

\PYG{n}{concentration\PYGZus{}data} \PYG{o}{=} \PYG{n}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{data}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Parser function}
\PYG{k}{def} \PYG{n+nf}{data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{template\PYGZus{}path}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{measurement\PYGZus{}data}\PYG{p}{:} \PYG{n}{np}\PYG{o}{.}\PYG{n}{ndarray}\PYG{p}{,}
    \PYG{n}{species\PYGZus{}id}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{data\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{time\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}
    \PYG{p}{)} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZgt{}} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{:}

    \PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{n}{template\PYGZus{}path}\PYG{p}{)}
    \PYG{k}{for} \PYG{n}{IDs}\PYG{p}{,} \PYG{n}{concentration} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{keys}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{n}{measurement\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{for} \PYG{n}{counter}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{rep} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{Replicate}\PYG{p}{(}
                \PYG{n+nb}{id}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Measurement}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{counter}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
                \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{n}{species\PYGZus{}id}\PYG{p}{,}
                \PYG{n}{data}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{replicate}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{n}{data\PYGZus{}unit}\PYG{p}{,}
                \PYG{n}{time}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{n}{time\PYGZus{}unit}\PYG{p}{)}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getMeasurement}\PYG{p}{(}\PYG{n}{IDs}\PYG{p}{)}\PYG{o}{.}\PYG{n}{addReplicates}\PYG{p}{(}\PYG{n}{rep}\PYG{p}{,} \PYG{n}{enzmldoc}\PYG{p}{)}
    \PYG{k}{return} \PYG{n}{enzmldoc}

\PYG{c+c1}{\PYGZsh{} Write experimantal data to EnzymeML document vis parser function}
\PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{template\PYGZus{}path}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/enzyme\PYGZus{}inactivation/EnzymeML\PYGZus{}CotA.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{measurement\PYGZus{}data}\PYG{o}{=}\PYG{n}{concentration\PYGZus{}data}\PYG{p}{,}
    \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{min}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize experimantal data}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,} \PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{measurement} \PYG{o+ow}{in} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{concentration} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{product} \PYG{o}{=} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{init\PYGZus{}substrate}\PYG{o}{=}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{init\PYGZus{}conc}
    \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{product}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
        \PYG{n}{concentration}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{std}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{n}{init\PYGZus{}substrate}\PYG{p}{,}\PYGZbs{}
        \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{.}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.5}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{std}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{n}{init\PYGZus{}substrate}\PYG{p}{,}\PYGZbs{}
        \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{o}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical concentration over CotA reaction time\PYGZhy{}course }\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical concentration within the first 5 min}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlim}\PYG{p}{(}\PYG{p}{[}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.2}\PYG{p}{,}\PYG{l+m+mf}{5.2}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylim}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mf}{0.01}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
        \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
        \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{fig}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{enzyme_inactivation_5_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig 14. ABTS radical concentration over the time\sphinxhyphen{}course of CotA reaction.}

\sphinxAtStartPar
The provided reaction data showed large deviations between individual repeats of reactions with identical initial ABTS concentrations. Especially for initial ABTS concentrations above 0.1 mM (Fig. 14A).
Furthermore, visual analysis of the initial measurement points indicated that reactions with an initial substrate concentration of 0.1 mM had an increased enzyme concentration compared to others. This is evident, because the mentioned reactions show the highest concentration at the initial measurement time (Fig. 14B). The order of initial measurement points is expected to reflect the initial substrate concentrations, since the substrate slightly absorbs at the product detection wavelength. The substrate of the mentioned reaction was likely applied correctly, since the final concentration at 70 min is between reactions with the next lower and next higher initial substrate concentration.


\subsection{Kinetic parameter estimation}
\label{\detokenize{scenarios/enzyme_inactivation:kinetic-parameter-estimation}}

\subsubsection{Quality control through modeling}
\label{\detokenize{scenarios/enzyme_inactivation:quality-control-through-modeling}}
\sphinxAtStartPar
An initial parameter estimation considering enzyme inactivation was conducted in order to check further systematic deviations. As a result, reactions with an initial ABTS concentration of 0.05 mM showed to deviate from all other reactions, since the product concentrations was lower than the model predicted. The mentioned measurements were the only ones where the measured concentration was distinctly and systematically below that of the model (Fig. 15A). Hence indicating, that either less substrate or enzyme than defined in the protocol was applied to the respective reactions. In consequence, data from reactions with an initial substrate concentration of 0.05 and 0.1 mM were excluded from parameter estimation, because the identified systematic deviations in reaction conditions would distort the results.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Parameter estimation considering time\PYGZhy{}dependent enzyme inactivation}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}first\PYGZus{}run} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}first\PYGZus{}run}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{l+m+mf}{0.02}\PYG{p}{,} \PYG{l+m+mf}{0.05}\PYG{p}{,} \PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+m+mf}{0.2}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mf}{1.0}\PYG{p}{,} \PYG{l+m+mf}{2.0}\PYG{p}{]}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Parameter estimation considering time\PYGZhy{}dependent enzyme inactivation}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{l+m+mf}{0.02}\PYG{p}{,} \PYG{l+m+mf}{0.2}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mf}{1.0}\PYG{p}{,} \PYG{l+m+mf}{2.0}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df\PYGZus{}inactivation} \PYG{o}{=} \PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat / Km [1/min * 1/mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{insert}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{True}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize fit of the models}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}first\PYGZus{}run}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Parameter estimation based on full dataset}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Parameter estimation based on data subset}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
        \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} 
        \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{enzyme_inactivation_7_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig 15: Identification of systematic deviations through kinetic modeling.}

\sphinxAtStartPar
By excluding the mentioned measurements with systematic issues, the model fit was improved (Fig 15B). Additionally, the correlation between parameter estimates for \(k_{cat}\) and \(K_{m}\) was reduced from \sphinxstyleemphasis{r2} = 0.518 to 0.204.


\subsubsection{Parameter estimation with and without considering enzyme inactivation}
\label{\detokenize{scenarios/enzyme_inactivation:parameter-estimation-with-and-without-considering-enzyme-inactivation}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Parameter estimation without time\PYGZhy{}dependent enzyme inactivation}
\PYG{n}{CotA\PYGZus{}kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{l+m+mf}{0.02}\PYG{p}{,} \PYG{l+m+mf}{0.2}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mf}{1.0}\PYG{p}{,} \PYG{l+m+mf}{2.0}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat / Km [1/min * 1/mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df}\PYG{o}{.}\PYG{n}{insert}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{False}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{results} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{df\PYGZus{}inactivation}\PYG{p}{)}\PYG{o}{.}\PYG{n}{sort\PYGZus{}values}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{AIC}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{results}\PYG{o}{.}\PYG{n}{style}\PYG{o}{.}\PYG{n}{set\PYGZus{}table\PYGZus{}attributes}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{style=}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{font\PYGZhy{}size: 12px}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fa328b79100\PYGZgt{}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
Kinetic parameters were estimated with and without considering time\sphinxhyphen{}dependent enzyme inactivation.
Modeling results are listed in the table above.
All models with an additional parameter for enzyme inactivation resulted in lower AIC values, indicating a better fit of the experimental data to inactivation models. When comparing models, an additional parameter in a model is informative, if the difference in AIC between the two models is larger than 2 . Accordingly, two additional parameters in a model compared to another model are informative, if the difference in AIC is larger than 4 (\hyperlink{cite.references:id16}{Arnold, 2010}). Thus, the additional parameters in inhibition models considering enzyme activation were uninformative and did not enhance the fit quality. Additionally, the mentioned models showed high standard deviations above 80\% for the respective \(K_{i}\) estimates.
Without considering enzyme inactivation, product inhibition models were favoured over the irreversible Michaelis\sphinxhyphen{}Menten model,
whereas irreversible Michaelis\sphinxhyphen{}Menten with enzyme inactivation showed the best fit statistic compared to all other models (Fig 16A). Thereby, \(k_{cat}\) was estimated to 3.003 min\sphinxhyphen{}1 ± 0.77\% whereas 30 µM ± 1.99\%  was estimated for \(K_{m}\). The respective half life of CotA was estimated at 95 min ± 4 min using equation {\hyperref[\detokenize{scenarios/SLAC_kinetic_characterization:equation-enzyme-halflife}]{\sphinxcrossref{(16)}}}, whereas \(k_{cat}\) and \(k_{inact}\) were highly correlated (\sphinxstyleemphasis{r2} > 0.95). Irreversible Michelis\sphinxhyphen{}Menten model without enzyme concentration did not fit the measurement data, since the reaction rate is initially underestimated, and overestimated in later stages of the reaction (Fig 16B).

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{9.2}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irrev. Michaelis\PYGZhy{}Menten model with time\PYGZhy{}dependent enzyme inactivation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irrev. Michaelis\PYGZhy{}Menten model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{non\PYGZhy{}competitive product inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{non\PYGZhy{}competitive product inhibition model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{uncompetitive product inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{uncompetitive product inhibition model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{p}{[}\PYG{n}{ax}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)} \PYG{k}{for} \PYG{n}{ax} \PYG{o+ow}{in} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{]}

\PYG{p}{[}\PYG{n}{ax}\PYG{o}{.}\PYG{n}{text}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mf}{1.1}\PYG{p}{,} \PYG{n}{string}\PYG{o}{.}\PYG{n}{ascii\PYGZus{}uppercase}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]}\PYG{p}{,} \PYG{n}{transform}\PYG{o}{=}\PYG{n}{ax}\PYG{o}{.}\PYG{n}{transAxes}\PYG{p}{,} \PYG{n}{size}\PYG{o}{=}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{n}{weight}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bold}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)} \PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{n}{ax} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{]}

\PYG{c+c1}{\PYGZsh{} axes[0]}
\PYG{c+c1}{\PYGZsh{} axes[1].text(1.1, 1.1, string.ascii\PYGZus{}uppercase[1], transform=axes[0].transAxes, }
\PYG{c+c1}{\PYGZsh{}         size=20, weight=\PYGZsq{}bold\PYGZsq{})}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{enzyme_inactivation_11_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 16: Different kinetic models fitted to CotA reaction data}

\sphinxAtStartPar
Out of the models without enzyme inactivation, product inhibition models showed the lowest AIC. Qualitatively, the inhibition models  described the progress curve of the reaction (Fig. 16C and 16D) similarly to the best fitting model (Fig. 16A)


\subsection{Project discussion}
\label{\detokenize{scenarios/enzyme_inactivation:project-discussion}}

\subsubsection{Data quality}
\label{\detokenize{scenarios/enzyme_inactivation:data-quality}}
\sphinxAtStartPar
Both datasets, for the absorption experiment and the enzyme kinetics experiment, showed quality issues. In the adsorption experiment the initial measurement showed large uncertainties as well as mixing issues, indicated by an apparent fluctuation in reaction rate. In the enzyme kinetics experiment, individual reactions were shown to have substrate and enzyme concentrations that deviated from the assay protocol. Furthermore, the measured product concentration showed large deviations among experimental repeats, indicating unequal enzyme concentrations. Hence, larger pipetting volume for the enzyme solution was suggested to reduce pipetting errors.
Furthermore, the project partners were advised to repeat the experiments.


\subsubsection{Enzyme inactivation}
\label{\detokenize{scenarios/enzyme_inactivation:enzyme-inactivation}}
\sphinxAtStartPar
The observed time\sphinxhyphen{}dependent decrease in enzyme activity can be explained by enzyme inactivation and product inhibition. In the case of time\sphinxhyphen{}dependent enzyme inactivation, the catalytic activity decreases, since the catalyst becomes inactive. Thereby, the decrease in relative reaction rate is independent from product concentration.
In the case of product inhibition, the formed product reduces the reaction rate. Hence, the observed enzyme inactivation is dependent on product concentration.
Since both phenomena express themselves in an apparent decrease of reaction rate, they bare the potential to be confused, as the results of the enzyme kinetics experiment show. Visually, product inhibition models without enzyme inactivation and irreversible Michaelis\sphinxhyphen{}Menten model with enzyme inactivation are able to qualitatively describe the observed progress curve (Fig 16C, 16D).

\sphinxAtStartPar
Both independent and methodologically different experiments resulted in a comparable time period, in which the enzyme’s activity is halved. Whereas in the inactivation experiment the enzyme’s activity was halved after approximately 105 min, half the activity was left in the enzyme kinetics experiment after 95 min. Due to the design of the incubation experiment, the loss of enzyme activity showed to be independent from the catalysis itself. In consequence, product inhibition is not a conclusive explanation for the observed behavior. Product inhibition is further disfavored based on the modeling results, since the AIC of all inhibition models was higher compared the irreversible Michaelis\sphinxhyphen{}Menten model with time\sphinxhyphen{}dependent enzyme inactivation.

\sphinxAtStartPar
In conclusion, the observed inactivation in potentially caused by hydrophobic interaction between the reaction vessel and the enzyme. Interactions between hydrophobic surfaces and enzymes leading to decreased enzymatic activity were already reported for ribonuclease A (\hyperlink{cite.references:id29}{Wei \sphinxstyleemphasis{et al.}, 2014}). In this work, enzyme inactivation was observed across multiple enzyme classes, where all reactions were carried out in MTPs. Therefore, MTPs might cause enzyme inactivation for a broad range of different enzymes.

\sphinxAtStartPar
Misinterpretation of the apparent decrease in enzymatic activity might lead to wrong conclusions and therefore negatively impacts the research process. As already mentioned the activity decrease might be wrongfully attributed to product inhibition. This attribution might lead to wrong design decisions if the enzyme in question should be applied to an enzymatic cascade reaction or make the enzyme entirely unfavorable for this particular application. Furthermore, screening applications might be impacted by enzyme inactivation, if yields of an enzymatic reactions should be determined over the time course of multiple hours. In consequence, the determined time\sphinxhyphen{}dependent yield might be estimated too low.
Therefore, enzyme inactivation in MTPs should be investigated in future experiments and the results should be reported on.


\subsection{Save modeling results}
\label{\detokenize{scenarios/enzyme_inactivation:save-modeling-results}}
\sphinxAtStartPar
Lastly, the modeling results were written to the EnzymeML documents and the files were exported.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Write modeling results to kinetic parameters}
\PYG{n}{k\PYGZus{}cat} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
    \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{value}\PYG{o}{=}\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
    \PYG{n}{unit}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{1 / }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{time\PYGZus{}unit}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{K\PYGZus{}m} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{enzymeml}\PYG{o}{.}\PYG{n}{models}\PYG{o}{.}\PYG{n}{KineticParameter}\PYG{p}{(}
    \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}m}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{value}\PYG{o}{=}\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{get\PYGZus{}model\PYGZus{}results}\PYG{p}{(}\PYG{p}{)}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{,}
    \PYG{n}{unit}\PYG{o}{=}\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{data\PYGZus{}conc\PYGZus{}unit}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Define kinetic model}
\PYG{n}{model} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{KineticModel}\PYG{p}{(}
    \PYG{n}{name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis\PYGZhy{}Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{equation}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZhy{}k\PYGZus{}cat * p0 * s0 / (K\PYGZus{}m + s0)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{parameters}\PYG{o}{=}\PYG{p}{[}\PYG{n}{k\PYGZus{}cat}\PYG{p}{,} \PYG{n}{K\PYGZus{}m}\PYG{p}{]}\PYG{p}{)}

\PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{model} \PYG{o}{=} \PYG{n}{model}

\PYG{n}{add\PYGZus{}enzyme\PYGZus{}inactivation\PYGZus{}model}\PYG{p}{(}
    \PYG{n}{result}\PYG{o}{=}\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{p}{,}
    \PYG{n}{enzmldoc}\PYG{o}{=}\PYG{n}{enzmldoc}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Export EnzymeML documents}
\PYG{n}{export} \PYG{o}{=} \PYG{k+kc}{False}
\PYG{k}{if} \PYG{n}{export}\PYG{p}{:}
    \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{toFile}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope


\chapter{Discussion}
\label{\detokenize{discussion:discussion}}\label{\detokenize{discussion::doc}}
\sphinxAtStartPar
Ensuring data integrity throughout an analysis workflow is important to yield trustworthy results. Hence, the analysis workflow was conceptualized starting from photometric measurement data in order to control all data processing steps. For that purpose, discussing the output format of the analytical device with the experimental partners proved to be valuable to plan the data acquisition process. Thus, a project specific parser function was established, which transferred the measurement data to the respective EnzymeML containing information on the measurement conditions. Thereby, error\sphinxhyphen{}prone and tedious manual copying between files was avoided, ensuring raw data integrity from the very start of the workflow.


\section{Data integrity of the analysis workflow}
\label{\detokenize{discussion:data-integrity-of-the-analysis-workflow}}
\sphinxAtStartPar
Another aspect of data integrity is accurate concentration calculation based on the measured absorption signal. The relation between concentration to absorption signal of an analyte can be linear as well as non\sphinxhyphen{}linear. Therefore, wrong assumption of a linear relationship, albeit the relation is better described non\sphinxhyphen{}linearly, leads to avoidable inaccuracy in concentration calculation (\hyperlink{cite.references:id14}{Hsu and Chen, 2010}) (\hyperlink{cite.references:id15}{Martin \sphinxstyleemphasis{et al.}, 2017}) and thus imprecise kinetic parameter estimates.
Hence, concentration calculation based on extinction coefficients was discouraged, if the underlying relation between absorption and concentration of an analyte was not self\sphinxhyphen{}determined for the respective reaction system.
To enable precise concentration calculation of linear and non\sphinxhyphen{}linear absorption to concentration relationships, CaliPytion was applied with analyte standards provided by the experimental partners.
All of the analyzed standard curves, except for scenario C ABTS at pH 5, were best described by non\sphinxhyphen{}linear calibration equations based on their fit quality determined by AIC. Hence, uncertainties from concentration calculation were reduced and thus their impact on the kinetic parameters.

\sphinxAtStartPar
Besides data integrity, high quality data is required for accurate parameter estimates, since unnoticed experimental errors unintentionally distort the resulting kinetic parameters. Therefore, quality control of measurement data is an important aspect of every data analysis process. EnzymePynetics allows visualization of measurement data together with the fitted kinetic model. Ultimately, systematic deviations of single measurements were identified by simultaneously visualizing measurement data and the fitted model.
By this method, errors with pipetting, mixing, and temperature equilibration during experimental preparation as well as issues with the experimental design itself were identified in different scenarios. Systematic pipetting errors were disclosed by systematic deviations between the measurement data and the fitted model.
In scenario D individual enzyme reactions showed to have substrate concentrations which deviated from the assay protocol. Parameter estimation without the presumably erroneous measurements yielded a better fit between data and model based on lower standard deviation for the kinetic parameters. Hence the experimental partners were advised to repeat the deviating measurement. In two projects the progress\sphinxhyphen{}curve of the reactions indicate a lag phase after reaction start. In one case, enzyme reactions were started by adding enzyme with low volume. In consequence, inhomogeneous mixing led to increasing reaction rates until the enzyme was distributed evenly. As a measure, the project partners were advised to increase the pipetting volume of the enzyme solution.
In another case, temperature incubation effects in MTP caused by prolonged assay preparation time resulted in an initial lag phase. Due to small reaction volumes and low mass, MTPs have a low heat capacity and thus high susceptibility to temperature change. Hence, project partners were advised to pre\sphinxhyphen{}incubate the MTP within the photometer at reaction temperature for 5 min and then start the reaction by adding enzyme as quick as possible.
In another project, modeling results of the estimated \(k_{cat}\) and \(K_{m}\) were used to iteratively improve the design of the enzyme assay. In addition, the correlation between the aforementioned parameters was used as a measure to ensure that the highest substrate concentration applied was sufficiently high and thus the kinetic parameters could be determined independently.
Thereby, the appropriate enzyme concentration and substrate concentration range were identified through multiple round of lab experiments with subsequent kinetic modeling.

\sphinxAtStartPar
Assessing the data quality through modeling and enhancing assay designs proved to be a strength of the implemented progress curve method on which the parameter estimation of this workflow is based. In contrast to the predominantly applied initial rates method (\hyperlink{cite.references:id25}{Tang and Leyh, 2010}), the progress curve method offers intrinsic advantages with regard to methods reproducibility.
In initial rate kinetics no consensus on the linear reaction period on which the kinetic parameters are estimated exists. Hence, the linear period is manually determined or only assumed. Ultimately, the resulting parameters are influenced by subjective choice of the linear period. For progress curve analysis the entire dataset of a continuous enzyme assay is used. If the fit statistics reveal that the data is not in accordance with the model, either the assay should be repeated due to an underlying issue or the model should be questioned. Contrarily, initial rates method can mostly be applied by arbitrarily choosing any linear subset or only the initial two data points of a time course data set. In discussions with project partners this showed to be a common practice although, scientifically at least, questionable.

\sphinxAtStartPar
The choice of the modeling method is just one decision which impacts kinetic parameters in the long chain of actions from enzyme expression all the way to data modeling. Each individual treatment step needs to be documented and later reported in order to make the results of the experiment reproducible. The STRENDA guidelines offer a valuable orientation on the minimum reporting standards in order to make an enzyme kinetics experiment reproducible. However, the guidelines are mainly focused on laboratory aspects and do not specify reporting standards for data treatment steps. Hence methodological reproducibility is not given by solely adhering to STRENDA guidelines, if kinetic parameters should be reproduced. Therefore, the method for concentration calculation, as well as the method for parameter estimation should be added to the STRENDA guidelines. As a result, the importance of data analytic aspects for methodologically reproducible kinetic parameters would receive more attention.


\section{Repeatability, reproducibility and FAIRness of the analysis workflow}
\label{\detokenize{discussion:repeatability-reproducibility-and-fairness-of-the-analysis-workflow}}
\sphinxAtStartPar
Availability of raw data is the prerequisite to repeat a data analysis process, which additionally reinforces trust in scientific findings and the process itself. The lack of raw data is therefore also described as one of the reasons for the reproducibility crisis, since reported findings cannot be verified (\hyperlink{cite.references:id21}{Miyakawa, 2020}).
A second necessity for reanalysis is documentation to provide information on data treatment steps that led to a particular result. Furthermore, detailed documentation makes the analysis comprehensive for others. The developed parameter estimation workflow unifies these two requirements. On the one hand, the workflow is based on raw data. On the other hand, the Jupyter Notebook facilitates the analysis, whilst containing documentation on all analysis steps. Thus, the workflow is repeatable and reproducible on a methods and results level. Moreover, by unifying data with executable code the implementation of the analysis workflow satisfies the highest reproducibility standards in computer science (\hyperlink{cite.references:id22}{Peng, 2011}).
In conclusion, the Jupyter Notebook with the implemented workflow can be understood as a graph, describing how experimental data is transform yielding kinetic parameters. As such, the measurement data and modeling results are stored within an EnzymeML document and the information of the modeling process is stored in the notebook file.

\sphinxAtStartPar
Starting from data, FAIR guiding principles were implemented on multiple levels of this thesis. On data level, all experimental data as well as the modeling results were stored as EnzymeML files, which are compliant with FAIR data principles (\hyperlink{cite.references:id4}{Pleiss, 2021}) (\hyperlink{cite.references:id18}{Range \sphinxstyleemphasis{et al.}, 2022}). On a methods level, the workflow and all of its components were designed in FAIR fashion. The Python packages are findable and accessible on PyPI and GitHub, which represent the most important distribution platforms for Python code. Additionally, the software is interoperable with other software. This was achieved by segregating the data model from the functionality of the software. Therefore, other Python tools can utilize the functionalities of CaliPytion and EnzymePynetics by serving the underlying data models, and thus no modifications on the software functionalities are necessary. Furthermore, the data model and its vocabulary is described in the specifications of \sphinxhref{https://github.com/FAIRChemistry/CaliPytion/blob/main/specifications/CalibrationModel.md}{CaliPytion} as well as \sphinxhref{https://github.com/haeussma/EnzymePynetics/blob/main/specifications/EnzymeKinetics.md}{EnzymePynetics}, making the data model comprehensive for others. Furthermore, the software packages are reusable, since the software can be installed and applied by anyone due to the documentation.

\sphinxAtStartPar
Since data integrity as well as reproducibility of analysis were key parts of this work, the thesis itself was conceptualized enabling reproducible data analysis. Therefore, the Jupyter Notebooks were integrated into the Jupyter Book, which constitutes this thesis. Thus, all data analysis is reproducible by launching the individual notebooks with Binder. As a result, a level of data integrity and transparency is achieved, which is not reachable with established word processing software. Furthermore, the Jupyter Book format allows to display computational content in its native form. Due to the toggle functionality of code cells, the readability of text and code is equally maintained, which is not possible with analog formats. Hence, preserving readability as well as reproducibility was weighted higher than printed readability.
The Jupyter Book formate allows this thesis to be findable and accessible on GibHub, whereas interoperability between book and notebooks makes all code content of this work reusable and reanalyzable. In conclusion, the Jupyter Book format allows scientific reporting of computational content which is compliant with FAIR guiding principles.

\sphinxstepscope


\chapter{Conclusion and Outlook}
\label{\detokenize{conclusion:conclusion-and-outlook}}\label{\detokenize{conclusion::doc}}
\sphinxAtStartPar
In this work, a reproducible and FAIR workflow for kinetic parameter estimation of enzyme reactions was established. Thereby, the output of an analytical device is directly used as the input of the modeling pipeline. By considering possible nonlinear relationships in the conversion of analytic data to concentration data, a possible source of inaccuracies is avoided. The parameter estimation is based on progress curve analysis. Since various kinetic models were implemented, models can be compared and insight into the kinetic mechanism can be obtained. Therefore, the implemented parameter estimator is suited for inhibition studies. After fitting of the models, visualization of the fit together with the underlying measurement data allows identification of individual systematic deviation between the model and data. Hence, providing a measure for quality control.
The workflow is capable to process EnzymeML files by using the associated PyEnzyme application programming interface. The analysis workflow was implemented in a Jupyter Notebook, thus the workflow is conclusive, repeatable and reproducible.
Due to the format and the associated Jupyter, Binder, and GitHub ecosystems, the analysis can be extensively documented, reanalyzed, and shared among scientists.
The developed workflow supports the scientific community by making accurate and scalable progress curve analysis of enzyme reactions available which is additionally compliant with FAIR guiding principles.
Application of the workflow to ongoing research projects demonstrated its capabilities to identify issues with the preparation of enzymatic assays, which might otherwise have remained unrecognized. Thus, the developments additionally support the field by potentially preventing the publication of incorrect kinetic parameters based on erroneous data.

\sphinxAtStartPar
In future developments, the workflow could be extended to enable parameter estimation of enzyme reactions with more than one substrate, by implementing more complex models. Furthermore, part of the quality control, which is currently carried out by visual analysis, could be automated. Thereby, the user would be informed about systematic errors in the dataset, which might have resulted from experimental errors. Moreover, tools could be implemented which assist scientists in assay design with focus on the design of minimal experiments. Hence, after an initial analysis round the user could be informed, if the given information is sufficient for parameter estimation, or if the data set contains to little information. In result, initial substrate concentrations for the next round of lab experiments could be suggested.

\sphinxstepscope

\begin{sphinxthebibliography}{Widsten }
\bibitem[Begley \& Ioannidis, 2015]{references:id33}
\sphinxAtStartPar
C Glenn Begley and John PA Ioannidis. Reproducibility in science: improving the standard for basic and preclinical research. \sphinxstyleemphasis{Circulation research}, 116(1):116–126, 2015.
\bibitem[Baker, 2016]{references:id32}
\sphinxAtStartPar
Monya Baker. Reproducibility crisis. \sphinxstyleemphasis{Nature}, 533(26):353–66, 2016.
\bibitem[Goodman et al., 2016]{references:id31}
\sphinxAtStartPar
Steven N Goodman, Daniele Fanelli, and John PA Ioannidis. What does research reproducibility mean? \sphinxstyleemphasis{Science translational medicine}, 8(341):341ps12–341ps12, 2016.
\bibitem[Munafo et al., 2017]{references:id34}
\sphinxAtStartPar
Marcus R Munafò, Brian A Nosek, Dorothy VM Bishop, Katherine S Button, Christopher D Chambers, Nathalie Percie du Sert, Uri Simonsohn, Eric\sphinxhyphen{}Jan Wagenmakers, Jennifer J Ware, and John Ioannidis. A manifesto for reproducible science. \sphinxstyleemphasis{Nature human behaviour}, 1(1):1–9, 2017.
\bibitem[Miyakawa, 2020]{references:id21}
\sphinxAtStartPar
Tsuyoshi Miyakawa. No raw data, no science: another possible source of the reproducibility crisis. 2020.
\bibitem[Bell et al., 2021]{references:id42}
\sphinxAtStartPar
Elizabeth L. Bell, William Finnigan, Scott P. France, Anthony P. Green, Martin A. Hayes, Lorna J. Hepworth, Sarah L. Lovelock, Haruka Niikura, S\textbackslash{}'ılvia Osuna, Elvira Romero, Katherine S. Ryan, Nicholas J. Turner, and Sabine L. Flitsch. Biocatalysis. \sphinxstyleemphasis{Nature Reviews Methods Primers}, June 2021. URL: \sphinxurl{https://doi.org/10.1038/s43586-021-00044-z}, \sphinxhref{https://doi.org/10.1038/s43586-021-00044-z}{doi:10.1038/s43586\sphinxhyphen{}021\sphinxhyphen{}00044\sphinxhyphen{}z}.
\bibitem[Tipton et al., 2014]{references:id11}
\sphinxAtStartPar
Keith F Tipton, Richard N Armstrong, Barbara M Bakker, Amos Bairoch, Athel Cornish\sphinxhyphen{}Bowden, Peter J Halling, Jan\sphinxhyphen{}Hendrik Hofmeyr, Thomas S Leyh, Carsten Kettner, Frank M Raushel, and others. Standards for reporting enzyme data: the strenda consortium: what it aims to do and why it should be helpful. \sphinxstyleemphasis{Perspectives in Science}, 1(1\sphinxhyphen{}6):131–137, 2014.
\bibitem[Wilkinson et al., 2016]{references:id40}
\sphinxAtStartPar
Mark D Wilkinson, Michel Dumontier, IJsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, Jan\sphinxhyphen{}Willem Boiten, Luiz Bonino da Silva Santos, Philip E Bourne, and others. The fair guiding principles for scientific data management and stewardship. \sphinxstyleemphasis{Scientific data}, 3(1):1–9, 2016.
\bibitem[Pleiss, 2021]{references:id4}
\sphinxAtStartPar
Jürgen Pleiss. Standardized data, scalable documentation, sustainable storage–enzymeml as a basis for fair data management in biocatalysis. \sphinxstyleemphasis{ChemCatChem}, 13(18):3909–3913, 2021.
\bibitem[Range et al., 2021]{references:id12}
\sphinxAtStartPar
Jan Range, Frank Bergmann, and Neil Swainston. Pyenzyme 1.1.0\sphinxhyphen{}alpha. June 2021. URL: \sphinxurl{https://doi.org/10.5281/zenodo.5021263}, \sphinxhref{https://doi.org/10.5281/zenodo.5021263}{doi:10.5281/zenodo.5021263}.
\bibitem[Newville et al., 2016]{references:id35}
\sphinxAtStartPar
Matthew Newville, Till Stensitzki, Daniel B Allen, Michal Rawlik, Antonino Ingargiola, and Andrew Nelson. Lmfit: non\sphinxhyphen{}linear least\sphinxhyphen{}square minimization and curve\sphinxhyphen{}fitting for python. \sphinxstyleemphasis{Astrophysics Source Code Library}, pages ascl–1606, 2016.
\bibitem[Bisswanger, 2017]{references:id5}
\sphinxAtStartPar
Hans Bisswanger. \sphinxstyleemphasis{Enzyme kinetics: principles and methods}. John Wiley \& Sons, 2017.
\bibitem[Caswell et al., 2020]{references:id36}
\sphinxAtStartPar
Thomas A Caswell, Michael Droettboom, Antony Lee, John Hunter, Eric Firing, Elliott Sales De Andrade, Tim Hoffmann, David Stansby, Jody Klymak, Nelle Varoquaux, and others. Matplotlib/matplotlib: rel: v3. 3.1. \sphinxstyleemphasis{Zenodo}, 2020.
\bibitem[Arnold, 2010]{references:id16}
\sphinxAtStartPar
Todd W Arnold. Uninformative parameters and model selection using akaike's information criterion. \sphinxstyleemphasis{The Journal of Wildlife Management}, 74(6):1175–1178, 2010.
\bibitem[Akaike, 1998]{references:id17}
\sphinxAtStartPar
Hirotogu Akaike. Information theory and an extension of the maximum likelihood principle. In \sphinxstyleemphasis{Selected papers of hirotugu akaike}, pages 199–213. Springer, 1998.
\bibitem[Kluyver et al., 2016]{references:id37}
\sphinxAtStartPar
Thomas Kluyver, Benjamin Ragan\sphinxhyphen{}Kelley, Fernando Pérez, Brian E Granger, Matthias Bussonnier, Jonathan Frederic, Kyle Kelley, Jessica B Hamrick, Jason Grout, Sylvain Corlay, and others. \sphinxstyleemphasis{Jupyter Notebooks\sphinxhyphen{}a publishing format for reproducible computational workflows.} Volume 2016. IOS Press, 2016.
\bibitem[Ragan\sphinxhyphen{}Kelley et al., 2018]{references:id20}
\sphinxAtStartPar
Benjamin Ragan\sphinxhyphen{}Kelley, Carol Willing, F Akici, D Lippa, D Niederhut, and M Pacer. Binder 2.0\sphinxhyphen{}reproducible, interactive, sharable environments for science at scale. In \sphinxstyleemphasis{Proceedings of the 17th python in science conference}, 113–120. F. Akici, D. Lippa, D. Niederhut, and M. Pacer, eds., 2018.
\bibitem[Community, 2020]{references:id38}
\sphinxAtStartPar
Executable Books Community. Jupyter book. February 2020. URL: \sphinxurl{https://doi.org/10.5281/zenodo.4539666}, \sphinxhref{https://doi.org/10.5281/zenodo.4539666}{doi:10.5281/zenodo.4539666}.
\bibitem[Hsu \& Chen, 2010]{references:id14}
\sphinxAtStartPar
Kang\sphinxhyphen{}Hung Hsu and Chiachung Chen. The effect of calibration equations on the uncertainty of uv–vis spectrophotometric measurements. \sphinxstyleemphasis{Measurement}, 43(10):1525–1531, 2010.
\bibitem[Martin et al., 2017]{references:id15}
\sphinxAtStartPar
Julia Martin, Alberto Romero Gracia, Agustín G Asuero, and others. Fitting nonlinear calibration curves: no models perfect. \sphinxstyleemphasis{Journal of Analytical Sciences, Methods and Instrumentation}, 7(01):1, 2017.
\bibitem[Busby et al., 1999]{references:id9}
\sphinxAtStartPar
William F Busby, Joseph M Ackermann, and Charles L Crespi. Effect of methanol, ethanol, dimethyl sulfoxide, and acetonitrile on in vitro activities of cdna\sphinxhyphen{}expressed human cytochromes p\sphinxhyphen{}450. \sphinxstyleemphasis{Drug Metabolism and Disposition}, 27(2):246–249, 1999.
\bibitem[Li et al., 2008]{references:id7}
\sphinxAtStartPar
Bo Li, Fei Lu, Xinjun Wei, and Ruixiang Zhao. Fucoidan: structure and bioactivity. \sphinxstyleemphasis{Molecules}, 13(8):1671–1695, 2008.
\bibitem[Daub et al., 2020]{references:id6}
\sphinxAtStartPar
Chantal Désirée Daub, Blessing Mabate, Samkelo Malgas, and Brett Ivan Pletschke. Fucoidan from ecklonia maxima is a powerful inhibitor of the diabetes\sphinxhyphen{}related enzyme, α\sphinxhyphen{}glucosidase. \sphinxstyleemphasis{International journal of biological macromolecules}, 151:412–420, 2020.
\bibitem[Shan et al., 2016]{references:id26}
\sphinxAtStartPar
Xindi Shan, Xin Liu, Jiejie Hao, Chao Cai, Fei Fan, Yunlou Dun, Xiaoliang Zhao, Xiaoxiao Liu, Chunxia Li, and Guangli Yu. In vitro and in vivo hypoglycemic effects of brown algal fucoidans. \sphinxstyleemphasis{International Journal of Biological Macromolecules}, 82:249–255, 2016.
\bibitem[Widsten \& Kandelbauer, 2008]{references:id13}
\sphinxAtStartPar
Petri Widsten and Andreas Kandelbauer. Laccase applications in the forest products industry: a review. \sphinxstyleemphasis{Enzyme and microbial technology}, 42(4):293–307, 2008.
\bibitem[Prins et al., 2015]{references:id30}
\sphinxAtStartPar
Alaric Prins, L Kleinsmidt, Nuraan Khan, Bronwyn Kirby, Tukayi Kudanga, Jannik Vollmer, Juergen Pleiss, Stephanie Burton, and M Le Roes\sphinxhyphen{}Hill. The effect of mutations near the t1 copper site on the biochemical characteristics of the small laccase from streptomyces coelicolor a3 (2). \sphinxstyleemphasis{Enzyme and microbial technology}, 68:23–32, 2015.
\bibitem[Buxbaum, 2015]{references:id8}
\sphinxAtStartPar
Engelbert Buxbaum. \sphinxstyleemphasis{Enzyme Kinetics: Special Cases}, pages 185–191. Springer International Publishing, Cham, 2015. URL: \sphinxurl{https://doi.org/10.1007/978-3-319-19920-7\_8}, \sphinxhref{https://doi.org/10.1007/978-3-319-19920-7\_8}{doi:10.1007/978\sphinxhyphen{}3\sphinxhyphen{}319\sphinxhyphen{}19920\sphinxhyphen{}7\_8}.
\bibitem[Wei et al., 2014]{references:id29}
\sphinxAtStartPar
Yang Wei, Aby A Thyparambil, Yonnie Wu, and Robert A Latour. Adsorption\sphinxhyphen{}induced changes in ribonuclease a structure and enzymatic activity on solid surfaces. \sphinxstyleemphasis{Langmuir}, 30(49):14849–14858, 2014.
\bibitem[Tang \& Leyh, 2010]{references:id25}
\sphinxAtStartPar
Qingxiu Tang and Thomas S Leyh. Precise, facile initial rate measurements. \sphinxstyleemphasis{The Journal of Physical Chemistry B}, 114(49):16131–16136, 2010.
\bibitem[Peng, 2011]{references:id22}
\sphinxAtStartPar
Roger D Peng. Reproducible research in computational science. \sphinxstyleemphasis{Science}, 334(6060):1226–1227, 2011.
\bibitem[Range et al., 2022]{references:id18}
\sphinxAtStartPar
Jan Range, Colin Halupczok, Jens Lohmann, Neil Swainston, Carsten Kettner, Frank T Bergmann, Andreas Weidemann, Ulrike Wittig, Santiago Schnell, and Jürgen Pleiss. Enzymeml—a data exchange format for biocatalysis and enzymology. \sphinxstyleemphasis{The FEBS Journal}, 289(19):5864–5874, 2022.
\end{sphinxthebibliography}







\renewcommand{\indexname}{Index}
\printindex
\end{document}