%% Generated by Sphinx.
\def\sphinxdocclass{jupyterBook}
\documentclass[letterpaper,10pt,english]{jupyterBook}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}
%% turn off hyperref patch of \index as sphinx.xdy xindy module takes care of
%% suitable \hyperpage mark-up, working around hyperref-xindy incompatibility
\PassOptionsToPackage{hyperindex=false}{hyperref}
%% memoir class requires extra handling
\makeatletter\@ifclassloaded{memoir}
{\ifdefined\memhyperindexfalse\memhyperindexfalse\fi}{}\makeatother

\PassOptionsToPackage{warn}{textcomp}

\catcode`^^^^00a0\active\protected\def^^^^00a0{\leavevmode\nobreak\ }
\usepackage{cmap}
\usepackage{fontspec}
\defaultfontfeatures[\rmfamily,\sffamily,\ttfamily]{}
\usepackage{amsmath,amssymb,amstext}
\usepackage{polyglossia}
\setmainlanguage{english}



\setmainfont{FreeSerif}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Italic,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldItalic
]
\setsansfont{FreeSans}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]
\setmonofont{FreeMono}[
  Extension      = .otf,
  UprightFont    = *,
  ItalicFont     = *Oblique,
  BoldFont       = *Bold,
  BoldItalicFont = *BoldOblique,
]



\usepackage[Bjarne]{fncychap}
\usepackage[,numfigreset=1,mathnumfig]{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}



        % Start of preamble defined in sphinx-jupyterbook-latex %
         \usepackage[Latin,Greek]{ucharclasses}
        \usepackage{unicode-math}
        % fixing title of the toc
        \addto\captionsenglish{\renewcommand{\contentsname}{Contents}}
        \hypersetup{
            pdfencoding=auto,
            psdextra
        }
        % End of preamble defined in sphinx-jupyterbook-latex %
        

\title{From raw data to kinetic parameters: an EnzymeML-based workflow for reproducible enzyme kinetics}
\date{Dec 08, 2022}
\release{}
\author{Max Häußler}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{welcome::doc}}


\sphinxAtStartPar
This is the root of the book.

\sphinxAtStartPar
Here is a list:
\begin{itemize}
\item {} 
\sphinxAtStartPar
A

\item {} 
\sphinxAtStartPar
B

\item {} 
\sphinxAtStartPar
C
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
1

\item {} 
\sphinxAtStartPar
2

\end{enumerate}

\item {} 
\sphinxAtStartPar
D

\end{itemize}

\sphinxstepscope


\chapter{Declaration of authorship}
\label{\detokenize{declaration:declaration-of-authorship}}\label{\detokenize{declaration::doc}}
\sphinxAtStartPar
I, Max Häußler, hereby declare, that…

\sphinxstepscope


\chapter{Abstract}
\label{\detokenize{abstract:abstract}}\label{\detokenize{abstract::doc}}
\sphinxAtStartPar
here


\section{English title}
\label{\detokenize{abstract:english-title}}

\section{German title}
\label{\detokenize{abstract:german-title}}
\sphinxstepscope


\chapter{Acknowledgements}
\label{\detokenize{acknowledgements:acknowledgements}}\label{\detokenize{acknowledgements::doc}}
\sphinxAtStartPar
Danke Merkel.

\sphinxstepscope


\chapter{Abbreviations}
\label{\detokenize{abbreviations:abbreviations}}\label{\detokenize{abbreviations::doc}}
\sphinxAtStartPar
:Alignment 1: If the field body starts on the first line…

\begin{sphinxVerbatim}[commandchars=\\\{\}]
          Then the entire field body must be indented the same.
\end{sphinxVerbatim}

\sphinxAtStartPar
:Alignment 2:
If the field body starts on a subsequent line…

\sphinxstepscope


\chapter{Introduction}
\label{\detokenize{introduction:introduction}}\label{\detokenize{introduction::doc}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Reproducibility crisis

\item {} 
\sphinxAtStartPar
Kinetic parameter estimation of enzyme reactions

\item {} 
\sphinxAtStartPar
Biology and big data

\end{itemize}

\sphinxstepscope


\chapter{Methods}
\label{\detokenize{methods:methods}}\label{\detokenize{methods::doc}}

\section{1. EnzymeML}
\label{\detokenize{methods:enzymeml}}
\sphinxAtStartPar
In its core, EnzymeML is a data model, structuring data and meta data of biocatalytic reactions. Thereby, properties of the reaction vessel, molecules involved in the reaction, as well as reaction properties like stoichiometry, pH, and temperature can be documented. Furthermore, properties of the protein catalyst like sequence, EC\sphinxhyphen{}number, UniProt ID can be stored alongside of potential reaction modifiers like inhibitory molecules. {[}\hyperlink{cite.references:id5}{Pleiss, 2021}{]}

\sphinxAtStartPar
Within the measurements section of the data model, measurement data of one or multiple measurements is stored. Each measurement contains data of identical measurement conditions as well as information about these conditions. Thus, data and meta data from experiments with differing conditions can be stored in a structured way.

\sphinxAtStartPar
The data model consists of descriptions for the reaction vessel,


\section{CaliPytion}
\label{\detokenize{methods:calipytion}}\label{\detokenize{methods:method-calipytion}}

\subsection{Data model for calibration data}
\label{\detokenize{methods:data-model-for-calibration-data}}

\subsection{Linear and non\sphinxhyphen{}linear fitting of calibration equations}
\label{\detokenize{methods:linear-and-non-linear-fitting-of-calibration-equations}}

\section{EnzymePynetics}
\label{\detokenize{methods:enzymepynetics}}\label{\detokenize{methods:method-enzymepynetics}}

\subsection{Overview}
\label{\detokenize{methods:overview}}
\sphinxAtStartPar
EnzymePynetics is a a python package, for kinetic parameter estimation of single\sphinxhyphen{}substrate enzyme reactions, which was developed during this thesis. The \sphinxcode{\sphinxupquote{ParameterEstimator}} of EnzymePynetics estimates the kinetic parameters \(k_{cat}\) and \(K_{m}\) by fitting time\sphinxhyphen{}course measurement data of enzyme reactions to different Michaelis\sphinxhyphen{}Menten models. Thereby, the residuals between measurement data and integrated Michaelis\sphinxhyphen{}Menten rate equations are minimized through a non\sphinxhyphen{}linear least\sphinxhyphen{}squares algorithm. Additionally, the inhibition constant \(K_{i}\) can be assessed for potential substrate or product inhibition. Furthermore, the inhibition constant of an enzyme inhibitor can be determined.


\subsection{Data model}
\label{\detokenize{methods:data-model}}
\sphinxAtStartPar
Data models build the backbone of applications, by defining the relations between informations in an hierarchical manner.
EnzymePynetics is based on a data model, resembling the experimental design of an enzyme kinetics assay. Thereby, all relevant data and meta data of an kinetic experiment are ordered in the base object \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}}. On the metadata side, the base object consists of the attributes temperature with its respective unit, pH, and the name of the measured substance. Additionally, it can be specified whether the measurement data originates from substrate or product measurements. On the data side, \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}} contains one or multiple \sphinxcode{\sphinxupquote{Measurements}}. Each measurement stores the information of an experimental condition, to which the enzyme was subjected. Therefore, each \sphinxcode{\sphinxupquote{Measurement}} contains information on the initial substrate concentration, enzyme concentration, and inhibitor concentration, if present, along with the respective concentration units. Each \sphinxcode{\sphinxupquote{Measurement}} contains the measured data, which itself consist of one or multiple replicates of the respective experimental condition.The data model was was generated using \sphinxhref{https://github.com/JR-1991/software-driven-rdm}{sdRDM}, a python tool allowing the creation and versioning of data models.

\sphinxAtStartPar
An extensive documentation of the data model can be accessed in the \sphinxhref{https://github.com/haeussma/EnzymePynetics/blob/main/specifications/EnzymeKinetics.md}{specifications} of the the software package.


\subsection{3.3 Kinetic models}
\label{\detokenize{methods:kinetic-models}}
\sphinxAtStartPar
Besides the irreversible Michaelis\sphinxhyphen{}Menten rate equation (Eq. 1) inhibition models for competitive (Eq. 2), uncompetitive (Eq. 3), and non\sphinxhyphen{}competitive) inhibition (Eq. 4) were implemented. Thereby, \(S\), \(E\), and \(I\) denote the concentration of substrate, enzyme, and inhibitor, respectively. In therms of kinetic parameters, \(k_{cat}\) denotes the turnover number, \(K_{m}\) the Michaelis\sphinxhyphen{}Menten constant of the substrate, whereas \(K_{ic}\) and \(K_{iu}\) describe the competitive and uncompetitive inhibition constant, respectively.
\begin{equation}\label{equation:methods:irreversible_mm}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} + S}\end{split}
\end{equation}\begin{equation}\label{equation:methods:competitive_inhibition}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{ic}}) + S}\end{split}
\end{equation}\begin{equation}\label{equation:methods:uncompetitive_inhibition}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{iu}}) * S}\end{split}
\end{equation}\begin{equation}\label{equation:methods:noncompetitive_inhibition}
\begin{split}\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{ic}}) + (1+\frac{I}{K_{iu}}) + S}\end{split}
\end{equation}
\sphinxAtStartPar
By default, \(E\) is assumed to be constant throughout the reaction. If required, the kinetic models can be extended by the parameter \(K_{inact}\), which describes the time\sphinxhyphen{}dependent inactivation rate of the enzyme. The decrease in active enzyme is modeled by Eq. 5:
\begin{equation}\label{equation:methods:enzyme_inactivation}
\begin{split}\frac{dE}{dt} = -K_{inact} * E\end{split}
\end{equation}

\subsection{3.4 Initialization}
\label{\detokenize{methods:initialization}}
\sphinxAtStartPar
Whereas the \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}} object solely serves a
The \sphinxcode{\sphinxupquote{ParameterEstimator}} harbors the functionalities for parameter estimation. Data can be provided by passing data as an \sphinxcode{\sphinxupquote{EnzymeKineticsExperiment}} object. Alternatively, an EnzymeML documents can be provided as the data source via \sphinxhref{https://github.com/EnzymeML/PyEnzyme}{PyEnzyme} software.Initially, product concentration is calculated, if the input data is from substrate measurements. Substrate is calculated, if product data was provided. The calculation is based on the initial substrate concentration, which needs to be provided for all measurements. The parameter estimation is then based on substrate data.In the background, rough estimates for \(k_{cat}\), \(K_{m}\), and \(K_{i}\) are calculated based on the fastest reaction rate in the provided dataset. Initial parameter estimates are needed as a starting point for the solver, whereas the parameter space is limited to ±1000\sphinxhyphen{}fold its estimated value for each parameter.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{parameterestimator} \PYG{k+kn}{import} \PYG{n}{ParameterEstimator}
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{core}\PYG{n+nn}{.}\PYG{n+nn}{enzymekineticsexperiment} \PYG{k+kn}{import} \PYG{n}{EnzymeKineticsExperiment}
\PYG{k+kn}{import} \PYG{n+nn}{pyenzyme}

\PYG{c+c1}{\PYGZsh{} Define data}
\PYG{n}{experimental\PYGZus{}data} \PYG{o}{=} \PYG{n}{EnzymeKineticsExperiment}\PYG{p}{(}
    \PYG{n}{pH}\PYG{o}{=}\PYG{l+m+mi}{7}\PYG{p}{,}
    \PYG{n}{reactant\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{test substance}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{o}{.}\PYG{o}{.}\PYG{o}{.}\PYG{p}{)}

\PYG{n}{enzymeml\PYGZus{}document} \PYG{o}{=} \PYG{n}{pyenzyme}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromFile}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{enzymeML\PYGZus{}dataset.omex}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Initialize the parameter estimator from an \PYGZsq{}EnzymeKineticsExperiment\PYGZsq{} instance}
\PYG{n}{estimator} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{p}{(}\PYG{n}{data}\PYG{o}{=}\PYG{n}{experimental\PYGZus{}data}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} ... or directly from an EnzymeML document.}
\PYG{n}{estimator} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{enzmldoc}\PYG{o}{=}\PYG{n}{enzymeml\PYGZus{}document}\PYG{p}{,}
    \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{inhibitor\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Fit experimental data to kinetic models and get the fit report of all models}
\PYG{n}{estimator}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize data with the best fitting model}
\PYG{n}{estimator}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}


\subsection{3.5 Visualization}
\label{\detokenize{methods:visualization}}

\section{4. Model selection}
\label{\detokenize{methods:model-selection}}
\sphinxAtStartPar
Akaike information criterion

\sphinxstepscope


\chapter{Results}
\label{\detokenize{results:results}}\label{\detokenize{results::doc}}

\section{Scenario\sphinxhyphen{}driven workflow development}
\label{\detokenize{results:scenario-driven-workflow-development}}
\sphinxAtStartPar
The development of the workflow was driven by different research scenarios of EnzymeML project partners. The goal of all project was to reliably estimate kinetic parameters of enzyme reactions. The collaboration consisted of multiple rounds of lab experiments and kinetic modeling of the respective data. Each round consisted of (i) wet lab experiments, (ii) kinetic modeling, (iii) design of follow\sphinxhyphen{}up experiments, and (iv) discussion of results with project partners. Hence, a short feedback loop between lab experiments and modeling\sphinxhyphen{}based experimental suggestions was established.In parallel, the python modules {\hyperref[\detokenize{methods:method-calipytion}]{\sphinxcrossref{\DUrole{std,std-ref}{CaliPytion}}}} and {\hyperref[\detokenize{methods:method-enzymepynetics}]{\sphinxcrossref{\DUrole{std,std-ref}{EnzymePynetics}}}} were developed to provide a workflow from analytical raw data to kinetic parameter estimates. Thereby, the individual requirements of each research scenario, fostered the implementation of different features. Hence, a generic workflow for parameter estimation of enzyme kinetics was established. The developed workflow is schematically visualized in figure 1.

\sphinxAtStartPar
\sphinxincludegraphics{{concept_workflow}.png}
\sphinxstyleemphasis{Fig. 1: Workflow for kinetic parameter estimation of enzyme reactions.}

\sphinxAtStartPar
The workflow was designed around EnzymeML, whereas it’s data model serves
\begin{itemize}
\item {} 
\sphinxAtStartPar
Data model vs. format

\item {} 
\sphinxAtStartPar
FAIR

\end{itemize}

\sphinxAtStartPar
The following chapters show the developed workflow applied in different research scenarios. Each each of the following sections is an executable JupyterNotebook. Hence all figures for data visualization were generated at runtime of the analysis.
Each notebook consists of a short description of the project’s background and methodology carried out by the project partners in the wet lab. Additionally kinetic modeling steps as well as the results are shown. Lastly, project specific results are discussed.

\sphinxstepscope


\section{\sphinxstylestrong{Scenario A:}Chymotrypsin inhibiton by a designed albumin fusion protein}
\label{\detokenize{scenarios/chymotrypsin_inhibition:scenario-a-br-chymotrypsin-inhibiton-by-a-designed-albumin-fusion-protein}}\label{\detokenize{scenarios/chymotrypsin_inhibition::doc}}
\sphinxAtStartPar
Data provided by Marwa Mohamed (Institute of Cell Biology and Immunology, University of Stuttgart, Stuttgart, Germany)


\subsection{Project background}
\label{\detokenize{scenarios/chymotrypsin_inhibition:project-background}}
\sphinxAtStartPar
In this scenaro, the inhibitory effect of a human serum albumin mutant on chymotrypsin was investigated. Thereby, the inhibition constant \(K_{i}\) of the HSA wild\sphinxhyphen{}type was compared to HSA(M3) mutant. Both HSAs were individually dimerized into fusion proteins through a huFc.Experimental data from the HSA(wt)\sphinxhyphen{}huFc and HSA(M3)\sphinxhyphen{}huFc originate from two independent experiments. In each experiment the initial substrate concentration was varied. Once with the respective inhibitor, and once without. Therefore, \(K_{i}\) and \(K_{m}\) were determined indenpendently from each other. Additionally, each individual reaction condition was prepared in duplicates to ensure repeatability.Since


\subsubsection{Experimental design}
\label{\detokenize{scenarios/chymotrypsin_inhibition:experimental-design}}
\sphinxAtStartPar
In order to assess the effect of the introduced mutations to the HSA(M3) variant, \(K_{i}\) of the HSA wild\sphinxhyphen{}type was compared to HSA(M3) variant. Enzyme activity was monitored by measuring the product formation of p\sphinxhyphen{}Nitroanilin (p\sphinxhyphen{}NA) photometrically at 410 nm for 30 min at 30°C. Therefore,  Succinyl\sphinxhyphen{}gly\sphinxhyphen{}gly\sphinxhyphen{}phe\sphinxhyphen{}p\sphinxhyphen{}nitroanilide (SGGPpNA) was applied as substrate in a concentration range of 0.25 \sphinxhyphen{} 2 mM. For concentration calculations, p\sphinxhyphen{}NA standard was prepared in the range of 0 \sphinxhyphen{} 0.3 mM in duplicates. \(K_{i}\) of HSA(wt) and HSA(M3) on chymotrypsin were investigated in independent experiments. Each experiment consisted of enzyme reactions with and without the respective HSA variant. The enzyme reactions contined dfhdfhdfh µM of enzyme and 26.88 µM of the respective HSA variant, if applied.


\subsubsection{Data management}
\label{\detokenize{scenarios/chymotrypsin_inhibition:data-management}}
\sphinxAtStartPar
Experimental data and meta data was filled in EnzymeML Excel templates for each of the two inhibition experiments respectively. Calibration data was stored as Excel files. Measurement data was already blanked.


\subsection{Data preparation}
\label{\detokenize{scenarios/chymotrypsin_inhibition:data-preparation}}

\subsubsection{Imports}
\label{\detokenize{scenarios/chymotrypsin_inhibition:imports}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{import} \PYG{n+nn}{pyenzyme} \PYG{k}{as} \PYG{n+nn}{pe}
\PYG{k+kn}{import} \PYG{n+nn}{copy}
\PYG{k+kn}{from} \PYG{n+nn}{IPython}\PYG{n+nn}{.}\PYG{n+nn}{display} \PYG{k+kn}{import} \PYG{n}{display}
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{parameterestimator} \PYG{k+kn}{import} \PYG{n}{ParameterEstimator}
\PYG{k+kn}{from} \PYG{n+nn}{CaliPytion}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{standardcurve} \PYG{k+kn}{import} \PYG{n}{StandardCurve}

\PYG{k+kn}{import} \PYG{n+nn}{warnings}
\PYG{n}{warnings}\PYG{o}{.}\PYG{n}{filterwarnings}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ignore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Concentration calculation}
\label{\detokenize{scenarios/chymotrypsin_inhibition:concentration-calculation}}
\sphinxAtStartPar
Product standard data was imported directly from an Excel file. Then, a standard curve was created.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{product\PYGZus{}standard} \PYG{o}{=} \PYG{n}{StandardCurve}\PYG{o}{.}\PYG{n}{from\PYGZus{}excel}\PYG{p}{(}
    \PYG{n}{path}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/chymotrypsin\PYGZus{}inhibition/pNA\PYGZhy{}standard.xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{wavelength}\PYG{o}{=}\PYG{l+m+mi}{410}\PYG{p}{,}
    \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{csv}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{concentration\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{temperature}\PYG{o}{=}\PYG{l+m+mi}{30}\PYG{p}{,}
    \PYG{n}{temperature\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    
\PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Calibration data was automatically blanked.
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7f8498027370\PYGZgt{}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_3_2}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
Based on the Akaike information criterion (AIC), the relation between concentration and absorption is best described by a quadratic function. The figure above visualizes the calibration measurements and the fitted calibration model.


\subsubsection{Experimantal data}
\label{\detokenize{scenarios/chymotrypsin_inhibition:experimantal-data}}
\sphinxAtStartPar
The EnzymeML documents were loaded and the standard curve was applied to the absorption data for concentration calculation.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load data from }
\PYG{n}{chymo\PYGZus{}HSAwt} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/chymotrypsin\PYGZus{}inhibition/chymo\PYGZus{}HSAwt.xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{chymo\PYGZus{}HSAM3} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/chymotrypsin\PYGZus{}inhibition/chymo\PYGZus{}HSA(M3).xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Apply standard curve to \PYGZsq{}EnzymeMLDocument\PYGZsq{}}
\PYG{n}{chymo\PYGZus{}HSAwt} \PYG{o}{=} \PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAwt}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{chymo\PYGZus{}HSAM3} \PYG{o}{=} \PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAM3}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Comparability of the experiments}
\label{\detokenize{scenarios/chymotrypsin_inhibition:comparability-of-the-experiments}}
\sphinxAtStartPar
Since the experimental data from the HSA wild\sphinxhyphen{}type and the HSA(M3) variant originate from independent experiments, the control reactions without the respective inhibitor were compared by performing a parameter estimation. Thereby, \(k_{cat}\) and \(K_{m}\) were highly correlated (corr > 0.98). Hence, catalytic efficiency \(\frac{k_{cat}}{K_{m}}\) was used to assess comparability between the data sets. High correlation indicates, that the highest initial substrate concentration is too low, compared to the true \(K_{m}\) of the enzyme under the given experimental conditions. In this case, higher substrate concentration were not applied for multiple reasons. On the one hand dimethyl sulfoxide (DMSO) was used as a co\sphinxhyphen{}solvent of the substrate, which inhibits enzyme activity Busby \sphinxstyleemphasis{et al.} {[}\hyperlink{cite.references:id9}{1999}{]}. Hence, higher initial substrate concentrations would have led to higher enzyme inhibition. On the other hand, high substrate viscosity denied the application of higher concentrations without sacrificing pipetting percision.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Create copys of the data sets and delete measuremnts with inhibitor.}
\PYG{n}{wt\PYGZus{}control} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAwt}\PYG{p}{)}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m4}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m5}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m6}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m7}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{n}{m3\PYGZus{}control} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAM3}\PYG{p}{)}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m4}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m5}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m6}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m7}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{c+c1}{\PYGZsh{} Estimate kinetic parameters of the control reactions of the HSA wild\PYGZhy{}type data set.}
\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{wt\PYGZus{}control}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Kinetic parameters of HSA(wt) chymotrypsin control reactions:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{p}{]}\PYG{p}{)}\PYG{p}{)}


\PYG{c+c1}{\PYGZsh{} Estimate kinetic parameters of the control reactions of the HSA(M3) data set.}
\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{m3\PYGZus{}control}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{Kinetic parameters of HSA(M3) chymotrypsin control reactions:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{p}{)}


\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{e}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{n}{kinetics\PYGZus{}wt\PYGZus{}control} \PYG{p}{,}\PYG{n}{kinetics\PYGZus{}m3\PYGZus{}control}\PYG{p}{]}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin control reactions HSA(wt)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin control reactions HSA(M3)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{doc}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{n}{title}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{4\PYGZhy{}nitroanilin [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time after reaction start [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xticks}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{,} \PYG{l+m+mi}{20}\PYG{p}{]}\PYG{p}{)}

\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{l+m+mi}{4}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial SGGPpNA  [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.15}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters of HSA(wt) chymotrypsin control reactions:
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                     AIC    Km [mmole / l]  \PYGZbs{}
irreversible Michaelis Menten      \PYGZhy{}1857   2.226 +/\PYGZhy{} 7.28\PYGZpc{}   
competitive product inhibition     \PYGZhy{}1856   2.342 +/\PYGZhy{} 9.88\PYGZpc{}   
uncompetitive product inhibition   \PYGZhy{}1855  2.276 +/\PYGZhy{} 10.65\PYGZpc{}   
substrate inhibition               \PYGZhy{}1855  2.233 +/\PYGZhy{} 35.51\PYGZpc{}   
non\PYGZhy{}competitive product inhibition \PYGZhy{}1854  2.331 +/\PYGZhy{} 10.00\PYGZpc{}   

                                   kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
irreversible Michaelis Menten                     25.353 +/\PYGZhy{} 8.57\PYGZpc{}   
competitive product inhibition                   25.714 +/\PYGZhy{} 13.37\PYGZpc{}   
uncompetitive product inhibition                 25.272 +/\PYGZhy{} 13.50\PYGZpc{}   
substrate inhibition                             25.339 +/\PYGZhy{} 45.93\PYGZpc{}   
non\PYGZhy{}competitive product inhibition               25.787 +/\PYGZhy{} 13.79\PYGZpc{}   

                                   Ki competitive [mmole / l]  \PYGZbs{}
irreversible Michaelis Menten                               \PYGZhy{}   
competitive product inhibition              0.725 +/\PYGZhy{} 112.65\PYGZpc{}   
uncompetitive product inhibition                            \PYGZhy{}   
substrate inhibition                                        \PYGZhy{}   
non\PYGZhy{}competitive product inhibition          0.702 +/\PYGZhy{} 117.81\PYGZpc{}   

                                   Ki uncompetitive [mmole / l]  
irreversible Michaelis Menten                                 \PYGZhy{}  
competitive product inhibition                                \PYGZhy{}  
uncompetitive product inhibition              3.175 +/\PYGZhy{} 428.92\PYGZpc{}  
substrate inhibition                      997.715 +/\PYGZhy{} 12275.11\PYGZpc{}  
non\PYGZhy{}competitive product inhibition          179.118 +/\PYGZhy{} 358.38\PYGZpc{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters of HSA(M3) chymotrypsin control reactions:
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                    AIC      kcat [1/min]   Km [mmole / l]  \PYGZbs{}
irreversible Michaelis Menten      \PYGZhy{}549  48.996 +/\PYGZhy{} 2.23\PYGZpc{}  1.978 +/\PYGZhy{} 3.74\PYGZpc{}   
competitive product inhibition     \PYGZhy{}549  51.227 +/\PYGZhy{} 4.39\PYGZpc{}  2.043 +/\PYGZhy{} 4.76\PYGZpc{}   
uncompetitive product inhibition   \PYGZhy{}547  49.384 +/\PYGZhy{} 5.33\PYGZpc{}  1.996 +/\PYGZhy{} 6.71\PYGZpc{}   
non\PYGZhy{}competitive product inhibition \PYGZhy{}547  51.241 +/\PYGZhy{} 5.62\PYGZpc{}  2.044 +/\PYGZhy{} 6.62\PYGZpc{}   
substrate inhibition               \PYGZhy{}545  50.368 +/\PYGZhy{} 3.78\PYGZpc{}  2.046 +/\PYGZhy{} 5.12\PYGZpc{}   

                                   kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
irreversible Michaelis Menten                     24.776 +/\PYGZhy{} 4.35\PYGZpc{}   
competitive product inhibition                    25.077 +/\PYGZhy{} 6.47\PYGZpc{}   
uncompetitive product inhibition                  24.743 +/\PYGZhy{} 8.57\PYGZpc{}   
non\PYGZhy{}competitive product inhibition                25.073 +/\PYGZhy{} 8.69\PYGZpc{}   
substrate inhibition                              24.623 +/\PYGZhy{} 6.36\PYGZpc{}   

                                   Ki competitive [mmole / l]  \PYGZbs{}
irreversible Michaelis Menten                               \PYGZhy{}   
competitive product inhibition               0.908 +/\PYGZhy{} 81.63\PYGZpc{}   
uncompetitive product inhibition                            \PYGZhy{}   
non\PYGZhy{}competitive product inhibition           0.914 +/\PYGZhy{} 83.56\PYGZpc{}   
substrate inhibition                                        \PYGZhy{}   

                                   Ki uncompetitive [mmole / l]  
irreversible Michaelis Menten                                 \PYGZhy{}  
competitive product inhibition                                \PYGZhy{}  
uncompetitive product inhibition              7.445 +/\PYGZhy{} 631.41\PYGZpc{}  
non\PYGZhy{}competitive product inhibition         121.556 +/\PYGZhy{} 2156.85\PYGZpc{}  
substrate inhibition                         87.549 +/\PYGZhy{} 121.66\PYGZpc{}  
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_8_4}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 1: Measurement data and fitted irreversible Michaelis\sphinxhyphen{}Menten model for chymotrypsin reactions without HSA inhibitor.}

\sphinxAtStartPar
The above figure visualizes the control reactions of each HSA inhibition experiment. Each dataset was fitted against the models listed in the above table.
Each dataset is best described by the irreversible Michaelis\sphinxhyphen{}Menten model in therms of AIC and standard deviation on the estimated parameters. Models with product or substrate inhibition resulted in large uncertanties above 80 \% on the parameter estimates. Thus, substrate and product inhibition were ruled out for the given reactions.  \(\frac{k_{cat}}{K_{m}}\) was estimated to be 25.353 min\sphinxhyphen{}1mM\sphinxhyphen{}1 ± 8.57\%  for the control reaction of the HSA(wt) data set and 24.776 min\sphinxhyphen{}1mM\sphinxhyphen{}1 ± 4.35\% for the HSA(M3) data set. As a result, the two experiments showed to be comparable, since the catalytic efficiency differs less than 3 \% between the two data sets.


\subsection{Determination and comparison of \protect\(K_{i}\protect\)}
\label{\detokenize{scenarios/chymotrypsin_inhibition:determination-and-comparison-of-k-i}}
\sphinxAtStartPar
The data set of chymotrypsin inhibition by HSA(M3) contained negative absorption values for the first measurement point. Presumably from an incorrect blank measurement. Therefore, only measurement data from the second data point (minute 5 and onward) was considered for parameter estimation. Additionally, measuremet
The tables below show the parameter estimastes for all applied kinetic models.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Parameter estimation for HSA(wt) data set}
\PYG{n}{kinetics\PYGZus{}HSAwt} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAwt}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{inhibitor\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}HSAwt}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{start\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(wt):}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}HSAwt}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Parameter estimation for HSA(M3) data set}
\PYG{n}{kinetics\PYGZus{}HSAM3} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{chymo\PYGZus{}HSAM3}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{inhibitor\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{,} \PYG{n}{stop\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{start\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(M3):}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize experimental data and fitted models}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{e}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{n}{kinetics\PYGZus{}HSAwt} \PYG{p}{,}\PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{p}{]}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin inhibition by HSA(WT)\PYGZhy{}huFc}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin inhibition by HSA(Chymo\PYGZhy{}M3)\PYGZhy{}huFc}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{doc}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{n}{title}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{4\PYGZhy{}nitroanilin [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time after reaction start [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xticks}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{,} \PYG{l+m+mi}{20}\PYG{p}{]}\PYG{p}{)}

\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial SGGPpNA  [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.2}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(wt):
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                   AIC kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
competitive inhibition           \PYGZhy{}3100                22.253 +/\PYGZhy{} 6.97\PYGZpc{}   
non\PYGZhy{}competitive inhibition       \PYGZhy{}3098                22.403 +/\PYGZhy{} 7.00\PYGZpc{}   
irreversible Michaelis Menten    \PYGZhy{}3088                21.622 +/\PYGZhy{} 7.15\PYGZpc{}   
uncompetitive inhibition         \PYGZhy{}3087                21.631 +/\PYGZhy{} 7.63\PYGZpc{}   
partially competitive inhibition \PYGZhy{}3084                21.643 +/\PYGZhy{} 8.23\PYGZpc{}   

                                 Ki competitive [mmole / l]  \PYGZbs{}
competitive inhibition                     0.460 +/\PYGZhy{} 27.24\PYGZpc{}   
non\PYGZhy{}competitive inhibition                 0.449 +/\PYGZhy{} 27.64\PYGZpc{}   
irreversible Michaelis Menten                             \PYGZhy{}   
uncompetitive inhibition                                  \PYGZhy{}   
partially competitive inhibition       796.196 +/\PYGZhy{} 1075.02\PYGZpc{}   

                                 Ki uncompetitive [mmole / l]  
competitive inhibition                                      \PYGZhy{}  
non\PYGZhy{}competitive inhibition                 79.193 +/\PYGZhy{} 516.52\PYGZpc{}  
irreversible Michaelis Menten                               \PYGZhy{}  
uncompetitive inhibition                     0.696 +/\PYGZhy{} 88.99\PYGZpc{}  
partially competitive inhibition           992.685 +/\PYGZhy{} 48.74\PYGZpc{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(M3):
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                  AIC kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
competitive inhibition           \PYGZhy{}808               23.031 +/\PYGZhy{} 10.48\PYGZpc{}   
non\PYGZhy{}competitive inhibition       \PYGZhy{}806               23.002 +/\PYGZhy{} 10.89\PYGZpc{}   
uncompetitive inhibition         \PYGZhy{}751               19.297 +/\PYGZhy{} 21.77\PYGZpc{}   
irreversible Michaelis Menten    \PYGZhy{}713               18.512 +/\PYGZhy{} 24.41\PYGZpc{}   
partially competitive inhibition \PYGZhy{}709               18.544 +/\PYGZhy{} 27.64\PYGZpc{}   

                                 Ki competitive [mmole / l]  \PYGZbs{}
competitive inhibition                      0.059 +/\PYGZhy{} 8.51\PYGZpc{}   
non\PYGZhy{}competitive inhibition                  0.060 +/\PYGZhy{} 9.04\PYGZpc{}   
uncompetitive inhibition                                  \PYGZhy{}   
irreversible Michaelis Menten                             \PYGZhy{}   
partially competitive inhibition         569.354 +/\PYGZhy{} 57.96\PYGZpc{}   

                                 Ki uncompetitive [mmole / l]  
competitive inhibition                                      \PYGZhy{}  
non\PYGZhy{}competitive inhibition                 44.965 +/\PYGZhy{} 302.77\PYGZpc{}  
uncompetitive inhibition                     0.035 +/\PYGZhy{} 21.33\PYGZpc{}  
irreversible Michaelis Menten                               \PYGZhy{}  
partially competitive inhibition        925.096 +/\PYGZhy{} 11733.22\PYGZpc{}  
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_10_4}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. 1: Measurement data and fitted product inhibition model for chymotrypsin reactions with respective HSA inhibitior.}

\sphinxAtStartPar
Both reaction systems are best described by the competitive inhibition model, which is indicated by the lowest AIC and standard deviation on the estimated parameters. Thereby, a \(K_{i}\) of 0.460 mM ± 27.24\% was estimated for HSA(wt) and 0.059 mM ± 8.51\% for HSA(M3). This resembles a roughly 7\sphinxhyphen{}fold increase in affinity of HSA(M3) to the enzyme compared to the HSA(wt). Since the competitive inhibition model describes the data the best, HSA(M3) presumably interacts with the enzyme in the active site region.

\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{kinetics\PYGZus{}HSAwt}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin inhibition by HSA(WT)\PYGZhy{}huFc}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for competitive inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 36
    \PYGZsh{} data points      = 256
    \PYGZsh{} variables        = 3
    chi\PYGZhy{}square         = 0.00137490
    reduced chi\PYGZhy{}square = 5.4344e\PYGZhy{}06
    Akaike info crit   = \PYGZhy{}3100.44585
    Bayesian info crit = \PYGZhy{}3089.81032
[[Variables]]
    k\PYGZus{}cat:  68.6263364 +/\PYGZhy{} 2.69078924 (3.92\PYGZpc{}) (init = 32.216)
    Km:     3.08387875 +/\PYGZhy{} 0.17764014 (5.76\PYGZpc{}) (init = 0.32216)
    K\PYGZus{}ic:   0.45950740 +/\PYGZhy{} 0.12515388 (27.24\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(k\PYGZus{}cat, Km) = 0.983
    C(Km, K\PYGZus{}ic)  = 0.152
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_12_1}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{kinetics\PYGZus{}HSAM3}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chymotrypsin inhibition by HSA(Chymo\PYGZhy{}M3)\PYGZhy{}huFc}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for competitive inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 59
    \PYGZsh{} data points      = 64
    \PYGZsh{} variables        = 3
    chi\PYGZhy{}square         = 1.9077e\PYGZhy{}04
    reduced chi\PYGZhy{}square = 3.1274e\PYGZhy{}06
    Akaike info crit   = \PYGZhy{}808.292970
    Bayesian info crit = \PYGZhy{}801.816321
[[Variables]]
    k\PYGZus{}cat:  56.8160006 +/\PYGZhy{} 3.20785889 (5.65\PYGZpc{}) (init = 25.93928)
    Km:     2.46689202 +/\PYGZhy{} 0.21789814 (8.83\PYGZpc{}) (init = 0.2593928)
    K\PYGZus{}ic:   0.05940519 +/\PYGZhy{} 0.00505739 (8.51\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(k\PYGZus{}cat, Km)   = 0.982
    C(Km, K\PYGZus{}ic)    = 0.388
    C(k\PYGZus{}cat, K\PYGZus{}ic) = 0.273
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{chymotrypsin_inhibition_13_1}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\sphinxstepscope


\section{\sphinxstylestrong{Scenario B:}\protect\(\alpha\protect\)\sphinxhyphen{}glucosidase inhibition by fucoidan}
\label{\detokenize{scenarios/glycosidase_inhibition:scenario-b-br-alpha-glucosidase-inhibition-by-fucoidan}}\label{\detokenize{scenarios/glycosidase_inhibition::doc}}
\sphinxAtStartPar
Data provided by Chantal Daub (Biochemistry, Rhodes University, Makhanda, South Africa)


\subsection{Project background}
\label{\detokenize{scenarios/glycosidase_inhibition:project-background}}
\sphinxAtStartPar
In this scenario, the inhibitory properties of fucoidan, a polysaccharide found in brown algae, on \(\alpha\)\sphinxhyphen{}glucosidase from \sphinxstyleemphasis{Saccharomyces cerevisiae} was investigated. Fucoidans are actively investigated in the fields of anit\sphinxhyphen{}cancer, anti\sphinxhyphen{}inflammation, and anti\sphinxhyphen{}coagulate, to name a fiew (Li \sphinxstyleemphasis{et al.} {[}\hyperlink{cite.references:id7}{2008}{]}). In a previous study (Daub \sphinxstyleemphasis{et al.} {[}\hyperlink{cite.references:id6}{2020}{]}), fucoidan from \sphinxstyleemphasis{E. maxima} showed an almost 2\sphinxhyphen{}fold lower IC50 value, compared to acarbose. Thus, fucoidan is a potential antidiabetic drug candidate for the treatment of diabetes mellitus.
In the following analysis, fucoidan from the brown algae species \sphinxstyleemphasis{Ecklonia maxima}, \sphinxstyleemphasis{Ecklonia radiata}, \sphinxstyleemphasis{Fucus vesiculosus}, S. CYM???????, and \sphinxstyleemphasis{Schimmelmannia elegans} were investigated for their inhibition constant \(K_{i}\) for \(\alpha\)\sphinxhyphen{}glucosidase inhibition.


\subsection{Experimental design}
\label{\detokenize{scenarios/glycosidase_inhibition:experimental-design}}
\sphinxAtStartPar
Extracted fucoidan from the mentioned brown algea species was applied in two different concentrations to enzyme reactions. Additionally, control reactions without inhibitor were performed. For the enzyme reactions, \sphinxstyleemphasis{p}\sphinxhyphen{}nitrophenyl\sphinxhyphen{}\(\alpha\)\sphinxhyphen{}D\sphinxhyphen{}glucopyranoside (pNPG) was applied as a substrate in a concentration range from 0.1 \sphinxhyphen{} 5 mM. Product accumulatin was followed photometrically in a micro titer plate at 405 nm for 20 min.


\subsection{Data preparation}
\label{\detokenize{scenarios/glycosidase_inhibition:data-preparation}}

\subsubsection{Imports}
\label{\detokenize{scenarios/glycosidase_inhibition:imports}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{typing} \PYG{k+kn}{import} \PYG{n}{Dict}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{colors} \PYG{k}{as} \PYG{n+nn}{mcolors}
\PYG{k+kn}{import} \PYG{n+nn}{re}
\PYG{k+kn}{import} \PYG{n+nn}{os}
\PYG{k+kn}{import} \PYG{n+nn}{pyenzyme} \PYG{k}{as} \PYG{n+nn}{pe}
\PYG{k+kn}{from} \PYG{n+nn}{CaliPytion}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{standardcurve} \PYG{k+kn}{import} \PYG{n}{StandardCurve}
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{parameterestimator} \PYG{k+kn}{import} \PYG{n}{ParameterEstimator}


\PYG{k+kn}{import} \PYG{n+nn}{warnings}
\PYG{n}{warnings}\PYG{o}{.}\PYG{n}{filterwarnings}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ignore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{colors} \PYG{o}{=} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{mcolors}\PYG{o}{.}\PYG{n}{TABLEAU\PYGZus{}COLORS}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Experimental data}
\label{\detokenize{scenarios/glycosidase_inhibition:experimental-data}}
\sphinxAtStartPar
Time\sphinxhyphen{}course data of all kinetic experiments was collected in an Excel file, whereas the meta data was specified in individual EnzymeML Excel templates for each origin species of fucoidan. In the following cell the experimental data is loaded and blanked to subtract the absorbance contribution of enzyme, buffer, and the respective inhibitor for each measurement. Thereafter, the data is written to EnzymeML documents.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{dataset\PYGZus{}path} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/glucosidase\PYGZus{}inhibition/experimental\PYGZus{}data.xlsx}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{template\PYGZus{}directory} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/glucosidase\PYGZus{}inhibition/EnzymeML\PYGZus{}templates}\PYG{l+s+s2}{\PYGZdq{}}   

\PYG{n}{inhibitors} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{pd}\PYG{o}{.}\PYG{n}{ExcelFile}\PYG{p}{(}\PYG{n}{dataset\PYGZus{}path}\PYG{p}{)}\PYG{o}{.}\PYG{n}{sheet\PYGZus{}names}\PYG{p}{)}
\PYG{n}{initial\PYGZus{}substrates} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mf}{2.5}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{]}


\PYG{n}{data\PYGZus{}dict} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
\PYG{k}{for} \PYG{n}{inhibitor} \PYG{o+ow}{in} \PYG{n}{inhibitors}\PYG{p}{:}
    \PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{dataset\PYGZus{}path}\PYG{p}{,} \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{n}{inhibitor}\PYG{p}{)}\PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{blanc\PYGZus{}no\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{buffer + enzyme}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{blank\PYGZus{}low\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{df}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{[}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{]}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{blank\PYGZus{}high\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{df}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{[}\PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{]}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{iloc}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{17}\PYG{p}{:}\PYG{p}{]}
    \PYG{n}{keys} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{df}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{)}
    \PYG{n}{df\PYGZus{}no\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{keys}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{12}\PYG{p}{]}\PYG{p}{]}
    \PYG{n}{df\PYGZus{}low\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{keys}\PYG{p}{[}\PYG{l+m+mi}{12}\PYG{p}{:}\PYG{l+m+mi}{24}\PYG{p}{]}\PYG{p}{]}
    \PYG{n}{df\PYGZus{}high\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{keys}\PYG{p}{[}\PYG{l+m+mi}{24}\PYG{p}{:}\PYG{p}{]}\PYG{p}{]}

    \PYG{n}{df\PYGZus{}no\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df\PYGZus{}no\PYGZus{}inhibitor}\PYG{o}{.}\PYG{n}{subtract}\PYG{p}{(}\PYG{n}{blanc\PYGZus{}no\PYGZus{}inhibitor}\PYG{p}{)}
    \PYG{n}{df\PYGZus{}low\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df\PYGZus{}low\PYGZus{}inhibitor}\PYG{o}{.}\PYG{n}{subtract}\PYG{p}{(}\PYG{n}{blank\PYGZus{}low\PYGZus{}inhibitor}\PYG{p}{)}
    \PYG{n}{df\PYGZus{}high\PYGZus{}inhibitor} \PYG{o}{=} \PYG{n}{df\PYGZus{}high\PYGZus{}inhibitor}\PYG{o}{.}\PYG{n}{subtract}\PYG{p}{(}\PYG{n}{blank\PYGZus{}high\PYGZus{}inhibitor}\PYG{p}{)}


    \PYG{n}{data} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{data}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{df\PYGZus{}no\PYGZus{}inhibitor}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{p}{)}
    \PYG{n}{data}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{df\PYGZus{}low\PYGZus{}inhibitor}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{p}{)}
    \PYG{n}{data}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{df\PYGZus{}high\PYGZus{}inhibitor}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{p}{)}

    \PYG{n}{data\PYGZus{}dict}\PYG{p}{[}\PYG{n}{inhibitor}\PYG{p}{]} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{data}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,}\PYG{l+m+mi}{20}\PYG{p}{)}

\PYG{n}{time} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{index}\PYG{o}{.}\PYG{n}{values}

\PYG{c+c1}{\PYGZsh{} Parse measurement data to EnzymeML documents}
\PYG{k}{def} \PYG{n+nf}{measurement\PYGZus{}data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{template\PYGZus{}path}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{measurement\PYGZus{}data}\PYG{p}{:} \PYG{n}{np}\PYG{o}{.}\PYG{n}{ndarray}\PYG{p}{,}
    \PYG{n}{species\PYGZus{}id}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{time}\PYG{p}{:} \PYG{n}{np}\PYG{o}{.}\PYG{n}{ndarray}\PYG{p}{,}
    \PYG{n}{data\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{time\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}
    \PYG{p}{)} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZgt{}} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{:}

    \PYG{n}{enzmldoc}\PYG{p}{:} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{n}{template\PYGZus{}path}\PYG{p}{)}

    \PYG{k}{for} \PYG{n}{IDs}\PYG{p}{,} \PYG{n}{concentration} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{keys}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{n}{measurement\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{for} \PYG{n}{counter}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{)}\PYG{p}{:}
            
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getMeasurement}\PYG{p}{(}\PYG{n}{IDs}\PYG{p}{)}\PYG{o}{.}\PYG{n}{addReplicates}\PYG{p}{(}\PYG{n}{pe}\PYG{o}{.}\PYG{n}{Replicate}\PYG{p}{(}
                \PYG{n+nb}{id}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Measurement}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{counter}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
                \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{n}{species\PYGZus{}id}\PYG{p}{,}
                \PYG{n}{data}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{replicate}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{n}{data\PYGZus{}unit}\PYG{p}{,}
                \PYG{n}{time}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{n}{time\PYGZus{}unit}\PYG{p}{)}\PYG{p}{,} \PYG{n}{enzmldoc}\PYG{p}{)}

    \PYG{k}{return} \PYG{n}{enzmldoc}

\PYG{n}{enzml\PYGZus{}docs} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{datas} \PYG{o}{=} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{data\PYGZus{}dict}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{file}\PYG{p}{,} \PYG{n}{data} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{template\PYGZus{}directory}\PYG{p}{)}\PYG{p}{)}\PYG{p}{,} \PYG{n}{datas}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{enzml\PYGZus{}docs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{measurement\PYGZus{}data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
        \PYG{n}{template\PYGZus{}path}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{template\PYGZus{}directory}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{file}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{measurement\PYGZus{}data}\PYG{o}{=}\PYG{n}{data}\PYG{p}{,}
        \PYG{n}{time}\PYG{o}{=}\PYG{n}{time}\PYG{p}{,}
        \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{min}\PYG{l+s+s2}{\PYGZdq{}}
    \PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Concentration calculation}
\label{\detokenize{scenarios/glycosidase_inhibition:concentration-calculation}}
\sphinxAtStartPar
Standard data of the product was loaded from an excel file, and a standard curve was created. Then, the standard curve was applied to the EnzymeML documents, containing the absorption measurements for concentration calculation.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{path\PYGZus{}calibration\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/glucosidase\PYGZus{}inhibition/p\PYGZhy{}NP\PYGZus{}standard.xlsx}\PYG{l+s+s2}{\PYGZdq{}}


\PYG{n}{product\PYGZus{}standard} \PYG{o}{=} \PYG{n}{StandardCurve}\PYG{o}{.}\PYG{n}{from\PYGZus{}excel}\PYG{p}{(}
    \PYG{n}{path}\PYG{o}{=}\PYG{n}{path\PYGZus{}calibration\PYGZus{}data}\PYG{p}{,}
    \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{csv}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{wavelength}\PYG{o}{=}\PYG{l+m+mi}{405}\PYG{p}{,} 
    \PYG{n}{concentration\PYGZus{}unit} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} 
    \PYG{n}{cutoff\PYGZus{}absorption}\PYG{o}{=}\PYG{l+m+mi}{2}\PYG{p}{)}

\PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Apply calibration curves to absorption EnzymeML documents}
\PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{:}
    \PYG{n}{product\PYGZus{}standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Calibration data was automatically blanked.
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                 AIC
3rd polynominal \PYGZhy{}135
Quadratic       \PYGZhy{}135
Rational        \PYGZhy{}135
Linear          \PYGZhy{}125
Exponential      \PYGZhy{}11
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_5_2}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Parameter estimation}
\label{\detokenize{scenarios/glycosidase_inhibition:parameter-estimation}}
\sphinxAtStartPar
Parameter estimation was perfomed with EnzymePynetics. Thereby, each data set was fitted to competitive, uncompetitive, and non\sphinxhyphen{}competitive inhibition models.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}
\begin{sphinxuseclass}{tag_output_scroll}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{kinetics} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{:}
    \PYG{n}{result} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{o}{=}\PYG{n}{enzmldoc}\PYG{p}{,}\PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{inhibitor\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{result}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
    \PYG{n}{result}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{plot\PYGZus{}means}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
    \PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive product inhibition model
 \PYGZhy{} uncompetitive product inhibition model
 \PYGZhy{} non\PYGZhy{}competitive product inhibition model
 \PYGZhy{} substrate inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                     AIC         kcat [1/min]  \PYGZbs{}
non\PYGZhy{}competitive product inhibition \PYGZhy{}2257  2480.867 +/\PYGZhy{} 10.41\PYGZpc{}   
competitive product inhibition     \PYGZhy{}2243   1756.430 +/\PYGZhy{} 2.72\PYGZpc{}   
uncompetitive product inhibition   \PYGZhy{}2187  2912.596 +/\PYGZhy{} 14.85\PYGZpc{}   
irreversible Michaelis Menten      \PYGZhy{}2146   1429.827 +/\PYGZhy{} 1.55\PYGZpc{}   
substrate inhibition               \PYGZhy{}2139   1455.278 +/\PYGZhy{} 1.73\PYGZpc{}   

                                      Km [mmole / l]  \PYGZbs{}
non\PYGZhy{}competitive product inhibition  0.545 +/\PYGZhy{} 19.96\PYGZpc{}   
competitive product inhibition      0.286 +/\PYGZhy{} 15.37\PYGZpc{}   
uncompetitive product inhibition    1.432 +/\PYGZhy{} 17.61\PYGZpc{}   
irreversible Michaelis Menten        0.603 +/\PYGZhy{} 4.92\PYGZpc{}   
substrate inhibition                 0.623 +/\PYGZhy{} 5.31\PYGZpc{}   

                                   kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
non\PYGZhy{}competitive product inhibition             4554.048 +/\PYGZhy{} 22.51\PYGZpc{}   
competitive product inhibition                 6132.817 +/\PYGZhy{} 15.61\PYGZpc{}   
uncompetitive product inhibition               2034.270 +/\PYGZhy{} 23.03\PYGZpc{}   
irreversible Michaelis Menten                   2372.038 +/\PYGZhy{} 5.16\PYGZpc{}   
substrate inhibition                            2334.831 +/\PYGZhy{} 5.59\PYGZpc{}   

                                   Ki competitive [mmole / l]  \PYGZbs{}
non\PYGZhy{}competitive product inhibition           0.040 +/\PYGZhy{} 24.85\PYGZpc{}   
competitive product inhibition               0.027 +/\PYGZhy{} 25.68\PYGZpc{}   
uncompetitive product inhibition                            \PYGZhy{}   
irreversible Michaelis Menten                               \PYGZhy{}   
substrate inhibition                                        \PYGZhy{}   

                                   Ki uncompetitive [mmole / l]  
non\PYGZhy{}competitive product inhibition             0.249 +/\PYGZhy{} 34.07\PYGZpc{}  
competitive product inhibition                                \PYGZhy{}  
uncompetitive product inhibition               0.108 +/\PYGZhy{} 29.22\PYGZpc{}  
irreversible Michaelis Menten                                 \PYGZhy{}  
substrate inhibition                         248.634 +/\PYGZhy{} 61.85\PYGZpc{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for non\PYGZhy{}competitive product inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 54
    \PYGZsh{} data points      = 237
    \PYGZsh{} variables        = 4
    chi\PYGZhy{}square         = 0.01676347
    reduced chi\PYGZhy{}square = 7.1946e\PYGZhy{}05
    Akaike info crit   = \PYGZhy{}2256.91734
    Bayesian info crit = \PYGZhy{}2243.04510
[[Variables]]
    k\PYGZus{}cat:  2480.86683 +/\PYGZhy{} 258.151536 (10.41\PYGZpc{}) (init = 3652.667)
    Km:     0.54476083 +/\PYGZhy{} 0.10872800 (19.96\PYGZpc{}) (init = 1.6784)
    K\PYGZus{}iu:   0.24899154 +/\PYGZhy{} 0.08483063 (34.07\PYGZpc{}) (init = 0.1)
    K\PYGZus{}ic:   0.04004974 +/\PYGZhy{} 0.00995050 (24.85\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(k\PYGZus{}cat, K\PYGZus{}iu) = \PYGZhy{}0.964
    C(Km, K\PYGZus{}iu)    = \PYGZhy{}0.808
    C(Km, K\PYGZus{}ic)    = 0.771
    C(k\PYGZus{}cat, Km)   = 0.731
    C(K\PYGZus{}iu, K\PYGZus{}ic)  = \PYGZhy{}0.324
    C(k\PYGZus{}cat, K\PYGZus{}ic) = 0.150
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_7_3}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive inhibition model
 \PYGZhy{} uncompetitive inhibition model
 \PYGZhy{} non\PYGZhy{}competitive inhibition model
 \PYGZhy{} partially competitive inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                   AIC        kcat [1/min]    Km [mmole / l]  \PYGZbs{}
non\PYGZhy{}competitive inhibition       \PYGZhy{}6142  1639.222 +/\PYGZhy{} 1.65\PYGZpc{}   0.552 +/\PYGZhy{} 5.44\PYGZpc{}   
competitive inhibition           \PYGZhy{}6055  1610.235 +/\PYGZhy{} 1.73\PYGZpc{}   0.526 +/\PYGZhy{} 5.80\PYGZpc{}   
partially competitive inhibition \PYGZhy{}6053  1610.208 +/\PYGZhy{} 1.74\PYGZpc{}   0.526 +/\PYGZhy{} 5.80\PYGZpc{}   
uncompetitive inhibition         \PYGZhy{}5819  1746.747 +/\PYGZhy{} 2.21\PYGZpc{}   0.742 +/\PYGZhy{} 6.49\PYGZpc{}   
irreversible Michaelis Menten    \PYGZhy{}4753   945.417 +/\PYGZhy{} 5.07\PYGZpc{}  0.809 +/\PYGZhy{} 15.18\PYGZpc{}   

                                 kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
non\PYGZhy{}competitive inhibition                    2971.298 +/\PYGZhy{} 5.68\PYGZpc{}   
competitive inhibition                        3062.265 +/\PYGZhy{} 6.05\PYGZpc{}   
partially competitive inhibition              3062.405 +/\PYGZhy{} 6.05\PYGZpc{}   
uncompetitive inhibition                      2353.001 +/\PYGZhy{} 6.86\PYGZpc{}   
irreversible Michaelis Menten                1168.173 +/\PYGZhy{} 16.00\PYGZpc{}   

                                 Ki competitive [g / l]  \PYGZbs{}
non\PYGZhy{}competitive inhibition             0.109 +/\PYGZhy{} 10.27\PYGZpc{}   
competitive inhibition                  0.052 +/\PYGZhy{} 5.48\PYGZpc{}   
partially competitive inhibition        0.052 +/\PYGZhy{} 5.75\PYGZpc{}   
uncompetitive inhibition                              \PYGZhy{}   
irreversible Michaelis Menten                         \PYGZhy{}   

                                 Ki uncompetitive [g / l]  
non\PYGZhy{}competitive inhibition               0.602 +/\PYGZhy{} 10.61\PYGZpc{}  
competitive inhibition                                  \PYGZhy{}  
partially competitive inhibition    1000.000 +/\PYGZhy{} 3817.44\PYGZpc{}  
uncompetitive inhibition                  0.214 +/\PYGZhy{} 4.57\PYGZpc{}  
irreversible Michaelis Menten                           \PYGZhy{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for non\PYGZhy{}competitive inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 31
    \PYGZsh{} data points      = 717
    \PYGZsh{} variables        = 4
    chi\PYGZhy{}square         = 0.13495950
    reduced chi\PYGZhy{}square = 1.8928e\PYGZhy{}04
    Akaike info crit   = \PYGZhy{}6142.32304
    Bayesian info crit = \PYGZhy{}6124.02274
[[Variables]]
    k\PYGZus{}cat:  1639.22182 +/\PYGZhy{} 27.1273263 (1.65\PYGZpc{}) (init = 4965.792)
    Km:     0.55168535 +/\PYGZhy{} 0.03000408 (5.44\PYGZpc{}) (init = 2.281781)
    K\PYGZus{}iu:   0.60181800 +/\PYGZhy{} 0.06386257 (10.61\PYGZpc{}) (init = 0.1)
    K\PYGZus{}ic:   0.10943768 +/\PYGZhy{} 0.01123454 (10.27\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(k\PYGZus{}cat, Km)   = 0.822
    C(K\PYGZus{}iu, K\PYGZus{}ic)  = \PYGZhy{}0.765
    C(Km, K\PYGZus{}ic)    = 0.469
    C(k\PYGZus{}cat, K\PYGZus{}ic) = 0.322
    C(k\PYGZus{}cat, K\PYGZus{}iu) = \PYGZhy{}0.306
    C(Km, K\PYGZus{}iu)    = \PYGZhy{}0.253
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_7_7}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive inhibition model
 \PYGZhy{} uncompetitive inhibition model
 \PYGZhy{} non\PYGZhy{}competitive inhibition model
 \PYGZhy{} partially competitive inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                   AIC        kcat [1/min]   Km [mmole / l]  \PYGZbs{}
partially competitive inhibition \PYGZhy{}7058  1075.485 +/\PYGZhy{} 1.11\PYGZpc{}  0.478 +/\PYGZhy{} 3.97\PYGZpc{}   
non\PYGZhy{}competitive inhibition       \PYGZhy{}6658  1059.810 +/\PYGZhy{} 1.56\PYGZpc{}  0.483 +/\PYGZhy{} 5.51\PYGZpc{}   
competitive inhibition           \PYGZhy{}6656  1046.201 +/\PYGZhy{} 1.46\PYGZpc{}  0.464 +/\PYGZhy{} 5.32\PYGZpc{}   
uncompetitive inhibition         \PYGZhy{}6260  1291.208 +/\PYGZhy{} 2.47\PYGZpc{}  1.151 +/\PYGZhy{} 6.12\PYGZpc{}   
irreversible Michaelis Menten    \PYGZhy{}5818   921.806 +/\PYGZhy{} 2.90\PYGZpc{}  1.070 +/\PYGZhy{} 8.06\PYGZpc{}   

                                 kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
partially competitive inhibition              2252.123 +/\PYGZhy{} 4.12\PYGZpc{}   
non\PYGZhy{}competitive inhibition                    2194.677 +/\PYGZhy{} 5.73\PYGZpc{}   
competitive inhibition                        2253.506 +/\PYGZhy{} 5.52\PYGZpc{}   
uncompetitive inhibition                      1121.648 +/\PYGZhy{} 6.60\PYGZpc{}   
irreversible Michaelis Menten                  861.615 +/\PYGZhy{} 8.57\PYGZpc{}   

                                 Ki competitive [g / l]  \PYGZbs{}
partially competitive inhibition      0.000 +/\PYGZhy{} 271.89\PYGZpc{}   
non\PYGZhy{}competitive inhibition              0.017 +/\PYGZhy{} 7.98\PYGZpc{}   
competitive inhibition                  0.015 +/\PYGZhy{} 5.33\PYGZpc{}   
uncompetitive inhibition                              \PYGZhy{}   
irreversible Michaelis Menten                         \PYGZhy{}   

                                 Ki uncompetitive [g / l]  
partially competitive inhibition        0.001 +/\PYGZhy{} 728.83\PYGZpc{}  
non\PYGZhy{}competitive inhibition               0.789 +/\PYGZhy{} 48.63\PYGZpc{}  
competitive inhibition                                  \PYGZhy{}  
uncompetitive inhibition                  0.065 +/\PYGZhy{} 5.45\PYGZpc{}  
irreversible Michaelis Menten                           \PYGZhy{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for partially competitive inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 272
    \PYGZsh{} data points      = 720
    \PYGZsh{} variables        = 4
    chi\PYGZhy{}square         = 0.03936609
    reduced chi\PYGZhy{}square = 5.4981e\PYGZhy{}05
    Akaike info crit   = \PYGZhy{}7058.15327
    Bayesian info crit = \PYGZhy{}7039.83626
[[Variables]]
    k\PYGZus{}cat:  1075.48537 +/\PYGZhy{} 11.8902583 (1.11\PYGZpc{}) (init = 1620.85)
    Km:     0.47754289 +/\PYGZhy{} 0.01896377 (3.97\PYGZpc{}) (init = 0.7447807)
    K\PYGZus{}iu:   6.1448e\PYGZhy{}04 +/\PYGZhy{} 0.00447849 (728.83\PYGZpc{}) (init = 0.1)
    K\PYGZus{}ic:   1.0000e\PYGZhy{}04 +/\PYGZhy{} 2.7189e\PYGZhy{}04 (271.89\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(K\PYGZus{}iu, K\PYGZus{}ic) = \PYGZhy{}1.000
    C(k\PYGZus{}cat, Km)  = 0.779
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_7_11}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive inhibition model
 \PYGZhy{} uncompetitive inhibition model
 \PYGZhy{} non\PYGZhy{}competitive inhibition model
 \PYGZhy{} partially competitive inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                   AIC        kcat [1/min]    Km [mmole / l]  \PYGZbs{}
competitive inhibition           \PYGZhy{}7326  1527.487 +/\PYGZhy{} 0.99\PYGZpc{}   0.938 +/\PYGZhy{} 2.87\PYGZpc{}   
non\PYGZhy{}competitive inhibition       \PYGZhy{}7324  1527.536 +/\PYGZhy{} 0.99\PYGZpc{}   0.938 +/\PYGZhy{} 2.88\PYGZpc{}   
partially competitive inhibition \PYGZhy{}7324  1527.908 +/\PYGZhy{} 0.99\PYGZpc{}   0.939 +/\PYGZhy{} 2.88\PYGZpc{}   
uncompetitive inhibition         \PYGZhy{}7028  1531.314 +/\PYGZhy{} 1.22\PYGZpc{}   0.947 +/\PYGZhy{} 3.53\PYGZpc{}   
irreversible Michaelis Menten    \PYGZhy{}4775  667.025 +/\PYGZhy{} 10.33\PYGZpc{}  1.455 +/\PYGZhy{} 26.07\PYGZpc{}   

                                 kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
competitive inhibition                        1628.994 +/\PYGZhy{} 3.04\PYGZpc{}   
non\PYGZhy{}competitive inhibition                    1628.803 +/\PYGZhy{} 3.04\PYGZpc{}   
partially competitive inhibition              1627.963 +/\PYGZhy{} 3.04\PYGZpc{}   
uncompetitive inhibition                      1616.521 +/\PYGZhy{} 3.74\PYGZpc{}   
irreversible Michaelis Menten                 458.453 +/\PYGZhy{} 28.04\PYGZpc{}   

                                 Ki competitive [g / l]  \PYGZbs{}
competitive inhibition                  0.001 +/\PYGZhy{} 4.85\PYGZpc{}   
non\PYGZhy{}competitive inhibition             0.001 +/\PYGZhy{} 14.89\PYGZpc{}   
partially competitive inhibition        0.001 +/\PYGZhy{} 7.82\PYGZpc{}   
uncompetitive inhibition                              \PYGZhy{}   
irreversible Michaelis Menten                         \PYGZhy{}   

                                 Ki uncompetitive [g / l]  
competitive inhibition                                  \PYGZhy{}  
non\PYGZhy{}competitive inhibition           15.494 +/\PYGZhy{} 36228.16\PYGZpc{}  
partially competitive inhibition   999.141 +/\PYGZhy{} 139876.39\PYGZpc{}  
uncompetitive inhibition                  0.002 +/\PYGZhy{} 7.52\PYGZpc{}  
irreversible Michaelis Menten                           \PYGZhy{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for competitive inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 34
    \PYGZsh{} data points      = 720
    \PYGZsh{} variables        = 3
    chi\PYGZhy{}square         = 0.02722244
    reduced chi\PYGZhy{}square = 3.7967e\PYGZhy{}05
    Akaike info crit   = \PYGZhy{}7325.73480
    Bayesian info crit = \PYGZhy{}7311.99704
[[Variables]]
    k\PYGZus{}cat:  1527.48678 +/\PYGZhy{} 15.1025514 (0.99\PYGZpc{}) (init = 2691.849)
    Km:     0.93768696 +/\PYGZhy{} 0.02693290 (2.87\PYGZpc{}) (init = 1.236905)
    K\PYGZus{}ic:   0.00134723 +/\PYGZhy{} 6.5379e\PYGZhy{}05 (4.85\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(k\PYGZus{}cat, Km)   = 0.865
    C(Km, K\PYGZus{}ic)    = 0.407
    C(k\PYGZus{}cat, K\PYGZus{}ic) = 0.294
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_7_15}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive inhibition model
 \PYGZhy{} uncompetitive inhibition model
 \PYGZhy{} non\PYGZhy{}competitive inhibition model
 \PYGZhy{} partially competitive inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                   AIC       kcat [1/min]   Km [mmole / l]  \PYGZbs{}
partially competitive inhibition \PYGZhy{}7611  885.193 +/\PYGZhy{} 0.65\PYGZpc{}  0.431 +/\PYGZhy{} 2.76\PYGZpc{}   
non\PYGZhy{}competitive inhibition       \PYGZhy{}7574  900.852 +/\PYGZhy{} 0.92\PYGZpc{}  0.471 +/\PYGZhy{} 3.29\PYGZpc{}   
competitive inhibition           \PYGZhy{}7568  885.729 +/\PYGZhy{} 0.67\PYGZpc{}  0.448 +/\PYGZhy{} 2.79\PYGZpc{}   
uncompetitive inhibition         \PYGZhy{}7324  975.489 +/\PYGZhy{} 1.06\PYGZpc{}  0.710 +/\PYGZhy{} 2.69\PYGZpc{}   
irreversible Michaelis Menten    \PYGZhy{}7096  874.452 +/\PYGZhy{} 0.94\PYGZpc{}  0.652 +/\PYGZhy{} 3.01\PYGZpc{}   

                                 kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
partially competitive inhibition              2051.891 +/\PYGZhy{} 2.83\PYGZpc{}   
non\PYGZhy{}competitive inhibition                    1914.077 +/\PYGZhy{} 3.41\PYGZpc{}   
competitive inhibition                        1979.019 +/\PYGZhy{} 2.87\PYGZpc{}   
uncompetitive inhibition                      1373.348 +/\PYGZhy{} 2.89\PYGZpc{}   
irreversible Michaelis Menten                 1340.611 +/\PYGZhy{} 3.16\PYGZpc{}   

                                 Ki competitive [g / l]  \PYGZbs{}
partially competitive inhibition       0.175 +/\PYGZhy{} 28.00\PYGZpc{}   
non\PYGZhy{}competitive inhibition              0.966 +/\PYGZhy{} 7.73\PYGZpc{}   
competitive inhibition                  0.841 +/\PYGZhy{} 5.36\PYGZpc{}   
uncompetitive inhibition                              \PYGZhy{}   
irreversible Michaelis Menten                         \PYGZhy{}   

                                 Ki uncompetitive [g / l]  
partially competitive inhibition         0.452 +/\PYGZhy{} 36.06\PYGZpc{}  
non\PYGZhy{}competitive inhibition              21.519 +/\PYGZhy{} 37.84\PYGZpc{}  
competitive inhibition                                  \PYGZhy{}  
uncompetitive inhibition                  3.737 +/\PYGZhy{} 6.80\PYGZpc{}  
irreversible Michaelis Menten                           \PYGZhy{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for partially competitive inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 124
    \PYGZsh{} data points      = 720
    \PYGZsh{} variables        = 4
    chi\PYGZhy{}square         = 0.01825580
    reduced chi\PYGZhy{}square = 2.5497e\PYGZhy{}05
    Akaike info crit   = \PYGZhy{}7611.41698
    Bayesian info crit = \PYGZhy{}7593.09998
[[Variables]]
    k\PYGZus{}cat:  885.192687 +/\PYGZhy{} 5.78448532 (0.65\PYGZpc{}) (init = 1279.274)
    Km:     0.43140343 +/\PYGZhy{} 0.01189835 (2.76\PYGZpc{}) (init = 0.5878264)
    K\PYGZus{}iu:   0.45219052 +/\PYGZhy{} 0.16305078 (36.06\PYGZpc{}) (init = 0.1)
    K\PYGZus{}ic:   0.17451878 +/\PYGZhy{} 0.04887071 (28.00\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(K\PYGZus{}iu, K\PYGZus{}ic) = 0.994
    C(k\PYGZus{}cat, Km)  = 0.671
    C(Km, K\PYGZus{}ic)   = 0.146
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_7_19}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive inhibition model
 \PYGZhy{} uncompetitive inhibition model
 \PYGZhy{} non\PYGZhy{}competitive inhibition model
 \PYGZhy{} partially competitive inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                   AIC        kcat [1/min]    Km [mmole / l]  \PYGZbs{}
non\PYGZhy{}competitive inhibition       \PYGZhy{}7110  1196.091 +/\PYGZhy{} 1.22\PYGZpc{}   0.693 +/\PYGZhy{} 3.86\PYGZpc{}   
competitive inhibition           \PYGZhy{}7106  1191.620 +/\PYGZhy{} 1.21\PYGZpc{}   0.686 +/\PYGZhy{} 3.85\PYGZpc{}   
partially competitive inhibition \PYGZhy{}7104  1191.646 +/\PYGZhy{} 1.21\PYGZpc{}   0.686 +/\PYGZhy{} 3.86\PYGZpc{}   
uncompetitive inhibition         \PYGZhy{}6381  1334.869 +/\PYGZhy{} 2.31\PYGZpc{}   1.095 +/\PYGZhy{} 6.22\PYGZpc{}   
irreversible Michaelis Menten    \PYGZhy{}5320   837.262 +/\PYGZhy{} 6.08\PYGZpc{}  1.586 +/\PYGZhy{} 14.95\PYGZpc{}   

                                 kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
non\PYGZhy{}competitive inhibition                    1724.867 +/\PYGZhy{} 4.04\PYGZpc{}   
competitive inhibition                        1736.014 +/\PYGZhy{} 4.04\PYGZpc{}   
partially competitive inhibition              1735.964 +/\PYGZhy{} 4.04\PYGZpc{}   
uncompetitive inhibition                      1218.643 +/\PYGZhy{} 6.63\PYGZpc{}   
irreversible Michaelis Menten                 527.980 +/\PYGZhy{} 16.14\PYGZpc{}   

                                 Ki competitive [g / l]  \PYGZbs{}
non\PYGZhy{}competitive inhibition              0.008 +/\PYGZhy{} 6.67\PYGZpc{}   
competitive inhibition                  0.007 +/\PYGZhy{} 3.65\PYGZpc{}   
partially competitive inhibition        0.007 +/\PYGZhy{} 9.27\PYGZpc{}   
uncompetitive inhibition                              \PYGZhy{}   
irreversible Michaelis Menten                         \PYGZhy{}   

                                 Ki uncompetitive [g / l]  
non\PYGZhy{}competitive inhibition               0.272 +/\PYGZhy{} 39.90\PYGZpc{}  
competitive inhibition                                  \PYGZhy{}  
partially competitive inhibition   999.732 +/\PYGZhy{} 110814.71\PYGZpc{}  
uncompetitive inhibition                  0.021 +/\PYGZhy{} 4.53\PYGZpc{}  
irreversible Michaelis Menten                           \PYGZhy{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for non\PYGZhy{}competitive inhibition model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 48
    \PYGZsh{} data points      = 720
    \PYGZsh{} variables        = 4
    chi\PYGZhy{}square         = 0.03661395
    reduced chi\PYGZhy{}square = 5.1137e\PYGZhy{}05
    Akaike info crit   = \PYGZhy{}7110.33552
    Bayesian info crit = \PYGZhy{}7092.01851
[[Variables]]
    k\PYGZus{}cat:  1196.09080 +/\PYGZhy{} 14.5632835 (1.22\PYGZpc{}) (init = 1686.748)
    Km:     0.69343953 +/\PYGZhy{} 0.02673319 (3.86\PYGZpc{}) (init = 0.7750608)
    K\PYGZus{}iu:   0.27178749 +/\PYGZhy{} 0.10844409 (39.90\PYGZpc{}) (init = 0.1)
    K\PYGZus{}ic:   0.00750637 +/\PYGZhy{} 5.0042e\PYGZhy{}04 (6.67\PYGZpc{}) (init = 0.1)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(k\PYGZus{}cat, Km)   = 0.834
    C(K\PYGZus{}iu, K\PYGZus{}ic)  = \PYGZhy{}0.825
    C(Km, K\PYGZus{}ic)    = 0.477
    C(k\PYGZus{}cat, K\PYGZus{}ic) = 0.334
    C(k\PYGZus{}cat, K\PYGZus{}iu) = \PYGZhy{}0.159
    C(Km, K\PYGZus{}iu)    = \PYGZhy{}0.135
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_7_23}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{\protect\(k_{cat}\protect\) and \protect\(K_{m}\protect\)}
\label{\detokenize{scenarios/glycosidase_inhibition:k-cat-and-k-m}}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Get kinetic parameters of all datasets}
\PYG{n}{kcat} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{corr\PYGZus{}kcat\PYGZus{}km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{result} \PYG{o+ow}{in} \PYG{n}{kinetics}\PYG{p}{:}
    \PYG{n}{params} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}parameter\PYGZus{}dict}\PYG{p}{(}\PYG{p}{)}

    \PYG{n}{kcat}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{kcat\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{Km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{Km\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}


    \PYG{n}{correlation} \PYG{o}{=} \PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{correl}
    \PYG{k}{if} \PYG{n}{correlation} \PYG{o}{==} \PYG{k+kc}{None}\PYG{p}{:}
        \PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n+nb}{float}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{nan}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{correlation}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}


\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{o}{.}\PYG{n}{from\PYGZus{}dict}\PYG{p}{(}\PYG{p}{\PYGZob{}}
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat [1/min]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km [mM]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{correlation kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{df}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
   kcat [1/min]  kcat stderr   Km [mM]  Km stderr  correlation kcat/Km
0   1650.543562    68.567687  0.537270   0.080309            \PYGZhy{}0.434760
1   1768.422752   102.033022  1.461047   0.200855             0.912771
2   1089.309474    15.733598  0.676981   0.031014             0.826747
3   1474.870342    11.446200  1.260928   0.025848             0.893857
4    824.901555     5.351402  0.510622   0.012484             0.722688
5   1163.327292    17.444942  0.916960   0.039877             0.861308
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{enzml\PYGZus{}docs}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}

\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m6}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m7}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m8}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m9}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m10}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m11}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m12}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m13}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m14}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m15}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m16}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m17}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{enzmldoc}\PYG{o}{=}\PYG{n}{enzmldoc}\PYG{p}{,}
    \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive product inhibition model
 \PYGZhy{} uncompetitive product inhibition model
 \PYGZhy{} non\PYGZhy{}competitive product inhibition model
 \PYGZhy{} substrate inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                     AIC         kcat [1/min]  \PYGZbs{}
competitive product inhibition     \PYGZhy{}2257   1879.010 +/\PYGZhy{} 2.96\PYGZpc{}   
non\PYGZhy{}competitive product inhibition \PYGZhy{}2255  2477.180 +/\PYGZhy{} 24.37\PYGZpc{}   
irreversible Michaelis Menten      \PYGZhy{}2231   1883.439 +/\PYGZhy{} 2.90\PYGZpc{}   
uncompetitive product inhibition   \PYGZhy{}2229   1883.604 +/\PYGZhy{} 3.08\PYGZpc{}   
substrate inhibition               \PYGZhy{}2227   1892.295 +/\PYGZhy{} 3.08\PYGZpc{}   

                                      Km [mmole / l]  \PYGZbs{}
competitive product inhibition      0.398 +/\PYGZhy{} 11.56\PYGZpc{}   
non\PYGZhy{}competitive product inhibition  0.544 +/\PYGZhy{} 30.17\PYGZpc{}   
irreversible Michaelis Menten        0.582 +/\PYGZhy{} 4.10\PYGZpc{}   
uncompetitive product inhibition     0.582 +/\PYGZhy{} 4.11\PYGZpc{}   
substrate inhibition                 0.587 +/\PYGZhy{} 4.17\PYGZpc{}   

                                   ki time\PYGZhy{}dep enzyme\PYGZhy{}inactiv. [1/min]  \PYGZbs{}
competitive product inhibition                        0.023 +/\PYGZhy{} 24.42\PYGZpc{}   
non\PYGZhy{}competitive product inhibition                  0.000 +/\PYGZhy{} 3906.23\PYGZpc{}   
irreversible Michaelis Menten                         0.043 +/\PYGZhy{} 10.25\PYGZpc{}   
uncompetitive product inhibition                      0.043 +/\PYGZhy{} 10.34\PYGZpc{}   
substrate inhibition                                  0.043 +/\PYGZhy{} 11.03\PYGZpc{}   

                                   kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
competitive product inhibition                 4716.196 +/\PYGZhy{} 11.93\PYGZpc{}   
non\PYGZhy{}competitive product inhibition             4554.158 +/\PYGZhy{} 38.78\PYGZpc{}   
irreversible Michaelis Menten                   3234.668 +/\PYGZhy{} 5.03\PYGZpc{}   
uncompetitive product inhibition                3234.343 +/\PYGZhy{} 5.14\PYGZpc{}   
substrate inhibition                            3222.920 +/\PYGZhy{} 5.18\PYGZpc{}   

                                   Ki competitive [mmole / l]  \PYGZbs{}
competitive product inhibition               0.065 +/\PYGZhy{} 32.40\PYGZpc{}   
non\PYGZhy{}competitive product inhibition           0.040 +/\PYGZhy{} 39.86\PYGZpc{}   
irreversible Michaelis Menten                               \PYGZhy{}   
uncompetitive product inhibition                            \PYGZhy{}   
substrate inhibition                                        \PYGZhy{}   

                                   Ki uncompetitive [mmole / l]  
competitive product inhibition                                \PYGZhy{}  
non\PYGZhy{}competitive product inhibition             0.251 +/\PYGZhy{} 99.55\PYGZpc{}  
irreversible Michaelis Menten                                 \PYGZhy{}  
uncompetitive product inhibition           844.290 +/\PYGZhy{} 4979.17\PYGZpc{}  
substrate inhibition                        999.940 +/\PYGZhy{} 156.07\PYGZpc{}  
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}
    \PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} 
    \PYG{n}{start\PYGZus{}time\PYGZus{}index}\PYG{o}{=}\PYG{l+m+mi}{8}\PYG{p}{,}
    \PYG{n}{initial\PYGZus{}substrate\PYGZus{}concs}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+m+mf}{0.25}\PYG{p}{,} \PYG{l+m+mf}{0.5}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{]}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
 \PYGZhy{} competitive product inhibition model
 \PYGZhy{} uncompetitive product inhibition model
 \PYGZhy{} non\PYGZhy{}competitive product inhibition model
 \PYGZhy{} substrate inhibition model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                                     AIC         kcat [1/min]  \PYGZbs{}
competitive product inhibition     \PYGZhy{}1367   1807.855 +/\PYGZhy{} 3.09\PYGZpc{}   
non\PYGZhy{}competitive product inhibition \PYGZhy{}1365  1808.173 +/\PYGZhy{} 17.82\PYGZpc{}   
irreversible Michaelis Menten      \PYGZhy{}1342   1716.141 +/\PYGZhy{} 3.08\PYGZpc{}   
uncompetitive product inhibition   \PYGZhy{}1340   1716.422 +/\PYGZhy{} 3.14\PYGZpc{}   
substrate inhibition               \PYGZhy{}1339   1725.403 +/\PYGZhy{} 4.13\PYGZpc{}   

                                      Km [mmole / l]  \PYGZbs{}
competitive product inhibition       0.660 +/\PYGZhy{} 8.33\PYGZpc{}   
non\PYGZhy{}competitive product inhibition  0.660 +/\PYGZhy{} 19.81\PYGZpc{}   
irreversible Michaelis Menten        0.907 +/\PYGZhy{} 3.62\PYGZpc{}   
uncompetitive product inhibition     0.907 +/\PYGZhy{} 4.10\PYGZpc{}   
substrate inhibition                 0.914 +/\PYGZhy{} 5.89\PYGZpc{}   

                                   ki time\PYGZhy{}dep enzyme\PYGZhy{}inactiv. [1/min]  \PYGZbs{}
competitive product inhibition                        0.018 +/\PYGZhy{} 42.60\PYGZpc{}   
non\PYGZhy{}competitive product inhibition                    0.018 +/\PYGZhy{} 54.17\PYGZpc{}   
irreversible Michaelis Menten                         0.032 +/\PYGZhy{} 25.03\PYGZpc{}   
uncompetitive product inhibition                      0.032 +/\PYGZhy{} 26.29\PYGZpc{}   
substrate inhibition                                  0.032 +/\PYGZhy{} 25.24\PYGZpc{}   

                                   kcat / Km [1/min * 1/mmole / l]  \PYGZbs{}
competitive product inhibition                  2738.363 +/\PYGZhy{} 8.88\PYGZpc{}   
non\PYGZhy{}competitive product inhibition             2738.303 +/\PYGZhy{} 26.65\PYGZpc{}   
irreversible Michaelis Menten                   1892.619 +/\PYGZhy{} 4.75\PYGZpc{}   
uncompetitive product inhibition                1892.485 +/\PYGZhy{} 5.17\PYGZpc{}   
substrate inhibition                            1886.983 +/\PYGZhy{} 7.19\PYGZpc{}   

                                   Ki competitive [mmole / l]  \PYGZbs{}
competitive product inhibition               0.145 +/\PYGZhy{} 28.45\PYGZpc{}   
non\PYGZhy{}competitive product inhibition           0.145 +/\PYGZhy{} 35.04\PYGZpc{}   
irreversible Michaelis Menten                               \PYGZhy{}   
uncompetitive product inhibition                            \PYGZhy{}   
substrate inhibition                                        \PYGZhy{}   

                                   Ki uncompetitive [mmole / l]  
competitive product inhibition                                \PYGZhy{}  
non\PYGZhy{}competitive product inhibition        954.402 +/\PYGZhy{} 86749.69\PYGZpc{}  
irreversible Michaelis Menten                                 \PYGZhy{}  
uncompetitive product inhibition          999.612 +/\PYGZhy{} 10450.38\PYGZpc{}  
substrate inhibition                        999.998 +/\PYGZhy{} 575.44\PYGZpc{}  
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{model\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{a\PYGZhy{}glucosidase reaction}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Fit report for irreversible Michaelis Menten model
[[Fit Statistics]]
    \PYGZsh{} fitting method   = leastsq
    \PYGZsh{} function evals   = 21
    \PYGZsh{} data points      = 117
    \PYGZsh{} variables        = 3
    chi\PYGZhy{}square         = 0.00116138
    reduced chi\PYGZhy{}square = 1.0188e\PYGZhy{}05
    Akaike info crit   = \PYGZhy{}1341.87794
    Bayesian info crit = \PYGZhy{}1333.59141
[[Variables]]
    k\PYGZus{}cat:  1716.14120 +/\PYGZhy{} 52.8030058 (3.08\PYGZpc{}) (init = 3652.667)
    Km:     0.90675489 +/\PYGZhy{} 0.03286791 (3.62\PYGZpc{}) (init = 1.6784)
    K\PYGZus{}ie:   0.03177073 +/\PYGZhy{} 0.00795342 (25.03\PYGZpc{}) (init = 0.01)
[[Correlations]] (unreported correlations are \PYGZlt{} 0.100)
    C(k\PYGZus{}cat, K\PYGZus{}ie) = 0.883
    C(Km, K\PYGZus{}ie)    = \PYGZhy{}0.210
    C(k\PYGZus{}cat, Km)   = 0.198
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{glycosidase_inhibition_13_1}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxstepscope


\section{\sphinxstylestrong{Scenario C:}SLAC characterization}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:scenario-c-br-slac-characterization}}\label{\detokenize{scenarios/SLAC_kinetic_characterization::doc}}
\sphinxAtStartPar
Data provided by Alaric Prins (Biocatalysis and Technical Biology, Cape Peninsula University of Technology, Capetown, South Africa)


\subsection{Project background}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:project-background}}

\subsubsection{Experimental design}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:experimental-design}}
\sphinxAtStartPar
In this scenario, the catalytic properties of the small laccase from \sphinxstyleemphasis{Streptomyces coelicolor} (SLAC) were investigated. Therefore, the enzymatic oxidation of 2,2’\sphinxhyphen{}Azino\sphinxhyphen{}bis(3\sphinxhyphen{}ethylbenzothiazoline\sphinxhyphen{}6\sphinxhyphen{}sulfonic acid) (ABTS) to its radical form ABTS+· was studied in the pH range between pH 3 \sphinxhyphen{} 5.5 and temperature range between 25°C \sphinxhyphen{} 45°C. In total 30 kinetic enzyme assays in a substrate range between 0 \sphinxhyphen{} 200 uM of ABTS were conducted. Additionally, for each enzyme reaction with a given initial substrate concentration, a control reaction without enzyme was prepared.For each pH \sphinxhyphen{} temperature condition a seperate ABTS standard curve and absorption spectrum was recorded, in order to account for varying ABTS absorption properties do to reaction conditions. Each enzyme reaction was followed for 15 min photometrically at two seperate wavelengths, measuring substrate depletion and product accumululation simultaneously. Therefore, preliminary experiments confirmed, that ABTS absorbs at 340 nm, whereas the ABTS+· radical product absorbs at 420 nm. Furthermore, cross absorbance of product at the substrate detection wavelength and vice versa was ruled out.


\subsubsection{Data management}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:data-management}}
\sphinxAtStartPar
Overall, the dataset consists of more than 100 000 individual absorbance reads. Thus, data preperation was automated by custom parser functions, which were tailored to the output of the used spectro photometer. Consequently, the tedious and error\sphinxhyphen{}prone manual copying of raw data was avoided. Information on the involved reactants, and the enzyme was filled in an EnzymeML Excel spreadsheet, which served as a meta data container. All other information was parsed from the .txt output of the photometer. Figure 1 illustrates the schematic data flow of this project.

\sphinxAtStartPar
\sphinxincludegraphics{{workflow_SLAC}.png}
\sphinxstyleemphasis{Fig. 1: Schematic data pipeline of the SLAC characterization.}


\subsection{Data preparation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:data-preparation}}

\subsubsection{Imports}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:imports}}

\subsubsection{Experimantal data}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:experimantal-data}}
\sphinxAtStartPar
Data from the enzyme reactions was loaded from the output files of the photometer and written to individual EnzymeML documents. Then infromation of the control reactions was used to subtract the absorption, contributed from enzyme and buffer from the substrate and product signal.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Specify the location of the data sets}
\PYG{n}{directory\PYGZus{}measurement\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/TimeCourseData}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{directory\PYGZus{}standard\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/StandardData}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{directory\PYGZus{}spectrum\PYGZus{}data} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/SpectrumData}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{path\PYGZus{}EnzymeML\PYGZus{}templates} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/SLAC\PYGZus{}kinetic\PYGZus{}characterization/EnzymeML\PYGZus{}templates}\PYG{l+s+s2}{\PYGZdq{}}

\PYG{c+c1}{\PYGZsh{} Define IDs for species, listed in the EnzymeML Excel template}
\PYG{n}{substrate\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{product\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{substrate\PYGZus{}control\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{product\PYGZus{}control\PYGZus{}id} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{n}{species\PYGZus{}ids} \PYG{o}{=} \PYG{p}{[}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{,} \PYG{n}{product\PYGZus{}id}\PYG{p}{,} \PYG{n}{substrate\PYGZus{}control\PYGZus{}id}\PYG{p}{,} \PYG{n}{product\PYGZus{}control\PYGZus{}id}\PYG{p}{]}


\PYG{c+c1}{\PYGZsh{} Parse measurement data from photometer output}
\PYG{n}{raw\PYGZus{}data\PYGZus{}dict} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
\PYG{k}{for} \PYG{n}{path} \PYG{o+ow}{in} \PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{directory\PYGZus{}measurement\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{data} \PYG{o}{=} \PYG{n}{read\PYGZus{}measurement\PYGZus{}data}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{directory\PYGZus{}measurement\PYGZus{}data}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{path}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{pH} \PYG{o}{=} \PYG{n}{data}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
    \PYG{n}{temp} \PYG{o}{=} \PYG{n}{data}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
    \PYG{n}{raw\PYGZus{}data\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{ }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{temp}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{data}

\PYG{n}{EnzymeML\PYGZus{}template\PYGZus{}dict} \PYG{o}{=} \PYG{p}{\PYGZob{}}
    \PYG{l+m+mf}{3.0}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH3.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{3.5}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH3\PYGZus{}5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{4.0}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH4.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{4.5}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH4\PYGZus{}5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{5.0}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{l+m+mf}{5.5}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{EnzymeML\PYGZus{}SLAC\PYGZus{}pH5\PYGZus{}5.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
\PYG{p}{\PYGZcb{}}

\PYG{c+c1}{\PYGZsh{} Write absorption data to EnzymeMLDocuments}
\PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{:} \PYG{n}{List}\PYG{p}{[}\PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{name}\PYG{p}{,} \PYG{n}{data} \PYG{o+ow}{in} \PYG{n}{raw\PYGZus{}data\PYGZus{}dict}\PYG{o}{.}\PYG{n}{items}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{pH} \PYG{o}{=} \PYG{n}{data}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
    \PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{measurement\PYGZus{}data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
        \PYG{n}{template\PYGZus{}path}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{path\PYGZus{}EnzymeML\PYGZus{}templates}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{EnzymeML\PYGZus{}template\PYGZus{}dict}\PYG{p}{[}\PYG{n}{pH}\PYG{p}{]}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{measurement\PYGZus{}data}\PYG{o}{=}\PYG{n}{data}\PYG{p}{,}
        \PYG{n}{species\PYGZus{}ids}\PYG{o}{=}\PYG{n}{species\PYGZus{}ids}\PYG{p}{,}
        \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{umole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Sort documents by ascending pH and temperature}
\PYG{n}{absortion\PYGZus{}enzymemldocs} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{,} \PYG{n}{key}\PYG{o}{=}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{ph}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Blanc measurement data}
\PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{:}
    \PYG{n}{blanc\PYGZus{}measurement} \PYG{o}{=} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}
    \PYG{n}{blanc} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{p}{[}\PYG{n}{repeat}\PYG{o}{.}\PYG{n}{data} \PYG{k}{for} \PYG{n}{repeat} \PYG{o+ow}{in} \PYG{n}{blanc\PYGZus{}measurement}\PYG{p}{]}\PYG{p}{)}

    \PYG{k}{for} \PYG{n+nb}{id}\PYG{p}{,} \PYG{n}{measurement} \PYG{o+ow}{in} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{items}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}
        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}

        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}
        \PYG{k}{for} \PYG{n}{rep}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{blanced\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{value} \PYG{o}{\PYGZhy{}} \PYG{n}{blanc} \PYG{k}{for} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{]}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{n+nb}{id}\PYG{p}{]}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s3}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{n}{rep}\PYG{p}{]}\PYG{o}{.}\PYG{n}{data} \PYG{o}{=} \PYG{n}{blanced\PYGZus{}data}
    
    \PYG{c+c1}{\PYGZsh{} Delete control measurement \PYGZsq{}m0\PYGZsq{}}
    \PYG{k}{del} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Concentration calculation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:concentration-calculation}}
\sphinxAtStartPar
Calibration data was loaded and converted into individual instances of the calibration data model. Some meta data of the calibration needed to be provided to the custom \sphinxcode{\sphinxupquote{read\_calibration\_data}} function, since the output of the used spectro photometer contained a minimum of information only. Thereafter, a StandardCurve was created for each calibration data set. Thereby, only absoption values below 3 were considered, since higher absorption values could not be converted into concentration accurately.
Lastly, the fit of each standard curves was visualized.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load calibration raw data}
\PYG{n}{calibration\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{standard\PYGZus{}directory} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{sort}\PYG{p}{(}\PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{directory\PYGZus{}standard\PYGZus{}data}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{spectrum\PYGZus{}directory} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{sort}\PYG{p}{(}\PYG{n}{os}\PYG{o}{.}\PYG{n}{listdir}\PYG{p}{(}\PYG{n}{directory\PYGZus{}spectrum\PYGZus{}data}\PYG{p}{)}\PYG{p}{)}

\PYG{k}{for} \PYG{n}{standard}\PYG{p}{,} \PYG{n}{spectrum} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{standard\PYGZus{}directory}\PYG{p}{,} \PYG{n}{spectrum\PYGZus{}directory}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{standard} \PYG{o}{=} \PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{directory\PYGZus{}standard\PYGZus{}data}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{standard}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}
    \PYG{n}{spectrum} \PYG{o}{=} \PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{directory\PYGZus{}spectrum\PYGZus{}data}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{/}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{spectrum}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}
    \PYG{n}{result} \PYG{o}{=} \PYG{n}{read\PYGZus{}calibration\PYGZus{}data}\PYG{p}{(}
        \PYG{n}{path\PYGZus{}standard}\PYG{o}{=}\PYG{n}{standard}\PYG{p}{,}
        \PYG{n}{path\PYGZus{}spectrum}\PYG{o}{=}\PYG{n}{spectrum}\PYG{p}{,}
        \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{,}
        \PYG{n}{wavelengths}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mi}{340}\PYG{p}{,} \PYG{l+m+mi}{420}\PYG{p}{]}\PYG{p}{,}
        \PYG{n}{concentrations}\PYG{o}{=}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{15}\PYG{p}{,}\PYG{l+m+mi}{25}\PYG{p}{,}\PYG{l+m+mi}{50}\PYG{p}{,}\PYG{l+m+mi}{75}\PYG{p}{,}\PYG{l+m+mi}{100}\PYG{p}{,}\PYG{l+m+mi}{125}\PYG{p}{,}\PYG{l+m+mi}{150}\PYG{p}{,}\PYG{l+m+mi}{175}\PYG{p}{,}\PYG{l+m+mi}{200}\PYG{p}{]}\PYG{p}{,}
        \PYG{n}{concentration\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{umole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{device\PYGZus{}manufacturer}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{MANUFACTURER}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{device\PYGZus{}model}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{SUPERMODEL}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
        \PYG{n}{spectrum\PYGZus{}reactant\PYGZus{}concentration}\PYG{o}{=}\PYG{l+m+mi}{69}
    \PYG{p}{)}
    \PYG{k}{for} \PYG{n}{pH} \PYG{o+ow}{in} \PYG{n}{result}\PYG{o}{.}\PYG{n}{keys}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{p}{[}\PYG{n}{pH}\PYG{p}{]}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Sort calibration data by ascending pH and temperature}
\PYG{n}{calibration\PYGZus{}data} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{calibration\PYGZus{}data}\PYG{p}{,} \PYG{n}{key} \PYG{o}{=} \PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Generate standard curves for ABTS calibration data}
\PYG{n}{standard\PYGZus{}curves}\PYG{p}{:} \PYG{n}{List}\PYG{p}{[}\PYG{n}{StandardCurve}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{calibration} \PYG{o+ow}{in} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{calibration\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{standard\PYGZus{}curves}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{StandardCurve}\PYG{p}{(}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{=}\PYG{n}{calibration}\PYG{p}{,} \PYG{n}{wavelength}\PYG{o}{=}\PYG{l+m+mi}{340}\PYG{p}{,} \PYG{n}{cutoff\PYGZus{}absorption}\PYG{o}{=}\PYG{l+m+mf}{3.2}\PYG{p}{,} \PYG{n}{show\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Sort standard curves by ascending pH and temperature.}
\PYG{n}{standard\PYGZus{}curves} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{standard\PYGZus{}curves}\PYG{p}{,} \PYG{n}{key} \PYG{o}{=} \PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize all fitted standard curves }
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{13}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{standard}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{standard\PYGZus{}curves}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{30}\PYG{p}{]}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{i}\PYG{o}{\PYGZpc{}}\PYG{k}{5}:
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{standard}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{standard}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{, }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{standard}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{°C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mi}{25}\PYG{p}{,}\PYG{l+m+mi}{26}\PYG{p}{,}\PYG{l+m+mi}{27}\PYG{p}{,}\PYG{l+m+mi}{28}\PYG{p}{,}\PYG{l+m+mi}{29}\PYG{p}{]}\PYG{p}{:}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_6_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The figure above shows the standard curves for all experimental conditions of this scenario. Since the absorption proterties ABTS change with pH, the calibration range differs between some pH values, due to the upper absorption limit. In consequence, the upper calibration limit for reactions at pH 3, pH 3.5, and pH 5.5 is at 150 uM of ABTS, whereas for all other pH values the upper limit is at 175 uM. The sulfonate groups of ABTS are deprotonated for more neutral pH values. Therefore, the absorption properties of ABTS might decrease. Calibration curve data at pH 5.5 showed larger variation between the repeats. In this case, pipetting of one of the the three repeats differs from the other two, which should be considered for kinetic parameter estimation. In contrast to pH, the temperature during calibration affected the calibration curve only marginally.

\sphinxAtStartPar
Nextup, the generate standard curves were used to convert the absorption measurement data into concentration data. Thereby, the respective concentration values were only calculated, if the measured absorption was within the respective calibration bounds.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_output_scroll}
\begin{sphinxuseclass}{tag_hide_cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Calculate concentrations by applying standard curves to \PYGZsq{}EnzymeMLDocuments\PYGZsq{}.}
\PYG{n}{concentration\PYGZus{}enzymemldocs} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{standard\PYGZus{}curve}\PYG{p}{,} \PYG{n}{abso\PYGZus{}enzmldoc} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{standard\PYGZus{}curves}\PYG{p}{,} \PYG{n}{absortion\PYGZus{}enzymemldocs}\PYG{p}{)}\PYG{p}{:}

    \PYG{c+c1}{\PYGZsh{} Validate, that measurement data is paired with respective standard curve}
    \PYG{k}{assert} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH} \PYG{o}{==} \PYG{n}{abso\PYGZus{}enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{ph}
    \PYG{k}{assert} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature} \PYG{o}{==} \PYG{n}{abso\PYGZus{}enzmldoc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{temperature}

    \PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{abso\PYGZus{}enzmldoc}\PYG{p}{)}
    \PYG{n}{conc\PYGZus{}enzmldoc} \PYG{o}{=} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ommit\PYGZus{}nan\PYGZus{}measurements}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
    \PYG{n}{conc\PYGZus{}enzmldoc} \PYG{o}{=} \PYG{n}{standard\PYGZus{}curve}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ommit\PYGZus{}nan\PYGZus{}measurements}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}

    \PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{conc\PYGZus{}enzmldoc}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m8\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
Measurements \PYGZsq{}[\PYGZsq{}m9\PYGZsq{}]\PYGZsq{} removed from document, since respective measurement values are out of calibration range.
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Quality control through mass balance analysis}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:quality-control-through-mass-balance-analysis}}
\sphinxAtStartPar
Since product and substrate of the SLAC reaction were simultaneously recorded, mass balance analysis was conductas as a control of quality. By assuming mass conservation, the following concentration balance can be established:
\begin{equation}\label{equation:scenarios/SLAC_kinetic_characterization:mass_balance}
\begin{split}0 = S_{(t)} + P_{(t)} - S_{0}\end{split}
\end{equation}
\sphinxAtStartPar
Thereby, \(S_{0}\) denotes the initial substrate concentration, whereas \(S_{(t)}\) and \(P_{(t)}\) describe the substrate and product concentration for each time point \(t\). \(S_{t}\) and \(S_{0}\) were individually measured. Thus each enzyme reaction with a given initial substrate concentration had a control reaction with identical substrate concentration. In contrast to the substrate, no calibration standard is available for the product. Therefore, an additional parameter §k§ was introdiced to the mass balance equation, assuming linear relationship between the product concentration and its signal:
\begin{equation}\label{equation:scenarios/SLAC_kinetic_characterization:mass_balance_with_k}
\begin{split}0 = S_{(t)} + P_{(t)}k - S_{0}\end{split}
\end{equation}
\sphinxAtStartPar
\(k\) was determined for each data set individually by a minimization algorithm. The minimization objective was to find the optimal \(k\), which minimizes all slopes of a given reaction condition. Thereafter, the mass balances of all measurements were visualized.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Defenition of parameter \PYGZsq{}k\PYGZsq{}}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{Parameters}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{params}\PYG{o}{.}\PYG{n}{add}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{value}\PYG{o}{=}\PYG{l+m+mi}{30}\PYG{p}{,} \PYG{n+nb}{min}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n+nb}{max}\PYG{o}{=}\PYG{l+m+mi}{200}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Target function for the minimizer}
\PYG{k}{def} \PYG{n+nf}{residual}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{x}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{k}\PYG{o}{=}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

    \PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,}  \PYG{n}{control} \PYG{o}{=} \PYG{n}{x}
    \PYG{n}{slopes} \PYG{o}{=} \PYG{n}{substrate}\PYG{p}{[}\PYG{p}{:}\PYG{p}{,}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{o}{*} \PYG{l+m+mf}{0.0}
    \PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{p}\PYG{p}{,} \PYG{n}{c}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,} \PYG{n}{control}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{sub} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{prod} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{p}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{cont} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{c}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}

        \PYG{n}{model} \PYG{o}{=} \PYG{n}{sub} \PYG{o}{+} \PYG{n}{prod}\PYG{o}{*}\PYG{n}{k} \PYG{o}{\PYGZhy{}} \PYG{n}{cont}

        \PYG{n}{slopes}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]} \PYG{o}{=} \PYG{n}{linregress}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{model}\PYG{p}{)}\PYG{p}{)}\PYG{p}{,} \PYG{n}{model}\PYG{p}{)}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}

    \PYG{k}{return} \PYG{n}{slopes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}


\PYG{n}{f} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.5}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{e}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{substrate} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{product} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{initial\PYGZus{}substrate} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{control} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{k}{for} \PYG{n}{measurement} \PYG{o+ow}{in} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{initial\PYGZus{}substrate}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{)}\PYG{o}{.}\PYG{n}{init\PYGZus{}conc}\PYG{p}{)}
        \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{n}{substrate\PYGZus{}id}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
            \PYG{n}{substrate}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}
        \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{n}{product\PYGZus{}id}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
            \PYG{n}{product}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}
        \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s2}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
            \PYG{n}{control}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}

    \PYG{n}{time} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{time}\PYG{p}{)}

    \PYG{n}{product} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{product}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{p}{(}\PYG{n}{product}\PYG{p}{)}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{3}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{11}\PYG{p}{)}
    \PYG{n}{substrate} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{)}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{3}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{11}\PYG{p}{)}
    \PYG{n}{control} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{control}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n+nb}{len}\PYG{p}{(}\PYG{p}{(}\PYG{n}{control}\PYG{p}{)}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{3}\PYG{p}{)}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{11}\PYG{p}{)}
    \PYG{n}{initial\PYGZus{}substrate} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{initial\PYGZus{}substrate}\PYG{p}{)}

    \PYG{n}{x} \PYG{o}{=} \PYG{p}{(}\PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,} \PYG{n}{control}\PYG{p}{)}

    \PYG{n}{result} \PYG{o}{=} \PYG{n}{minimize}\PYG{p}{(}\PYG{n}{residual}\PYG{p}{,} \PYG{n}{params}\PYG{p}{,} \PYG{n}{args}\PYG{o}{=}\PYG{p}{(}\PYG{n}{x}\PYG{p}{,}\PYG{p}{)}\PYG{p}{)}

    \PYG{n}{factor} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
    \PYG{n}{f}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{factor}\PYG{p}{)}

    \PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{p}\PYG{p}{,} \PYG{n}{c}\PYG{p}{,} \PYG{n}{l}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{substrate}\PYG{p}{,} \PYG{n}{product}\PYG{p}{,}  \PYG{n}{control}\PYG{p}{,} \PYG{n}{initial\PYGZus{}substrate}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{init} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{c}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{sub} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{s}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{prod} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{p}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{cont} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{c}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
        \PYG{n}{balance} \PYG{o}{=} \PYG{p}{(}\PYG{n}{sub} \PYG{o}{+} \PYG{n}{prod}\PYG{o}{*}\PYG{n}{factor} \PYG{o}{\PYGZhy{}} \PYG{n}{cont}\PYG{p}{)}

        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{doc}\PYG{o}{.}\PYG{n}{getMeasurement}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{m1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{time}\PYG{p}{,} \PYG{n}{balance}\PYG{p}{,} \PYG{n}{label} \PYG{o}{=} \PYG{n}{l}\PYG{p}{)}
        \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{i}\PYG{o}{\PYGZpc{}}\PYG{k}{5}:
            \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mass balance [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{k}{if} \PYG{n}{i} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mi}{24}\PYG{p}{,} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{l+m+mi}{26}\PYG{p}{,} \PYG{l+m+mi}{27}\PYG{p}{,} \PYG{l+m+mi}{28}\PYG{p}{,} \PYG{l+m+mi}{29}\PYG{p}{]}\PYG{p}{:}
            \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{pH} \PYG{o}{=} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{ph}
        \PYG{n}{temp} \PYG{o}{=} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{temperature}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{, }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{temp}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{°C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{c+c1}{\PYGZsh{}, factor: \PYGZob{}factor:.0f\PYGZcb{}\PYGZdq{})}
    \PYG{k}{if} \PYG{n}{e} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
        \PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{)}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.03}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_10_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The figure above shows the mass balances for all experiments. For all reactions, except for pH 4, 45°C and pH 4.5, 35°C, the product and substrate is in balance over the reaction time\sphinxhyphen{}course. Mass balances, diverging from 0 uM indicate, result likely from pipetting errors, eighter increasing or decreasing the substrate concentration in the enzyme\sphinxhyphen{} or control reaction. In the reactions at pH 4, 45°C and pH 4.5, 35°C the mass balance slopes are not linear. This indicates issues with the measurement, presumably from the spectro photometer. Therefore, the respective measurements might result in wrong parameter estimations.


\subsection{Kinetic parameter estimation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:kinetic-parameter-estimation}}

\subsubsection{Choice of kinetic model}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:choice-of-kinetic-model}}
\sphinxAtStartPar
Model selection is vital for parameter estimation. Thus, different settings for the parameter estimator were tested. Firstly, substrate and product inhibition models were excluded, since in some of the experiments the actual initial substrate concentration was higher than the specified initial substrate concentration. This results calculated product concentrations with negative value. Hence, product inhibition models were excluede, since no acurate information on product concentration was available.Secondly, substrate inhibition models were excluded, since the model was not able to describe the observed reaction kinetics. This was evident from a higher Akaike information criterion as well as more than 100 \% standard deviation on the estimated parameters.Lastly, the irreversible Michaelis\sphinxhyphen{}Menten model with and without time\sphinxhyphen{}dependent enzyme inactivation was compared, since enzyme inactivation was observed in previous experiments. Estimated parameters and the fitted models are shown for the experimental data at pH 3 and 25°C. Once without enzyme inactivation and once with.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mi}{10}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{n}{ax} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{if} \PYG{n}{i} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
        \PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Estimated parameters without time\PYGZhy{}dependent enzyme inactivation:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{without enzyme inactivation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Estimated parameters with time\PYGZhy{}dependent enzyme inactivation:}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{with enzyme inactivation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{)}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.09}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
Estimated parameters without time\PYGZhy{}dependent enzyme inactivation:
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                               AIC    Km [umole / l]  \PYGZbs{}
irreversible Michaelis Menten \PYGZhy{}162  35.473 +/\PYGZhy{} 5.52\PYGZpc{}   

                              kcat / Km [1/s * 1/umole / l]       kcat [1/s]  
irreversible Michaelis Menten               0.009 +/\PYGZhy{} 5.89\PYGZpc{}  0.308 +/\PYGZhy{} 2.07\PYGZpc{}  
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Estimated parameters with time\PYGZhy{}dependent enzyme inactivation:
Fitting data to:
 \PYGZhy{} irreversible Michaelis Menten model
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
                               AIC    Km [umole / l]  \PYGZbs{}
irreversible Michaelis Menten \PYGZhy{}476  35.137 +/\PYGZhy{} 3.00\PYGZpc{}   

                              kcat / Km [1/s * 1/umole / l]       kcat [1/s]  \PYGZbs{}
irreversible Michaelis Menten               0.013 +/\PYGZhy{} 3.58\PYGZpc{}  0.460 +/\PYGZhy{} 1.95\PYGZpc{}   

                              ki time\PYGZhy{}dep enzyme\PYGZhy{}inactiv. [1/s]  
irreversible Michaelis Menten                   0.001 +/\PYGZhy{} 4.47\PYGZpc{}  
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_13_4}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
Both plots in the above figure contain the same experimental data. The left plot shows the fitted model with enzyme inactivation and the right without enzyme inactivation, which is shown as solid lines. Visually, the model with enzyme inactinvation describes the data better, since the model intersects the data points more directly, compared to the model without enzyme inactivation. Statistically, this is characterized through a lower Akaike information criterion, as well as lower standard deviations on the estimated parameters.
In absolute therms, the estimated \(K_{m}\) of both models is approximately identical, whereas the \(k_{cat}\) estimate is approximately 33\% lower for the model without enzyme inactivation. Hence, the model without enzyme inactivation underestimates the turnover number of the enzyme.

\sphinxAtStartPar
Thus, irreversible Michaelis\sphinxhyphen{}Menten model with time\sphinxhyphen{}dependent enzyme inactivation was selected for the parameter estimation for all data sets.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Run parameter estimator for all datasets, utilizing multi\PYGZhy{}processing.}
\PYG{k}{def} \PYG{n+nf}{run\PYGZus{}ParameterEstimator}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{:} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{)}\PYG{p}{:}    
    \PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
    \PYG{k}{return} \PYG{n}{kinetics}

\PYG{n}{results} \PYG{o}{=} \PYG{n}{Parallel}\PYG{p}{(}\PYG{n}{n\PYGZus{}jobs}\PYG{o}{=}\PYG{l+m+mi}{8}\PYG{p}{)}\PYG{p}{(}\PYG{n}{delayed}\PYG{p}{(}\PYG{n}{run\PYGZus{}ParameterEstimator}\PYG{p}{)}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{)} \PYG{k}{for} \PYG{n}{enzmldoc} \PYG{o+ow}{in} \PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{)}
\PYG{n}{results} \PYG{o}{=} \PYG{n+nb}{sorted}\PYG{p}{(}\PYG{n}{results}\PYG{p}{,} \PYG{n}{key}\PYG{o}{=}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{p}{(}\PYG{n}{x}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{,} \PYG{n}{x}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize all fitted models}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.5}\PYG{p}{,} \PYG{l+m+mi}{15}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{ax}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{results}\PYG{p}{,} \PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{ph} \PYG{o}{=} \PYG{n}{doc}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}
    \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{i}\PYG{o}{\PYGZpc{}}\PYG{k}{5}:
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{doc}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{,} \PYG{n}{title} \PYG{o}{=} \PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{doc}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{, }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{doc}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{temperature}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{°C}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mi}{25}\PYG{p}{,}\PYG{l+m+mi}{26}\PYG{p}{,}\PYG{l+m+mi}{27}\PYG{p}{,}\PYG{l+m+mi}{28}\PYG{p}{,}\PYG{l+m+mi}{29}\PYG{p}{]}\PYG{p}{:}
        \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{k}{if} \PYG{n}{i} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
        \PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels} \PYG{o}{=} \PYG{n}{ax}\PYG{o}{.}\PYG{n}{get\PYGZus{}legend\PYGZus{}handles\PYGZus{}labels}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{fig}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{handles}\PYG{p}{,} \PYG{n}{labels}\PYG{p}{,} \PYG{n}{loc}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{lower center}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ncol}\PYG{o}{=}\PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{labels}\PYG{p}{)}\PYG{p}{,} \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{bbox\PYGZus{}to\PYGZus{}anchor}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{0.5}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.04}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_15_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The above plot shows all kinetic experiments, fitted to the irreversible Michaelis\sphinxhyphen{}Menten model with time\sphinxhyphen{}dependent enzyme inactivation. Based on the reaction slopes, no catalytic avtivity is observable for reactions at pH 5 or higher.
In the following cells, the resulting kinetic parameters were extracted and visualized. As in the mass balance analysis, the reactions at pH 4, 45°C and pH 4.5, 35°C differ from other experiments of the respective pH. In both cases, every applied substrate concentration is higher than intended in the design of the experiment. Furthermore, the resulting parameter estimates have a high uncertanty. Therefore, the respective measurements are excluded from further analysis. Additionaly, the results from pH 5.5, 35°C were excluded, since the uncertainty of the parameters could not be estimated.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_cell}
\begin{sphinxuseclass}{tag_output_croll}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Extract kinetic parameters of all datasets}
\PYG{n}{kcat} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{pH} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{temperature} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{corr\PYGZus{}kcat\PYGZus{}km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{result} \PYG{o+ow}{in} \PYG{n}{results}\PYG{p}{:}
    \PYG{n}{params} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}parameter\PYGZus{}dict}\PYG{p}{(}\PYG{p}{)}

    \PYG{n}{kcat}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{kcat\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{Km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{Km\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{ki}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{ki\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}  

    \PYG{n}{correlation} \PYG{o}{=} \PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{correl}
    \PYG{k}{if} \PYG{n}{correlation} \PYG{o}{==} \PYG{k+kc}{None}\PYG{p}{:}
        \PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n+nb}{float}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{nan}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{correlation}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}

    \PYG{n}{pH}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{pH}\PYG{p}{)}
    \PYG{n}{temperature}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Organize kinetic and experimantal parameters in a \PYGZsq{}DataFrame\PYGZsq{}}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{o}{.}\PYG{n}{from\PYGZus{}dict}\PYG{p}{(}\PYG{p}{\PYGZob{}}
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pH}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{pH}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature [C]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{temperature}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat [1/s]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km [uM]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki}\PYG{p}{,}
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki\PYGZus{}std}\PYG{p}{,}
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{correlation kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{corr\PYGZus{}kcat\PYGZus{}km}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{/} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr} \PYG{o}{=}\PYG{p}{(}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{o}{+}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mf}{0.5} \PYG{o}{*} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr}

\PYG{c+c1}{\PYGZsh{} Exclued results from failed experiments}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{25}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} reached parameter boundaries \PYGZhy{}\PYGZhy{}\PYGZgt{} inactive}
\PYG{c+c1}{\PYGZsh{}df = df.drop(index=12) \PYGZsh{}Km really high stddev}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{17}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{}kcat outlier, mass balance outlier}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{14}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{}kcat outlier, mass balance outlier}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{zeros}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{zeros}\PYG{p}{(}\PYG{l+m+mi}{27}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Calculate halflife of enzyme}
\PYG{k}{def} \PYG{n+nf}{calulate\PYGZus{}halflife}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{return} \PYG{n}{np}\PYG{o}{.}\PYG{n}{log}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{/}\PYG{n}{x}

\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{apply}\PYG{p}{(}\PYG{n}{calulate\PYGZus{}halflife}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{60}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{/} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{*} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{c+c1}{\PYGZsh{} Delete inactivation rates}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
     pH  temperature [C]  kcat [1/s]  kcat stderr     Km [uM]  Km stderr  \PYGZbs{}
0   3.0             25.0    0.459962     0.008966   35.137357   1.053968   
1   3.0             30.0    0.470366     0.009855   32.114083   1.037088   
2   3.0             35.0    0.546029     0.016598   37.454496   1.765891   
3   3.0             40.0    0.712488     0.021754   46.441229   2.155285   
4   3.0             45.0    0.815724     0.022451   47.967746   2.047138   
5   3.5             25.0    0.420480     0.045642   89.752447  13.848584   
6   3.5             30.0    0.399649     0.007455   61.224383   1.674235   
7   3.5             35.0    0.246773     0.009247   28.237534   1.572755   
8   3.5             40.0    0.650031     0.012397   79.432405   2.189303   
9   3.5             45.0    0.978983     0.026556  103.784356   4.040850   
10  4.0             25.0    0.115133     0.008752   65.852866   2.476592   
11  4.0             30.0    0.192374     0.020265   95.314777   5.569713   
12  4.0             35.0    0.282991     0.029279  100.647793  11.576143   
13  4.0             40.0    0.225429     0.012766   97.966631   1.677116   
15  4.5             25.0    0.368730     0.025801  110.902372   0.830267   
16  4.5             30.0    0.486402     0.036084  148.193267   3.440830   
18  4.5             40.0    0.549232     0.021351  178.481651   2.544896   
19  4.5             45.0    0.235716     0.013338   64.718239   0.037679   
20  5.0             25.0    0.008085     0.006871   15.067578  20.447918   
21  5.0             30.0    0.040665     0.005659   50.772187   3.784251   
22  5.0             35.0    0.061677     0.006158   68.874565   0.267258   
23  5.0             40.0    0.131196     0.017131   91.314407   2.756719   
24  5.0             45.0    0.129399     0.014577   96.735544   2.562607   
26  5.5             30.0    0.014374     0.007402   96.125085   0.393705   
27  5.5             35.0    0.124370     0.023968   81.996103  22.257271   
28  5.5             40.0    0.055892     0.016036   48.740460  19.833207   
29  5.5             45.0    0.075249     0.015052  115.390172  27.541849   

    correlation kcat/Km  kcat/Km [1/s * uM]  kcat/Km stderr  zeros  \PYGZbs{}
0              0.535141            0.013090        0.000468    0.0   
1              0.522963            0.014647        0.000564    0.0   
2              0.558865            0.014578        0.000818    0.0   
3              0.608947            0.015342        0.000852    0.0   
4              0.623270            0.017006        0.000864    0.0   
5              0.746575            0.004685        0.000884    0.0   
6              0.666682            0.006528        0.000216    0.0   
7              0.554406            0.008739        0.000587    0.0   
8              0.731830            0.008183        0.000274    0.0   
9              0.774126            0.009433        0.000448    0.0   
10             0.247717            0.001748        0.000148    0.0   
11            \PYGZhy{}0.718208            0.002018        0.000243    0.0   
12             0.608286            0.002812        0.000435    0.0   
13             0.168743            0.002301        0.000136    0.0   
15            \PYGZhy{}0.736267            0.003325        0.000234    0.0   
16             0.816213            0.003282        0.000255    0.0   
18             0.084106            0.003077        0.000127    0.0   
19            \PYGZhy{}0.113239            0.003642        0.000206    0.0   
20             0.276523            0.000537        0.000859    0.0   
21             0.009319            0.000801        0.000126    0.0   
22             0.299400            0.000895        0.000089    0.0   
23            \PYGZhy{}0.426692            0.001437        0.000193    0.0   
24            \PYGZhy{}0.696719            0.001338        0.000155    0.0   
26            \PYGZhy{}0.200038            0.000150        0.000077    0.0   
27             0.771045            0.001517        0.000505    0.0   
28             0.552798            0.001147        0.000571    0.0   
29             0.604743            0.000652        0.000203    0.0   

    half life [min]  half life std  
0          9.139562       0.408568  
1          9.522170       0.476531  
2         10.304462       0.776312  
3          9.127765       0.600883  
4          9.087343       0.530573  
5          8.824513       1.680327  
6         14.545045       0.804521  
7         18.269277       2.766875  
8         11.026573       0.453021  
9          9.110325       0.425387  
10       111.702355     222.064200  
11        23.819987      10.874634  
12        14.929497       4.974473  
13        29.441552      12.288914  
15        22.536023       6.247470  
16        19.046651       4.109261  
18        34.946634      11.931005  
19       108.053681     160.970528  
20       115.522134    2593.086797  
21       115.512956     453.016208  
22       115.519891     286.705892  
23         8.399863       2.519856  
24        11.438272       2.986966  
26       115.523248    1471.972292  
27         5.841192       1.400781  
28         6.132460       2.957303  
29        10.834329       5.343971  
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{High correlation between \protect\(k_{cat}\protect\) and \protect\(K_{m}\protect\)}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:high-correlation-between-k-cat-and-k-m}}
\sphinxAtStartPar
The parameter estimates for \(k_{cat}\) and \(K_{m}\) showed correlation. For experiments at pH 3 and pH 3.5, the parameters were correlated between 0.5 \sphinxhyphen{} 0.85. This indeicates, that the highest initial substrate concentration, which was applied for parameter estimation was not sufficently high. Ideally, the highest initial substrate concentration applied for a kinetic experiment should be 10\sphinxhyphen{}fold higher than \(K_{m}\) to allow independent estimates for \(k_{cat}\) and \(K_{m}\). In this scenario, only reactions with an initial substrate concentration of 150 uM were used for parameter estimation, due to the limited photometric measurement range of ABTS. Hence, ABTS was only applied 1.5\sphinxhyphen{}fold to 6\sphinxhyphen{}fold of the estimated \(K_{m}\).For reactions at pH values of 4 and above, positive as well as negative correlations were observed.
As the figure above on the right shows, \(K_{m}\) was estimated in the range of 25 \sphinxhyphen{} 105 uM in the mentioned pH range.
Low and also negative correlations were observed for reactions with pH values at and above pH 4.

\sphinxAtStartPar
The figure on the left shows \(k_{cat}\) over \(K_{m}\) for all experimental conditions, whereas the pH is color\sphinxhyphen{}coded. For reactions with identical pH values, \(k_{cat}\) and \(K_{m}\) both increase with temperatures. This might be attributed to the high correleation between the parameters. Thus, the true change of \(k_{cat}\) or \(K_{m}\) through pH or temperature cannot be assessed. Therefore,

\sphinxAtStartPar
For pH values of 5 and above, almost no catalytic activity was observed. Therefore, data of these experimental conditions is excluded from further analysis.


\subsubsection{Parameter estimates for \protect\(k_{cat}\protect\) and \protect\(K_{m}\protect\)}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:parameter-estimates-for-k-cat-and-k-m}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Visualize estimated parameters}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,} \PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{n}{ax} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{axes}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{if} \PYG{n}{i}\PYG{o}{==}\PYG{l+m+mi}{1}\PYG{p}{:}
        \PYG{n}{plot}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{xdata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ydata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{xerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{colors}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{)}
    \PYG{k}{else}\PYG{p}{:}
        \PYG{n}{plot}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{xdata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ydata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{xerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{zeros}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{colors}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature [C]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ax}\PYG{o}{=}\PYG{n}{ax}\PYG{p}{)}
    
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_19_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The figure on the left shows the catalytic efficiency \(\frac{k_{cat}}{K_{m}}\) over the pH value of each reaction. The highest catalytic efficiency was observed at pH 3 and 45°C. Increasing the pH by 0.5 reduced \(\frac{k_{cat}}{K_{m}}\) approximatel by half. For higher pH values, \(\frac{k_{cat}}{K_{m}}\) is even more decreased. The catalytic efficiency is therefore highly sensitive to the pH. This might source from different causes. Eighter deprotonation ABTS sulfonate groups for more neutral pH values or protonation changes of the enzyme might hinder catalysis at higher pH values.In terms of temperature, higher \(\frac{k_{cat}}{K_{m}}\) was achieved at higher temperatures. SLAC might even be more active at higher temperatures and pH values.


\subsubsection{Time\sphinxhyphen{}dependent enzyme inactivation}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:time-dependent-enzyme-inactivation}}
\sphinxAtStartPar
The time\sphinxhyphen{}dependent enzyme inactivaten rate \(k_{inact}\), was estimated for each experiment, since the irreversible Michaels\sphinxhyphen{}Menten model did not describe the observed time\sphinxhyphen{}course data accurately. The halfife of the enzyme was calculated as shown in equation \eqref{equation:scenarios/SLAC_kinetic_characterization:enzyme_halflife} {[}\hyperlink{cite.references:id8}{Buxbaum, 2015}{]}.
\begin{equation}\label{equation:scenarios/SLAC_kinetic_characterization:enzyme_halflife}
\begin{split}t_{\frac{1}{2}} = \frac{ln(2)}{k_{inact}}\end{split}
\end{equation}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Exclued inactive enzyme reactions above pH 4.5 and discard measurements with more than 100 \PYGZpc{} standard deviation on enzyme inactivation parameter}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{:}\PYG{l+m+mi}{19}\PYG{p}{]}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{10}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} more than 100 \PYGZpc{} standard deviation on enzyme inactivation parameter}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{index}\PYG{o}{=}\PYG{l+m+mi}{19}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} more than 100 \PYGZpc{} standard deviation on enzyme inactivation parameter}

\PYG{c+c1}{\PYGZsh{} Plot half life over catalytic efficiency}
\PYG{n}{plot}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{xdata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ydata}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{xerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yerror}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{colors}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{SLAC_kinetic_characterization_21_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The enzyme’s calculated half life was between 8 \sphinxhyphen{} 18 min for reactions at pH 3 and pH 3.5, whereas enzyme at higher pH values showed a half life between 15 \sphinxhyphen{} 35 min. Therefore, enzyme at reaction conditions with higher catalytic efficieny showed generally a shorter half life compared to enzyme at reaction conditions with lower catalytic efficiency.In order to check for corrections between \(k_{inact}\), \(k_{cat}\), \(K_{m}\), \(\frac{k_{cat}}{K_{m}}\), as well as the reaction temperature and pH, a correlation analysis was conducted.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Calculate correlations}
\PYG{n}{df\PYGZus{}corr} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{correlation kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{zeros}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{half life std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{rho} \PYG{o}{=} \PYG{n}{df\PYGZus{}corr}\PYG{o}{.}\PYG{n}{corr}\PYG{p}{(}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize correlation heatmap}
\PYG{c+c1}{\PYGZsh{}sns.heatmap(rho, }
\PYG{c+c1}{\PYGZsh{}        xticklabels=rho.columns,}
\PYG{c+c1}{\PYGZsh{}        yticklabels=rho.columns,}
\PYG{c+c1}{\PYGZsh{}        cmap = sns.color\PYGZus{}palette(\PYGZdq{}vlag\PYGZdq{}, as\PYGZus{}cmap=True))}
\PYG{c+c1}{\PYGZsh{}plt.show()}

\PYG{c+c1}{\PYGZsh{} Calculate confidence intercal for correlations}
\PYG{n}{pval} \PYG{o}{=} \PYG{n}{df\PYGZus{}corr}\PYG{o}{.}\PYG{n}{corr}\PYG{p}{(}\PYG{n}{method}\PYG{o}{=}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{,} \PYG{n}{y}\PYG{p}{:} \PYG{n}{pearsonr}\PYG{p}{(}\PYG{n}{x}\PYG{p}{,} \PYG{n}{y}\PYG{p}{)}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)} \PYG{o}{\PYGZhy{}} \PYG{n}{np}\PYG{o}{.}\PYG{n}{eye}\PYG{p}{(}\PYG{o}{*}\PYG{n}{rho}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{)}
\PYG{n}{p} \PYG{o}{=} \PYG{n}{pval}\PYG{o}{.}\PYG{n}{applymap}\PYG{p}{(}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{.}\PYG{n}{join}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{*}\PYG{l+s+s1}{\PYGZsq{}} \PYG{k}{for} \PYG{n}{t} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mf}{.05}\PYG{p}{,} \PYG{l+m+mf}{.01}\PYG{p}{,} \PYG{l+m+mf}{.001}\PYG{p}{]} \PYG{k}{if} \PYG{n}{x}\PYG{o}{\PYGZlt{}}\PYG{o}{=}\PYG{n}{t}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{rho}\PYG{o}{.}\PYG{n}{round}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{.}\PYG{n}{astype}\PYG{p}{(}\PYG{n+nb}{str}\PYG{p}{)} \PYG{o}{+} \PYG{n}{p}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
                          pH temperature [C] kcat [1/s]   Km [uM]  \PYGZbs{}
pH                    1.0***           \PYGZhy{}0.15      \PYGZhy{}0.39   0.88***   
temperature [C]        \PYGZhy{}0.15          1.0***       0.6*      0.07   
kcat [1/s]             \PYGZhy{}0.39            0.6*     1.0***     \PYGZhy{}0.03   
Km [uM]              0.88***            0.07      \PYGZhy{}0.03    1.0***   
kcat/Km [1/s * uM]  \PYGZhy{}0.89***            0.27      0.58*  \PYGZhy{}0.75***   
half life [min]       0.8***            0.03      \PYGZhy{}0.49    0.68**   

                   kcat/Km [1/s * uM] half life [min]  
pH                           \PYGZhy{}0.89***          0.8***  
temperature [C]                  0.27            0.03  
kcat [1/s]                      0.58*           \PYGZhy{}0.49  
Km [uM]                      \PYGZhy{}0.75***          0.68**  
kcat/Km [1/s * uM]             1.0***         \PYGZhy{}0.71**  
half life [min]               \PYGZhy{}0.71**          1.0***  
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
The above table shows correlation between between kinetic parameters and experimental conditions. The half life of the enzyme is significantly (p < 0.05) correlated to \(\frac{k_{cat}}{K_{m}}\) as well as to pH. Due to correlation between \(k_{cat}\) and \(K_{m}\), the enzyme’s half life is also correlated to \(K_{m}\). Multiple reasons might be the reason for this observation. On the one hand, the observed correlation might be of technical origine. Hence the system of ordinary differential equations, describing the change in substrate and enzyme concentration, are not an accurate model for the reaction system. Therefore, high cross\sphinxhyphen{}correlation occures, since observations cannot be attributed to single parameter. On the other hand, enzyme activity and enzyme inactivation might might be causually related. In this case, the formed ABTS radical might in activate the enzyme by potential suicide inhibition, or the enzyme deteriorates through catalysis.


\subsection{Conclusion}
\label{\detokenize{scenarios/SLAC_kinetic_characterization:conclusion}}
\sphinxAtStartPar
\sphinxstylestrong{Data management}The established data pipeline allows scalable data preparation and analysis of enzyme kinetics experiments. In futer experiments this workflow could be used to further expand the experimental parameter space. As a result, different enzymes with different substrates in diffrent buffers at different temperatures and pH values could be analyzed.

\sphinxAtStartPar
\sphinxstylestrong{Correlation between parameters}The highest catalytic efficieny of SLAC was observed at pH 3 at 45°C and therefore on the edge of the investigated prameter space. Optimal reaction conditions might therefore be at an even lower pH value and higher temperature. \(k_{cat}\) and \(K_{m}\) were correlated, which is likely the result of a too low initial substrate concentration in relation to the true \(K_{m}\) of the enzyme at a given experimental condition. Hence, in future experiments the product signal could be used for parameter estimation, since the concentration to absorbance ratio is lower compared to the product. However, a reliable method for product quantification would need to be established first.

\sphinxAtStartPar
\sphinxstylestrong{Enzyme inactivation}SLAC showed a low half life of approximately 10 min at reaction conditions with the highest catalytic efficiency. Potential reasons for enzyme inactivation are discussed in the next chapter.

\sphinxstepscope


\section{\sphinxstylestrong{Scenario D:}Time\sphinxhyphen{}dependent enzyme inactivation}
\label{\detokenize{scenarios/enzyme_inactivation:scenario-d-br-time-dependent-enzyme-inactivation}}\label{\detokenize{scenarios/enzyme_inactivation::doc}}
\sphinxAtStartPar
Data provided by Paulo Durão (Microbial \& Enzyme Technology, Instituto de Tecnologia Química e Biológica, Oeiras, Portugal)


\subsection{Project background}
\label{\detokenize{scenarios/enzyme_inactivation:project-background}}
\sphinxAtStartPar
All investigated enzyme reaction without inhibitor showed progress curves behavior, which was not explainable by irreversible Michaelis\sphinxhyphen{}Menten kinetics. SUBSTRATE PRODUCT INHIBITION. All experiments had in commmon, that enzyme reactions were carried out in 96\sphinxhyphen{}well polystyrene micro titer plates (MTP), whereas the change in substrate and or product absorption was monitored photometrically.
One hypothesis for the observed time\sphinxhyphen{}dependent decrease of enzyme activity is potential hydrophobic interaction between the enzyme and the MTP suface. Thereby, hydrophobic regions of the enzyme’s surface might interact with the hydrophobic reaction vessle, ultimately preventing substrate access to the active site of the enzyme.In order to test the hypothesis, an absorption experiment in which enzyme was incubated in MTP wells prior to reaction start was performed. Thereby, the enzyme activity should decrease with regard to the prior incubation time. If the hypothesis is correct, the calculated half life from the adsorption experiment
should match with the half life of an enzyme kinetics experiment which was conducted in parallel.

\sphinxAtStartPar
For this experiment Co

\sphinxAtStartPar
\sphinxstylestrong{Show results of enzyme inactication across projects} \#TODO

\sphinxAtStartPar
All investigated enzyme reactions in this thesis showed a time\sphinxhyphen{}dependent decrease in catalytic activity, which was not explainable by the irreversible Michaelis\sphinxhyphen{}Menten model.


\subsubsection{Experimental design}
\label{\detokenize{scenarios/enzyme_inactivation:experimental-design}}
\sphinxAtStartPar
CotA laccase from \sphinxstyleemphasis{Bacillus subtilis}
Both experiments wer conducted with CotA from

\sphinxAtStartPar
\sphinxstylestrong{Determination of enzyme inactivation trough adsorption}In order to test the hypothesis of time\sphinxhyphen{}dependent enzyme inactivation through adsorption to the MTP surface, the following experiment was conducted. Thereby, enzyme solution was incubated in individual MTP wells up to 1 h, prior to reaction start.  Then, individual enzyme reactions were started in triplicates by transferring 2 µL of incubated enzyme in 10 min increments. Each proceeding enzyme reaction contained 256 nM CotA, 1 mM ABTS and was buffered in acetate buffer at pH 4. Product formation was followed photometrically at 420 nm and 25°C for 5 min, whereas concentrations were calculated assuming an extincion coefficient of ε = 36000 M\sphinxhyphen{}1cm\sphinxhyphen{}1 for the ABTS radical product.

\sphinxAtStartPar
\sphinxstylestrong{Enzyme kinetics experiment}Enzymatic oxidation of ABTS to its radical form was followed photometrically at 420 nm at 25°C for 70 min.
Thereby, ABTS was applied in a range from 0.01 mM \sphinxhyphen{} 2 mM. Each proceeding enzyme reaction contained 256 nM CotA, and was buffered in acetate buffer at pH 4.
Repeats of 4.


\subsubsection{Experimantal data}
\label{\detokenize{scenarios/enzyme_inactivation:experimantal-data}}
\sphinxAtStartPar
Experimantal data was provided as an Excel file, containing time\sphinxhyphen{}course absorption data. Meta data was written into the EnzymeML Excel template. Then, the experimantal data was written to an EnzymeML document by a parser function. Concentrations were calculated via the provided extinction coefficient of the ABTS radical (ε = 36000 M\sphinxhyphen{}1cm\sphinxhyphen{}1).


\subsection{Adsorption of enzyme to micro titer plate surface}
\label{\detokenize{scenarios/enzyme_inactivation:adsorption-of-enzyme-to-micro-titer-plate-surface}}

\subsubsection{Imports}
\label{\detokenize{scenarios/enzyme_inactivation:imports}}
\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{from} \PYG{n+nn}{scipy}\PYG{n+nn}{.}\PYG{n+nn}{stats} \PYG{k+kn}{import} \PYG{n}{linregress}
\PYG{k+kn}{import} \PYG{n+nn}{pyenzyme} \PYG{k}{as} \PYG{n+nn}{pe}
\PYG{k+kn}{from} \PYG{n+nn}{EnzymePynetics}\PYG{n+nn}{.}\PYG{n+nn}{tools}\PYG{n+nn}{.}\PYG{n+nn}{parameterestimator} \PYG{k+kn}{import} \PYG{n}{ParameterEstimator}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{cm} \PYG{k}{as} \PYG{n+nn}{cm}
\PYG{k+kn}{from} \PYG{n+nn}{IPython}\PYG{n+nn}{.}\PYG{n+nn}{display} \PYG{k+kn}{import} \PYG{n}{display}


\PYG{c+c1}{\PYGZsh{}TODO warning in EnzymePynetics}

\PYG{k+kn}{import} \PYG{n+nn}{warnings}
\PYG{n}{warnings}\PYG{o}{.}\PYG{n}{filterwarnings}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ignore}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsubsection{Experimantal data}
\label{\detokenize{scenarios/enzyme_inactivation:id1}}
\sphinxAtStartPar
Experimental data was provided as an Excel file. Data was loaded into an pandas DataFrame and the slopes of each incubation condition were calculated through linear regression. The resulting initial rates were used to calculate by which the enzyme activity decreases in regard to the incubation time tin the reaction vessel. Lastly the half life of the enzyme was calculated with equation \eqref{equation:scenarios/SLAC_kinetic_characterization:enzyme_halflife}.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load excel}
\PYG{n}{path} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{../../data/enzyme\PYGZus{}inactivation/Slide 2 \PYGZhy{} Activity effect of incubating CotA in MTP.xlsx}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{n}{path}\PYG{p}{,} \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{time (min)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} replace values of \PYGZsq{}0**\PYGZsq{} with nan\PYGZhy{}values, since the measurement is incorrect}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{0**}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{nan}



\PYG{c+c1}{\PYGZsh{} Get data from Excel file}
\PYG{n}{columns} \PYG{o}{=} \PYG{p}{[}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{df}\PYG{o}{.}\PYG{n}{columns}\PYG{p}{)} \PYG{k}{if} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)}\PYG{o}{.}\PYG{n}{endswith}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{]}
\PYG{n}{time} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{index}\PYG{o}{.}\PYG{n}{values}
\PYG{n}{absorption} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{7}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{22}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Calculate concentrations}
\PYG{n}{extinction\PYGZus{}coefficient} \PYG{o}{=} \PYG{l+m+mi}{36} \PYG{c+c1}{\PYGZsh{} (1/mM * 1/cm)}
\PYG{n}{optical\PYGZus{}length} \PYG{o}{=} \PYG{l+m+mf}{0.65} \PYG{c+c1}{\PYGZsh{} cm}

\PYG{k}{def} \PYG{n+nf}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{abso}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{return} \PYG{n}{abso} \PYG{o}{/} \PYG{p}{(}\PYG{n}{extinction\PYGZus{}coefficient}\PYG{o}{*}\PYG{n}{optical\PYGZus{}length}\PYG{p}{)}

\PYG{n}{concentration} \PYG{o}{=} \PYG{n}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{absorption}\PYG{p}{)}
\PYG{n}{concentration\PYGZus{}mean} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{nanmean}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{concentration\PYGZus{}std} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{nanstd}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Linear regression}
\PYG{n}{slopes} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{intercepts} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{stderrs} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{time\PYGZus{}set} \PYG{o+ow}{in} \PYG{n}{concentration}\PYG{p}{:}
    \PYG{n}{mask} \PYG{o}{=} \PYG{o}{\PYGZti{}}\PYG{n}{np}\PYG{o}{.}\PYG{n}{isnan}\PYG{p}{(}\PYG{n}{time\PYGZus{}set}\PYG{p}{)}
    \PYG{n}{time\PYGZus{}regression} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{tile}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{)}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{n}{mask}\PYG{o}{.}\PYG{n}{shape}\PYG{p}{)}\PYG{p}{[}\PYG{n}{mask}\PYG{p}{]}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{data\PYGZus{}regression} \PYG{o}{=} \PYG{n}{time\PYGZus{}set}\PYG{p}{[}\PYG{n}{mask}\PYG{p}{]}\PYG{o}{.}\PYG{n}{flatten}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{slope}\PYG{p}{,}\PYG{n}{intercept}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{stderr} \PYG{o}{=} \PYG{p}{(}\PYG{n}{linregress}\PYG{p}{(}\PYG{n}{time\PYGZus{}regression}\PYG{p}{,} \PYG{n}{data\PYGZus{}regression}\PYG{p}{)}\PYG{p}{)}
    \PYG{n}{slopes}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{slope}\PYG{p}{)}
    \PYG{n}{intercepts}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{intercept}\PYG{p}{)}
    \PYG{n}{stderrs}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{stderr}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Regression of the slopes between 10 and 60 mincolumns}
\PYG{n}{slope\PYGZus{}of\PYGZus{}slopes}\PYG{p}{,} \PYG{n}{intercept\PYGZus{}of\PYGZus{}slopes}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{\PYGZus{}}\PYG{p}{,}\PYG{n}{\PYGZus{}} \PYG{o}{=} \PYG{n}{linregress}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{,} \PYG{n}{slopes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Half life in hours}
\PYG{n}{t12} \PYG{o}{=} \PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{log}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{/}\PYG{o}{\PYGZhy{}}\PYG{n}{slope\PYGZus{}of\PYGZus{}slopes}\PYG{p}{)}\PYG{o}{/}\PYG{l+m+mi}{60}

\PYG{c+c1}{\PYGZsh{} Plot results}
\PYG{n}{colors} \PYG{o}{=} \PYG{n}{cm}\PYG{o}{.}\PYG{n}{tab10}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{linspace}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{absorption}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}

\PYG{k}{for} \PYG{n}{mean}\PYG{p}{,} \PYG{n}{std}\PYG{p}{,} \PYG{n}{label}\PYG{p}{,} \PYG{n}{slope}\PYG{p}{,} \PYG{n}{intercept}\PYG{p}{,} \PYG{n}{color} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{concentration\PYGZus{}mean}\PYG{p}{,} \PYG{n}{concentration\PYGZus{}std}\PYG{p}{,} \PYG{n}{columns}\PYG{p}{,} \PYG{n}{slopes}\PYG{p}{,} \PYG{n}{intercepts}\PYG{p}{,} \PYG{n}{colors}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{n}{mean}\PYG{p}{,} \PYG{n}{std}\PYG{p}{,} \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{n}{label}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=} \PYG{l+m+mf}{0.4}\PYG{p}{,} \PYG{n}{color}\PYG{o}{=}\PYG{n}{color}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{o}{*}\PYG{n}{slope}\PYG{o}{+}\PYG{n}{intercept}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZhy{}\PYGZhy{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{color}\PYG{o}{=}\PYG{n}{color}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{CotA incubation}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{in MTP\PYGZhy{}well [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ABTS radical mM}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{time [min]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Experimental data with calculated regression line}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}


\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{,} \PYG{n}{slopes}\PYG{p}{,} \PYG{n}{stderrs}\PYG{p}{,} \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial rates within the first 5 min}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{CotA incubation time}\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s2}{in MTP\PYGZhy{}well [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{initial rate [mM / min]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}\PYG{p}{,} \PYG{n}{slope\PYGZus{}of\PYGZus{}slopes}\PYG{o}{*}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{n}{columns}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{]}\PYG{p}{)}\PYG{o}{+}\PYG{n}{intercept\PYGZus{}of\PYGZus{}slopes}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZhy{}\PYGZhy{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{color}\PYG{o}{=}\PYG{n}{colors}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{regression}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Initial rates of CotA reaction}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}

\PYG{n+nb}{print}\PYG{p}{(}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Calculated enzyme half life based on regression slope: }\PYG{l+s+si}{\PYGZob{}}\PYG{n}{t12}\PYG{l+s+si}{:}\PYG{l+s+s2}{.2f}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{ h}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{enzyme_inactivation_3_0}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
Calculated enzyme half life based on regression slope: 693.73 h
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. XXX: Experimental data and regression results of CotA reaction with different enzyme incubation periods.}

\sphinxAtStartPar
The left plot of figure XXX the change in product concentration is shown over the first 5 minutes of the enzyme reaction. Therein, different enzyme incubation times prior to reaction start are color\sphinxhyphen{}coded. The first reaction without prior incubation shows large standard deviations between the experimental repeats. This might be the result of insufficient mixing or inconsistent ammount of enzyme between the repeats. All other slopes showed a gradually decreasing slope for increasing incubation times.The plot on the right of figure XXX shows the initial rates of the enzyme reactions in relation to prior enzyme incubation time. Based on the calculated rate of deactivation, the half life of the enzyme is approximately 28 day. Due to the high standard deviation between the repeats of the reaction withour prior incubation. The experiments were not considered for the calculation of the inactivation rate.


\subsection{Enzyme kinetics experiment}
\label{\detokenize{scenarios/enzyme_inactivation:enzyme-kinetics-experiment}}

\subsubsection{Experimental data}
\label{\detokenize{scenarios/enzyme_inactivation:experimental-data}}
\sphinxAtStartPar
Experimental data was provided as excel files, whereas meta data of the experiment was filled in to an EnzymeML Excel spreadsheet. Measurement data was written to the EnzymeML document by a parser function, whereas concentrations were calculated via the provided extinction coefficient.Kinetic parameters were estimated with and without considering enzyme inactivation.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Load experimental data from excel file}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}excel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/enzyme\PYGZus{}inactivation/Repetition CotA ABTS kinetics higher volumes 2nd time.xlsx}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{sheet\PYGZus{}name}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{csv}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{set\PYGZus{}index}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Time(min)}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{data} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{values}\PYG{o}{.}\PYG{n}{T}\PYG{o}{.}\PYG{n}{reshape}\PYG{p}{(}\PYG{l+m+mi}{8}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{,}\PYG{l+m+mi}{72}\PYG{p}{)}
\PYG{n}{time} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{index}\PYG{o}{.}\PYG{n}{values}

\PYG{c+c1}{\PYGZsh{} Calculate concentrations}
\PYG{n}{extinction\PYGZus{}coefficient} \PYG{o}{=} \PYG{l+m+mi}{36} \PYG{c+c1}{\PYGZsh{} (1/mM * 1/cm)}
\PYG{n}{optical\PYGZus{}length} \PYG{o}{=} \PYG{l+m+mf}{0.65} \PYG{c+c1}{\PYGZsh{} cm}

\PYG{k}{def} \PYG{n+nf}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{abso}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{return} \PYG{n}{abso} \PYG{o}{/} \PYG{p}{(}\PYG{n}{extinction\PYGZus{}coefficient}\PYG{o}{*}\PYG{n}{optical\PYGZus{}length}\PYG{p}{)}

\PYG{n}{concentration\PYGZus{}data} \PYG{o}{=} \PYG{n}{absorption\PYGZus{}to\PYGZus{}concentration}\PYG{p}{(}\PYG{n}{data}\PYG{p}{)}


\PYG{c+c1}{\PYGZsh{} Parser function}
\PYG{k}{def} \PYG{n+nf}{data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{template\PYGZus{}path}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{measurement\PYGZus{}data}\PYG{p}{:} \PYG{n}{np}\PYG{o}{.}\PYG{n}{ndarray}\PYG{p}{,}
    \PYG{n}{species\PYGZus{}id}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{data\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}\PYG{p}{,}
    \PYG{n}{time\PYGZus{}unit}\PYG{p}{:} \PYG{n+nb}{str}
    \PYG{p}{)} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZgt{}} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{p}{:}

    \PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{EnzymeMLDocument}\PYG{o}{.}\PYG{n}{fromTemplate}\PYG{p}{(}\PYG{n}{template\PYGZus{}path}\PYG{p}{)}
    \PYG{k}{for} \PYG{n}{IDs}\PYG{p}{,} \PYG{n}{concentration} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{keys}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{n}{measurement\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
        \PYG{k}{for} \PYG{n}{counter}\PYG{p}{,} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{)}\PYG{p}{:}
            \PYG{n}{rep} \PYG{o}{=} \PYG{n}{pe}\PYG{o}{.}\PYG{n}{Replicate}\PYG{p}{(}
                \PYG{n+nb}{id}\PYG{o}{=}\PYG{l+s+sa}{f}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Measurement}\PYG{l+s+si}{\PYGZob{}}\PYG{n}{counter}\PYG{l+s+si}{\PYGZcb{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
                \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{n}{species\PYGZus{}id}\PYG{p}{,}
                \PYG{n}{data}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{replicate}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{n}{data\PYGZus{}unit}\PYG{p}{,}
                \PYG{n}{time}\PYG{o}{=}\PYG{n+nb}{list}\PYG{p}{(}\PYG{n}{time}\PYG{p}{)}\PYG{p}{,}
                \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{n}{time\PYGZus{}unit}\PYG{p}{)}
            \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{getMeasurement}\PYG{p}{(}\PYG{n}{IDs}\PYG{p}{)}\PYG{o}{.}\PYG{n}{addReplicates}\PYG{p}{(}\PYG{n}{rep}\PYG{p}{,} \PYG{n}{enzmldoc}\PYG{p}{)}
    \PYG{k}{return} \PYG{n}{enzmldoc}

\PYG{c+c1}{\PYGZsh{} Write experimantal data to EnzymeML document vis parser function}
\PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{data\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}
    \PYG{n}{template\PYGZus{}path}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{../../data/enzyme\PYGZus{}inactivation/EnzymeML\PYGZus{}CotA.xlsm}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{measurement\PYGZus{}data}\PYG{o}{=}\PYG{n}{concentration\PYGZus{}data}\PYG{p}{,}
    \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{data\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{mmole / l}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}
    \PYG{n}{time\PYGZus{}unit}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{min}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Visualize experimantal data}
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,} \PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{measurement} \PYG{o+ow}{in} \PYG{n}{enzmldoc}\PYG{o}{.}\PYG{n}{measurement\PYGZus{}dict}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
    \PYG{n}{concentration} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
    \PYG{n}{product} \PYG{o}{=} \PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{init\PYGZus{}substrate}\PYG{o}{=}\PYG{n}{measurement}\PYG{o}{.}\PYG{n}{getReactant}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{init\PYGZus{}conc}
    \PYG{k}{for} \PYG{n}{replicate} \PYG{o+ow}{in} \PYG{n}{product}\PYG{o}{.}\PYG{n}{replicates}\PYG{p}{:}
        \PYG{n}{concentration}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{replicate}\PYG{o}{.}\PYG{n}{data}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{std}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{n}{init\PYGZus{}substrate}\PYG{p}{,}\PYGZbs{}
        \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{.}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{alpha}\PYG{o}{=}\PYG{l+m+mf}{0.5}\PYG{p}{)}
    \PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{time}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{mean}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{np}\PYG{o}{.}\PYG{n}{std}\PYG{p}{(}\PYG{n}{concentration}\PYG{p}{,} \PYG{n}{axis}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}\PYG{p}{,} \PYG{n}{label}\PYG{o}{=}\PYG{n}{init\PYGZus{}substrate}\PYG{p}{,}\PYGZbs{}
        \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{o}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical concentration over CotA reaction time\PYGZhy{}course }\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical concentration within the first 5 min}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlim}\PYG{p}{(}\PYG{p}{[}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.2}\PYG{p}{,}\PYG{l+m+mf}{5.2}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylim}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mf}{0.01}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{fig}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{enzyme_inactivation_5_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig XXX. ABTS radical concentration over the time\sphinxhyphen{}course of CotA reaction.}

\sphinxAtStartPar
The provided reaction data showed large deviations between individual repeats of reactions with identical initial ABTS concentrations. Expecially for initial ABTS concentrations above 0.1 mM (Fig. XXX, left). Additionally, measurement data was not blanked, since the first measured data points at 0 min have too high concentration values in relation to their respective slopes.
Furthermore, the order of initial measurement points is expected to reflect the initial substrate concentrations (Fig. XXX right). Thereby, higher initial substrate concentrations should also show a slightly increased product concentration, since ABTS marginally absorbs at the product detection wavelength.


\subsubsection{Kinetic parameter estimation}
\label{\detokenize{scenarios/enzyme_inactivation:kinetic-parameter-estimation}}
\sphinxAtStartPar
Kinetic parameters were estimated for all models with and without considering enzyme inactivation.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Parameter estimation without time\PYGZhy{}dependent enzyme inactivation}
\PYG{n}{CotA\PYGZus{}kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{df} \PYG{o}{=} \PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat / Km [1/min * 1/mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df}\PYG{o}{.}\PYG{n}{insert}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{False}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Parameter estimation considering time\PYGZhy{}dependent enzyme inactivation}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s1}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{product}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{display\PYGZus{}output}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{df\PYGZus{}inactivation} \PYG{o}{=} \PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{result\PYGZus{}dict}\PYG{o}{.}\PYG{n}{drop}\PYG{p}{(}\PYG{n}{columns}\PYG{o}{=}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat / Km [1/min * 1/mmole / l]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{df\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{insert}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{True}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{results} \PYG{o}{=} \PYG{n}{df}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{df\PYGZus{}inactivation}\PYG{p}{)}\PYG{o}{.}\PYG{n}{sort\PYGZus{}values}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{AIC}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{display}\PYG{p}{(}\PYG{n}{results}\PYG{o}{.}\PYG{n}{style}\PYG{o}{.}\PYG{n}{set\PYGZus{}table\PYGZus{}attributes}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{style=}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{font\PYGZhy{}size: 12px}\PYG{l+s+s1}{\PYGZdq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZlt{}pandas.io.formats.style.Styler at 0x7fbabba40a90\PYGZgt{}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
Kinetic parameteres were estimated with and without considering time\sphinxhyphen{}dependent enzyme inactivation.
Modeling results are listed in table XXX. All models with an additional parameter for enzyme inactivation resulted in lower AIC values, indicating a better fit of the experimantal data to the respective models.
Estimates for \(k_{cat}\) ranged from 2.58 min\sphinxhyphen{}1 to 3.41 min\sphinxhyphen{}1, where \(K_{m}\) estimates ranged from 23 µM to 62 µM depending on the kinetic model.
Based on the resulting \(k_{inact}\), the enzyme’s half life was estimated to 95 min, using equation xxx.
Since AIC does not consider uncertanties of parameteres, they need to be assessed for model evaluaton.


\paragraph{Kinetic models with enzyme inactivation}
\label{\detokenize{scenarios/enzyme_inactivation:kinetic-models-with-enzyme-inactivation}}
\sphinxAtStartPar
For models with enzyme inactivation, the substrate inhibiton model showed the lowest AIC and low standard deviations on the estimated parameters. The irreversible Michaelis\sphinxhyphen{}Menten model had the least ammount of parameters and showed even lower standard deviations on the parameter estimates, compared to the substrate inhibition model. Product inhibition models showed generally higher standard deviation on the estimated inhibition parameters. Thus, the two models shown in figure xxx describe the data the best in therms of kinetic parameter standard deviations.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Substrate inhibition model with enzyme inactivation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics\PYGZus{}with\PYGZus{}inactivation}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis\PYGZhy{}Menten model with enzyme inactivation}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{enzyme_inactivation_9_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}
\sphinxAtStartPar
\sphinxstyleemphasis{Fig. XXX: Substrate inhibition and irreversible Michealis\sphinxhyphen{}Menten model fitted to CotA reaction data}

\sphinxAtStartPar
Visually, both models in figure xxx describe the observed concentration time\sphinxhyphen{}course of ABTS radical formation. For both models, measurement data from enzyme reactions with intial substrate concentrations below 0.1 mM deviate more strongly from the model compared to higher concentrations. Since the measured absorption values were not blanked, all calculated concentrations are falsely increased. This effect is more pronounced for low initial substrate concentrations, since the absoprtion contribution from buffer is lager compared to the accumulating product for low concentrations.For reactions with initial substrate concentrations above 0.1 mM, both models describe the progress curve of the reaction. Due to high standard deviation between the prepeat, the true reaction rate at high substrate concentrations is unknown. Therefore, neighter substrate inhibition nor irreversible Michaelis\sphinxhyphen{}Menten model were able to clearly represent the experimental data.


\paragraph{Kinetic models without enzyme inactivation}
\label{\detokenize{scenarios/enzyme_inactivation:kinetic-models-without-enzyme-inactivation}}
\sphinxAtStartPar
In therms of AIC, product inhibition models without considering enzyme inactivation described the data the best, whereas substrate inhibition and irreversible Michaelis\sphinxhyphen{}Menten model showed resulted in the highest AIC of all models. In therms of parameter standard deviation, irreversible Michaelis\sphinxhyphen{}Menten model showed
Non\sphinxhyphen{}competitive product inhibition
and irreversible Michaelis\sphinxhyphen{}Menten model both showed low standerd deviations on the estimated parameters, whereas product inhibition models showed higher uncertanties on the inhibition parameters.
Visualizations of the fitted

\sphinxAtStartPar
All models, except for non\sphinxhyphen{}competitive and uncompetitice product inhibition with enzyme inactivation resulted in parameter uncertanties of more than 100 \% of the estimated value.

\begin{sphinxuseclass}{cell}
\begin{sphinxuseclass}{tag_hide_input}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fig}\PYG{p}{,} \PYG{n}{axes} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mf}{12.8}\PYG{p}{,}\PYG{l+m+mf}{4.8}\PYG{p}{)}\PYG{p}{,} \PYG{n}{sharey}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sharex}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{uncompetitive product inhibition}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{uncompetitive product inhibition model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{CotA\PYGZus{}kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis Menten}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{n}{ax}\PYG{o}{=}\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{,} \PYG{n}{alpha} \PYG{o}{=}\PYG{l+m+mf}{.2}\PYG{p}{,}\PYGZbs{}
    \PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{irreversible Michaelis\PYGZhy{}Menten model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}

\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ABTS radical [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{time [min]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{axes}\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{initial ABTS [mM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{tight\PYGZus{}layout}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}
\begin{sphinxVerbatimOutput}

\begin{sphinxuseclass}{cell_output}
\noindent\sphinxincludegraphics{{enzyme_inactivation_11_0}.png}

\end{sphinxuseclass}\end{sphinxVerbatimOutput}

\end{sphinxuseclass}
\end{sphinxuseclass}

\subsection{Conclusion}
\label{\detokenize{scenarios/enzyme_inactivation:conclusion}}
\sphinxAtStartPar
\sphinxstylestrong{Low data quality}

\sphinxAtStartPar
Experimental data of the data set for parameter estimation showed low quality. This was indicated by high deviations between experimental repeats as well as

\sphinxAtStartPar
\sphinxstylestrong{Enzyme inactivation}
Parameter estimation with kinetic models considering enzyme inactivation leads to opposing conclusion of the kinetic mechanism.
Not considerin enzyme inactivation would lead to the assumption of product inhibition

\sphinxstepscope


\section{Effect of calibration equation on kinetic parameters}
\label{\detokenize{scenarios/effect_calibration_model:effect-of-calibration-equation-on-kinetic-parameters}}\label{\detokenize{scenarios/effect_calibration_model::doc}}
\sphinxAtStartPar
In this analysis the influence of calibration model selection on the resulting parameters should was investigated. Each parameter estimation scenario in this work relies on the conversion from absorption data into concentration data. Therefore, a standard of the analyte, which was used to measure the enzyme activty, was provided with each kinetic experiment. Usually, a linear relationship between the absobance signal and concentration is assumed by experimentalists. Therefore, a linear calibration is applied.

\sphinxAtStartPar
Experimentalists often restrict themselfs to only use the so called ‘linear’ range of the photometer below an absorption of 1. Thus, sacrificing a significant ammount of the measurement range of the analytical device.


\subsection{Data}
\label{\detokenize{scenarios/effect_calibration_model:data}}\begin{itemize}
\item {} 
\sphinxAtStartPar
chymotrypsin

\item {} 
\sphinxAtStartPar
SLAC

\item {} 
\sphinxAtStartPar
trametes

\item {} 
\sphinxAtStartPar
glucosidase

\item {} 
\end{itemize}


\subsection{Calibration}
\label{\detokenize{scenarios/effect_calibration_model:calibration}}

\subsection{Inactivation}
\label{\detokenize{scenarios/effect_calibration_model:inactivation}}

\subsubsection{Effekt of calibration temperature and pH on kinetic parameters}
\label{\detokenize{scenarios/effect_calibration_model:effekt-of-calibration-temperature-and-ph-on-kinetic-parameters}}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{enzmldoc\PYGZus{}absorption\PYGZus{}dict}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{3.0 45.0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{n}{standards\PYGZus{}variations} \PYG{o}{=} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n+nb}{filter}\PYG{p}{(}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH} \PYG{o}{==} \PYG{l+m+mf}{3.0}\PYG{p}{,} \PYG{n}{standard\PYGZus{}list}\PYG{p}{)}\PYG{p}{)}
\PYG{p}{[}\PYG{n}{standards\PYGZus{}variations}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n+nb}{list}\PYG{p}{(}\PYG{n+nb}{filter}\PYG{p}{(}\PYG{k}{lambda} \PYG{n}{x}\PYG{p}{:} \PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature} \PYG{o}{==} \PYG{l+m+mi}{45} \PYG{o+ow}{and} \PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH} \PYG{o}{\PYGZlt{}} \PYG{l+m+mi}{6}\PYG{p}{,} \PYG{n}{standard\PYGZus{}list}\PYG{p}{)}\PYG{p}{)}\PYG{p}{]}\PYG{p}{;}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]

\PYG{n}{kinetics} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{standard} \PYG{o+ow}{in} \PYG{n}{standards\PYGZus{}variations}\PYG{p}{:}
    \PYG{n+nb}{print}\PYG{p}{(}\PYG{n}{copy}\PYG{o}{.}\PYG{n+nv+vm}{\PYGZus{}\PYGZus{}file\PYGZus{}\PYGZus{}}\PYG{p}{)}
    \PYG{n}{doc} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{)}
    \PYG{n}{standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ommit\PYGZus{}nan\PYGZus{}measurements}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
    \PYG{n}{kinetic} \PYG{o}{=} \PYG{p}{(}\PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}
    \PYG{n}{kinetic}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
    \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{kinetic}\PYG{p}{)}


\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}

\subsection{Using linear calibration instead of 3rd polynominal model}
\label{\detokenize{scenarios/effect_calibration_model:using-linear-calibration-instead-of-3rd-polynominal-model}}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{models} \PYG{o}{=} \PYG{p}{[}\PYG{k+kc}{None}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Linear}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{kinetics}\PYG{o}{=}\PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{model} \PYG{o+ow}{in} \PYG{n}{models}\PYG{p}{:}
    \PYG{n+nb}{print}\PYG{p}{(}\PYG{n}{copy}\PYG{o}{.}\PYG{n+nv+vm}{\PYGZus{}\PYGZus{}file\PYGZus{}\PYGZus{}}\PYG{p}{)}
    \PYG{n}{doc} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{)}
    \PYG{n}{standards\PYGZus{}variations}\PYG{p}{[}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ommit\PYGZus{}nan\PYGZus{}measurements}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{model\PYGZus{}name}\PYG{o}{=}\PYG{n}{model}\PYG{p}{)}
    \PYG{n}{kinetic} \PYG{o}{=} \PYG{p}{(}\PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{doc}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}
    \PYG{n}{kinetic}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
    \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{kinetic}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{} Get kinetic parameters of all datasets}
\PYG{n}{kcat} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{temperature} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{45}\PYG{p}{,}\PYG{l+m+mi}{45}\PYG{p}{]}
\PYG{n}{pH} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{result} \PYG{o+ow}{in} \PYG{n}{kinetics}\PYG{p}{:}
    \PYG{n}{params} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}parameter\PYGZus{}dict}\PYG{p}{(}\PYG{p}{)}

    \PYG{n}{kcat}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{kcat\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{Km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{Km\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{ki}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{ki\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}


\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{o}{.}\PYG{n}{from\PYGZus{}dict}\PYG{p}{(}\PYG{p}{\PYGZob{}}
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pH}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{pH}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{temperature}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat [1/s]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km [uM]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki}\PYG{p}{,}
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki\PYGZus{}std}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{/} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr} \PYG{o}{=}\PYG{p}{(}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{o}{+}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mf}{0.5} \PYG{o}{*} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr}

\PYG{n}{ph\PYGZus{}map} \PYG{o}{=} \PYG{n}{sns}\PYG{o}{.}\PYG{n}{color\PYGZus{}palette}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{husl}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{ph\PYGZus{}colors} \PYG{o}{=} \PYG{p}{[}\PYG{n}{ph\PYGZus{}map}\PYG{p}{[}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)}\PYG{p}{]} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{)}\PYG{p}{]}
\PYG{n}{pH\PYGZus{}dict} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mf}{3.0}\PYG{p}{,} \PYG{l+m+mf}{3.5}\PYG{p}{,} \PYG{l+m+mf}{4.0}\PYG{p}{,} \PYG{l+m+mf}{4.5}\PYG{p}{,} \PYG{l+m+mf}{5.0}\PYG{p}{,} \PYG{l+m+mf}{5.5}\PYG{p}{]}\PYG{p}{,} \PYG{n}{ph\PYGZus{}colors}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{pH\PYGZus{}tuples} \PYG{o}{=} \PYG{p}{[}\PYG{n}{pH\PYGZus{}dict}\PYG{p}{[}\PYG{n}{x}\PYG{p}{]} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{]}

\PYG{n}{temperature\PYGZus{}map} \PYG{o}{=} \PYG{n}{matplotlib}\PYG{o}{.}\PYG{n}{cm}\PYG{o}{.}\PYG{n}{get\PYGZus{}cmap}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{coolwarm}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{colors} \PYG{o}{=} \PYG{p}{[}\PYG{n}{temperature\PYGZus{}map}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{np}\PYG{o}{.}\PYG{n}{linspace}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{)}\PYG{p}{]}
\PYG{n}{color\PYGZus{}dict} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{25}\PYG{p}{,} \PYG{l+m+mi}{30}\PYG{p}{,} \PYG{l+m+mi}{35}\PYG{p}{,} \PYG{l+m+mi}{40} \PYG{p}{,}\PYG{l+m+mi}{45}\PYG{p}{]}\PYG{p}{,} \PYG{n}{colors}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{color\PYGZus{}tuples} \PYG{o}{=} \PYG{p}{[}\PYG{n}{color\PYGZus{}dict}\PYG{p}{[}\PYG{n}{x}\PYG{p}{]} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{]}

\PYG{n}{ax} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{gca}\PYG{p}{(}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{km}\PYG{p}{,} \PYG{n}{kcat}\PYG{p}{,} \PYG{n}{km\PYGZus{}error}\PYG{p}{,} \PYG{n}{kcat\PYGZus{}error}\PYG{p}{,} \PYG{n}{color}\PYG{p}{,} \PYG{n}{label}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{pH\PYGZus{}tuples}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}

    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{scatter}\PYG{p}{(}\PYG{n}{km}\PYG{p}{,} \PYG{n}{kcat}\PYG{p}{,} \PYG{n}{color} \PYG{o}{=} \PYG{n}{color}\PYG{p}{,} \PYG{n}{label} \PYG{o}{=} \PYG{n}{label} \PYG{k}{if} \PYG{o+ow}{not} \PYG{n}{i} \PYG{o}{\PYGZpc{}} \PYG{l+m+mi}{5} \PYG{k}{else} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{km}\PYG{p}{,} \PYG{n}{kcat}\PYG{p}{,} \PYG{n}{yerr}\PYG{o}{=}\PYG{n}{kcat\PYGZus{}error}\PYG{p}{,}\PYG{n}{xerr}\PYG{o}{=}\PYG{n}{km\PYGZus{}error}\PYG{p}{,} \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{.}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ecolor}\PYG{o}{=}\PYG{n}{color}\PYG{p}{,} \PYG{n}{markersize}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlim}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{200}\PYG{p}{]}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{subset\PYGZus{}enzymeml} \PYG{o}{=} \PYG{p}{[}\PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{enzmldoc\PYGZus{}absorption\PYGZus{}dict}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)} \PYG{k}{if} \PYG{n}{x}\PYG{o}{.}\PYG{n}{getReaction}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{r0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{o}{.}\PYG{n}{temperature} \PYG{o}{==} \PYG{l+m+mi}{45}\PYG{p}{]}
\PYG{n}{models} \PYG{o}{=} \PYG{p}{[}\PYG{k+kc}{None}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Linear}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{calibration\PYGZus{}data} \PYG{o}{=} \PYG{p}{[}\PYG{n}{x} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{standard\PYGZus{}curve\PYGZus{}dict}\PYG{o}{.}\PYG{n}{values}\PYG{p}{(}\PYG{p}{)} \PYG{k}{if} \PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{temperature} \PYG{o}{==} \PYG{l+m+mi}{45} \PYG{o+ow}{and} \PYG{n}{x}\PYG{o}{.}\PYG{n}{calibration\PYGZus{}data}\PYG{o}{.}\PYG{n}{pH} \PYG{o}{\PYGZlt{}} \PYG{l+m+mi}{6}\PYG{p}{]}
\PYG{n}{applied\PYGZus{}model} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{best model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:} \PYG{k+kc}{None}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Linear}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Linear}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{\PYGZcb{}}

\PYG{n}{kinetics} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{model\PYGZus{}names} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{k}{for} \PYG{n}{doc}\PYG{p}{,} \PYG{n}{standard} \PYG{o+ow}{in} \PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{subset\PYGZus{}enzymeml}\PYG{p}{,} \PYG{n}{calibration\PYGZus{}data}\PYG{p}{)}\PYG{p}{:}
    \PYG{k}{for} \PYG{n}{name}\PYG{p}{,} \PYG{n}{value} \PYG{o+ow}{in} \PYG{n}{applied\PYGZus{}model}\PYG{o}{.}\PYG{n}{items}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{d} \PYG{o}{=} \PYG{n}{copy}\PYG{o}{.}\PYG{n}{deepcopy}\PYG{p}{(}\PYG{n}{doc}\PYG{p}{)}
        \PYG{n}{standard}\PYG{o}{.}\PYG{n}{apply\PYGZus{}to\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{d}\PYG{p}{,} \PYG{n}{species\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{ommit\PYGZus{}nan\PYGZus{}measurements}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{model\PYGZus{}name}\PYG{o}{=}\PYG{n}{value}\PYG{p}{)}
        \PYG{n}{kinetic} \PYG{o}{=} \PYG{p}{(}\PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{d}\PYG{p}{,} \PYG{n}{reactant\PYGZus{}id}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{measured\PYGZus{}species}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}\PYG{p}{)}
        \PYG{n}{kinetic}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}\PYG{p}{;}
        \PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{kinetic}\PYG{p}{)}
        \PYG{n}{kinetic}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{model\PYGZus{}names}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{name}\PYG{p}{)}



\PYG{n}{model\PYGZus{}names}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{kcat} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{Km\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{ki\PYGZus{}std} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{temp} \PYG{o}{=} \PYG{p}{[}\PYG{p}{]}
\PYG{n}{calibration\PYGZus{}model} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{tile}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{best model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{linear calibration}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{)}
\PYG{n}{pH} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{repeat}\PYG{p}{(}\PYG{n}{np}\PYG{o}{.}\PYG{n}{array}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{,}\PYG{l+m+mf}{3.5}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{,}\PYG{l+m+mf}{4.5}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{,}\PYG{l+m+mf}{5.5}\PYG{p}{]}\PYG{p}{)}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{result} \PYG{o+ow}{in} \PYG{n}{kinetics}\PYG{p}{:}
    \PYG{n}{params} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{get\PYGZus{}parameter\PYGZus{}dict}\PYG{p}{(}\PYG{p}{)}
    \PYG{n}{temp}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{data}\PYG{o}{.}\PYG{n}{temperature}\PYG{p}{)}

    \PYG{n}{kcat}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{kcat\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k\PYGZus{}cat}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{Km}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{Km\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}

    \PYG{n}{ki}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}
    \PYG{n}{ki\PYGZus{}std}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{K\PYGZus{}ie}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}\PYG{p}{)}


\PYG{n}{df} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{o}{.}\PYG{n}{from\PYGZus{}dict}\PYG{p}{(}\PYG{p}{\PYGZob{}}
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pH}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{pH}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{calibration model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{calibration\PYGZus{}model}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat [1/s]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{kcat stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{kcat\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km [uM]}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km}\PYG{p}{,} 
    \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Km stderr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{Km\PYGZus{}std}\PYG{p}{,} 
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki}\PYG{p}{,}
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:} \PYG{n}{temp}\PYG{p}{,}
    \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Enzyme inactivation std}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{:}\PYG{n}{ki\PYGZus{}std}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{/} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr} \PYG{o}{=}\PYG{p}{(}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{o}{+}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{/}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Km [uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mi}{2}\PYG{p}{)}\PYG{o}{*}\PYG{o}{*}\PYG{l+m+mf}{0.5} \PYG{o}{*} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{kcat\PYGZus{}km\PYGZus{}stderr}

\PYG{n}{df}
        
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{ph\PYGZus{}map} \PYG{o}{=} \PYG{n}{sns}\PYG{o}{.}\PYG{n}{color\PYGZus{}palette}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{husl}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{ph\PYGZus{}colors} \PYG{o}{=} \PYG{p}{[}\PYG{n}{ph\PYGZus{}map}\PYG{p}{[}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)}\PYG{p}{]} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{l+m+mi}{6}\PYG{p}{)}\PYG{p}{]}
\PYG{n}{pH\PYGZus{}dict} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mf}{3.0}\PYG{p}{,} \PYG{l+m+mf}{3.5}\PYG{p}{,} \PYG{l+m+mf}{4.0}\PYG{p}{,} \PYG{l+m+mf}{4.5}\PYG{p}{,} \PYG{l+m+mf}{5.0}\PYG{p}{,} \PYG{l+m+mf}{5.5}\PYG{p}{]}\PYG{p}{,} \PYG{n}{ph\PYGZus{}colors}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{pH\PYGZus{}tuples} \PYG{o}{=} \PYG{p}{[}\PYG{n}{pH\PYGZus{}dict}\PYG{p}{[}\PYG{n}{x}\PYG{p}{]} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{]}

\PYG{n}{temperature\PYGZus{}map} \PYG{o}{=} \PYG{n}{matplotlib}\PYG{o}{.}\PYG{n}{cm}\PYG{o}{.}\PYG{n}{get\PYGZus{}cmap}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{coolwarm}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{colors} \PYG{o}{=} \PYG{p}{[}\PYG{n}{temperature\PYGZus{}map}\PYG{p}{(}\PYG{n}{x}\PYG{p}{)} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{np}\PYG{o}{.}\PYG{n}{linspace}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{)}\PYG{p}{]}
\PYG{n}{color\PYGZus{}dict} \PYG{o}{=} \PYG{n+nb}{dict}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{p}{[}\PYG{l+m+mi}{25}\PYG{p}{,} \PYG{l+m+mi}{30}\PYG{p}{,} \PYG{l+m+mi}{35}\PYG{p}{,} \PYG{l+m+mi}{40} \PYG{p}{,}\PYG{l+m+mi}{45}\PYG{p}{]}\PYG{p}{,} \PYG{n}{colors}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{color\PYGZus{}tuples} \PYG{o}{=} \PYG{p}{[}\PYG{n}{color\PYGZus{}dict}\PYG{p}{[}\PYG{n}{x}\PYG{p}{]} \PYG{k}{for} \PYG{n}{x} \PYG{o+ow}{in} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{temperature}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{]}

\PYG{n}{cools} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{b}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{orange}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{b}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{orange}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{b}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{orange}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{b}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{orange}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{b}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{orange}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{b}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{orange}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{n}{ax} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{gca}\PYG{p}{(}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{i}\PYG{p}{,} \PYG{p}{(}\PYG{n}{kcatkm}\PYG{p}{,} \PYG{n}{kcatkm\PYGZus{}error}\PYG{p}{,} \PYG{n}{temp}\PYG{p}{,} \PYG{n}{color}\PYG{p}{,} \PYG{n}{label}\PYG{p}{)} \PYG{o+ow}{in} \PYG{n+nb}{enumerate}\PYG{p}{(}\PYG{n+nb}{zip}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat [1/s]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat stderr}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]} \PYG{p}{,} \PYG{n}{cools}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{calibration model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}

    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{scatter}\PYG{p}{(}\PYG{n}{temp}\PYG{p}{,} \PYG{n}{kcatkm}\PYG{p}{,} \PYG{n}{color} \PYG{o}{=} \PYG{n}{color}\PYG{p}{,} \PYG{n}{label} \PYG{o}{=} \PYG{n}{label} \PYG{k}{if} \PYG{n}{i} \PYG{o+ow}{in} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{k}{else} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
    \PYG{n}{ax}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{temp}\PYG{p}{,} \PYG{n}{kcatkm}\PYG{p}{,} \PYG{n}{yerr}\PYG{o}{=}\PYG{n}{kcatkm\PYGZus{}error}\PYG{p}{,} \PYG{n}{fmt}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{.}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{color}\PYG{o}{=} \PYG{n}{color}\PYG{p}{,} \PYG{n}{markersize}\PYG{o}{=}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{n}{title} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{calibration model}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{kcat/Km [1/s * uM]}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{pH}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{title}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Datasets of various pHs and 45°C calibrated with different standard curve fits}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\begin{sphinxuseclass}{cell}\begin{sphinxVerbatimInput}

\begin{sphinxuseclass}{cell_input}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{enzmldoc} \PYG{o}{=} \PYG{n}{concentration\PYGZus{}enzymemldocs}\PYG{p}{[}\PYG{l+m+mi}{4}\PYG{p}{]}

\PYG{n}{kinetics} \PYG{o}{=} \PYG{n}{ParameterEstimator}\PYG{o}{.}\PYG{n}{from\PYGZus{}EnzymeML}\PYG{p}{(}\PYG{n}{enzmldoc}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{s0}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{substrate}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{fit\PYGZus{}models}\PYG{p}{(}\PYG{n}{only\PYGZus{}irrev\PYGZus{}MM}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{enzyme\PYGZus{}inactivation}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{kinetics}\PYG{o}{.}\PYG{n}{visualize}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\end{sphinxuseclass}\end{sphinxVerbatimInput}

\end{sphinxuseclass}
\sphinxstepscope


\chapter{Discussion}
\label{\detokenize{discussion:discussion}}\label{\detokenize{discussion::doc}}
\sphinxAtStartPar
Discuss both viewpoints: biologist and RSE.


\section{Jupyter notebooks for documentation of scientific analysis}
\label{\detokenize{discussion:jupyter-notebooks-for-documentation-of-scientific-analysis}}
\sphinxAtStartPar
Jupyter notebooks enable enable


\section{Reproducibility in enzyme kinetics}
\label{\detokenize{discussion:reproducibility-in-enzyme-kinetics}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Need for documentation without limiting experimental possibilities

\end{itemize}


\section{Data management / lab digitalization}
\label{\detokenize{discussion:data-management-lab-digitalization}}\begin{itemize}
\item {} 
\sphinxAtStartPar
temperature in MTPs

\end{itemize}


\section{Jupyter notebooks}
\label{\detokenize{discussion:jupyter-notebooks}}\begin{itemize}
\item {} 
\sphinxAtStartPar
great documentation (FAIR)

\end{itemize}

\sphinxAtStartPar
In this project, a robust and FAIR workflow from analytical raw\sphinxhyphen{}data to kinetic parameters was established. Thereby, the output of a analytical device was used directly as the input of the modeling pipeline.

\sphinxAtStartPar
Despite the robust workflow, after the data is written to an EnzymeML document, manual copying of data into the EnzymeML spreadsheet is prone to human error. Especially, for large datasets with many time points and reactants. Hence, manufacturers should enable direct access to the measurement device, for instance via an API. (Control and laer experiments)

\sphinxAtStartPar
Despite the easy applicability of the workflow, after the data is initialized,


\section{Current state in kinetic modeling of enzyme kinetics}
\label{\detokenize{discussion:current-state-in-kinetic-modeling-of-enzyme-kinetics}}

\section{Model selection in enzyme kinetics: calibration and kinetic model}
\label{\detokenize{discussion:model-selection-in-enzyme-kinetics-calibration-and-kinetic-model}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Best\sphinxhyphen{}practices kinetic modeling
\begin{itemize}
\item {} 
\sphinxAtStartPar
initial rates vs progress\sphinxhyphen{}curve analysis
\begin{itemize}
\item {} 
\sphinxAtStartPar
initial rates not suited to estimate conversion after 24 h.

\end{itemize}

\end{itemize}

\item {} 
\sphinxAtStartPar
Cost of additional parameters

\item {} 
\sphinxAtStartPar
Model selection criteria

\item {} 
\sphinxAtStartPar
How to assess the quality of a model
\begin{itemize}
\item {} 
\sphinxAtStartPar
what does the model represent (not newtonian mechanics)

\end{itemize}

\end{itemize}


\section{Inactivation}
\label{\detokenize{discussion:inactivation}}\begin{itemize}
\item {} 
\sphinxAtStartPar
deterioration during catalysis

\item {} 
\sphinxAtStartPar
adsorption to MTP surface (hydrophobic interaction)

\end{itemize}


\section{Percision of kinetic parameters}
\label{\detokenize{discussion:percision-of-kinetic-parameters}}\begin{itemize}
\item {} 
\sphinxAtStartPar
how percise should parameters be

\item {} 
\sphinxAtStartPar
randomized order in MTPs

\end{itemize}

\sphinxstepscope


\chapter{Conclusion}
\label{\detokenize{conclusion:conclusion}}\label{\detokenize{conclusion::doc}}
\sphinxAtStartPar
Nice.:::\{figure\sphinxhyphen{}md\} markdown\sphinxhyphen{}fig


\sphinxAtStartPar
This is a caption in \sphinxstylestrong{Markdown}!

\sphinxstepscope


\chapter{References}
\label{\detokenize{references:references}}\label{\detokenize{references::doc}}
\begin{sphinxthebibliography}{DMMP20}
\bibitem[BAC99]{references:id9}
\sphinxAtStartPar
William F Busby, Joseph M Ackermann, and Charles L Crespi. Effect of methanol, ethanol, dimethyl sulfoxide, and acetonitrile on in vitro activities of cdna\sphinxhyphen{}expressed human cytochromes p\sphinxhyphen{}450. \sphinxstyleemphasis{Drug Metabolism and Disposition}, 27(2):246–249, 1999.
\bibitem[Bux15]{references:id8}
\sphinxAtStartPar
Engelbert Buxbaum. \sphinxstyleemphasis{Enzyme Kinetics: Special Cases}, pages 185–191. Springer International Publishing, Cham, 2015. URL: \sphinxurl{https://doi.org/10.1007/978-3-319-19920-7\_8}, \sphinxhref{https://doi.org/10.1007/978-3-319-19920-7\_8}{doi:10.1007/978\sphinxhyphen{}3\sphinxhyphen{}319\sphinxhyphen{}19920\sphinxhyphen{}7\_8}.
\bibitem[DMMP20]{references:id6}
\sphinxAtStartPar
Chantal Désirée Daub, Blessing Mabate, Samkelo Malgas, and Brett Ivan Pletschke. Fucoidan from ecklonia maxima is a powerful inhibitor of the diabetes\sphinxhyphen{}related enzyme, α\sphinxhyphen{}glucosidase. \sphinxstyleemphasis{International journal of biological macromolecules}, 151:412–420, 2020.
\bibitem[LLWZ08]{references:id7}
\sphinxAtStartPar
Bo Li, Fei Lu, Xinjun Wei, and Ruixiang Zhao. Fucoidan: structure and bioactivity. \sphinxstyleemphasis{Molecules}, 13(8):1671–1695, 2008.
\bibitem[Ple21]{references:id5}
\sphinxAtStartPar
Jürgen Pleiss. Standardized data, scalable documentation, sustainable storage–enzymeml as a basis for fair data management in biocatalysis. \sphinxstyleemphasis{ChemCatChem}, 13(18):3909–3913, 2021.
\end{sphinxthebibliography}







\renewcommand{\indexname}{Index}
\printindex
\end{document}