
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>From raw data to kinetic parameters: an EnzymeML-based workflow for reproducible enzyme kinetics</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint single-page" id="site-navigation">
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>
<a href="https://github.com/haeussma/masterthesis"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="bottom"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>

</a>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-acknowledgements">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-abstract">
   Abstract
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-introduction">
   Introduction
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-methods">
   Methods
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-results">
   Results
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/chymotrypsin_inhibition">
     <strong>
      Scenario A:
     </strong>
     <br/>
     Chymotrypsin inhibiton by an
     <em>
      in silico
     </em>
     designed albumin fusion protein
    </a>
   </li>
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/glycosidase_inhibition">
     <strong>
      Scenario B:
     </strong>
     <br/>
     α-glucosidase inhibition by fucoidan
    </a>
   </li>
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/SLAC_kinetic_characterization">
     <strong>
      Scenario C:
     </strong>
     <br/>
     SLAC characterization
    </a>
   </li>
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/enzyme_inactivation">
     <strong>
      Scenario D:
     </strong>
     <br/>
     Time-dependent enzyme inactivation
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-discussion">
   Discussion
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-conclusion">
   Conclusion and Outlook
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-references">
   References
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Welcome to my master thesis</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-acknowledgements">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-abstract">
   Abstract
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-introduction">
   Introduction
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-methods">
   Methods
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-results">
   Results
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/chymotrypsin_inhibition">
     <strong>
      Scenario A:
     </strong>
     <br/>
     Chymotrypsin inhibiton by an
     <em>
      in silico
     </em>
     designed albumin fusion protein
    </a>
   </li>
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/glycosidase_inhibition">
     <strong>
      Scenario B:
     </strong>
     <br/>
     α-glucosidase inhibition by fucoidan
    </a>
   </li>
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/SLAC_kinetic_characterization">
     <strong>
      Scenario C:
     </strong>
     <br/>
     SLAC characterization
    </a>
   </li>
   <li class="toctree-l2 toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="welcome.html#document-scenarios/enzyme_inactivation">
     <strong>
      Scenario D:
     </strong>
     <br/>
     Time-dependent enzyme inactivation
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-discussion">
   Discussion
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-conclusion">
   Conclusion and Outlook
  </a>
 </li>
 <li class="toctree-l1 toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="welcome.html#document-references">
   References
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="welcome-to-my-master-thesis">
<h1>Welcome to my master thesis<a class="headerlink" href="#welcome-to-my-master-thesis" title="Permalink to this headline">#</a></h1>
<p>This JupyterBook constitutes my master thesis. Within this thesis I developed a workflow for kinetic parameter estimation of enzyme reactions. Within multiple Jupyter Notebooks, which are included in this work, i show my developments and their capabilities in action, as I applied them to different ongoing EnzymeML projects.</p>
<p>Except for this page and the capability to execute Jupyter Notebooks, the printed version this book is identical to this Jupyter Book.</p>
<p><img alt="cover" src="_images/cover_master_thesis.jpg" /></p>
<div class="toctree-wrapper compound">
<span id="document-acknowledgements"></span><section class="tex2jax_ignore mathjax_ignore" id="acknowledgements">
<h2>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permalink to this headline">#</a></h2>
<p>First, I would like to thank Prof. Dr. Jürgen Pleiss for the opportunity to work on this project and the support throughout the work of this thesis.
Furthermore, I would like to thank Associate Professor Marilize Le Roes-Hill together with Prof. Dr. Jürgen Pleiss for organizing and enabling my research visit to South Africa, which fueled the collaboration on the projects of this thesis.</p>
<p>My special thanks go to my project partners Alaric Prins, Marwa Mohamed, Chantal Daub, and Paulo Durão who provided the experimental data for the projects of this thesis as well as the fruitful and pleasant project discussions.</p>
<p>I would like to thank Jan Range and Torsten Gieß for their support as well as the whole group for inspiring and motivating seminar discussions.</p>
</section>
<span id="document-abstract"></span><section class="tex2jax_ignore mathjax_ignore" id="abstract">
<h2>Abstract<a class="headerlink" href="#abstract" title="Permalink to this headline">#</a></h2>
<p>In this work, an EnzymeML-based workflow for kinetic parameter estimation was developed, enabling a continuous data flow from raw data to kinetic parameters.
For this purpose, two Python modules were developed and published on PyPI: CaliPytion facilitating precise calculation of concentrations based on analytical absorption data, whereas EnzymePynetics facilitates kinetic parameter estimation of single substrate enzyme reactions. Furthermore, EnzymePynetics is suited for inhibition studies.
Moreover, the workflow was implemented in a Jupyter Notebook environment, enabling conclusive documentation of each data processing step. Hence, the workflow is transparent, additionally the application of the workflow is reproducible. FAIR data principles were implemented on multiple levels of this thesis. Consequently, this thesis was conceptualized as a Jupyter Book, making all data analysis and methodologies of this work Findable, Acessible, Interoperable and Reusable to the most possible degree.</p>
<section id="von-rohdaten-zu-kinetischen-parametern-ein-enzymeml-basierter-workflow-fur-reproduzierbare-enzymkinetik">
<h3>Von Rohdaten zu kinetischen Parametern: ein EnzymeML basierter Workflow für reproduzierbare Enzymkinetik<a class="headerlink" href="#von-rohdaten-zu-kinetischen-parametern-ein-enzymeml-basierter-workflow-fur-reproduzierbare-enzymkinetik" title="Permalink to this headline">#</a></h3>
<p>In dieser Arbeit wurde ein EnzymeML basierter Workflow zur Bestimmung kinetischer Parameter entwickelt, welcher einen kontinuierlichen Datenfluss von Rohdaten zu kinetischen Parametern ermöglicht.
Zu diesem Zweck wurden zwei Python-Module entwickelt und auf PyPI veröffentlicht: CaliPytion, welches die präzise Berechnung von Konzentrationen auf der Grundlage analytischer Absorptionsdaten ermöglicht und EnzymePynetics, welches die Schätzung kinetischer Parameter von Enzymreaktionen mit einem Substrat ermöglicht. Außerdem ist EnzymePynetics für Inhibitionsstudien geeignet.
Darüber hinaus wurde der Workflow in einer Jupyter-Notebook-Umgebung implementiert, was eine schlüssige Dokumentation der einzelnen Datenverarbeitungsschritte ermöglicht. Somit ist der Workflow transparent, während die Anwendung des Workflows reproduzierbar ist. Die FAIR-Prinzipien wurden in dieser Arbeit auf mehreren Ebenen umgesetzt. Daher wurde diese Arbeit als Jupyter Book konzipiert, um alle Datenanalysen und Methoden dieser Arbeit so weit wie möglich den FAIR-Kriterien Auffindbarkeit, Zugänglichkeit, Interoperabilität und Wiederverwendbarkeit entsprechend zu machen.</p>
</section>
</section>
<span id="document-introduction"></span><section class="tex2jax_ignore mathjax_ignore" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">#</a></h2>
<p>Reproducible research is the cornerstone of scientific advances and builds the core principle of the scientific method itself. Within the last decade, an increasing number of reports raised attention on unreproducible results across various scientific disciplines <span id="id1">[<a class="reference internal" href="welcome.html#id33" title="C Glenn Begley and John PA Ioannidis. Reproducibility in science: improving the standard for basic and preclinical research. Circulation research, 116(1):116–126, 2015.">Begley and Ioannidis, 2015</a>]</span>. Reportedly, more than 70 % of researchers were not able to reproduce the results of another researcher, whereas more than 50 % were not able to reproduce their own findings <span id="id2">[<a class="reference internal" href="welcome.html#id32" title="Monya Baker. Reproducibility crisis. Nature, 533(26):353–66, 2016.">Baker, 2016</a>]</span>.
Since reproducibility is a non-standardized term, it often serves as a vague umbrella term for mingling (i) repeatable analysis based on given information by the same scientist, (ii) reproducible analysis based on given information by another scientist, or (iii) replicable analysis with new information by another scientist.
Therefore, reproducibility is not an absolute term but rather relative, depending on the scope of the notion <span id="id3">[<a class="reference internal" href="welcome.html#id31" title="Steven N Goodman, Daniele Fanelli, and John PA Ioannidis. What does research reproducibility mean? Science translational medicine, 8(341):341ps12–341ps12, 2016.">Goodman <em>et al.</em>, 2016</a>]</span>. Goodman et. al. proposed to combine the word reproducibility with its relative scope, hence subdividing reproducibility in “methods reproducibility”, “results reproducibility” and “inferential reproducibility”. According to Goodman, ‘methods reproducibility refers to the provision of enough detail about study procedures and data so the same procedures could, in theory or in actuality, be exactly repeated.’. Results reproducibility requires methodological reproducibility, so that the same results are obtained from an independent study. Inferential reproducibility is given, if qualitatively similar conclusions are drawn based on the results of independent studies or reanalysis of a given study. Therefore, inferential reproducibility constitutes the highest level of reproducibility.<br />
Possible reasons for non-reproducible experiments are broadly discussed and range from study design, to publication bias and malpractice <span id="id4">[<a class="reference internal" href="welcome.html#id34" title="Marcus R Munafò, Brian A Nosek, Dorothy VM Bishop, Katherine S Button, Christopher D Chambers, Nathalie Percie du Sert, Uri Simonsohn, Eric-Jan Wagenmakers, Jennifer J Ware, and John Ioannidis. A manifesto for reproducible science. Nature human behaviour, 1(1):1–9, 2017.">Munafò <em>et al.</em>, 2017</a>]</span>, <span id="id5">[<a class="reference internal" href="welcome.html#id21" title="Tsuyoshi Miyakawa. No raw data, no science: another possible source of the reproducibility crisis. 2020.">Miyakawa, 2020</a>]</span> <span id="id6">[<a class="reference internal" href="welcome.html#id32" title="Monya Baker. Reproducibility crisis. Nature, 533(26):353–66, 2016.">Baker, 2016</a>]</span>. Once published, the results of non-reproducible studies are harmful for science, since the reported results along the generated data assets are likely false.</p>
<p>Low reproducibility also affects the field of enzymology, where varying experimental standards are described as the leading cause <span id="id7">[<a class="reference internal" href="welcome.html#id42" title="Elizabeth L. Bell, William Finnigan, Scott P. France, Anthony P. Green, Martin A. Hayes, Lorna J. Hepworth, Sarah L. Lovelock, Haruka Niikura, S\'ılvia Osuna, Elvira Romero, Katherine S. Ryan, Nicholas J. Turner, and Sabine L. Flitsch. Biocatalysis. Nature Reviews Methods Primers, June 2021. URL: https://doi.org/10.1038/s43586-021-00044-z, doi:10.1038/s43586-021-00044-z.">Bell <em>et al.</em>, 2021</a>]</span>. With the aim to increase reproducibility on biocatalytic experiments the STRENDA commission has defined standards for reporting enzymology data (STRENDA) <span id="id8">[<a class="reference internal" href="welcome.html#id11" title="Keith F Tipton, Richard N Armstrong, Barbara M Bakker, Amos Bairoch, Athel Cornish-Bowden, Peter J Halling, Jan-Hendrik Hofmeyr, Thomas S Leyh, Carsten Kettner, Frank M Raushel, and others. Standards for reporting enzyme data: the strenda consortium: what it aims to do and why it should be helpful. Perspectives in Science, 1(1-6):131–137, 2014.">Tipton <em>et al.</em>, 2014</a>]</span>. The <a class="reference external" href="https://www.beilstein-institut.de/en/projects/strenda/guidelines/">guidelines</a> recommend reporting on identity information of the enzyme, enzyme preparation steps, storage conditions, assay conditions and method, as well as enzyme activity with the corresponding kinetic parameters and units. Furthermore, the precision of measurements and preferably the raw data itself should be shared. Hence, the STRENDA guidelines pave the way for reproducible and therefore reusable functional enzymology data.<br />
Besides the reproducibility of data, availability of data assets must be given in order to enable reuse of extant data. This is especially important to enable the application of big data technologies, like machine learning which rely on large amounts of high quality data. In the field of biocatalysis, machine learning bares the potential to investigate how the structures of an enzyme is related to its catalytic properties. Thereby, valuable conclusions for enzyme engineering might be drawn.
One prerequisite for applying big data technologies and therefore use their potential in biocatalysis is the availability of data. With the scope to enhance data reusability, FAIR guiding principles for scientific data management were established <span id="id9">[<a class="reference internal" href="welcome.html#id40" title="Mark D Wilkinson, Michel Dumontier, IJsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, Jan-Willem Boiten, Luiz Bonino da Silva Santos, Philip E Bourne, and others. The fair guiding principles for scientific data management and stewardship. Scientific data, 3(1):1–9, 2016.">Wilkinson <em>et al.</em>, 2016</a>]</span>. The acronym FAIR denotes four foundational principles for contemporary data management in a Findable, Accessible, Interoperable and Reusable fashion. Special emphasis is placed on machines being able to find and use the data, due to their fundamental importance in modern science. Thereby, (meta)data should be findable by a globally unique identifier. Data should be accessible by a standardized and open protocol. Interoperability is achieved, if data contains qualitative references to other (meta)data and uses conclusive vocabulary. For reusability, the data needs to be extensively describes with accurate and relevant attributes <span id="id10">[<a class="reference internal" href="welcome.html#id40" title="Mark D Wilkinson, Michel Dumontier, IJsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, Jan-Willem Boiten, Luiz Bonino da Silva Santos, Philip E Bourne, and others. The fair guiding principles for scientific data management and stewardship. Scientific data, 3(1):1–9, 2016.">Wilkinson <em>et al.</em>, 2016</a>]</span>. Broad application of the stated principles will promote the reuse of data and ultimately amplify the knowledge which can be deduced from each published dataset.</p>
<p>To enable data exchange in biocatalysis the standardized exchange format EnzymeML was developed, abiding by FAIR data principles as well as STRENDA guidelines <span id="id11">[<a class="reference internal" href="welcome.html#id4" title="Jürgen Pleiss. Standardized data, scalable documentation, sustainable storage–enzymeml as a basis for fair data management in biocatalysis. ChemCatChem, 13(18):3909–3913, 2021.">Pleiss, 2021</a>]</span>. Within an EnzymeML document information on the reaction conditions, obtained measurement data, as well as modeling results are stored. As such, reaction conditions contain information on the pH value, temperature and the reaction vessel. Each species within a reaction like substrate, product or inhibition species is uniquely labeled with an identifier, which allows referencing the specific species in databases. Thereby, proteins are labeled with their UniProt id, and reactants are labeled with their respective SMILES or InChI code.<br />
A special focus of the EnzymeML format is to store the modeling results which are extracted from the experimental data which are stored alongside.
Modeling results and information on model parameters and model equations can be stored alongside the respective measurement data of which the results are based on. In summary, data and meta data of biocatalytic experiments can be stored and shared between scientists and databases compliant with contemporary data management practices.</p>
<p>In this work, a computational workflow based on EnzymeML for kinetic parameters estimation of enzyme reactions was developed with the following requirements: The workflow should enable comprehensive data analysis starting from raw data of enzyme kinetics experiments to kinetic parameter estimates. The workflow should be iteratively developed by applied by workflow on ongoing research projects of EnzymeML project partners. Therefore, two software packages should be developed, which facilitate the parameter estimation process. Finally, the estimated parameters should be written back to the EnzymeML file, unifying measurement data with information on the applied estimation model as well as the resulting kinetic parameters.</p>
</section>
<span id="document-methods"></span><section class="tex2jax_ignore mathjax_ignore" id="methods">
<h2>Methods<a class="headerlink" href="#methods" title="Permalink to this headline">#</a></h2>
<section id="enzymeml-and-pyenzyme">
<h3>EnzymeML and PyEnzyme<a class="headerlink" href="#enzymeml-and-pyenzyme" title="Permalink to this headline">#</a></h3>
<p>In this work, EnzymeML documents were read, edited, and written via the Python API PyEnzyme <span id="id1">[<a class="reference internal" href="welcome.html#id12" title="Jan Range, Frank Bergmann, and Neil Swainston. Pyenzyme 1.1.0-alpha. June 2021. URL: https://doi.org/10.5281/zenodo.5021263, doi:10.5281/zenodo.5021263.">Range <em>et al.</em>, 2021</a>]</span>, providing the possibility to integrate PyEnzyme in Python-based data analysis workflows. Thereby, EnzymeML documents were used as a data source, or modeling results were written back to the document.</p>
<section id="creation-of-enzymeml-documents">
<h4>Creation of EnzymeML documents<a class="headerlink" href="#creation-of-enzymeml-documents" title="Permalink to this headline">#</a></h4>
<p>EnzymeML documents were created using the EnzymeML Excel template, in combination with the <code class="docutils literal notranslate"><span class="pre">.fromTemplate()</span></code> method of PyEnzyme. Within the spreadsheet, data and metadata of an experiment were filled in the respective sheets. Alternatively, only metadata was entered to the spreadsheet, whereas measurement data was parsed by a custom Python function from the output file of the analytical device to the measurement data section of the EnzymeML document.</p>
</section>
<section id="saving-of-modeling-results">
<h4>Saving of modeling results<a class="headerlink" href="#saving-of-modeling-results" title="Permalink to this headline">#</a></h4>
<p>Modeling results from parameter estimation were written back to the EnzymeML document, which contained the measurement data by using PyEnzyme.</p>
</section>
</section>
<section id="calipytion">
<h3>CaliPytion<a class="headerlink" href="#calipytion" title="Permalink to this headline">#</a></h3>
<p>CaliPytion was developed to provide an easy way to find the best model by which the relationship between analytical signal to analyte concentration of photometric measurements can be determined. The resulting <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> is created based on a provided analyte standard. Besides a linear calibration equation, non-linear calibration models were implemented. All implemented calibration equations <a class="reference internal" href="#equation-linear">(1)</a>, <a class="reference internal" href="#equation-quadratic">(2)</a> <a class="reference internal" href="#equation-3rd-polynominal">(3)</a>, <a class="reference internal" href="#equation-poly-e">(4)</a> <a class="reference internal" href="#equation-fractional">(5)</a> are listed below, whereas the <span class="math notranslate nohighlight">\(A\)</span> denotes the absorption, <span class="math notranslate nohighlight">\(x\)</span> the concentration, and <span class="math notranslate nohighlight">\(a, b, c\)</span> the respective parameters of each equation.
During the fitting process, all calibration equations are fitted against the analyte standard by non-linear least-square minimization. Therefore, the Python library lmfit <span id="id2">[<a class="reference internal" href="welcome.html#id35" title="Matthew Newville, Till Stensitzki, Daniel B Allen, Michal Rawlik, Antonino Ingargiola, and Andrew Nelson. Lmfit: non-linear least-square minimization and curve-fitting for python. Astrophysics Source Code Library, pages ascl–1606, 2016.">Newville <em>et al.</em>, 2016</a>]</span> was utilized. The best fitting model is then selected based on the lowest Akaike information criterion (AIC). Furthermore the fit of the models can be visually assessed by calling the <code class="docutils literal notranslate"><span class="pre">.visualize()</span></code> method of the <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> class. After the calibration curves are fitted, concentration of unknown samples are determined by calculating the root of the fitted calibration model.</p>
<div class="math notranslate nohighlight" id="equation-linear">
<span class="eqno">(1)<a class="headerlink" href="#equation-linear" title="Permalink to this equation">#</a></span>\[A = ax\]</div>
<div class="math notranslate nohighlight" id="equation-quadratic">
<span class="eqno">(2)<a class="headerlink" href="#equation-quadratic" title="Permalink to this equation">#</a></span>\[A = ax^2 + bx\]</div>
<div class="math notranslate nohighlight" id="equation-3rd-polynominal">
<span class="eqno">(3)<a class="headerlink" href="#equation-3rd-polynominal" title="Permalink to this equation">#</a></span>\[A = ax^3 + bx^2 + cx\]</div>
<div class="math notranslate nohighlight" id="equation-poly-e">
<span class="eqno">(4)<a class="headerlink" href="#equation-poly-e" title="Permalink to this equation">#</a></span>\[A = ae^{\frac{b}{x}}\]</div>
<div class="math notranslate nohighlight" id="equation-fractional">
<span class="eqno">(5)<a class="headerlink" href="#equation-fractional" title="Permalink to this equation">#</a></span>\[A = \frac{ax}{b+x}\]</div>
<section id="data-model">
<span id="calipytion-datamodel"></span><h4>Data model<a class="headerlink" href="#data-model" title="Permalink to this headline">#</a></h4>
<p>The <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> functionality of CaliPythion is based on a <a class="reference external" href="https://github.com/FAIRChemistry/CaliPytion/blob/main/specifications/CalibrationModel.md">data model</a>, structuring data and metadata of calibration measurements. Thereby, calibration conditions like temperature, pH, and information about the analytical device can be provided. Furthermore, the name as well as an ID can be specified for the analyzed substance. All of this information is stored in the <code class="docutils literal notranslate"><span class="pre">Calibration</span></code> root object. Additionally, the root object can contain a <code class="docutils literal notranslate"><span class="pre">Standard</span></code> and a <code class="docutils literal notranslate"><span class="pre">Spectrum</span></code>.<br />
A <code class="docutils literal notranslate"><span class="pre">Standard</span></code> contains measurements of multiple predefined concentrations and a wavelength at which the measurement was conducted. Additionally, a <code class="docutils literal notranslate"><span class="pre">Spectrum</span></code> can be defined by providing measurement data as well as the respective wavelengths at which the data was measured. The data model was generated using <a class="reference external" href="https://github.com/JR-1991/software-driven-rdm">sdRDM</a>, a python tool allowing the creation and versioning of data models.</p>
</section>
<section id="initialization">
<h4>Initialization<a class="headerlink" href="#initialization" title="Permalink to this headline">#</a></h4>
<p>A <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> can be initialized directly from an Excel file by calling <code class="docutils literal notranslate"><span class="pre">StandardCurve.from_Excel()</span></code>
Therefore, the first column must contain concentration data, whereas all following columns contain the according measurement data. The first row is not considered and should be used to label the data.
Alternatively a <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> can be initialized by providing a <code class="docutils literal notranslate"><span class="pre">Calibration</span></code> object, which contains the calibration data.</p>
<p>After <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> initialization, concentrations can be calculated by calling the <code class="docutils literal notranslate"><span class="pre">.get_concentration()</span></code> method. Alternatively, a <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> can be directly applied to an <code class="docutils literal notranslate"><span class="pre">EnzymeMLDocument</span></code> by calling the <code class="docutils literal notranslate"><span class="pre">.apply_to_EnzymeML()</span></code> method. The code cell below demonstrates schematically, how absorption data in an <code class="docutils literal notranslate"><span class="pre">EnzymeMLDocument</span></code> is transformed to concentration data by applying <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code>.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">enzmldoc_absorption</span> <span class="o">=</span> <span class="n">EnzymeMLDocument</span><span class="p">()</span>

<span class="c1"># Create standard curve</span>
<span class="n">product_standard</span> <span class="o">=</span> <span class="n">StandardCurve</span><span class="p">()</span>

<span class="c1"># Apply standard curve to species &#39;s1&#39; of the EnzymeMl document</span>
<span class="n">enzmldoc_concentration</span> <span class="o">=</span> <span class="n">product_standard</span><span class="o">.</span><span class="n">apply_to_EnzymeML</span><span class="p">(</span>
    <span class="n">enzmldoc</span><span class="o">=</span><span class="n">enzmldoc_absorption</span><span class="p">,</span>
    <span class="n">species_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="code-availability">
<h4>Code availability<a class="headerlink" href="#code-availability" title="Permalink to this headline">#</a></h4>
<p>CaliPytion was published on the Python packaging index (<a class="reference external" href="https://pypi.org/project/CaliPytion/">PyPI</a>) whereas the source code is accessible on <a class="reference external" href="https://github.com/FAIRChemistry/CaliPytion">GitHub</a>.</p>
</section>
</section>
<section id="enzymepynetics">
<span id="method-enzymepynetics"></span><h3>EnzymePynetics<a class="headerlink" href="#enzymepynetics" title="Permalink to this headline">#</a></h3>
<p>EnzymePynetics is a Python package for kinetic parameter estimation of single-substrate enzyme reactions. The <code class="docutils literal notranslate"><span class="pre">ParameterEstimator</span></code> of EnzymePynetics estimates the turnover number and Michaelis-Menten constant by fitting time-course measurement data of enzyme reactions to different Michaelis-Menten models. Thereby, the residuals between measurement data and integrated Michaelis-Menten rate equations are minimized through a non-linear least-squares algorithm utilizing the lmfit library. Additionally, the inhibition constant <span class="math notranslate nohighlight">\(K_{i}\)</span> can be assessed for potential substrate or product inhibition. Furthermore, <span class="math notranslate nohighlight">\(K_{i}\)</span> of an enzyme inhibitor apart from substrate or product can be determined for inhibitor studies.</p>
<section id="enzymepynetics-datamodel">
<span id="id3"></span><h4>Data model<a class="headerlink" href="#enzymepynetics-datamodel" title="Permalink to this headline">#</a></h4>
<p>EnzymePynetics is based on a data model, resembling the experimental design of an enzyme kinetics assay. Thereby, all relevant data and metadata of a kinetic experiment are structured in the <code class="docutils literal notranslate"><span class="pre">EnzymeKineticsExperiment</span></code> base object. Thereby, reaction temperature with its respective unit, pH value, and the name of substance from which the measurement data originates. An <code class="docutils literal notranslate"><span class="pre">EnzymeKineticsExperiment</span></code> contains one or multiple <code class="docutils literal notranslate"><span class="pre">Measurements</span></code>, which contain the measurement data. Each measurement stores information of the experimental conditions of an enzyme reaction. Therefore, each <code class="docutils literal notranslate"><span class="pre">Measurement</span></code> contains information on the initial substrate concentration, enzyme concentration, and inhibitor concentration, along with the respective concentration units. Each <code class="docutils literal notranslate"><span class="pre">Measurement</span></code> contains measured data, which itself consist of one or multiple replicates of the respective experimental condition.<br />
The data model was generated using <a class="reference external" href="https://github.com/JR-1991/software-driven-rdm">sdRDM</a>.</p>
<p>Extensive documentation of the data model can be accessed in the <a class="reference external" href="https://github.com/haeussma/EnzymePynetics/blob/main/specifications/EnzymeKinetics.md">specifications</a> of the software package.</p>
</section>
<section id="kinetic-models">
<h4>Kinetic models<a class="headerlink" href="#kinetic-models" title="Permalink to this headline">#</a></h4>
<p>Besides the irreversible Michaelis-Menten rate equation <a class="reference internal" href="#equation-irreversible-mm">(6)</a> inhibition models for competitive <a class="reference internal" href="#equation-competitive-inhibition">(7)</a>, uncompetitive <a class="reference internal" href="#equation-uncompetitive-inhibition">(8)</a>, non-competitive <a class="reference internal" href="#equation-noncompetitive-inhibition">(9)</a> and partially competitive inhibition <a class="reference internal" href="#equation-partially-inhibition">(10)</a> were implemented, according to <span id="id4">[<a class="reference internal" href="welcome.html#id5" title="Hans Bisswanger. Enzyme kinetics: principles and methods. John Wiley &amp; Sons, 2017.">Bisswanger, 2017</a>]</span>. Thereby, <span class="math notranslate nohighlight">\(S\)</span>, <span class="math notranslate nohighlight">\(E\)</span>, and <span class="math notranslate nohighlight">\(I\)</span> respectively denote the concentration of substrate, enzyme, and inhibitor. In terms of kinetic parameters, <span class="math notranslate nohighlight">\(k_{cat}\)</span> denotes the turnover number, <span class="math notranslate nohighlight">\(K_{m}\)</span> the Michaelis-Menten constant of the substrate, whereas <span class="math notranslate nohighlight">\(K_{ic}\)</span> and <span class="math notranslate nohighlight">\(K_{iu}\)</span> respectively describe the competitive and uncompetitive inhibition constant.</p>
<div class="math notranslate nohighlight" id="equation-irreversible-mm">
<span class="eqno">(6)<a class="headerlink" href="#equation-irreversible-mm" title="Permalink to this equation">#</a></span>\[\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} + S}\]</div>
<div class="math notranslate nohighlight" id="equation-competitive-inhibition">
<span class="eqno">(7)<a class="headerlink" href="#equation-competitive-inhibition" title="Permalink to this equation">#</a></span>\[\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{ic}}) + S}\]</div>
<div class="math notranslate nohighlight" id="equation-uncompetitive-inhibition">
<span class="eqno">(8)<a class="headerlink" href="#equation-uncompetitive-inhibition" title="Permalink to this equation">#</a></span>\[\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{iu}}) * S}\]</div>
<div class="math notranslate nohighlight" id="equation-noncompetitive-inhibition">
<span class="eqno">(9)<a class="headerlink" href="#equation-noncompetitive-inhibition" title="Permalink to this equation">#</a></span>\[\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * (1+\frac{I}{K_{ic}}) + (1+\frac{I}{K_{iu}}) + S}\]</div>
<div class="math notranslate nohighlight" id="equation-partially-inhibition">
<span class="eqno">(10)<a class="headerlink" href="#equation-partially-inhibition" title="Permalink to this equation">#</a></span>\[\frac{dS}{dt} = -\frac{k_{cat} * E * S}{K_{m} * \frac{(1+\frac{I}{K_{ic}})}{(1+\frac{I}{K_{iu}})} + S}\]</div>
<p>By default, <span class="math notranslate nohighlight">\(E\)</span> is assumed to be constant throughout the reaction. If required, the kinetic models can be extended by the parameter <span class="math notranslate nohighlight">\(k_{inact}\)</span>, which describes the time-dependent inactivation rate of the enzyme. The decrease in active enzyme concentration is modeled by <a class="reference internal" href="#equation-enzyme-inactivation">(11)</a>:</p>
<div class="math notranslate nohighlight" id="equation-enzyme-inactivation">
<span class="eqno">(11)<a class="headerlink" href="#equation-enzyme-inactivation" title="Permalink to this equation">#</a></span>\[\frac{dE}{dt} = -k_{inact} * E\]</div>
<p>Each kinetic models therefore consists of a set of ordinary differential equations, describing the change in concentration of substrate, product and enzyme.</p>
</section>
<section id="run-parameter-estimation">
<h4>Run parameter estimation<a class="headerlink" href="#run-parameter-estimation" title="Permalink to this headline">#</a></h4>
<p><strong>Initialization</strong><br />
The <code class="docutils literal notranslate"><span class="pre">EnzymeKineticsExperiment</span></code> object solely serves as a data container, whereas
the <code class="docutils literal notranslate"><span class="pre">ParameterEstimator</span></code> harbors the functionalities for parameter estimation. Data can be provided as an <code class="docutils literal notranslate"><span class="pre">EnzymeKineticsExperiment</span></code> object. Alternatively, an <code class="docutils literal notranslate"><span class="pre">EnzymeMLDocument</span></code> document can be provided as the data source via PyEnzyme by calling the <code class="docutils literal notranslate"><span class="pre">.from_EnzymeML()</span></code> method. Within the method, the id of the measured species needs to be specified according to the nomenclature of the <code class="docutils literal notranslate"><span class="pre">EnzymeMLDocument</span></code>. If the document contains information on an inhibitor, the respective id needs to be provided as well, if the inhibition constant should be estimated for the respective inhibitor.</p>
<p>Initially, missing data for the fitting process is calculated based on the assumption of mass conservation. As such, missing product concentration <span class="math notranslate nohighlight">\(P\)</span> is calculated based on the given substrate measurement data and the specified initial substrate concentration <span class="math notranslate nohighlight">\(S_{0}\)</span> for each measurement <span class="math notranslate nohighlight">\(t\)</span> (<a class="reference internal" href="#equation-missing-product">(12)</a>). If product data is provided, missing substrate concentrations are calculated accordingly (<a class="reference internal" href="#equation-missing-substrate">(13)</a>).</p>
<div class="math notranslate nohighlight" id="equation-missing-product">
<span class="eqno">(12)<a class="headerlink" href="#equation-missing-product" title="Permalink to this equation">#</a></span>\[P_{t} = S_{0} - S_{t}\]</div>
<div class="math notranslate nohighlight" id="equation-missing-substrate">
<span class="eqno">(13)<a class="headerlink" href="#equation-missing-substrate" title="Permalink to this equation">#</a></span>\[S_{t} = S_{0} - P_{t}\]</div>
<p><strong>Fitting of models</strong><br />
After the <code class="docutils literal notranslate"><span class="pre">ParameterEstimator</span></code> is initialized, all kinetic models are initialized by calling the <code class="docutils literal notranslate"><span class="pre">.fit_models()</span></code> method. Furthermore, it can be determined whether enzyme inactivation should be considered for modeling. Depending on whether an inhibitor was specified, the models are initialized accordingly. If no inhibitor was specified, product and substrate inhibition models are initialized besides the irreversible Michaelis model. If inhibitor data was provided, the inhibition models are initialized with concentration data of the inhibitor. Thereafter, kinetic parameters of each model are initialized with estimates based on the highest reaction rate in the dataset. For minimization the parameter space is limited to ± 1000-fold of the initial parameter estimates.<br />
After all models are fitted, an overview table is printed which lists all of the kinetic parameters of each model together with the respective 1σ standard deviation in percent. The table is sorted by ascending AIC.</p>
<p><strong>Visualization</strong><br />
After model fitting, the modeling results can be visualized by calling the <code class="docutils literal notranslate"><span class="pre">.visualize()</span></code> method, which was implemented utilizing Matplotlib <span id="id5">[<a class="reference internal" href="welcome.html#id36" title="Thomas A Caswell, Michael Droettboom, Antony Lee, John Hunter, Eric Firing, Elliott Sales De Andrade, Tim Hoffmann, David Stansby, Jody Klymak, Nelle Varoquaux, and others. Matplotlib/matplotlib: rel: v3. 3.1. Zenodo, 2020.">Caswell <em>et al.</em>, 2020</a>]</span>.
Thereby, the measurement data is visualized together with the fitted kinetic model. If the experiment was carried out in replicates, mean values of the measurement data with the corresponding standard deviations are shown. If the experiment data contained reactions with different inhibitor concentrations, individual inhibitor concentrations are denoted by unique markers.<br />
By default the best fitting model according AIC is visualized. Different models can be visualized by passing the name of the respective model to the function call. Besides the figure, a detailed statistical fit report is printed, if not specified otherwise.</p>
<p><strong>Data units</strong><br />
So far, EnzymePynetics neither converts nor validates units of the measurement data. Hence, all data needs to be provided using the same molar or mass unit. Only the concentration unit of the inhibitor may differ from all other data units.</p>
</section>
<section id="id6">
<h4>Code availability<a class="headerlink" href="#id6" title="Permalink to this headline">#</a></h4>
<p>EnzymePynetics was published on the Python packaging index (<a class="reference external" href="https://pypi.org/project/EnzymePynetics/">PyPI</a>) whereas the source code is available on <a class="reference external" href="https://github.com/haeussma/EnzymePynetics">GitHub</a>.</p>
</section>
</section>
<section id="model-comparison">
<h3>Model comparison<a class="headerlink" href="#model-comparison" title="Permalink to this headline">#</a></h3>
<p>The kinetic model, which best describes the experimental data, was selected based on AIC, standard deviation of parameter estimates, and visual fit between measurement data and fitted model. AIC serves as a statistical metric for information loss, which allows to relatively compare different models with differing number of parameters for a given data set. Hence, AIC can be applied for model selection <span id="id7">[<a class="reference internal" href="welcome.html#id16" title="Todd W Arnold. Uninformative parameters and model selection using akaike's information criterion. The Journal of Wildlife Management, 74(6):1175–1178, 2010.">Arnold, 2010</a>]</span>, <span id="id8">[<a class="reference internal" href="welcome.html#id17" title="Hirotogu Akaike. Information theory and an extension of the maximum likelihood principle. In Selected papers of hirotugu akaike, pages 199–213. Springer, 1998.">Akaike, 1998</a>]</span>. Thereby, models with a lower AIC indicate less information loss.
Since AIC is based on the chi-square statistic, it does not consider the standard deviation of the estimated parameters of a model. Therefore, standard deviation was additionally considered in model selection. Models with low standard deviation were therefore preferred over models with high standard deviation. Furthermore, fit quality was assessed visually by confirming that the model describes the progress curve of the experimental data.</p>
</section>
<section id="jupyter-notebook">
<h3>Jupyter Notebook<a class="headerlink" href="#jupyter-notebook" title="Permalink to this headline">#</a></h3>
<p>All developed tools were deployed in Jupyter Notebooks <span id="id9">[<a class="reference internal" href="welcome.html#id37" title="Thomas Kluyver, Benjamin Ragan-Kelley, Fernando Pérez, Brian E Granger, Matthias Bussonnier, Jonathan Frederic, Kyle Kelley, Jessica B Hamrick, Jason Grout, Sylvain Corlay, and others. Jupyter Notebooks-a publishing format for reproducible computational workflows. Volume 2016. IOS Press, 2016.">Kluyver <em>et al.</em>, 2016</a>]</span> for data analysis within a Python environment. A Jupyter Notebook is a digital document containing code cells and text cells. Thereby, executable code with its corresponding output can be supplemented with narrative text, figures, and equations.
Besides the documentation capabilities, notebooks can be uploaded to platforms like GitHub or be deployed with Binder <span id="id10">[<a class="reference internal" href="welcome.html#id20" title="Benjamin Ragan-Kelley, Carol Willing, F Akici, D Lippa, D Niederhut, and M Pacer. Binder 2.0-reproducible, interactive, sharable environments for science at scale. In Proceedings of the 17th python in science conference, 113–120. F. Akici, D. Lippa, D. Niederhut, and M. Pacer, eds., 2018.">Ragan-Kelley <em>et al.</em>, 2018</a>]</span>.
GitHub is a cloud-based Git repository, which enables versioning as well as collaborative working on projects involving any kind of code. Besides GitHub, sharing of Jupiter Notebooks is possible with the open-source project Binder, which allows hosting of notebooks. Thus, Jupyter Notebooks can be shared in an executable form, which is independent from the operating system and the location of the user. Only an internet connection and a browser is required.
As a result, Jupyter Notebooks, which are executable through Binder, form a user-friendly environment for scientific data analysis. Due to the extensive documentation capabilities of Jupyter Notebooks, the analysis workflow is comprehensible and accessible for programmers as well as for non-programmers.</p>
</section>
<section id="jupyter-book">
<h3>Jupyter Book<a class="headerlink" href="#jupyter-book" title="Permalink to this headline">#</a></h3>
<p>The written thesis, except for the cover page, was entirely conceptualized as a Jupyter Book. Jupyter Book is an open-source tool that allows to incorporate computational material in publication-quality books <span id="id11">[<a class="reference internal" href="welcome.html#id38" title="Executable Books Community. Jupyter book. February 2020. URL: https://doi.org/10.5281/zenodo.4539666, doi:10.5281/zenodo.4539666.">Community, 2020</a>]</span>. Thereby, Jupyter Notebooks can natively be integrated in Jupyter Books, thus making all advantages of the notebook also available in the book.
Jupyter Books allow text formatting with Markdown, which is a simple markdown language. Furthermore, Jupyter Book supports cross-referencing, citations, and numbering of equations.
Together with all data, the final and interactive book was deployed on GitHub Pages whereas the print version was generated by exporting the html-version of the book to pdf.</p>
</section>
</section>
<span id="document-results"></span><section class="tex2jax_ignore mathjax_ignore" id="results">
<h2>Results<a class="headerlink" href="#results" title="Permalink to this headline">#</a></h2>
<section id="workflow-for-enzyme-kinetics-parameter-estimation">
<h3>Workflow for enzyme kinetics parameter estimation<a class="headerlink" href="#workflow-for-enzyme-kinetics-parameter-estimation" title="Permalink to this headline">#</a></h3>
<p>In this work, a Python-based analysis workflow for kinetic parameter estimation of enzyme reactions was developed. The workflow itself consists of the steps (i) data acquisition, (ii) raw data preparation, (iii) concentration calculation, (iv) quality control, (v) modeling, and (vi) saving of results. For precise concentration calculations as well as kinetic parameter estimation the two Python modules CaliPytion and EnzymePynetics were respectively developed over the course of this work.
The two packages were designed to seamlessly interact with EnzymeML documents which were modified using PyEnzyme and thus allowing a continuous data stream from analytical raw data to kinetic parameters.<br />
Starting from raw data of photometric measurements, which are stored in an EnzymeML document, concentration values are calculated by CaliPytion, whereupon kinetic parameters are determined by EnzymePynetics. Finally, the modeling results are written back to the EnzymeML document via PyEnzyme (Fig. 1).</p>
<p><img alt="Fig. 1" src="_images/concept_workflow.png" />
<em>Fig. 1: Components of the kinetic parameter estimation workflow for enzyme reaction data.</em></p>
<p>A key aspect of the workflow design was to enable reproducible analysis. Therefore, documentation of raw data treatment prior to modeling is vital. As a result, Jupyter Notebooks were used as a platform, since they allow to write text and code within the same document.</p>
<section id="calipytion">
<h4>CaliPytion<a class="headerlink" href="#calipytion" title="Permalink to this headline">#</a></h4>
<p>CaliPytion was developed to facilitate robust conversion of measurement data to concentration data. Based on an analyte standard, a calibration curve is created. Since linear and non-linear relations between analyte concentration and analytic signal may occur, <span id="id1">[<a class="reference internal" href="welcome.html#id14" title="Kang-Hung Hsu and Chiachung Chen. The effect of calibration equations on the uncertainty of uv–vis spectrophotometric measurements. Measurement, 43(10):1525–1531, 2010.">Hsu and Chen, 2010</a>]</span>, <span id="id2">[<a class="reference internal" href="welcome.html#id15" title="Julia Martin, Alberto Romero Gracia, Agustín G Asuero, and others. Fitting nonlinear calibration curves: no models perfect. Journal of Analytical Sciences, Methods and Instrumentation, 7(01):1, 2017.">Martin <em>et al.</em>, 2017</a>]</span>, linear, polynomial, exponential, and rational calibration equations were implemented. After fitting, the best model is preselected based on Akaike information criterion <span id="id3">[<a class="reference internal" href="welcome.html#id17" title="Hirotogu Akaike. Information theory and an extension of the maximum likelihood principle. In Selected papers of hirotugu akaike, pages 199–213. Springer, 1998.">Akaike, 1998</a>]</span>, offering a metric to compare models with different number of parameters. Thereby, AIC describes the information loss relative to alternative models, while penalizing the use of additional parameters. CaliPytion was designed to work with PyEnzyme. Hence, a <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> of CaliPytion can natively by applied to an <code class="docutils literal notranslate"><span class="pre">EnzymeMLDocument</span></code> of PyEnzyme. As a result, absorption data of an EnzymeML document can be converted into concentration data based on the previously generated standard curve.</p>
</section>
<section id="enzymepynetics">
<h4>EnzymePynetics<a class="headerlink" href="#enzymepynetics" title="Permalink to this headline">#</a></h4>
<p>EnzymePynetics was developed to enable easy applicable parameter estimation of single substrate enzyme reactions directly from EnzymeML documents. At its core, the <code class="docutils literal notranslate"><span class="pre">ParameterEstimator</span></code> of EnzymePynetics fits experimental data to different Michaels-Menten type models. Besides the irreversible Michaelis-Menten model, competitive, non-competitive, uncompetitive, and partially competitive inhibition models were implemented. Hence, the inhibition constant of potentially inhibiting substrate or product can be estimated. Furthermore, the inhibition constant of an applied inhibitor can be quantified. Since, the experimental data is fitted against multiple kinetic models, the best fitting model is suggested based on AIC. Hence, a proposal of the enzyme’s kinetic mechanism is given based on the best fitting model.
For quality control, visualization of experimental data as well as the fitted kinetic model was a priority during development. Thereby, detection of systematic deviation between model and measurement data is possible. Furthermore, the <code class="docutils literal notranslate"><span class="pre">ParameterEstimator</span></code> allows to subset the measurement data by time as well as initial substrate concentration without deleting data. Hence, identified systematic deviations (e.g. lag-phases or measurements with incorrect enzyme concentrations) can be excluded from parameter estimation.<br />
Like the <code class="docutils literal notranslate"><span class="pre">StandardCurve</span></code> of CaliPytion, the <code class="docutils literal notranslate"><span class="pre">ParameterEstimator</span></code> of EnzymePynetics natively supports <code class="docutils literal notranslate"><span class="pre">EnzymeMLDocument</span></code>s as a data source.</p>
</section>
</section>
<section id="scenario-driven-workflow-development">
<h3>Scenario-driven workflow development<a class="headerlink" href="#scenario-driven-workflow-development" title="Permalink to this headline">#</a></h3>
<p>The development of the workflow was driven by different research scenarios of EnzymeML project partners. The goal for all projects was to reliably determine kinetic parameters of enzyme reactions.
The collaboration consisted of multiple rounds of lab experiments (preformed by the experimental partners) and kinetic modeling of the experimental data. Each round included (i) wet lab experiments, (ii) kinetic modeling, (iii) design of follow-up experiments, and (iv) discussion of results with the respective project partners. Hence, a short feedback loop between lab experiments and modeling-based experimental suggestions was established.<br />
In parallel, the Python modules CaliPytion and EnzymePynetics were developed. Thereby, individual requirements of each research scenario fostered the implementation of various functionalities, resulting in a feature-rich yet generic workflow.</p>
</section>
<section id="fair-data-analysis-with-jupyter-notebooks">
<h3>FAIR data analysis with Jupyter Notebooks<a class="headerlink" href="#fair-data-analysis-with-jupyter-notebooks" title="Permalink to this headline">#</a></h3>
<p>All data analysis of the project was carried out in Jupyter Notebook by applying the developed workflow for kinetic parameter estimation. Within the next chapters, the latest results of all four experimental scenarios are shown. Thereby, the applicability of the workflow is demonstrated while analyzing and discussing the results of the respective scenario.
Since the analysis was conducted in Jupyter Notebooks, each of the following chapter is a executed Jupyter Notebook. Hence all figures and tables for data visualization are generated at runtime of the analysis.
Thereby, the applicability of Jupyter Notebooks for reproducible and comprehensive computational workflows in a scientific environment is demonstrated.
Every scenario consists of a short description of the project’s background and methodology carried out by the project partners in the laboratory. Additionally, data preparation, kinetic modeling steps, as well as the results are shown and project specific results are discussed.
All notebooks can be launched interactively by clicking on the <span class="fa fa-rocket"></span>-icon in the upper section of each scenario page. As a result, the analysis itself is transparent, and the developed workflow is demonstrated in its native environment. Therefore, it is highly recommended to <a class="reference external" href="https://haeussma.github.io/masterthesis/welcome.html">read this work in its conceptualized form</a>.</p>
</section>
<div class="toctree-wrapper compound">
<span id="document-scenarios/chymotrypsin_inhibition"></span><section class="tex2jax_ignore mathjax_ignore" id="scenario-a-br-chymotrypsin-inhibiton-by-an-in-silico-designed-albumin-fusion-protein">
<h3><strong>Scenario A:</strong><br>Chymotrypsin inhibiton by an <em>in silico</em> designed albumin fusion protein<a class="headerlink" href="#scenario-a-br-chymotrypsin-inhibiton-by-an-in-silico-designed-albumin-fusion-protein" title="Permalink to this headline">#</a></h3>
<p>Data provided by Marwa Mohamed (Institute of Cell Biology and Immunology, University of Stuttgart, Germany)</p>
<section id="project-background">
<h4>Project background<a class="headerlink" href="#project-background" title="Permalink to this headline">#</a></h4>
<p>In this scenario, the binding of an <em>in silico</em> designed protein to an enzyme was assessed by determining the  inhibitory constant <span class="math notranslate nohighlight">\(K_{i}\)</span>. Thereby, the efficiency of the newly developed approach for computational design of protein binders to deliberately chosen targets was demonstrated.
This was done by comparing <span class="math notranslate nohighlight">\(K_{i}\)</span> of the designed human serum albumin variant (HSA(M3)) to the wild-type (HSA(WT)) by respectively applying the proteins to chymotrypsin enzyme reactions. Both designed proteins were individually dimerized into fusion proteins through a huFc.</p>
<section id="experimental-design">
<h5>Experimental design<a class="headerlink" href="#experimental-design" title="Permalink to this headline">#</a></h5>
<p>Enzyme activity was monitored by measuring the product formation of p-Nitroanilin (p-NA) photometrically at 410 nm for 30 min at 30°C. Therefore, Succinyl-gly-gly-phe-p-nitroanilide (SGGPpNA) was applied as substrate in a concentration range of 0.25 - 2 mM. For concentration calculations, p-NA standard was prepared in the range of 0 - 0.3 mM in duplicates. <span class="math notranslate nohighlight">\(K_{i}\)</span> of HSA(WT)-huFc and HSA(M3)-huFc on chymotrypsin were investigated in independent experiments. Each experiment consisted of enzyme reactions with and without the respective HSA variant. Each enzyme reaction contained 0.2 µM of enzyme and 26.88 µM of the respective HSA variant, if inhibitor was applied. All enzyme reactions were prepared in duplicates.</p>
</section>
<section id="data-management">
<h5>Data management<a class="headerlink" href="#data-management" title="Permalink to this headline">#</a></h5>
<p>Experimental data and meta data was filled in EnzymeML Excel templates for each of the two inhibition experiments respectively. Calibration data was stored as Excel files. Measurement data was already blanked.</p>
</section>
</section>
<section id="data-preparation">
<h4>Data preparation<a class="headerlink" href="#data-preparation" title="Permalink to this headline">#</a></h4>
<section id="imports">
<h5>Imports<a class="headerlink" href="#imports" title="Permalink to this headline">#</a></h5>
<div class="cell tag_hide_cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">pyenzyme</span> <span class="k">as</span> <span class="nn">pe</span>
<span class="kn">import</span> <span class="nn">copy</span>
<span class="kn">import</span> <span class="nn">string</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span>
<span class="kn">from</span> <span class="nn">EnzymePynetics.tools.parameterestimator</span> <span class="kn">import</span> <span class="n">ParameterEstimator</span>
<span class="kn">from</span> <span class="nn">CaliPytion</span> <span class="o">.</span><span class="n">tools</span><span class="o">.</span><span class="n">standardcurve</span> <span class="kn">import</span> <span class="nn">StandardCurve</span>

<span class="kn">import</span> <span class="nn">warnings</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s1">&#39;ignore&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="concentration-calculation">
<h5>Concentration calculation<a class="headerlink" href="#concentration-calculation" title="Permalink to this headline">#</a></h5>
<p>Product standard data was imported directly from an Excel file. Then, a standard curve was created.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">product_standard</span> <span class="o">=</span> <span class="n">StandardCurve</span><span class="o">.</span><span class="n">from_excel</span><span class="p">(</span>
    <span class="n">path</span><span class="o">=</span><span class="s2">&quot;../../data/chymotrypsin_inhibition/pNA-standard.xlsx&quot;</span><span class="p">,</span>
    <span class="n">reactant_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">,</span>
    <span class="n">wavelength</span><span class="o">=</span><span class="mi">410</span><span class="p">,</span>
    <span class="n">sheet_name</span><span class="o">=</span><span class="s2">&quot;csv&quot;</span><span class="p">,</span> 
    <span class="n">concentration_unit</span><span class="o">=</span><span class="s2">&quot;mmole / l&quot;</span><span class="p">,</span>
    <span class="n">temperature</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
    <span class="n">temperature_unit</span><span class="o">=</span><span class="s2">&quot;C&quot;</span><span class="p">)</span>
    
<span class="n">product_standard</span><span class="o">.</span><span class="n">visualize</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Calibration data was automatically blanked.
</pre></div>
</div>
<div class="output text_html"><style type="text/css">
</style>
<table id="T_0453b" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_0453b_level0_col0" class="col_heading level0 col0" >AIC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_0453b_level0_row0" class="row_heading level0 row0" >Quadratic</th>
      <td id="T_0453b_row0_col0" class="data row0 col0" >-161</td>
    </tr>
    <tr>
      <th id="T_0453b_level0_row1" class="row_heading level0 row1" >3rd polynominal</th>
      <td id="T_0453b_row1_col0" class="data row1 col0" >-161</td>
    </tr>
    <tr>
      <th id="T_0453b_level0_row2" class="row_heading level0 row2" >Linear</th>
      <td id="T_0453b_row2_col0" class="data row2 col0" >-143</td>
    </tr>
    <tr>
      <th id="T_0453b_level0_row3" class="row_heading level0 row3" >Rational</th>
      <td id="T_0453b_row3_col0" class="data row3 col0" >-141</td>
    </tr>
    <tr>
      <th id="T_0453b_level0_row4" class="row_heading level0 row4" >Exponential</th>
      <td id="T_0453b_row4_col0" class="data row4 col0" >-17</td>
    </tr>
  </tbody>
</table>
</div><img alt="_images/chymotrypsin_inhibition_3_2.png" src="_images/chymotrypsin_inhibition_3_2.png" />
</div>
</div>
<p><em>Fig. 2: Fitted quadratic calibration model to standard data of p-NA.</em></p>
<p>Based on the Akaike information criterion (AIC), the relation between concentration and absorption is best described by a quadratic function. The calibration measurements and the fitted calibration model are shown in Fig. 2.</p>
</section>
<section id="experimental-data">
<h5>Experimental data<a class="headerlink" href="#experimental-data" title="Permalink to this headline">#</a></h5>
<p>The EnzymeML documents of each experiment were loaded and the standard curve was applied to the absorption data to calculate concentrations.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load data from </span>
<span class="n">chymo_HSAwt</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="o">.</span><span class="n">fromTemplate</span><span class="p">(</span><span class="s2">&quot;../../data/chymotrypsin_inhibition/chymo_HSAwt.xlsx&quot;</span><span class="p">)</span>
<span class="n">chymo_HSAM3</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="o">.</span><span class="n">fromTemplate</span><span class="p">(</span><span class="s2">&quot;../../data/chymotrypsin_inhibition/chymo_HSA(M3).xlsx&quot;</span><span class="p">)</span>

<span class="c1"># Apply standard curve to &#39;EnzymeMLDocument&#39;</span>
<span class="n">chymo_HSAwt</span> <span class="o">=</span> <span class="n">product_standard</span><span class="o">.</span><span class="n">apply_to_EnzymeML</span><span class="p">(</span><span class="n">chymo_HSAwt</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">)</span>
<span class="n">chymo_HSAM3</span> <span class="o">=</span> <span class="n">product_standard</span><span class="o">.</span><span class="n">apply_to_EnzymeML</span><span class="p">(</span><span class="n">chymo_HSAM3</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="comparability-of-the-experiments">
<h4>Comparability of the experiments<a class="headerlink" href="#comparability-of-the-experiments" title="Permalink to this headline">#</a></h4>
<p>Since experimental data with HSA(WT)-huFc and HSA(M3)-huFc originate from independent experiments, the control reactions without the respective inhibitor were compared by performing a parameter estimation (Fig. 3). Thereby, catalytic efficiency <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span> was used to assess comparability between the data sets, since <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> were highly correlated (corr &gt; 0.98). High correlations between parameters indicate that the parameters cannot be determined independently with certainty. In this case, the highest initial substrate concentration is presumably too low, compared to the true <span class="math notranslate nohighlight">\(K_{m}\)</span> of the enzyme under the given experimental conditions. However, higher substrate concentration were not applied for multiple reasons. On the one hand dimethyl sulfoxide (DMSO) was used as a co-solvent of the substrate, which inhibits enzyme activity <span id="id1">[<a class="reference internal" href="welcome.html#id9" title="William F Busby, Joseph M Ackermann, and Charles L Crespi. Effect of methanol, ethanol, dimethyl sulfoxide, and acetonitrile on in vitro activities of cdna-expressed human cytochromes p-450. Drug Metabolism and Disposition, 27(2):246–249, 1999.">Busby <em>et al.</em>, 1999</a>]</span>. Hence, higher initial substrate concentrations would have led to higher enzyme inhibition, which would have distorted the assessment of <span class="math notranslate nohighlight">\(K_{i}\)</span>.
On the other hand, high substrate viscosity denied the application of higher concentrations without sacrificing pipetting precision.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create copies of the data sets and delete measurements with inhibitor.</span>
<span class="n">wt_control</span> <span class="o">=</span> <span class="n">copy</span><span class="o">.</span><span class="n">deepcopy</span><span class="p">(</span><span class="n">chymo_HSAwt</span><span class="p">)</span>
<span class="k">del</span> <span class="n">wt_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m4&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">wt_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m5&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">wt_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m6&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">wt_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m7&quot;</span><span class="p">]</span>

<span class="n">m3_control</span> <span class="o">=</span> <span class="n">copy</span><span class="o">.</span><span class="n">deepcopy</span><span class="p">(</span><span class="n">chymo_HSAM3</span><span class="p">)</span>
<span class="k">del</span> <span class="n">m3_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m4&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">m3_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m5&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">m3_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m6&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">m3_control</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m7&quot;</span><span class="p">]</span>

<span class="c1"># Estimate kinetic parameters of the control reactions of the HSA wild-type data set.</span>
<span class="n">kinetics_wt_control</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">wt_control</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="s2">&quot;product&quot;</span><span class="p">)</span>
<span class="n">kinetics_wt_control</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">stop_time_index</span><span class="o">=-</span><span class="mi">1</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Kinetic parameters of HSA(wt) chymotrypsin control reactions:&quot;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">kinetics_wt_control</span><span class="o">.</span><span class="n">result_dict</span>\
    <span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">set_table_attributes</span><span class="p">(</span><span class="s1">&#39;style=&quot;font-size: 12px&quot;&#39;</span><span class="p">))</span>


<span class="c1"># Estimate kinetic parameters of the control reactions of the HSA(M3) data set.</span>
<span class="n">kinetics_m3_control</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">m3_control</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="s2">&quot;product&quot;</span><span class="p">)</span>
<span class="n">kinetics_m3_control</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">stop_time_index</span><span class="o">=-</span><span class="mi">1</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">Kinetic parameters of HSA(M3) chymotrypsin control reactions:&quot;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">kinetics_m3_control</span><span class="o">.</span><span class="n">result_dict</span>\
    <span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">set_table_attributes</span><span class="p">(</span><span class="s1">&#39;style=&quot;font-size: 12px&quot;&#39;</span><span class="p">))</span>


<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.8</span><span class="p">,</span><span class="mf">4.8</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">for</span> <span class="n">e</span><span class="p">,</span> <span class="p">(</span><span class="n">doc</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">([</span><span class="n">kinetics_wt_control</span> <span class="p">,</span><span class="n">kinetics_m3_control</span><span class="p">],</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">(),</span> <span class="p">[</span><span class="s2">&quot;chymotrypsin control reactions HSA(wt)&quot;</span><span class="p">,</span> <span class="s2">&quot;chymotrypsin control reactions HSA(M3)&quot;</span><span class="p">])):</span>
    <span class="n">doc</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;4-nitroanilin [mM]&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time after reaction start [min]&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([</span><span class="mi">5</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="mi">15</span><span class="p">,</span> <span class="mi">20</span><span class="p">])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="n">e</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
            <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>

<span class="n">handles</span><span class="p">,</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>

<span class="n">fig</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">handles</span><span class="p">,</span> <span class="n">labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;lower center&quot;</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial SGGPpNA  [mM]&quot;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">0.5</span><span class="p">,</span><span class="o">-</span><span class="mf">0.15</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Kinetic parameters of HSA(wt) chymotrypsin control reactions:
</pre></div>
</div>
<div class="output text_html"><style type="text/css">
</style>
<table id="T_03a22" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_03a22_level0_col0" class="col_heading level0 col0" >AIC</th>
      <th id="T_03a22_level0_col1" class="col_heading level0 col1" >kcat [1/min]</th>
      <th id="T_03a22_level0_col2" class="col_heading level0 col2" >Km [mmole / l]</th>
      <th id="T_03a22_level0_col3" class="col_heading level0 col3" >kcat / Km [1/min * 1/mmole / l]</th>
      <th id="T_03a22_level0_col4" class="col_heading level0 col4" >Ki competitive [mmole / l]</th>
      <th id="T_03a22_level0_col5" class="col_heading level0 col5" >Ki uncompetitive [mmole / l]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_03a22_level0_row0" class="row_heading level0 row0" >irreversible Michaelis Menten</th>
      <td id="T_03a22_row0_col0" class="data row0 col0" >-1857</td>
      <td id="T_03a22_row0_col1" class="data row0 col1" >56.440 +/- 4.52%</td>
      <td id="T_03a22_row0_col2" class="data row0 col2" >2.226 +/- 7.28%</td>
      <td id="T_03a22_row0_col3" class="data row0 col3" >25.353 +/- 8.57%</td>
      <td id="T_03a22_row0_col4" class="data row0 col4" >-</td>
      <td id="T_03a22_row0_col5" class="data row0 col5" >-</td>
    </tr>
    <tr>
      <th id="T_03a22_level0_row1" class="row_heading level0 row1" >competitive product inhibition</th>
      <td id="T_03a22_row1_col0" class="data row1 col0" >-1856</td>
      <td id="T_03a22_row1_col1" class="data row1 col1" >60.225 +/- 9.01%</td>
      <td id="T_03a22_row1_col2" class="data row1 col2" >2.342 +/- 9.88%</td>
      <td id="T_03a22_row1_col3" class="data row1 col3" >25.714 +/- 13.37%</td>
      <td id="T_03a22_row1_col4" class="data row1 col4" >0.725 +/- 112.65%</td>
      <td id="T_03a22_row1_col5" class="data row1 col5" >-</td>
    </tr>
    <tr>
      <th id="T_03a22_level0_row2" class="row_heading level0 row2" >uncompetitive product inhibition</th>
      <td id="T_03a22_row2_col0" class="data row2 col0" >-1855</td>
      <td id="T_03a22_row2_col1" class="data row2 col1" >57.529 +/- 8.30%</td>
      <td id="T_03a22_row2_col2" class="data row2 col2" >2.276 +/- 10.65%</td>
      <td id="T_03a22_row2_col3" class="data row2 col3" >25.272 +/- 13.50%</td>
      <td id="T_03a22_row2_col4" class="data row2 col4" >-</td>
      <td id="T_03a22_row2_col5" class="data row2 col5" >3.175 +/- 428.92%</td>
    </tr>
    <tr>
      <th id="T_03a22_level0_row3" class="row_heading level0 row3" >substrate inhibition</th>
      <td id="T_03a22_row3_col0" class="data row3 col0" >-1855</td>
      <td id="T_03a22_row3_col1" class="data row3 col1" >56.585 +/- 29.14%</td>
      <td id="T_03a22_row3_col2" class="data row3 col2" >2.233 +/- 35.51%</td>
      <td id="T_03a22_row3_col3" class="data row3 col3" >25.339 +/- 45.93%</td>
      <td id="T_03a22_row3_col4" class="data row3 col4" >-</td>
      <td id="T_03a22_row3_col5" class="data row3 col5" >997.715 +/- 12275.11%</td>
    </tr>
    <tr>
      <th id="T_03a22_level0_row4" class="row_heading level0 row4" >non-competitive product inhibition</th>
      <td id="T_03a22_row4_col0" class="data row4 col0" >-1854</td>
      <td id="T_03a22_row4_col1" class="data row4 col1" >60.110 +/- 9.50%</td>
      <td id="T_03a22_row4_col2" class="data row4 col2" >2.331 +/- 10.00%</td>
      <td id="T_03a22_row4_col3" class="data row4 col3" >25.787 +/- 13.79%</td>
      <td id="T_03a22_row4_col4" class="data row4 col4" >0.702 +/- 117.81%</td>
      <td id="T_03a22_row4_col5" class="data row4 col5" >179.118 +/- 358.38%</td>
    </tr>
  </tbody>
</table>
</div><div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Kinetic parameters of HSA(M3) chymotrypsin control reactions:
</pre></div>
</div>
<div class="output text_html"><style type="text/css">
</style>
<table id="T_b0bea" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_b0bea_level0_col0" class="col_heading level0 col0" >AIC</th>
      <th id="T_b0bea_level0_col1" class="col_heading level0 col1" >kcat [1/min]</th>
      <th id="T_b0bea_level0_col2" class="col_heading level0 col2" >Km [mmole / l]</th>
      <th id="T_b0bea_level0_col3" class="col_heading level0 col3" >kcat / Km [1/min * 1/mmole / l]</th>
      <th id="T_b0bea_level0_col4" class="col_heading level0 col4" >Ki competitive [mmole / l]</th>
      <th id="T_b0bea_level0_col5" class="col_heading level0 col5" >Ki uncompetitive [mmole / l]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_b0bea_level0_row0" class="row_heading level0 row0" >irreversible Michaelis Menten</th>
      <td id="T_b0bea_row0_col0" class="data row0 col0" >-549</td>
      <td id="T_b0bea_row0_col1" class="data row0 col1" >48.996 +/- 2.23%</td>
      <td id="T_b0bea_row0_col2" class="data row0 col2" >1.978 +/- 3.74%</td>
      <td id="T_b0bea_row0_col3" class="data row0 col3" >24.776 +/- 4.35%</td>
      <td id="T_b0bea_row0_col4" class="data row0 col4" >-</td>
      <td id="T_b0bea_row0_col5" class="data row0 col5" >-</td>
    </tr>
    <tr>
      <th id="T_b0bea_level0_row1" class="row_heading level0 row1" >competitive product inhibition</th>
      <td id="T_b0bea_row1_col0" class="data row1 col0" >-549</td>
      <td id="T_b0bea_row1_col1" class="data row1 col1" >51.227 +/- 4.39%</td>
      <td id="T_b0bea_row1_col2" class="data row1 col2" >2.043 +/- 4.76%</td>
      <td id="T_b0bea_row1_col3" class="data row1 col3" >25.077 +/- 6.47%</td>
      <td id="T_b0bea_row1_col4" class="data row1 col4" >0.908 +/- 81.63%</td>
      <td id="T_b0bea_row1_col5" class="data row1 col5" >-</td>
    </tr>
    <tr>
      <th id="T_b0bea_level0_row2" class="row_heading level0 row2" >uncompetitive product inhibition</th>
      <td id="T_b0bea_row2_col0" class="data row2 col0" >-547</td>
      <td id="T_b0bea_row2_col1" class="data row2 col1" >49.384 +/- 5.33%</td>
      <td id="T_b0bea_row2_col2" class="data row2 col2" >1.996 +/- 6.71%</td>
      <td id="T_b0bea_row2_col3" class="data row2 col3" >24.743 +/- 8.57%</td>
      <td id="T_b0bea_row2_col4" class="data row2 col4" >-</td>
      <td id="T_b0bea_row2_col5" class="data row2 col5" >7.445 +/- 631.41%</td>
    </tr>
    <tr>
      <th id="T_b0bea_level0_row3" class="row_heading level0 row3" >non-competitive product inhibition</th>
      <td id="T_b0bea_row3_col0" class="data row3 col0" >-547</td>
      <td id="T_b0bea_row3_col1" class="data row3 col1" >51.241 +/- 5.62%</td>
      <td id="T_b0bea_row3_col2" class="data row3 col2" >2.044 +/- 6.62%</td>
      <td id="T_b0bea_row3_col3" class="data row3 col3" >25.073 +/- 8.69%</td>
      <td id="T_b0bea_row3_col4" class="data row3 col4" >0.914 +/- 83.56%</td>
      <td id="T_b0bea_row3_col5" class="data row3 col5" >121.556 +/- 2156.85%</td>
    </tr>
    <tr>
      <th id="T_b0bea_level0_row4" class="row_heading level0 row4" >substrate inhibition</th>
      <td id="T_b0bea_row4_col0" class="data row4 col0" >-545</td>
      <td id="T_b0bea_row4_col1" class="data row4 col1" >50.368 +/- 3.78%</td>
      <td id="T_b0bea_row4_col2" class="data row4 col2" >2.046 +/- 5.12%</td>
      <td id="T_b0bea_row4_col3" class="data row4 col3" >24.623 +/- 6.36%</td>
      <td id="T_b0bea_row4_col4" class="data row4 col4" >-</td>
      <td id="T_b0bea_row4_col5" class="data row4 col5" >87.549 +/- 121.66%</td>
    </tr>
  </tbody>
</table>
</div><img alt="_images/chymotrypsin_inhibition_8_4.png" src="_images/chymotrypsin_inhibition_8_4.png" />
</div>
</div>
<p><em>Fig. 3: Measurement data and fitted irreversible Michaelis-Menten model for chymotrypsin reactions without HSA inhibitor.</em></p>
<p>The two datasets were fitted against kinetic models, which are displayed in the tables above.
Each dataset is best described by the irreversible Michaelis-Menten model in terms of AIC and standard deviation on the estimated parameters. Models with product or substrate inhibition resulted in large uncertainties above 80 % on the parameter estimates. Therefore, irreversible Michaelis-Menten model was utilized for comparison of parameters. <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span> was estimated to be 25.353 min<sup>-1</sup>mM<sup>-1</sup> ± 8.57% for the control reaction of the HSA(WT)-huFc data set and 24.776 min<sup>-1</sup>mM<sup>-1</sup> ± 4.35% for the HSA(M3)-huFc data set. As a result, the two experiments showed to be comparable, since the catalytic efficiency differs less than 3 % between the two data sets.</p>
</section>
<section id="determination-and-comparison-of-k-i">
<h4>Determination and comparison of <span class="math notranslate nohighlight">\(K_{i}\)</span><a class="headerlink" href="#determination-and-comparison-of-k-i" title="Permalink to this headline">#</a></h4>
<p>Experimental data of chymotrypsin inhibition by HSA(M3)-huFc contained negative absorption values for the first measurement point, presumably sourcing from an incorrect blank measurement. Therefore, only measurement data from the second data point (minute 5 and onward) was considered for parameter estimation.
Parameter estimates for all applied kinetic models are displayed in the output below and visualized in Fig. 4.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Parameter estimation for HSA(wt) data set</span>
<span class="n">kinetics_HSAwt</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">chymo_HSAwt</span><span class="p">,</span> <span class="n">reactant_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="n">inhibitor_id</span><span class="o">=</span><span class="s2">&quot;s2&quot;</span><span class="p">,</span> <span class="n">measured_species</span><span class="o">=</span><span class="s2">&quot;product&quot;</span><span class="p">)</span>
<span class="n">kinetics_HSAwt</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">initial_substrate_concs</span><span class="o">=</span><span class="p">[</span><span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">],</span> <span class="n">stop_time_index</span><span class="o">=-</span><span class="mi">1</span><span class="p">,</span> <span class="n">start_time_index</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(WT)-huFc:&quot;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">kinetics_HSAwt</span><span class="o">.</span><span class="n">result_dict</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;kcat [1/min]&quot;</span><span class="p">,</span> <span class="s2">&quot;Km [mmole / l]&quot;</span><span class="p">])</span>\
    <span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">set_table_attributes</span><span class="p">(</span><span class="s1">&#39;style=&quot;font-size: 12px&quot;&#39;</span><span class="p">))</span>

<span class="c1"># Parameter estimation for HSA(M3) data set</span>
<span class="n">kinetics_HSAM3</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">chymo_HSAM3</span><span class="p">,</span> <span class="n">reactant_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="n">inhibitor_id</span><span class="o">=</span><span class="s2">&quot;s3&quot;</span><span class="p">,</span> <span class="n">measured_species</span><span class="o">=</span><span class="s2">&quot;product&quot;</span><span class="p">)</span>
<span class="n">kinetics_HSAM3</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">initial_substrate_concs</span><span class="o">=</span><span class="p">[</span><span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">],</span> <span class="n">stop_time_index</span><span class="o">=-</span><span class="mi">1</span><span class="p">,</span> <span class="n">start_time_index</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(M3)-huFc:&quot;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">kinetics_HSAM3</span><span class="o">.</span><span class="n">result_dict</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;kcat [1/min]&quot;</span><span class="p">,</span> <span class="s2">&quot;Km [mmole / l]&quot;</span><span class="p">])</span>\
    <span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">set_table_attributes</span><span class="p">(</span><span class="s1">&#39;style=&quot;font-size: 12px&quot;&#39;</span><span class="p">))</span>

<span class="c1"># Visualize experimental data and fitted models</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.8</span><span class="p">,</span><span class="mf">4.8</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">for</span> <span class="n">e</span><span class="p">,</span> <span class="p">(</span><span class="n">doc</span><span class="p">,</span> <span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">([</span><span class="n">kinetics_HSAwt</span> <span class="p">,</span><span class="n">kinetics_HSAM3</span><span class="p">],</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">(),</span> <span class="p">[</span><span class="s2">&quot;chymotrypsin inhibition by HSA(WT)-huFc&quot;</span><span class="p">,</span> <span class="s2">&quot;chymotrypsin inhibition by HSA(M3)-huFc&quot;</span><span class="p">])):</span>
    <span class="n">doc</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;4-nitroanilin [mM]&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time after reaction start [min]&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([</span><span class="mi">5</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="mi">15</span><span class="p">,</span> <span class="mi">20</span><span class="p">])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="n">e</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
            <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>

<span class="n">handles</span><span class="p">,</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>

<span class="n">fig</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">handles</span><span class="p">,</span> <span class="n">labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;lower center&quot;</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial SGGPpNA  [mM]&quot;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">0.5</span><span class="p">,</span><span class="o">-</span><span class="mf">0.2</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(WT)-huFc:
</pre></div>
</div>
<div class="output text_html"><style type="text/css">
</style>
<table id="T_d096d" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_d096d_level0_col0" class="col_heading level0 col0" >AIC</th>
      <th id="T_d096d_level0_col1" class="col_heading level0 col1" >kcat / Km [1/min * 1/mmole / l]</th>
      <th id="T_d096d_level0_col2" class="col_heading level0 col2" >Ki competitive [mmole / l]</th>
      <th id="T_d096d_level0_col3" class="col_heading level0 col3" >Ki uncompetitive [mmole / l]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_d096d_level0_row0" class="row_heading level0 row0" >competitive inhibition</th>
      <td id="T_d096d_row0_col0" class="data row0 col0" >-3100</td>
      <td id="T_d096d_row0_col1" class="data row0 col1" >22.253 +/- 6.97%</td>
      <td id="T_d096d_row0_col2" class="data row0 col2" >0.460 +/- 27.24%</td>
      <td id="T_d096d_row0_col3" class="data row0 col3" >-</td>
    </tr>
    <tr>
      <th id="T_d096d_level0_row1" class="row_heading level0 row1" >non-competitive inhibition</th>
      <td id="T_d096d_row1_col0" class="data row1 col0" >-3098</td>
      <td id="T_d096d_row1_col1" class="data row1 col1" >22.403 +/- 7.00%</td>
      <td id="T_d096d_row1_col2" class="data row1 col2" >0.449 +/- 27.64%</td>
      <td id="T_d096d_row1_col3" class="data row1 col3" >79.193 +/- 516.52%</td>
    </tr>
    <tr>
      <th id="T_d096d_level0_row2" class="row_heading level0 row2" >irreversible Michaelis Menten</th>
      <td id="T_d096d_row2_col0" class="data row2 col0" >-3088</td>
      <td id="T_d096d_row2_col1" class="data row2 col1" >21.622 +/- 7.15%</td>
      <td id="T_d096d_row2_col2" class="data row2 col2" >-</td>
      <td id="T_d096d_row2_col3" class="data row2 col3" >-</td>
    </tr>
    <tr>
      <th id="T_d096d_level0_row3" class="row_heading level0 row3" >uncompetitive inhibition</th>
      <td id="T_d096d_row3_col0" class="data row3 col0" >-3087</td>
      <td id="T_d096d_row3_col1" class="data row3 col1" >21.631 +/- 7.63%</td>
      <td id="T_d096d_row3_col2" class="data row3 col2" >-</td>
      <td id="T_d096d_row3_col3" class="data row3 col3" >0.696 +/- 88.99%</td>
    </tr>
    <tr>
      <th id="T_d096d_level0_row4" class="row_heading level0 row4" >partially competitive inhibition</th>
      <td id="T_d096d_row4_col0" class="data row4 col0" >-3084</td>
      <td id="T_d096d_row4_col1" class="data row4 col1" >21.643 +/- 8.23%</td>
      <td id="T_d096d_row4_col2" class="data row4 col2" >796.196 +/- 1075.02%</td>
      <td id="T_d096d_row4_col3" class="data row4 col3" >992.685 +/- 48.74%</td>
    </tr>
  </tbody>
</table>
</div><div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Kinetic parameters estimates for all models of chymotrypsin inhibition by HSA(M3)-huFc:
</pre></div>
</div>
<div class="output text_html"><style type="text/css">
</style>
<table id="T_d6dca" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_d6dca_level0_col0" class="col_heading level0 col0" >AIC</th>
      <th id="T_d6dca_level0_col1" class="col_heading level0 col1" >kcat / Km [1/min * 1/mmole / l]</th>
      <th id="T_d6dca_level0_col2" class="col_heading level0 col2" >Ki competitive [mmole / l]</th>
      <th id="T_d6dca_level0_col3" class="col_heading level0 col3" >Ki uncompetitive [mmole / l]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_d6dca_level0_row0" class="row_heading level0 row0" >competitive inhibition</th>
      <td id="T_d6dca_row0_col0" class="data row0 col0" >-808</td>
      <td id="T_d6dca_row0_col1" class="data row0 col1" >23.031 +/- 10.48%</td>
      <td id="T_d6dca_row0_col2" class="data row0 col2" >0.059 +/- 8.51%</td>
      <td id="T_d6dca_row0_col3" class="data row0 col3" >-</td>
    </tr>
    <tr>
      <th id="T_d6dca_level0_row1" class="row_heading level0 row1" >non-competitive inhibition</th>
      <td id="T_d6dca_row1_col0" class="data row1 col0" >-806</td>
      <td id="T_d6dca_row1_col1" class="data row1 col1" >23.002 +/- 10.89%</td>
      <td id="T_d6dca_row1_col2" class="data row1 col2" >0.060 +/- 9.04%</td>
      <td id="T_d6dca_row1_col3" class="data row1 col3" >44.965 +/- 302.77%</td>
    </tr>
    <tr>
      <th id="T_d6dca_level0_row2" class="row_heading level0 row2" >uncompetitive inhibition</th>
      <td id="T_d6dca_row2_col0" class="data row2 col0" >-751</td>
      <td id="T_d6dca_row2_col1" class="data row2 col1" >19.297 +/- 21.77%</td>
      <td id="T_d6dca_row2_col2" class="data row2 col2" >-</td>
      <td id="T_d6dca_row2_col3" class="data row2 col3" >0.035 +/- 21.33%</td>
    </tr>
    <tr>
      <th id="T_d6dca_level0_row3" class="row_heading level0 row3" >irreversible Michaelis Menten</th>
      <td id="T_d6dca_row3_col0" class="data row3 col0" >-713</td>
      <td id="T_d6dca_row3_col1" class="data row3 col1" >18.512 +/- 24.41%</td>
      <td id="T_d6dca_row3_col2" class="data row3 col2" >-</td>
      <td id="T_d6dca_row3_col3" class="data row3 col3" >-</td>
    </tr>
    <tr>
      <th id="T_d6dca_level0_row4" class="row_heading level0 row4" >partially competitive inhibition</th>
      <td id="T_d6dca_row4_col0" class="data row4 col0" >-709</td>
      <td id="T_d6dca_row4_col1" class="data row4 col1" >18.544 +/- 27.64%</td>
      <td id="T_d6dca_row4_col2" class="data row4 col2" >569.354 +/- 57.96%</td>
      <td id="T_d6dca_row4_col3" class="data row4 col3" >925.096 +/- 11733.22%</td>
    </tr>
  </tbody>
</table>
</div><img alt="_images/chymotrypsin_inhibition_10_4.png" src="_images/chymotrypsin_inhibition_10_4.png" />
</div>
</div>
<p><em>Fig. 4: Measurement data and fitted product inhibition model for chymotrypsin reactions with respective HSA inhibitior.</em></p>
<p>Both reaction systems are best described by the competitive inhibition model, which is indicated by the lowest AIC and standard deviation on the estimated parameters. Thereby, a <span class="math notranslate nohighlight">\(K_{i}\)</span> of 0.460 mM ± 27.24% was estimated for HSA(WT)-huFc and 0.059 mM ± 8.51% for HSA(M3)-huFc. This resembles a roughly 7-fold increase in affinity of HSA(M3) to the enzyme compared to the HSA(wt).
Since the competitive inhibition model describes the data the best, HSA(M3)-huFc presumably interacts with the enzyme in the active site region.</p>
</section>
<section id="save-modeling-results">
<h4>Save modeling results<a class="headerlink" href="#save-modeling-results" title="Permalink to this headline">#</a></h4>
<p>Lastly, the modeling results are written to the EnzymeML documents and the files are exported.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">result</span><span class="p">,</span> <span class="n">enzmldoc</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">([</span><span class="n">kinetics_HSAwt</span><span class="p">,</span> <span class="n">kinetics_HSAM3</span><span class="p">],</span> <span class="p">[</span><span class="n">chymo_HSAwt</span><span class="p">,</span> <span class="n">chymo_HSAM3</span><span class="p">]):</span>

    <span class="c1"># Write modeling results to kinetic parameters</span>
    <span class="n">k_cat</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;k_cat&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;k_cat&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;1 / </span><span class="si">{</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">time_unit</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

    <span class="n">K_m</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;K_m&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;Km&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">data_conc_unit</span><span class="p">)</span>

    <span class="n">K_ic</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;K_ic&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;K_ic&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">data_conc_unit</span><span class="p">)</span>

    <span class="c1"># Define kinetic model</span>
    <span class="n">model</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">KineticModel</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;competitive inhibition&quot;</span><span class="p">,</span>
        <span class="n">equation</span><span class="o">=</span><span class="s2">&quot;-k_cat * p0 * s0 / (K_m*(1+(s2 / K_ic))+s0)&quot;</span><span class="p">,</span>
        <span class="n">parameters</span><span class="o">=</span><span class="p">[</span><span class="n">k_cat</span><span class="p">,</span> <span class="n">K_m</span><span class="p">,</span> <span class="n">K_ic</span><span class="p">])</span>

    <span class="n">enzmldoc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">model</span> <span class="o">=</span> <span class="n">model</span>

    <span class="c1"># Export EnzymeML documents</span>
    <span class="n">export</span> <span class="o">=</span> <span class="kc">False</span>
    <span class="k">if</span> <span class="n">export</span><span class="p">:</span>
        <span class="n">enzmldoc</span><span class="o">.</span><span class="n">toFile</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<span id="document-scenarios/glycosidase_inhibition"></span><section class="tex2jax_ignore mathjax_ignore" id="scenario-b-br-glucosidase-inhibition-by-fucoidan">
<h3><strong>Scenario B:</strong><br>α-glucosidase inhibition by fucoidan<a class="headerlink" href="#scenario-b-br-glucosidase-inhibition-by-fucoidan" title="Permalink to this headline">#</a></h3>
<p>Data provided by Chantal Daub (Biochemistry, Rhodes University, Makhanda, South Africa)</p>
<section id="project-background">
<h4>Project background<a class="headerlink" href="#project-background" title="Permalink to this headline">#</a></h4>
<p>In this scenario, the inhibitory properties of fucoidan on <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase from <em>Saccharomyces cerevisiae</em> were analyzed. Fucoidan is a sulfated polysaccharide found in various brown algae. The polysaccharide is investigated as a potential active compound in the fields of anti-cancer, anti-inflammation, and anti-coagulate research, among others <span id="id1">[<a class="reference internal" href="welcome.html#id7" title="Bo Li, Fei Lu, Xinjun Wei, and Ruixiang Zhao. Fucoidan: structure and bioactivity. Molecules, 13(8):1671–1695, 2008.">Li <em>et al.</em>, 2008</a>]</span>. Furthermore, fucoidan showed to be a promising drug for diabetes mellitus treatment, since it effectively inhibits <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase. Recently, <span id="id2">Daub <em>et al.</em> [<a class="reference internal" href="welcome.html#id6" title="Chantal Désirée Daub, Blessing Mabate, Samkelo Malgas, and Brett Ivan Pletschke. Fucoidan from ecklonia maxima is a powerful inhibitor of the diabetes-related enzyme, α-glucosidase. International journal of biological macromolecules, 151:412–420, 2020.">2020</a>]</span> showed that <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase inhibition by fucoidan from <em>E. maxima</em> exhibited significantly stronger inhibition, compared to the established diabetes drug acarbose.<br />
In the following analysis, <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase was exposed to fucoidan from <em>Ecklonia maxima</em>, <em>Ecklonia radiata</em>, <em>Fucus vesiculosus</em>, and <em>Schimmelmannia elegans</em> to test their respective capability of <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase inhibiton by assessing their respective <span class="math notranslate nohighlight">\(K_{i}\)</span> values.</p>
<section id="experimental-design">
<h5>Experimental design<a class="headerlink" href="#experimental-design" title="Permalink to this headline">#</a></h5>
<p><span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase reactions, catalyzing the hydrolysis of p-nitrophenyl glucopyranoside (p-NPG) to p-nitrophenol (p-NP) were conducted with and without fucoidan from each seaweed species as well as acarbose. Fucoidan was applied in two different concentrations. p-NPG was applied in a range from 0.1 mM to 5 mM, to enzyme reactions containing 9.19 µM <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase.
Product formation was recorded photometrically at 405 nm and 37°C for 20 min. Product concentrations were calculated utilizing a photometric p-NP standard. Additionally, control reactions without enzyme were prepared to subtract the absorption contribution of the respective inhibitor, buffer, enzyme and substrate from each measurement.</p>
</section>
</section>
<section id="data-preparation">
<h4>Data preparation<a class="headerlink" href="#data-preparation" title="Permalink to this headline">#</a></h4>
<section id="imports-and-parser-function">
<h5>Imports and parser function<a class="headerlink" href="#imports-and-parser-function" title="Permalink to this headline">#</a></h5>
<div class="cell tag_hide_cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">typing</span> <span class="kn">import</span> <span class="n">Dict</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">matplotlib.colors</span> <span class="k">as</span> <span class="nn">mcolors</span>
<span class="kn">import</span> <span class="nn">re</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pyenzyme</span> <span class="k">as</span> <span class="nn">pe</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span>
<span class="kn">from</span> <span class="nn">CaliPytion.tools.standardcurve</span> <span class="kn">import</span> <span class="n">StandardCurve</span>
<span class="kn">from</span> <span class="nn">EnzymePynetics.tools.parameterestimator</span> <span class="kn">import</span> <span class="n">ParameterEstimator</span>


<span class="kn">import</span> <span class="nn">warnings</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s1">&#39;ignore&#39;</span><span class="p">)</span>

<span class="n">colors</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">mcolors</span><span class="o">.</span><span class="n">TABLEAU_COLORS</span><span class="o">.</span><span class="n">values</span><span class="p">())</span>

<span class="c1"># Parser</span>
<span class="k">def</span> <span class="nf">measurement_data_to_EnzymeML</span><span class="p">(</span>
    <span class="n">template_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">measurement_data</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">,</span>
    <span class="n">species_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">time</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">,</span>
    <span class="n">data_unit</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">time_unit</span><span class="p">:</span> <span class="nb">str</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="p">:</span>

    <span class="n">enzmldoc</span><span class="p">:</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="o">.</span><span class="n">fromTemplate</span><span class="p">(</span><span class="n">template_path</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">IDs</span><span class="p">,</span> <span class="n">concentration</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="o">.</span><span class="n">keys</span><span class="p">(),</span> <span class="n">measurement_data</span><span class="p">):</span>
        <span class="k">for</span> <span class="n">counter</span><span class="p">,</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">concentration</span><span class="p">):</span>
            
            <span class="n">enzmldoc</span><span class="o">.</span><span class="n">getMeasurement</span><span class="p">(</span><span class="n">IDs</span><span class="p">)</span><span class="o">.</span><span class="n">addReplicates</span><span class="p">(</span><span class="n">pe</span><span class="o">.</span><span class="n">Replicate</span><span class="p">(</span>
                <span class="nb">id</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;Measurement</span><span class="si">{</span><span class="n">counter</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
                <span class="n">species_id</span><span class="o">=</span><span class="n">species_id</span><span class="p">,</span>
                <span class="n">data</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">replicate</span><span class="p">),</span>
                <span class="n">data_unit</span><span class="o">=</span><span class="n">data_unit</span><span class="p">,</span>
                <span class="n">time</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">time</span><span class="p">),</span>
                <span class="n">time_unit</span><span class="o">=</span><span class="n">time_unit</span><span class="p">),</span> <span class="n">enzmldoc</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">enzmldoc</span>

<span class="c1"># Ignore hidden files in file stystem</span>
<span class="k">def</span> <span class="nf">listdir_nohidden</span><span class="p">(</span><span class="n">path</span><span class="p">):</span>
    <span class="k">for</span> <span class="n">f</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">path</span><span class="p">):</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">f</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;.&#39;</span><span class="p">):</span>
            <span class="k">yield</span> <span class="n">f</span>
</pre></div>
</div>
</div>
</div>
<p>Measurement data was provided as an Excel file, whereas metadata was filled in EnzymeML Excel templates for each fucoidan seaweed species and acarbose respectively. In preliminary experiments, p-NPG showed to slightly absorb at the product detection wavelength. Therefore, the absorbance contribution of substrate at the product detection wavelength was subtracted as well as the contributions of enzyme, buffer and inhibitor. Then, the blanked absorbance data was written to the EnzymeML documents by a parser function.</p>
<div class="cell tag_hide_cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dataset_path</span> <span class="o">=</span> <span class="s2">&quot;../../data/glucosidase_inhibition/experimental_data_real.xlsx&quot;</span>
<span class="n">template_directory</span> <span class="o">=</span> <span class="s2">&quot;../../data/glucosidase_inhibition/EnzymeML_templates&quot;</span>   

<span class="n">enzml_docs</span> <span class="o">=</span> <span class="p">[]</span>
<span class="c1"># Load experimental data from Excel</span>
<span class="n">excel_sheets</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">ExcelFile</span><span class="p">(</span><span class="n">dataset_path</span><span class="p">)</span><span class="o">.</span><span class="n">sheet_names</span><span class="p">)</span>
<span class="n">inhibitors</span> <span class="o">=</span> <span class="n">excel_sheets</span><span class="p">[:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">substrate_controls</span> <span class="o">=</span> <span class="n">excel_sheets</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
<span class="n">initial_substrates</span> <span class="o">=</span> <span class="p">[</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mf">2.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">]</span> <span class="c1"># mM</span>

<span class="c1"># Blank data</span>
<span class="c1">## Absorption contribution from substrate</span>
<span class="n">substrate_absorption_data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">dataset_path</span><span class="p">,</span> <span class="n">sheet_name</span><span class="o">=</span><span class="n">substrate_controls</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;time&quot;</span><span class="p">)</span>
<span class="n">buffer_enzyme_absorption</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">substrate_absorption_data</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="mi">0</span><span class="p">])</span>
<span class="n">substrate_absorptions</span> <span class="o">=</span> <span class="n">substrate_absorption_data</span><span class="o">.</span><span class="n">subtract</span><span class="p">(</span><span class="n">buffer_enzyme_absorption</span><span class="p">)</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;Buffer+ Enzyme&quot;</span><span class="p">])</span>
<span class="n">substrate_absorptions</span> <span class="o">=</span> <span class="n">substrate_absorptions</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">21</span><span class="p">)</span>
<span class="n">substrate_absorptions</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">substrate_absorptions</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">substrate_absorptions</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">substrate_absorptions</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">mapper_substrate_enzyme_absorption</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">initial_substrates</span><span class="p">,</span> <span class="n">substrate_absorptions</span><span class="p">))</span>

<span class="k">for</span> <span class="n">inhibitor</span><span class="p">,</span> <span class="n">template</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="nb">sorted</span><span class="p">(</span><span class="n">inhibitors</span><span class="p">),</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">listdir_nohidden</span><span class="p">(</span><span class="n">template_directory</span><span class="p">))):</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">dataset_path</span><span class="p">,</span> <span class="n">sheet_name</span><span class="o">=</span><span class="n">inhibitor</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;time&quot;</span><span class="p">)</span>
    <span class="n">time</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">values</span>
    <span class="n">inhibitor_controls</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="mi">4</span><span class="p">]</span>
    <span class="n">inhibitor_concs</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">([</span><span class="nb">float</span><span class="p">(</span><span class="n">conc</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)[</span><span class="o">-</span><span class="mi">2</span><span class="p">])</span> <span class="k">for</span> <span class="n">conc</span> <span class="ow">in</span> <span class="n">inhibitor_controls</span><span class="o">.</span><span class="n">columns</span><span class="p">])</span>
    <span class="n">inhibitor_absorptions</span> <span class="o">=</span> <span class="n">inhibitor_controls</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">21</span><span class="p">)</span>
    <span class="n">inhibitor_absorptions</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">inhibitor_absorptions</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
    <span class="n">inhibitor_absorptions</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">inhibitor_absorptions</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
    <span class="n">mapper_inhibitor_absorption</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">inhibitor_concs</span><span class="p">,</span> <span class="n">inhibitor_absorptions</span><span class="p">))</span>
    <span class="n">mapper_inhibitor_absorption</span><span class="p">[</span><span class="mf">0.0</span><span class="p">]</span> <span class="o">=</span> <span class="n">buffer_enzyme_absorption</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="mi">4</span><span class="p">:]</span>
    <span class="k">for</span> <span class="n">column</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">init_substrate</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">column</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)[</span><span class="mi">4</span><span class="p">])</span>
        <span class="n">inhibitor_conc</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">column</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">)[</span><span class="mi">1</span><span class="p">])</span>
        <span class="n">df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">-</span> <span class="n">mapper_substrate_enzyme_absorption</span><span class="p">[</span><span class="n">init_substrate</span><span class="p">]</span> 
        <span class="n">df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">-</span> <span class="n">mapper_inhibitor_absorption</span><span class="p">[</span><span class="n">inhibitor_conc</span><span class="p">]</span>
    
    <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">21</span><span class="p">)</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">moveaxis</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">)</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="mi">18</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">21</span><span class="p">)</span>

    <span class="c1"># Parse measurement data to EnzymeML documents</span>
    <span class="n">enzml_docs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">measurement_data_to_EnzymeML</span><span class="p">(</span>
        <span class="n">template_path</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">template_directory</span><span class="si">}</span><span class="s2">/</span><span class="si">{</span><span class="n">template</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">measurement_data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
        <span class="n">time</span><span class="o">=</span><span class="n">time</span><span class="p">,</span>
        <span class="n">species_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">,</span>
        <span class="n">data_unit</span><span class="o">=</span><span class="s2">&quot;mmole / l&quot;</span><span class="p">,</span>
        <span class="n">time_unit</span><span class="o">=</span><span class="s2">&quot;min&quot;</span>
    <span class="p">))</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="data-quality">
<h5>Data quality<a class="headerlink" href="#data-quality" title="Permalink to this headline">#</a></h5>
<p>In the next cell, the blanked absorption data of each EnzymeML document is visualized with the <code class="docutils literal notranslate"><span class="pre">.visualize()</span></code>-method of <code class="docutils literal notranslate"><span class="pre">PyEnzyme</span></code> for quality control.</p>
<div class="cell tag_output_scroll tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">doc</span> <span class="ow">in</span> <span class="n">enzml_docs</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">doc</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>
    <span class="n">doc</span><span class="o">.</span><span class="n">visualize</span><span class="p">()</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>a-glucosidase inhibition by acarbose
</pre></div>
</div>
<img alt="_images/glycosidase_inhibition_5_1.png" src="_images/glycosidase_inhibition_5_1.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>a-glucosidase inhibition by fucoidan from E. maxima
</pre></div>
</div>
<img alt="_images/glycosidase_inhibition_5_3.png" src="_images/glycosidase_inhibition_5_3.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>a-glucosidase inhibition by fucoidan from E. radiata
</pre></div>
</div>
<img alt="_images/glycosidase_inhibition_5_5.png" src="_images/glycosidase_inhibition_5_5.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>a-glucosidase inhibition by fucoidan from F. vesiculosus
</pre></div>
</div>
<img alt="_images/glycosidase_inhibition_5_7.png" src="_images/glycosidase_inhibition_5_7.png" />
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>a-glucosidase inhibition by fucoidan from S. elegans
</pre></div>
</div>
<img alt="_images/glycosidase_inhibition_5_9.png" src="_images/glycosidase_inhibition_5_9.png" />
</div>
</div>
<p>The technical output of the cell above visualizes the blanked product absorbance data of each dataset. Therein, the individual measurements are labeled from m0 - m17, which represent individual experimental conditions: measurements m0 - m5 are from reactions without inhibitor, m6 - m11 reactions with the lower inhibitor concentration, and m12 - m17 originate from reactions with the higher inhibitor concentration. Each subplot contains the data of two experimental repeats.<br />
In most reactions a decrease followed by an increase of reaction rate is visible around minute 5. Since each dataset originates from a continuous photometric measurement carried out on a single MTP, the observed behavior likely sources from an analytical device malfunction. Additionally, one repeat of measurement ‘m8’ from the acarbose dataset shows low absorption, which is inconsistent with comparable measurements. Therefore, ‘m8’ was excluded from the dataset.</p>
<div class="cell tag_hide_cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">del</span> <span class="n">enzml_docs</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m8&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s1&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="concentration-calculation">
<h5>Concentration calculation<a class="headerlink" href="#concentration-calculation" title="Permalink to this headline">#</a></h5>
<p>Standard data of p-NP was loaded from an excel file, and a standard curve was created. Then, the standard curve was applied to the EnzymeML documents.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">path_calibration_data</span> <span class="o">=</span> <span class="s2">&quot;../../data/glucosidase_inhibition/p-NP_standard.xlsx&quot;</span>

<span class="n">product_standard</span> <span class="o">=</span> <span class="n">StandardCurve</span><span class="o">.</span><span class="n">from_excel</span><span class="p">(</span>
    <span class="n">path</span><span class="o">=</span><span class="n">path_calibration_data</span><span class="p">,</span>
    <span class="n">reactant_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">,</span> 
    <span class="n">sheet_name</span><span class="o">=</span><span class="s2">&quot;csv&quot;</span><span class="p">,</span> 
    <span class="n">wavelength</span><span class="o">=</span><span class="mi">405</span><span class="p">,</span> 
    <span class="n">concentration_unit</span> <span class="o">=</span> <span class="s2">&quot;mmole / l&quot;</span><span class="p">,</span> 
    <span class="n">cutoff_absorption</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>

<span class="n">product_standard</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="s2">&quot;Quadratic&quot;</span><span class="p">)</span>

<span class="c1"># Apply calibration curves to absorption EnzymeML documents</span>
<span class="k">for</span> <span class="n">enzmldoc</span> <span class="ow">in</span> <span class="n">enzml_docs</span><span class="p">:</span>
    <span class="n">product_standard</span><span class="o">.</span><span class="n">apply_to_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="s2">&quot;Quadratic&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Calibration data was automatically blanked.
</pre></div>
</div>
<div class="output text_html"><style type="text/css">
</style>
<table id="T_fb943" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_fb943_level0_col0" class="col_heading level0 col0" >AIC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_fb943_level0_row0" class="row_heading level0 row0" >3rd polynominal</th>
      <td id="T_fb943_row0_col0" class="data row0 col0" >-135</td>
    </tr>
    <tr>
      <th id="T_fb943_level0_row1" class="row_heading level0 row1" >Quadratic</th>
      <td id="T_fb943_row1_col0" class="data row1 col0" >-135</td>
    </tr>
    <tr>
      <th id="T_fb943_level0_row2" class="row_heading level0 row2" >Rational</th>
      <td id="T_fb943_row2_col0" class="data row2 col0" >-135</td>
    </tr>
    <tr>
      <th id="T_fb943_level0_row3" class="row_heading level0 row3" >Linear</th>
      <td id="T_fb943_row3_col0" class="data row3 col0" >-125</td>
    </tr>
    <tr>
      <th id="T_fb943_level0_row4" class="row_heading level0 row4" >Exponential</th>
      <td id="T_fb943_row4_col0" class="data row4 col0" >-11</td>
    </tr>
  </tbody>
</table>
</div><img alt="_images/glycosidase_inhibition_9_2.png" src="_images/glycosidase_inhibition_9_2.png" />
</div>
</div>
<p><em>Fig. 5: Fitted quadratic calibration model to standard data of p-NP (s1).</em></p>
<p>Based on AIC, 3<sup>rd</sup>-degree polynomial, quadratic and rational calibration models describe the relation between absorption to concentration ration of p-NP equally. Since the 3<sup>rd</sup>-degree polynomial model has one additional parameter compared to the quadratic model, the calibration model with less parameters is preferred, since the additional parameter did not give additional information. The fit between quadratic calibration model and calibration standard is visualized in Fig. 5.</p>
</section>
</section>
<section id="parameter-estimation">
<h4>Parameter estimation<a class="headerlink" href="#parameter-estimation" title="Permalink to this headline">#</a></h4>
<p>Comparison between different inhibition models and thus determining the kinetic mechanism was not possible, as no model was consistently favored over the other based on AIC and standard deviations of the parameter estimates.</p>
<p>Thus, different inhibition models cannot be compared, since information provided by the initial slopes is insufficient to distinguish kinetic inhibition mechanisms by modeling. In consequence, kinetic parameters were estimated based on initial rates, assuming competitive inhibition, since this kinetic mechanism has been proposed in literature <span id="id3">[<a class="reference internal" href="welcome.html#id26" title="Xindi Shan, Xin Liu, Jiejie Hao, Chao Cai, Fei Fan, Yunlou Dun, Xiaoliang Zhao, Xiaoxiao Liu, Chunxia Li, and Guangli Yu. In vitro and in vivo hypoglycemic effects of brown algal fucoidans. International Journal of Biological Macromolecules, 82:249–255, 2016.">Shan <em>et al.</em>, 2016</a>]</span>.<br />
Additionally, the faulty measurement with an initial substrate concentration of 0.5 mM from the ‘a-glucosidase inhibition by acarbose’ data set was excluded from parameter estimation.
Based on an initial parameter estimation run with subsequent visual analysis, severals measurements from the <em>E. maxima</em> dataset were excluded. Thereby, measurements from both applied inhibitor concentrations and an initial substrate concentration of 2.5 mM and 5 mM showed similar activity compared to reactions without inhibitor. In contrast, the reactions with inhibitor present showed inhibition, according to the difference in slope.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Discard measurement with 2.5 mM and 5 mM and lower inhibition concentration from E. maxima dataset</span>
<span class="k">del</span> <span class="n">enzml_docs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m10&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">enzml_docs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m11&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">enzml_docs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m16&quot;</span><span class="p">]</span>
<span class="k">del</span> <span class="n">enzml_docs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m17&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Run parameter estimation for all data sets</span>
<span class="n">results</span><span class="o">=</span><span class="p">[]</span>
<span class="k">for</span> <span class="n">enzmldoc</span> <span class="ow">in</span> <span class="n">enzml_docs</span><span class="p">:</span>
    <span class="n">result</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="o">=</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="n">reactant_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="n">inhibitor_id</span><span class="o">=</span><span class="s2">&quot;s2&quot;</span><span class="p">,</span> <span class="n">measured_species</span><span class="o">=</span><span class="s2">&quot;product&quot;</span><span class="p">)</span>
    <span class="n">result</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">enzyme_inactivation</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">stop_time_index</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
    <span class="n">results</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>

<span class="c1"># Display results for competitive inhibition model</span>
<span class="n">labels</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="s2">&quot;Acarbose&quot;</span><span class="p">,</span> <span class="s2">&quot;Fucoidan (E. maxima)&quot;</span><span class="p">,</span> <span class="s2">&quot;Fucoidan (E. radiata)&quot;</span><span class="p">,</span> <span class="s2">&quot;Fucoidan (F. vesiculosus)&quot;</span><span class="p">,</span> <span class="s2">&quot;Fucoidan (S. elegans)&quot;</span><span class="p">])</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">result</span><span class="o">.</span><span class="n">result_dict</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s2">&quot;competitive inhibition&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">to_frame</span><span class="p">()</span><span class="o">.</span><span class="n">T</span> <span class="k">for</span> <span class="n">result</span> <span class="ow">in</span> <span class="n">results</span><span class="p">],</span>\
    <span class="n">keys</span><span class="o">=</span><span class="n">labels</span><span class="p">)</span>\
        <span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>\
        <span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;level_1&quot;</span><span class="p">,</span> <span class="s2">&quot;AIC&quot;</span><span class="p">,</span> <span class="s2">&quot;Ki uncompetitive [g / l]&quot;</span><span class="p">])</span>\
        <span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;level_0&quot;</span><span class="p">)</span>\
        <span class="o">.</span><span class="n">rename_axis</span><span class="p">(</span><span class="s2">&quot;Inhibitor&quot;</span><span class="p">,</span> <span class="n">axis</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Kinetic parameters based on competitive inhibition model:&quot;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">set_table_attributes</span><span class="p">(</span><span class="s1">&#39;style=&quot;font-size: 12px&quot;&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Kinetic parameters based on competitive inhibition model:
</pre></div>
</div>
<div class="output text_html"><style type="text/css">
</style>
<table id="T_18fbf" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_18fbf_level0_col0" class="col_heading level0 col0" >kcat [1/min]</th>
      <th id="T_18fbf_level0_col1" class="col_heading level0 col1" >Km [mmole / l]</th>
      <th id="T_18fbf_level0_col2" class="col_heading level0 col2" >kcat / Km [1/min * 1/mmole / l]</th>
      <th id="T_18fbf_level0_col3" class="col_heading level0 col3" >Ki competitive [g / l]</th>
    </tr>
    <tr>
      <th class="index_name level0" >Inhibitor</th>
      <th class="blank col0" >&nbsp;</th>
      <th class="blank col1" >&nbsp;</th>
      <th class="blank col2" >&nbsp;</th>
      <th class="blank col3" >&nbsp;</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_18fbf_level0_row0" class="row_heading level0 row0" >Acarbose</th>
      <td id="T_18fbf_row0_col0" class="data row0 col0" >1.903 +/- 2.77%</td>
      <td id="T_18fbf_row0_col1" class="data row0 col1" >0.341 +/- 10.79%</td>
      <td id="T_18fbf_row0_col2" class="data row0 col2" >5.581 +/- 11.14%</td>
      <td id="T_18fbf_row0_col3" class="data row0 col3" >0.112 +/- 10.80%</td>
    </tr>
    <tr>
      <th id="T_18fbf_level0_row1" class="row_heading level0 row1" >Fucoidan (E. maxima)</th>
      <td id="T_18fbf_row1_col0" class="data row1 col0" >2.317 +/- 2.14%</td>
      <td id="T_18fbf_row1_col1" class="data row1 col1" >0.406 +/- 7.64%</td>
      <td id="T_18fbf_row1_col2" class="data row1 col2" >5.701 +/- 7.93%</td>
      <td id="T_18fbf_row1_col3" class="data row1 col3" >0.874 +/- 14.25%</td>
    </tr>
    <tr>
      <th id="T_18fbf_level0_row2" class="row_heading level0 row2" >Fucoidan (E. radiata)</th>
      <td id="T_18fbf_row2_col0" class="data row2 col0" >1.343 +/- 2.75%</td>
      <td id="T_18fbf_row2_col1" class="data row2 col1" >0.180 +/- 14.57%</td>
      <td id="T_18fbf_row2_col2" class="data row2 col2" >7.461 +/- 14.83%</td>
      <td id="T_18fbf_row2_col3" class="data row2 col3" >0.028 +/- 19.29%</td>
    </tr>
    <tr>
      <th id="T_18fbf_level0_row3" class="row_heading level0 row3" >Fucoidan (F. vesiculosus)</th>
      <td id="T_18fbf_row3_col0" class="data row3 col0" >1.467 +/- 2.07%</td>
      <td id="T_18fbf_row3_col1" class="data row3 col1" >0.415 +/- 7.37%</td>
      <td id="T_18fbf_row3_col2" class="data row3 col2" >3.531 +/- 7.66%</td>
      <td id="T_18fbf_row3_col3" class="data row3 col3" >0.003 +/- 7.80%</td>
    </tr>
    <tr>
      <th id="T_18fbf_level0_row4" class="row_heading level0 row4" >Fucoidan (S. elegans)</th>
      <td id="T_18fbf_row4_col0" class="data row4 col0" >1.474 +/- 2.42%</td>
      <td id="T_18fbf_row4_col1" class="data row4 col1" >0.251 +/- 10.79%</td>
      <td id="T_18fbf_row4_col2" class="data row4 col2" >5.863 +/- 11.06%</td>
      <td id="T_18fbf_row4_col3" class="data row4 col3" >0.016 +/- 11.88%</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>The output above displays estimated kinetic parameters for all data sets, assuming competitive inhibition between enzyme and inhibitor. As a result <span class="math notranslate nohighlight">\(k_{cat}\)</span> was estimated between 2.317 min<sup>-1</sup> ± 2.14% and 1.343 min<sup>-1</sup> ± 2.75%, whereas <span class="math notranslate nohighlight">\(K_{m}\)</span> was estimated between 0.415 mM ± 7.37% and 0.180 mM ± 14.57%.
Since all measurements were conducted under equal conditions, <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> should be similar across experiments. Deviations in parameter values might therefore result from pipetting or deviations in reaction temperature.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Get K_i parameter results</span>
<span class="k">def</span> <span class="nf">get_parameter</span><span class="p">(</span><span class="n">result</span><span class="p">,</span> <span class="n">model_name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">parameter</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">result</span><span class="o">.</span><span class="n">models</span><span class="p">[</span><span class="n">model_name</span><span class="p">]</span><span class="o">.</span><span class="n">result</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="n">parameter</span><span class="p">]</span>

<span class="n">ki</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">ki_stderr</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">result</span> <span class="ow">in</span> <span class="n">results</span><span class="p">:</span>
    <span class="n">ki</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">get_parameter</span><span class="p">(</span><span class="n">result</span><span class="p">,</span> <span class="s2">&quot;competitive inhibition&quot;</span><span class="p">,</span> <span class="s2">&quot;K_ic&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">value</span><span class="p">)</span>
    <span class="n">ki_stderr</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">get_parameter</span><span class="p">(</span><span class="n">result</span><span class="p">,</span> <span class="s2">&quot;competitive inhibition&quot;</span><span class="p">,</span> <span class="s2">&quot;K_ic&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">stderr</span><span class="p">)</span>
<span class="n">ki</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">ki</span><span class="p">)</span>
<span class="n">ki_stderr</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">ki_stderr</span><span class="p">)</span>

<span class="c1"># Visualize</span>
<span class="n">kis</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">labels</span><span class="p">,</span><span class="n">ki</span><span class="p">))</span>
<span class="n">f</span><span class="p">,</span> <span class="p">(</span><span class="n">a0</span><span class="p">,</span> <span class="n">a1</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">gridspec_kw</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;width_ratios&#39;</span><span class="p">:</span> <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mf">1.4</span><span class="p">]})</span>
<span class="n">a0</span><span class="o">.</span><span class="n">bar</span><span class="p">(</span><span class="n">labels</span><span class="p">[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">]],</span> <span class="n">ki</span><span class="p">[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">]],</span> <span class="n">yerr</span><span class="o">=</span><span class="n">ki_stderr</span><span class="p">[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">]],</span> <span class="n">align</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">capsize</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
<span class="n">a0</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">labels</span><span class="p">[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">]],</span> <span class="n">rotation</span> <span class="o">=</span> <span class="mi">90</span><span class="p">)</span>
<span class="n">a1</span><span class="o">.</span><span class="n">bar</span><span class="p">(</span><span class="n">labels</span><span class="p">[:</span><span class="mi">2</span><span class="p">],</span> <span class="n">ki</span><span class="p">[:</span><span class="mi">2</span><span class="p">],</span> <span class="n">yerr</span><span class="o">=</span><span class="n">ki_stderr</span><span class="p">[:</span><span class="mi">2</span><span class="p">],</span> <span class="n">align</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">capsize</span><span class="o">=</span><span class="mi">3</span><span class="p">)</span>
<span class="n">a1</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">labels</span><span class="p">[:</span><span class="mi">2</span><span class="p">],</span> <span class="n">rotation</span> <span class="o">=</span> <span class="mi">90</span><span class="p">)</span>
<span class="n">a0</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;$K_</span><span class="si">{i}</span><span class="s2">$ [$mg$ $ml^{-1}$]&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/glycosidase_inhibition_14_0.png" src="_images/glycosidase_inhibition_14_0.png" />
</div>
</div>
<p><em>Fig. 6: Estimated inhibitory constants for <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase inhibition by fucoidan from different algae species and acarbose</em></p>
<p>Fucoidan from <em>E. radiata</em>, <em>F. vesiculosus</em>, and <em>S. elegans</em> all showed a lower <span class="math notranslate nohighlight">\(K_{i}\)</span> (28, 3, and 16 µg ml<sup>-1</sup> respectively) compared to the acarbose reference (112 µg ml<sup>-1</sup>). This resembles a 37-fold higher binding affinity of fucoidan from <em>F. vesiculosus</em> to <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase compared to acarbose. In contrast to the previous study, fucoidan from <em>E. maxima</em> showed a more than 7-fold higher <span class="math notranslate nohighlight">\(K_{i}\)</span> (858 µg ml<sup>-1</sup>) compared to acarbose. This likely originates from inconsistencies during experimental preparation. Additionally, 4 of 10 measurements with applied inhibitor were excluded from parameter estimation, since they showed higher activity compared to the respective reactions without inhibitor. Therefore, these measurement should be repeated. Estimated inhibition constants for each tested inhibitor are visualized in Fig. 6.</p>
</section>
<section id="project-discussion">
<h4>Project discussion<a class="headerlink" href="#project-discussion" title="Permalink to this headline">#</a></h4>
<p>In general, the estimated parameters are not reliable on a quantitative level, since the malfunction of the analytical device undermines the validity of the whole dataset. Therefore, it cannot be ruled out, that some of the calculated concentrations deviate from the real concentrations. Furthermore, all estimated parameters are only estimated based on three time points of measure. In addition each experimental condition was applied as duplicate repeats, which additionally showed deviations.
In conclusion the estimated parameters qualitatively confirm that fucoidan from different host organism, inhibits <span class="math notranslate nohighlight">\(\alpha\)</span>-glucosidase more strongly, compared to acarbose.</p>
</section>
<section id="save-modeling-results">
<h4>Save modeling results<a class="headerlink" href="#save-modeling-results" title="Permalink to this headline">#</a></h4>
<p>Lastly, the modeling results are written to the EnzymeML documents and the files are exported.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">result</span><span class="p">,</span> <span class="n">enzmldoc</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">results</span><span class="p">,</span> <span class="n">enzml_docs</span><span class="p">):</span>

    <span class="c1"># Write modeling results to kinetic parameters</span>
    <span class="n">k_cat</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;k_cat&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;k_cat&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;1 / </span><span class="si">{</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">time_unit</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

    <span class="n">K_m</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;K_m&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;Km&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">data_conc_unit</span><span class="p">)</span>

    <span class="n">K_ic</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;K_ic&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;K_ic&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">data_conc_unit</span><span class="p">)</span>

    <span class="c1"># Define kinetic model</span>
    <span class="n">model</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">KineticModel</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;competitive inhibition&quot;</span><span class="p">,</span>
        <span class="n">equation</span><span class="o">=</span><span class="s2">&quot;-k_cat * p0 * s0 / (K_m*(1+(s2 / K_ic))+s0)&quot;</span><span class="p">,</span>
        <span class="n">parameters</span><span class="o">=</span><span class="p">[</span><span class="n">k_cat</span><span class="p">,</span> <span class="n">K_m</span><span class="p">,</span> <span class="n">K_ic</span><span class="p">])</span>

    <span class="n">enzmldoc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">model</span> <span class="o">=</span> <span class="n">model</span>

    <span class="c1"># Export EnzymeML documents</span>
    <span class="n">export</span> <span class="o">=</span> <span class="kc">False</span>
    <span class="k">if</span> <span class="n">export</span><span class="p">:</span>
        <span class="n">enzmldoc</span><span class="o">.</span><span class="n">toFile</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<span id="document-scenarios/SLAC_kinetic_characterization"></span><section class="tex2jax_ignore mathjax_ignore" id="scenario-c-br-slac-characterization">
<h3><strong>Scenario C:</strong><br>SLAC characterization<a class="headerlink" href="#scenario-c-br-slac-characterization" title="Permalink to this headline">#</a></h3>
<p>Data provided by Alaric Prins (Applied Microbial and Health Biotechnology Institute, Cape Peninsula University of Technology, Capetown, South Africa)</p>
<section id="project-background">
<h4>Project background<a class="headerlink" href="#project-background" title="Permalink to this headline">#</a></h4>
<p>The small laccase from <em>Streptomyces coelicolor</em> (SLAC) is a member of the muticopper oxidase family. Due to their potential to oxidize various electron-rich substrates, laccases are of interest for industrial application for lignin degradation <span id="id1">[<a class="reference internal" href="welcome.html#id13" title="Petri Widsten and Andreas Kandelbauer. Laccase applications in the forest products industry: a review. Enzyme and microbial technology, 42(4):293–307, 2008.">Widsten and Kandelbauer, 2008</a>]</span>. SLAC is currently used in ongoing biocatalysis and synergy projects. Recently, the wild-type SLAC was synthesised (previously cloned by restriction cloning) and codon-optimized for expression in <em>E. coli</em>. Therefore, the catalytic properties of the codon-optimized SLAC were investigated in a broad pH and temperature range, for rapid and accurate determination of biochemical data, which is useful for downstream applications.</p>
<section id="experimental-design">
<h5>Experimental design<a class="headerlink" href="#experimental-design" title="Permalink to this headline">#</a></h5>
<p>In this scenario the enzymatic oxidation of 2,2’-Azino-bis(3-ethylbenzothiazoline-6-sulfonic acid) (ABTS) to its radical form ABTS<sup>+·</sup> was studied in the pH range of pH 3 - pH 5.5 and temperature range of 25°C - 45°C. In total 30 kinetic enzyme assays in a substrate range between 0 - 200 uM of ABTS were conducted. Additionally, for each enzyme reaction with a given initial substrate concentration, a control reaction without enzyme was prepared.<br />
For each pH - temperature condition an individual ABTS standard curve and absorption spectrum was recorded to account for varying ABTS absorption properties due to reaction conditions. Each enzyme reaction was followed for 15 min photometrically at two wavelengths, measuring substrate depletion and product accumulation simultaneously. Preliminary experiments confirmed, that ABTS absorbs at 340 nm, whereas the ABTS<sup>+·</sup> absorbs at 420 nm in contrast to the substrate. Furthermore, cross absorbance of product at the substrate detection wavelength and vice versa was ruled out.</p>
</section>
<section id="data-management">
<h5>Data management<a class="headerlink" href="#data-management" title="Permalink to this headline">#</a></h5>
<p>Overall, the dataset consists of more than 100 000 individual absorbance reads. Thus, data preparation was automated by custom parser functions, which were tailored to the output of the used spectrophotometer. Consequently, the tedious and error-prone manual copying of raw data was avoided. Information on the involved reactants, and the enzyme was filled in an EnzymeML Excel spreadsheet, which served as a meta data container. All other information was parsed from the output of the spectrophotometer. Fig. 7 illustrates the schematic data flow of this project.</p>
<p><img alt="Fig. 7" src="_images/workflow_SLAC.png" />
<em>Fig. 7: Schematic data pipeline of the SLAC project.</em></p>
</section>
</section>
<section id="data-preparation">
<h4>Data preparation<a class="headerlink" href="#data-preparation" title="Permalink to this headline">#</a></h4>
<section id="imports">
<h5>Imports<a class="headerlink" href="#imports" title="Permalink to this headline">#</a></h5>
<div class="cell tag_hide-cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">typing</span> <span class="kn">import</span> <span class="n">Dict</span><span class="p">,</span> <span class="n">List</span>
<span class="kn">import</span> <span class="nn">pyenzyme</span> <span class="k">as</span> <span class="nn">pe</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">matplotlib</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">copy</span>
<span class="kn">import</span> <span class="nn">string</span>
<span class="kn">from</span> <span class="nn">lmfit</span> <span class="kn">import</span> <span class="n">Parameters</span><span class="p">,</span> <span class="n">minimize</span>
<span class="kn">from</span> <span class="nn">scipy.stats</span> <span class="kn">import</span> <span class="n">linregress</span><span class="p">,</span> <span class="n">pearsonr</span>
<span class="kn">from</span> <span class="nn">joblib</span> <span class="kn">import</span> <span class="n">Parallel</span><span class="p">,</span> <span class="n">delayed</span>
<span class="kn">from</span> <span class="nn">CaliPytion.tools.standardcurve</span> <span class="kn">import</span> <span class="n">StandardCurve</span>
<span class="kn">from</span> <span class="nn">EnzymePynetics.tools.parameterestimator</span> <span class="kn">import</span> <span class="n">ParameterEstimator</span>

<span class="c1"># Custom functions for data mapping</span>
<span class="kn">from</span> <span class="nn">parser_functions</span> <span class="kn">import</span> <span class="n">measurement_data_to_EnzymeML</span><span class="p">,</span> <span class="n">plot</span>
<span class="kn">from</span> <span class="nn">parser_functions</span> <span class="kn">import</span> <span class="n">read_measurement_data</span><span class="p">,</span> <span class="n">read_calibration_data</span>
<span class="kn">from</span> <span class="nn">parser_functions</span> <span class="kn">import</span> <span class="n">add_enzyme_inactivation_model</span>

<span class="kn">import</span> <span class="nn">warnings</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s1">&#39;ignore&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="experimental-data">
<h5>Experimental data<a class="headerlink" href="#experimental-data" title="Permalink to this headline">#</a></h5>
<p>Data from SLAC reactions was loaded from the output files of the photometer and written to individual EnzymeML documents. Then, information of the control reactions was used to subtract the absorption contribution from enzyme and buffer from the substrate and product signal.</p>
<div class="cell tag_hide_cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Specify the location of the data sets</span>
<span class="n">directory_measurement_data</span> <span class="o">=</span> <span class="s2">&quot;../../data/SLAC_kinetic_characterization/TimeCourseData&quot;</span>
<span class="n">directory_standard_data</span> <span class="o">=</span> <span class="s2">&quot;../../data/SLAC_kinetic_characterization/StandardData&quot;</span>
<span class="n">directory_spectrum_data</span> <span class="o">=</span> <span class="s2">&quot;../../data/SLAC_kinetic_characterization/SpectrumData&quot;</span>
<span class="n">path_EnzymeML_templates</span> <span class="o">=</span> <span class="s2">&quot;../../data/SLAC_kinetic_characterization/EnzymeML_templates&quot;</span>

<span class="c1"># Define IDs for species, listed in the EnzymeML Excel template</span>
<span class="n">substrate_id</span> <span class="o">=</span> <span class="s2">&quot;s0&quot;</span>
<span class="n">product_id</span> <span class="o">=</span> <span class="s2">&quot;s1&quot;</span>
<span class="n">substrate_control_id</span> <span class="o">=</span> <span class="s2">&quot;s2&quot;</span>
<span class="n">product_control_id</span> <span class="o">=</span> <span class="s2">&quot;s3&quot;</span>
<span class="n">species_ids</span> <span class="o">=</span> <span class="p">[</span><span class="n">substrate_id</span><span class="p">,</span> <span class="n">product_id</span><span class="p">,</span> <span class="n">substrate_control_id</span><span class="p">,</span> <span class="n">product_control_id</span><span class="p">]</span>


<span class="c1"># Parse measurement data from photometer output</span>
<span class="n">raw_data_dict</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">path</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">directory_measurement_data</span><span class="p">):</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">read_measurement_data</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">directory_measurement_data</span><span class="si">}</span><span class="s2">/</span><span class="si">{</span><span class="n">path</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">pH</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s2">&quot;pH&quot;</span><span class="p">]</span>
    <span class="n">temp</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s2">&quot;temperature&quot;</span><span class="p">]</span>
    <span class="n">raw_data_dict</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">pH</span><span class="si">}</span><span class="s2"> </span><span class="si">{</span><span class="n">temp</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span>

<span class="n">EnzymeML_template_dict</span> <span class="o">=</span> <span class="p">{</span>
    <span class="mf">3.0</span><span class="p">:</span> <span class="s2">&quot;EnzymeML_SLAC_pH3.xlsm&quot;</span><span class="p">,</span>
    <span class="mf">3.5</span><span class="p">:</span> <span class="s2">&quot;EnzymeML_SLAC_pH3_5.xlsm&quot;</span><span class="p">,</span>
    <span class="mf">4.0</span><span class="p">:</span> <span class="s2">&quot;EnzymeML_SLAC_pH4.xlsm&quot;</span><span class="p">,</span>
    <span class="mf">4.5</span><span class="p">:</span> <span class="s2">&quot;EnzymeML_SLAC_pH4_5.xlsm&quot;</span><span class="p">,</span>
    <span class="mf">5.0</span><span class="p">:</span> <span class="s2">&quot;EnzymeML_SLAC_pH5.xlsm&quot;</span><span class="p">,</span>
    <span class="mf">5.5</span><span class="p">:</span> <span class="s2">&quot;EnzymeML_SLAC_pH5_5.xlsm&quot;</span><span class="p">,</span>
<span class="p">}</span>

<span class="c1"># Write absorption data to EnzymeMLDocuments</span>
<span class="n">absortion_enzymemldocs</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="p">]</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">name</span><span class="p">,</span> <span class="n">data</span> <span class="ow">in</span> <span class="n">raw_data_dict</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
    <span class="n">pH</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s2">&quot;pH&quot;</span><span class="p">]</span>
    <span class="n">absortion_enzymemldocs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">measurement_data_to_EnzymeML</span><span class="p">(</span>
        <span class="n">template_path</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">path_EnzymeML_templates</span><span class="si">}</span><span class="s2">/</span><span class="si">{</span><span class="n">EnzymeML_template_dict</span><span class="p">[</span><span class="n">pH</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
        <span class="n">measurement_data</span><span class="o">=</span><span class="n">data</span><span class="p">,</span>
        <span class="n">species_ids</span><span class="o">=</span><span class="n">species_ids</span><span class="p">,</span>
        <span class="n">data_unit</span><span class="o">=</span><span class="s2">&quot;umole / l&quot;</span><span class="p">,</span>
        <span class="n">time_unit</span><span class="o">=</span><span class="s2">&quot;s&quot;</span><span class="p">))</span>

<span class="c1"># Sort documents by ascending pH and temperature</span>
<span class="n">absortion_enzymemldocs</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">absortion_enzymemldocs</span><span class="p">,</span> <span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">ph</span><span class="p">,</span> <span class="n">x</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">temperature</span><span class="p">))</span>

<span class="c1"># Blanc measurement data</span>
<span class="k">for</span> <span class="n">enzmldoc</span> <span class="ow">in</span> <span class="n">absortion_enzymemldocs</span><span class="p">:</span>
    <span class="n">blanc_measurement</span> <span class="o">=</span> <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m0&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span>
    <span class="n">blanc</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">([</span><span class="n">repeat</span><span class="o">.</span><span class="n">data</span> <span class="k">for</span> <span class="n">repeat</span> <span class="ow">in</span> <span class="n">blanc_measurement</span><span class="p">])</span>

    <span class="k">for</span> <span class="nb">id</span><span class="p">,</span> <span class="n">measurement</span> <span class="ow">in</span> <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">for</span> <span class="n">rep</span><span class="p">,</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">):</span>
            <span class="n">blanced_data</span> <span class="o">=</span> <span class="p">[</span><span class="n">value</span> <span class="o">-</span> <span class="n">blanc</span> <span class="k">for</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">]</span>
            <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="nb">id</span><span class="p">]</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">[</span><span class="n">rep</span><span class="p">]</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="n">blanced_data</span>
        <span class="k">for</span> <span class="n">rep</span><span class="p">,</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s2&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">):</span>
            <span class="n">blanced_data</span> <span class="o">=</span> <span class="p">[</span><span class="n">value</span> <span class="o">-</span> <span class="n">blanc</span> <span class="k">for</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">]</span>
            <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="nb">id</span><span class="p">]</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s2&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">[</span><span class="n">rep</span><span class="p">]</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="n">blanced_data</span>

        <span class="k">for</span> <span class="n">rep</span><span class="p">,</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s1&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">):</span>
            <span class="n">blanced_data</span> <span class="o">=</span> <span class="p">[</span><span class="n">value</span> <span class="o">-</span> <span class="n">blanc</span> <span class="k">for</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">]</span>
            <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="nb">id</span><span class="p">]</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s1&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">[</span><span class="n">rep</span><span class="p">]</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="n">blanced_data</span>
        <span class="k">for</span> <span class="n">rep</span><span class="p">,</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s3&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">):</span>
            <span class="n">blanced_data</span> <span class="o">=</span> <span class="p">[</span><span class="n">value</span> <span class="o">-</span> <span class="n">blanc</span> <span class="k">for</span> <span class="n">value</span> <span class="ow">in</span> <span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">]</span>
            <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="nb">id</span><span class="p">]</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s3&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">[</span><span class="n">rep</span><span class="p">]</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="n">blanced_data</span>
    
    <span class="c1"># Delete control measurement &#39;m0&#39;</span>
    <span class="k">del</span> <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="p">[</span><span class="s2">&quot;m0&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="concentration-calculation">
<h5>Concentration calculation<a class="headerlink" href="#concentration-calculation" title="Permalink to this headline">#</a></h5>
<p>Calibration data was loaded and converted into individual instances of the calibration data model. Some meta data of the calibration needed to be provided to the custom <code class="docutils literal notranslate"><span class="pre">read_calibration_data</span></code> function, since the output of the used spectrophotometer only contained a minimum of information. Thereafter, a StandardCurve was created for each calibration data set. Thereby, only absorption values below 3.2 were considered, since higher absorption values could not be converted into concentration accurately.
Lastly, the fit of each standard curves was visualized.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load calibration raw data</span>
<span class="n">calibration_data</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">standard_directory</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">sort</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">directory_standard_data</span><span class="p">))</span>
<span class="n">spectrum_directory</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">sort</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">directory_spectrum_data</span><span class="p">))</span>

<span class="k">for</span> <span class="n">standard</span><span class="p">,</span> <span class="n">spectrum</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">standard_directory</span><span class="p">,</span> <span class="n">spectrum_directory</span><span class="p">):</span>
    <span class="n">standard</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">directory_standard_data</span><span class="si">}</span><span class="s2">/</span><span class="si">{</span><span class="n">standard</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="n">spectrum</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">directory_spectrum_data</span><span class="si">}</span><span class="s2">/</span><span class="si">{</span><span class="n">spectrum</span><span class="si">}</span><span class="s2">&quot;</span>
    <span class="n">result</span> <span class="o">=</span> <span class="n">read_calibration_data</span><span class="p">(</span>
        <span class="n">path_standard</span><span class="o">=</span><span class="n">standard</span><span class="p">,</span>
        <span class="n">path_spectrum</span><span class="o">=</span><span class="n">spectrum</span><span class="p">,</span>
        <span class="n">species_id</span><span class="o">=</span><span class="n">substrate_id</span><span class="p">,</span>
        <span class="n">wavelengths</span><span class="o">=</span><span class="p">[</span><span class="mi">340</span><span class="p">,</span> <span class="mi">420</span><span class="p">],</span>
        <span class="n">concentrations</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">10</span><span class="p">,</span><span class="mi">15</span><span class="p">,</span><span class="mi">25</span><span class="p">,</span><span class="mi">50</span><span class="p">,</span><span class="mi">75</span><span class="p">,</span><span class="mi">100</span><span class="p">,</span><span class="mi">125</span><span class="p">,</span><span class="mi">150</span><span class="p">,</span><span class="mi">175</span><span class="p">,</span><span class="mi">200</span><span class="p">],</span>
        <span class="n">concentration_unit</span><span class="o">=</span><span class="s2">&quot;umole / l&quot;</span><span class="p">,</span>
        <span class="n">device_manufacturer</span><span class="o">=</span><span class="s2">&quot;Molecular Devices&quot;</span><span class="p">,</span>
        <span class="n">device_model</span><span class="o">=</span><span class="s2">&quot;SpectraMax i3x&quot;</span><span class="p">,</span>
        <span class="n">spectrum_reactant_concentration</span><span class="o">=</span><span class="mi">100</span>
    <span class="p">)</span>
    <span class="k">for</span> <span class="n">pH</span> <span class="ow">in</span> <span class="n">result</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
        <span class="n">calibration_data</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">result</span><span class="p">[</span><span class="n">pH</span><span class="p">])</span>

<span class="c1"># Sort calibration data by ascending pH and temperature</span>
<span class="n">calibration_data</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">calibration_data</span><span class="p">,</span> <span class="n">key</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">pH</span><span class="p">,</span> <span class="n">x</span><span class="o">.</span><span class="n">temperature</span><span class="p">))</span>

<span class="c1"># Generate standard curves for ABTS calibration data</span>
<span class="n">standard_curves</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="n">StandardCurve</span><span class="p">]</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">calibration</span> <span class="ow">in</span> <span class="nb">list</span><span class="p">(</span><span class="n">calibration_data</span><span class="p">):</span>
    <span class="n">standard_curves</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">StandardCurve</span><span class="p">(</span><span class="n">calibration_data</span><span class="o">=</span><span class="n">calibration</span><span class="p">,</span> <span class="n">wavelength</span><span class="o">=</span><span class="mi">340</span><span class="p">,</span> <span class="n">cutoff_absorption</span><span class="o">=</span><span class="mf">3.2</span><span class="p">,</span> <span class="n">show_output</span><span class="o">=</span><span class="kc">False</span><span class="p">))</span>

<span class="c1"># Sort standard curves by ascending pH and temperature.</span>
<span class="n">standard_curves</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">standard_curves</span><span class="p">,</span> <span class="n">key</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">calibration_data</span><span class="o">.</span><span class="n">pH</span><span class="p">,</span> <span class="n">x</span><span class="o">.</span><span class="n">calibration_data</span><span class="o">.</span><span class="n">temperature</span><span class="p">))</span>

<span class="c1"># Visualize all fitted standard curves </span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">13</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="p">(</span><span class="n">standard</span><span class="p">,</span> <span class="n">ax</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">standard_curves</span><span class="p">[:</span><span class="mi">30</span><span class="p">],</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">())):</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">i</span><span class="o">%</span><span class="k">5</span>:
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS [uM]&quot;</span><span class="p">)</span>
    <span class="n">standard</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;pH </span><span class="si">{</span><span class="n">standard</span><span class="o">.</span><span class="n">calibration_data</span><span class="o">.</span><span class="n">pH</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">standard</span><span class="o">.</span><span class="n">calibration_data</span><span class="o">.</span><span class="n">temperature</span><span class="si">}</span><span class="s2">°C&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">i</span> <span class="ow">in</span> <span class="p">[</span><span class="mi">25</span><span class="p">,</span><span class="mi">26</span><span class="p">,</span><span class="mi">27</span><span class="p">,</span><span class="mi">28</span><span class="p">,</span><span class="mi">29</span><span class="p">]:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [s]&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/SLAC_kinetic_characterization_6_0.png" src="_images/SLAC_kinetic_characterization_6_0.png" />
</div>
</div>
<p><em>Fig. 8: ABTS standard curves for different experimental condition.</em></p>
<p>Since the absorption characteristics of ABTS change with pH, the calibration range differs between pH values, due to the upper absorption limit (Fig. 8). In consequence, the upper calibration limit for reactions at pH 3, pH 3.5, and pH 5.5 is at 150 uM of ABTS, whereas for all other pH values the upper limit is at 175 uM. This might source from the protonation state of ABTS, since the sulfonate groups of ABTS are deprotonated for less acidic pH values. Therefore, the absorption properties of ABTS can decrease. Calibration curve data at pH 5.5 showed larger variation between the repeats. In this case, pipetting of one of the three repeats differs from the other two, which should be considered for kinetic parameter estimation. In contrast to pH, the temperature during calibration affected the calibration curve only marginally.</p>
<p>The generated standard curves were used to convert the absorption measurement data into concentration data. Thereby, the respective concentration values were only calculated, if the measured absorption was within the respective calibration bounds to avoid extrapolation.</p>
<div class="cell tag_output_scroll tag_hide_cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Calculate concentrations by applying standard curves to &#39;EnzymeMLDocuments&#39;.</span>
<span class="n">concentration_enzymemldocs</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">standard_curve</span><span class="p">,</span> <span class="n">abso_enzmldoc</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">standard_curves</span><span class="p">,</span> <span class="n">absortion_enzymemldocs</span><span class="p">):</span>

    <span class="c1"># Validate, that measurement data is paired with respective standard curve</span>
    <span class="k">assert</span> <span class="n">standard_curve</span><span class="o">.</span><span class="n">calibration_data</span><span class="o">.</span><span class="n">pH</span> <span class="o">==</span> <span class="n">abso_enzmldoc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">ph</span>
    <span class="k">assert</span> <span class="n">standard_curve</span><span class="o">.</span><span class="n">calibration_data</span><span class="o">.</span><span class="n">temperature</span> <span class="o">==</span> <span class="n">abso_enzmldoc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">temperature</span>

    <span class="n">enzmldoc</span> <span class="o">=</span> <span class="n">copy</span><span class="o">.</span><span class="n">deepcopy</span><span class="p">(</span><span class="n">abso_enzmldoc</span><span class="p">)</span>
    <span class="n">conc_enzmldoc</span> <span class="o">=</span> <span class="n">standard_curve</span><span class="o">.</span><span class="n">apply_to_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="s2">&quot;s0&quot;</span><span class="p">,</span> <span class="n">ommit_nan_measurements</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="n">conc_enzmldoc</span> <span class="o">=</span> <span class="n">standard_curve</span><span class="o">.</span><span class="n">apply_to_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="s2">&quot;s2&quot;</span><span class="p">,</span> <span class="n">ommit_nan_measurements</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

    <span class="n">concentration_enzymemldocs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">conc_enzmldoc</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m8&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m8&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m8&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m8&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m8&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
Measurements &#39;[&#39;m9&#39;]&#39; removed from document, since respective measurement values are out of calibration range.
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="quality-control-through-mass-balance-analysis">
<h4>Quality control through mass balance analysis<a class="headerlink" href="#quality-control-through-mass-balance-analysis" title="Permalink to this headline">#</a></h4>
<p>Since product and substrate of the SLAC reaction were simultaneously recorded, mass balance analysis was conducted as a control of quality. By assuming mass conservation, the following concentration balance can be established:</p>
<div class="math notranslate nohighlight" id="equation-mass-balance">
<span class="eqno">(14)<a class="headerlink" href="#equation-mass-balance" title="Permalink to this equation">#</a></span>\[0 = S_{(t)} + P_{(t)} - S_{0}\]</div>
<p>Where, <span class="math notranslate nohighlight">\(S_{0}\)</span> denotes the initial substrate concentration, whereas <span class="math notranslate nohighlight">\(S_{(t)}\)</span> and <span class="math notranslate nohighlight">\(P_{(t)}\)</span> describe the substrate and product concentration for each time point <span class="math notranslate nohighlight">\(t\)</span>. <span class="math notranslate nohighlight">\(S_{t}\)</span> and <span class="math notranslate nohighlight">\(S_{0}\)</span> were individually measured. Thus each enzyme reaction with a given initial substrate concentration had a control reaction with identical substrate concentration. In contrast to the substrate, no calibration standard was available for the product. Therefore, an additional parameter <span class="math notranslate nohighlight">\(k\)</span> was introduced to the mass balance equation, assuming linear relationship between the product concentration and its signal:</p>
<div class="math notranslate nohighlight" id="equation-mass-balance-with-k">
<span class="eqno">(15)<a class="headerlink" href="#equation-mass-balance-with-k" title="Permalink to this equation">#</a></span>\[0 = S_{(t)} + P_{(t)}k - S_{0}\]</div>
<p><span class="math notranslate nohighlight">\(k\)</span> was determined for each data set individually by a minimization algorithm. The minimization objective was to find the optimal <span class="math notranslate nohighlight">\(k\)</span>, which minimizes all slopes of an experiment. If the slopes are zero, mass conservation is given over the observed time-course. Mass balances of all measurements are visualized in Fig. 9.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Defenition of parameter &#39;k&#39;</span>
<span class="n">params</span> <span class="o">=</span> <span class="n">Parameters</span><span class="p">()</span>
<span class="n">params</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="s2">&quot;k&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span> <span class="nb">min</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="nb">max</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>

<span class="c1"># Target function for the minimizer</span>
<span class="k">def</span> <span class="nf">residual</span><span class="p">(</span><span class="n">params</span><span class="p">,</span> <span class="n">x</span><span class="p">):</span>
    <span class="n">k</span><span class="o">=</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;k&quot;</span><span class="p">]</span>

    <span class="n">substrate</span><span class="p">,</span> <span class="n">product</span><span class="p">,</span>  <span class="n">control</span> <span class="o">=</span> <span class="n">x</span>
    <span class="n">slopes</span> <span class="o">=</span> <span class="n">substrate</span><span class="p">[:,</span><span class="mi">0</span><span class="p">]</span> <span class="o">*</span> <span class="mf">0.0</span>
    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="p">(</span><span class="n">s</span><span class="p">,</span> <span class="n">p</span><span class="p">,</span> <span class="n">c</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">substrate</span><span class="p">,</span> <span class="n">product</span><span class="p">,</span> <span class="n">control</span><span class="p">)):</span>
        <span class="n">sub</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">s</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="n">prod</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">p</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="n">cont</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

        <span class="n">model</span> <span class="o">=</span> <span class="n">sub</span> <span class="o">+</span> <span class="n">prod</span><span class="o">*</span><span class="n">k</span> <span class="o">-</span> <span class="n">cont</span>

        <span class="n">slopes</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">=</span> <span class="n">linregress</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">model</span><span class="p">)),</span> <span class="n">model</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>

    <span class="k">return</span> <span class="n">slopes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>


<span class="n">f</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.5</span><span class="p">,</span> <span class="mi">15</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">for</span> <span class="n">e</span><span class="p">,</span> <span class="p">(</span><span class="n">doc</span><span class="p">,</span> <span class="n">ax</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">concentration_enzymemldocs</span><span class="p">,</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">())):</span>
    <span class="n">substrate</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">product</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">initial_substrate</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">control</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">measurement</span> <span class="ow">in</span> <span class="n">doc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
        <span class="n">initial_substrate</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="n">substrate_id</span><span class="p">)</span><span class="o">.</span><span class="n">init_conc</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="n">substrate_id</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">:</span>
            <span class="n">substrate</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="n">product_id</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">:</span>
            <span class="n">product</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s2&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">:</span>
            <span class="n">control</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">)</span>

    <span class="n">time</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">replicate</span><span class="o">.</span><span class="n">time</span><span class="p">)</span>

    <span class="n">product</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">product</span><span class="p">)</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="nb">len</span><span class="p">((</span><span class="n">product</span><span class="p">))</span><span class="o">/</span><span class="mi">3</span><span class="p">),</span><span class="mi">3</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>
    <span class="n">substrate</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">substrate</span><span class="p">)</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="nb">len</span><span class="p">((</span><span class="n">substrate</span><span class="p">))</span><span class="o">/</span><span class="mi">3</span><span class="p">),</span><span class="mi">3</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>
    <span class="n">control</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">control</span><span class="p">)</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="nb">len</span><span class="p">((</span><span class="n">control</span><span class="p">))</span><span class="o">/</span><span class="mi">3</span><span class="p">),</span><span class="mi">3</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>
    <span class="n">initial_substrate</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">initial_substrate</span><span class="p">)</span>

    <span class="n">x</span> <span class="o">=</span> <span class="p">(</span><span class="n">substrate</span><span class="p">,</span> <span class="n">product</span><span class="p">,</span> <span class="n">control</span><span class="p">)</span>

    <span class="n">result</span> <span class="o">=</span> <span class="n">minimize</span><span class="p">(</span><span class="n">residual</span><span class="p">,</span> <span class="n">params</span><span class="p">,</span> <span class="n">args</span><span class="o">=</span><span class="p">(</span><span class="n">x</span><span class="p">,))</span>

    <span class="n">factor</span> <span class="o">=</span> <span class="n">result</span><span class="o">.</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;k&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span>
    <span class="n">f</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">factor</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="p">(</span><span class="n">s</span><span class="p">,</span> <span class="n">p</span><span class="p">,</span> <span class="n">c</span><span class="p">,</span> <span class="n">l</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">substrate</span><span class="p">,</span> <span class="n">product</span><span class="p">,</span>  <span class="n">control</span><span class="p">,</span> <span class="n">initial_substrate</span><span class="p">)):</span>
        <span class="n">init</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="n">sub</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">s</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="n">prod</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">p</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="n">cont</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">c</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="n">balance</span> <span class="o">=</span> <span class="p">(</span><span class="n">sub</span> <span class="o">+</span> <span class="n">prod</span><span class="o">*</span><span class="n">factor</span> <span class="o">-</span> <span class="n">cont</span><span class="p">)</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">doc</span><span class="o">.</span><span class="n">getMeasurement</span><span class="p">(</span><span class="s2">&quot;m1&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">replicates</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">time</span><span class="p">,</span> <span class="n">balance</span><span class="p">,</span> <span class="n">label</span> <span class="o">=</span> <span class="n">l</span><span class="p">)</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">i</span><span class="o">%</span><span class="k">5</span>:
            <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;mass balance [uM]&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">i</span> <span class="ow">in</span> <span class="p">[</span><span class="mi">24</span><span class="p">,</span> <span class="mi">25</span><span class="p">,</span> <span class="mi">26</span><span class="p">,</span> <span class="mi">27</span><span class="p">,</span> <span class="mi">28</span><span class="p">,</span> <span class="mi">29</span><span class="p">]:</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [s]&quot;</span><span class="p">)</span>
        <span class="n">pH</span> <span class="o">=</span> <span class="n">doc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">ph</span>
        <span class="n">temp</span> <span class="o">=</span> <span class="n">doc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">temperature</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;pH </span><span class="si">{</span><span class="n">pH</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">temp</span><span class="si">}</span><span class="s2">°C&quot;</span><span class="p">)</span><span class="c1">#, factor: {factor:.0f}&quot;)</span>
    <span class="k">if</span> <span class="n">e</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">handles</span><span class="p">,</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">handles</span><span class="p">,</span> <span class="n">labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;lower center&quot;</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">labels</span><span class="p">),</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial ABTS [uM]&quot;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">0.5</span><span class="p">,</span><span class="o">-</span><span class="mf">0.03</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/SLAC_kinetic_characterization_10_0.png" src="_images/SLAC_kinetic_characterization_10_0.png" />
</div>
</div>
<p><em>Fig. 9: Quality control of each experiment through modeled mass balance.</em></p>
<p>For all experiments, except for pH 4, 45°C and pH 4.5, 35°C, the product and substrate is, qualitatively, in balance over the reaction time-course (Fig. 9). Mass balances, differing from 0 uM originate from differing substrate concentration in the enzyme reaction and the substrate control. In the reactions at pH 4, 45°C and pH 4.5, 35°C the mass balance slopes are not linear. This indicates issues with the measurement. Therefore, the respective measurements likely result in wrong parameter estimations.</p>
</section>
<section id="kinetic-parameter-estimation">
<h4>Kinetic parameter estimation<a class="headerlink" href="#kinetic-parameter-estimation" title="Permalink to this headline">#</a></h4>
<section id="choice-of-kinetic-model">
<h5>Choice of kinetic model<a class="headerlink" href="#choice-of-kinetic-model" title="Permalink to this headline">#</a></h5>
<p>Model selection is vital for parameter estimation. Thus, different settings for the parameter estimator were tested. Firstly, product inhibition models were excluded, since in some of the experiments the measured initial substrate concentration was higher than the specified one, which is used to calculate product concentrations. This resulted in calculated product concentrations with negative values.<br />
Secondly, substrate inhibition models were excluded, since the model was not able to describe the observed reaction kinetics. This was evident from a higher AIC as well as more than 100 % standard deviation on the estimated parameters.<br />
Lastly, the irreversible Michaelis-Menten model with and without time-dependent enzyme inactivation was compared, since enzyme inactivation was observed in previous experiments. Estimated parameters and the fitted models are shown for the experimental data at pH 3 and 25˚C under two scenarios - taking enzyme inactivation into consideration and, conversely, without enzyme inactivation (Fig. 10).</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">5</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">ax</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()):</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
            <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">i</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Estimated parameters without time-dependent enzyme inactivation:&quot;</span><span class="p">)</span>
        <span class="n">kinetics</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">concentration_enzymemldocs</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="s2">&quot;s0&quot;</span><span class="p">,</span> <span class="s2">&quot;substrate&quot;</span><span class="p">)</span>
        <span class="n">kinetics</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">only_irrev_MM</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
        <span class="n">kinetics</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;without enzyme inactivation&quot;</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS [uM]&quot;</span><span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Estimated parameters with time-dependent enzyme inactivation:&quot;</span><span class="p">)</span>
        <span class="n">kinetics</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">concentration_enzymemldocs</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="s2">&quot;s0&quot;</span><span class="p">,</span> <span class="s2">&quot;substrate&quot;</span><span class="p">)</span>
        <span class="n">kinetics</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">enzyme_inactivation</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">only_irrev_MM</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
        <span class="n">kinetics</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;with enzyme inactivation&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [s]&quot;</span><span class="p">)</span>
<span class="n">handles</span><span class="p">,</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">handles</span><span class="p">,</span> <span class="n">labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;lower center&quot;</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">labels</span><span class="p">),</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial ABTS [uM]&quot;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">0.5</span><span class="p">,</span><span class="o">-</span><span class="mf">0.09</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Estimated parameters without time-dependent enzyme inactivation:
Fitting data to:
 - irreversible Michaelis Menten model
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>AIC</th>
      <th>Km [umole / l]</th>
      <th>kcat / Km [1/s * 1/umole / l]</th>
      <th>kcat [1/s]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>irreversible Michaelis Menten</th>
      <td>-162</td>
      <td>35.473 +/- 5.52%</td>
      <td>0.009 +/- 5.89%</td>
      <td>0.308 +/- 2.07%</td>
    </tr>
  </tbody>
</table>
</div></div><div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Estimated parameters with time-dependent enzyme inactivation:
Fitting data to:
 - irreversible Michaelis Menten model
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>AIC</th>
      <th>Km [umole / l]</th>
      <th>kcat / Km [1/s * 1/umole / l]</th>
      <th>kcat [1/s]</th>
      <th>ki time-dep enzyme-inactiv. [1/s]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>irreversible Michaelis Menten</th>
      <td>-476</td>
      <td>35.137 +/- 3.00%</td>
      <td>0.013 +/- 3.58%</td>
      <td>0.460 +/- 1.95%</td>
      <td>0.001 +/- 4.47%</td>
    </tr>
  </tbody>
</table>
</div></div><img alt="_images/SLAC_kinetic_characterization_13_4.png" src="_images/SLAC_kinetic_characterization_13_4.png" />
</div>
</div>
<p><em>Fig. 10: Fitted irreversible Michaelis-Menten model to experimental data of SLAC reaction at pH 3 and 25°C with and without considering time-dependent enzyme inactivation.</em></p>
<p>The model with enzyme inactivation describes the data better (Fig. 10B). This is characterized through a lower AIC, as well as lower standard deviations on the estimated parameters.
In absolute terms, the estimated <span class="math notranslate nohighlight">\(K_{m}\)</span> of both models is approximately identical, whereas the <span class="math notranslate nohighlight">\(k_{cat}\)</span> estimate is approximately 33% lower for the model without enzyme inactivation. Hence, the model without enzyme inactivation underestimates the turnover number of the enzyme. As a result, irreversible Michaelis-Menten model with time-dependent enzyme inactivation was selected for the parameter estimation for all data sets.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Run parameter estimator for all datasets, utilizing multi-processing.</span>
<span class="k">def</span> <span class="nf">run_ParameterEstimator</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">:</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="p">):</span>    
    <span class="n">kinetics</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="s2">&quot;s0&quot;</span><span class="p">,</span> <span class="s2">&quot;substrate&quot;</span><span class="p">)</span>
    <span class="n">kinetics</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">enzyme_inactivation</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">only_irrev_MM</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">kinetics</span>

<span class="n">results</span> <span class="o">=</span> <span class="n">Parallel</span><span class="p">(</span><span class="n">n_jobs</span><span class="o">=</span><span class="mi">8</span><span class="p">)(</span><span class="n">delayed</span><span class="p">(</span><span class="n">run_ParameterEstimator</span><span class="p">)(</span><span class="n">enzmldoc</span><span class="p">)</span> <span class="k">for</span> <span class="n">enzmldoc</span> <span class="ow">in</span> <span class="n">concentration_enzymemldocs</span><span class="p">)</span>
<span class="n">results</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">results</span><span class="p">,</span> <span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">pH</span><span class="p">,</span> <span class="n">x</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">temperature</span><span class="p">))</span>

<span class="c1"># Visualize all fitted models</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.5</span><span class="p">,</span> <span class="mi">15</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="p">(</span><span class="n">doc</span><span class="p">,</span> <span class="n">ax</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">results</span><span class="p">,</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">())):</span>
    <span class="n">ph</span> <span class="o">=</span> <span class="n">doc</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">pH</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">i</span><span class="o">%</span><span class="k">5</span>:
        <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS [uM]&quot;</span><span class="p">)</span>
    <span class="n">doc</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">title</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;pH </span><span class="si">{</span><span class="n">doc</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">pH</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">doc</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">temperature</span><span class="si">}</span><span class="s2">°C&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">i</span> <span class="ow">in</span> <span class="p">[</span><span class="mi">25</span><span class="p">,</span><span class="mi">26</span><span class="p">,</span><span class="mi">27</span><span class="p">,</span><span class="mi">28</span><span class="p">,</span><span class="mi">29</span><span class="p">]:</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [s]&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">i</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">handles</span><span class="p">,</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>
        <span class="n">fig</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">handles</span><span class="p">,</span> <span class="n">labels</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;lower center&quot;</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">labels</span><span class="p">),</span> <span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial ABTS [uM]&quot;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">0.5</span><span class="p">,</span><span class="o">-</span><span class="mf">0.04</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/SLAC_kinetic_characterization_15_0.png" src="_images/SLAC_kinetic_characterization_15_0.png" />
</div>
</div>
<p><em>Fig. 11: Experimental data and fitted irreversible Michaelis-Menten model with time-dependent enzyme inactivation of SLAC reactions under various conditions.</em></p>
<p>Experimental data as well as the fitted model are visualized in Fig 11. Based on the reaction slopes, no catalytic activity was observed for reactions at pH 5 or higher.
As in the mass balance analysis, the reactions at pH 4, 45°C and pH 4.5, 35°C differed from other experiments with identical pH. In both cases, all applied substrate concentrations were higher than intended in the design of the experiment. Furthermore, the resulting parameter estimates have a high uncertainty. Therefore, the respective measurements were excluded from further analysis. Additionally, the results from pH 5.5, 35°C were excluded, since the uncertainty of the parameters could not be estimated. All calculated kinetic parameters are listed in the table below.</p>
<div class="cell tag_hide_cell tag_output_croll docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Extract kinetic parameters of all datasets</span>
<span class="n">kcat</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">kcat_std</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">Km</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">Km_std</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">ki</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">ki_std</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">pH</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">temperature</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">corr_kcat_km</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">result</span> <span class="ow">in</span> <span class="n">results</span><span class="p">:</span>
    <span class="n">params</span> <span class="o">=</span> <span class="n">result</span><span class="o">.</span><span class="n">get_parameter_dict</span><span class="p">()</span>

    <span class="n">kcat</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;k_cat&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">)</span>
    <span class="n">kcat_std</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;k_cat&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">stderr</span><span class="p">)</span>

    <span class="n">Km</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;Km&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">)</span>
    <span class="n">Km_std</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;Km&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">stderr</span><span class="p">)</span>

    <span class="n">ki</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;K_ie&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">)</span>
    <span class="n">ki_std</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="s2">&quot;K_ie&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">stderr</span><span class="p">)</span>  

    <span class="n">correlation</span> <span class="o">=</span> <span class="n">params</span><span class="p">[</span><span class="s2">&quot;k_cat&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">correl</span>
    <span class="k">if</span> <span class="n">correlation</span> <span class="o">==</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">corr_kcat_km</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">float</span><span class="p">(</span><span class="s2">&quot;nan&quot;</span><span class="p">))</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">corr_kcat_km</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">correlation</span><span class="p">[</span><span class="s2">&quot;Km&quot;</span><span class="p">])</span>

    <span class="n">pH</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">pH</span><span class="p">)</span>
    <span class="n">temperature</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">temperature</span><span class="p">)</span>

<span class="c1"># Organize kinetic and experimantal parameters in a &#39;DataFrame&#39;</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_dict</span><span class="p">({</span>
    <span class="s1">&#39;pH&#39;</span><span class="p">:</span><span class="n">pH</span><span class="p">,</span> 
    <span class="s2">&quot;temperature [C]&quot;</span><span class="p">:</span><span class="n">temperature</span><span class="p">,</span> 
    <span class="s1">&#39;kcat [1/s]&#39;</span><span class="p">:</span><span class="n">kcat</span><span class="p">,</span> 
    <span class="s1">&#39;kcat stderr&#39;</span><span class="p">:</span><span class="n">kcat_std</span><span class="p">,</span> 
    <span class="s1">&#39;Km [uM]&#39;</span><span class="p">:</span><span class="n">Km</span><span class="p">,</span> 
    <span class="s1">&#39;Km stderr&#39;</span><span class="p">:</span><span class="n">Km_std</span><span class="p">,</span> 
    <span class="s2">&quot;Enzyme inactivation [1/s]&quot;</span><span class="p">:</span><span class="n">ki</span><span class="p">,</span>
    <span class="s2">&quot;Enzyme inactivation std&quot;</span><span class="p">:</span><span class="n">ki_std</span><span class="p">,</span>
    <span class="s2">&quot;correlation kcat/Km&quot;</span><span class="p">:</span><span class="n">corr_kcat_km</span><span class="p">})</span>

<span class="n">df</span><span class="p">[</span><span class="s2">&quot;kcat/Km [1/s * uM]&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;kcat [1/s]&quot;</span><span class="p">]</span> <span class="o">/</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;Km [uM]&quot;</span><span class="p">]</span>
<span class="n">kcat_km_stderr</span> <span class="o">=</span><span class="p">((</span><span class="n">df</span><span class="p">[</span><span class="s2">&quot;kcat stderr&quot;</span><span class="p">]</span><span class="o">/</span><span class="n">df</span><span class="p">[</span><span class="s2">&quot;kcat [1/s]&quot;</span><span class="p">])</span><span class="o">**</span><span class="mi">2</span><span class="o">+</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s2">&quot;Km stderr&quot;</span><span class="p">]</span><span class="o">/</span><span class="n">df</span><span class="p">[</span><span class="s2">&quot;Km [uM]&quot;</span><span class="p">])</span><span class="o">**</span><span class="mi">2</span><span class="p">)</span><span class="o">**</span><span class="mf">0.5</span> <span class="o">*</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;kcat/Km [1/s * uM]&quot;</span><span class="p">]</span>
<span class="n">df</span><span class="p">[</span><span class="s2">&quot;kcat/Km stderr&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">kcat_km_stderr</span>

<span class="c1"># Exclued results from failed experiments</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="mi">25</span><span class="p">)</span> <span class="c1"># reached parameter boundaries --&gt; inactive</span>
<span class="c1">#df = df.drop(index=12) #Km really high stddev</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="mi">17</span><span class="p">)</span> <span class="c1">#kcat outlier, mass balance outlier</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span> <span class="c1">#kcat outlier, mass balance outlier</span>
<span class="n">df</span><span class="p">[</span><span class="s2">&quot;zeros&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">(</span><span class="mi">27</span><span class="p">)</span>

<span class="c1"># Calculate halflife of enzyme</span>
<span class="k">def</span> <span class="nf">calulate_halflife</span><span class="p">(</span><span class="n">x</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">log</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span><span class="o">/</span><span class="n">x</span>

<span class="n">df</span><span class="p">[</span><span class="s2">&quot;half life [min]&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;Enzyme inactivation [1/s]&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">calulate_halflife</span><span class="p">)</span><span class="o">/</span><span class="mi">60</span>
<span class="n">df</span><span class="p">[</span><span class="s2">&quot;half life std&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;Enzyme inactivation std&quot;</span><span class="p">]</span> <span class="o">/</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;Enzyme inactivation [1/s]&quot;</span><span class="p">]</span> <span class="o">*</span> <span class="n">df</span><span class="p">[</span><span class="s2">&quot;half life [min]&quot;</span><span class="p">]</span>

<span class="c1"># Delete inactivation rates</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;Enzyme inactivation [1/s]&quot;</span><span class="p">,</span> <span class="s2">&quot;Enzyme inactivation std&quot;</span><span class="p">])</span>
<span class="n">df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>pH</th>
      <th>temperature [C]</th>
      <th>kcat [1/s]</th>
      <th>kcat stderr</th>
      <th>Km [uM]</th>
      <th>Km stderr</th>
      <th>correlation kcat/Km</th>
      <th>kcat/Km [1/s * uM]</th>
      <th>kcat/Km stderr</th>
      <th>zeros</th>
      <th>half life [min]</th>
      <th>half life std</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>3.0</td>
      <td>25.0</td>
      <td>0.459962</td>
      <td>0.008966</td>
      <td>35.137357</td>
      <td>1.053968</td>
      <td>0.535141</td>
      <td>0.013090</td>
      <td>0.000468</td>
      <td>0.0</td>
      <td>9.139562</td>
      <td>0.408568</td>
    </tr>
    <tr>
      <th>1</th>
      <td>3.0</td>
      <td>30.0</td>
      <td>0.470366</td>
      <td>0.009855</td>
      <td>32.114083</td>
      <td>1.037088</td>
      <td>0.522963</td>
      <td>0.014647</td>
      <td>0.000564</td>
      <td>0.0</td>
      <td>9.522170</td>
      <td>0.476531</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3.0</td>
      <td>35.0</td>
      <td>0.546029</td>
      <td>0.016598</td>
      <td>37.454496</td>
      <td>1.765891</td>
      <td>0.558865</td>
      <td>0.014578</td>
      <td>0.000818</td>
      <td>0.0</td>
      <td>10.304462</td>
      <td>0.776312</td>
    </tr>
    <tr>
      <th>3</th>
      <td>3.0</td>
      <td>40.0</td>
      <td>0.712488</td>
      <td>0.021754</td>
      <td>46.441229</td>
      <td>2.155285</td>
      <td>0.608947</td>
      <td>0.015342</td>
      <td>0.000852</td>
      <td>0.0</td>
      <td>9.127765</td>
      <td>0.600883</td>
    </tr>
    <tr>
      <th>4</th>
      <td>3.0</td>
      <td>45.0</td>
      <td>0.815724</td>
      <td>0.022451</td>
      <td>47.967746</td>
      <td>2.047138</td>
      <td>0.623270</td>
      <td>0.017006</td>
      <td>0.000864</td>
      <td>0.0</td>
      <td>9.087343</td>
      <td>0.530573</td>
    </tr>
    <tr>
      <th>5</th>
      <td>3.5</td>
      <td>25.0</td>
      <td>0.420480</td>
      <td>0.045642</td>
      <td>89.752447</td>
      <td>13.848584</td>
      <td>0.746575</td>
      <td>0.004685</td>
      <td>0.000884</td>
      <td>0.0</td>
      <td>8.824513</td>
      <td>1.680327</td>
    </tr>
    <tr>
      <th>6</th>
      <td>3.5</td>
      <td>30.0</td>
      <td>0.399649</td>
      <td>0.007455</td>
      <td>61.224383</td>
      <td>1.674235</td>
      <td>0.666682</td>
      <td>0.006528</td>
      <td>0.000216</td>
      <td>0.0</td>
      <td>14.545045</td>
      <td>0.804521</td>
    </tr>
    <tr>
      <th>7</th>
      <td>3.5</td>
      <td>35.0</td>
      <td>0.246773</td>
      <td>0.009247</td>
      <td>28.237534</td>
      <td>1.572755</td>
      <td>0.554406</td>
      <td>0.008739</td>
      <td>0.000587</td>
      <td>0.0</td>
      <td>18.269277</td>
      <td>2.766875</td>
    </tr>
    <tr>
      <th>8</th>
      <td>3.5</td>
      <td>40.0</td>
      <td>0.650031</td>
      <td>0.012397</td>
      <td>79.432405</td>
      <td>2.189303</td>
      <td>0.731830</td>
      <td>0.008183</td>
      <td>0.000274</td>
      <td>0.0</td>
      <td>11.026573</td>
      <td>0.453021</td>
    </tr>
    <tr>
      <th>9</th>
      <td>3.5</td>
      <td>45.0</td>
      <td>0.978983</td>
      <td>0.026556</td>
      <td>103.784356</td>
      <td>4.040850</td>
      <td>0.774126</td>
      <td>0.009433</td>
      <td>0.000448</td>
      <td>0.0</td>
      <td>9.110325</td>
      <td>0.425387</td>
    </tr>
    <tr>
      <th>10</th>
      <td>4.0</td>
      <td>25.0</td>
      <td>0.115133</td>
      <td>0.008752</td>
      <td>65.852866</td>
      <td>2.476592</td>
      <td>0.247717</td>
      <td>0.001748</td>
      <td>0.000148</td>
      <td>0.0</td>
      <td>111.702355</td>
      <td>222.064200</td>
    </tr>
    <tr>
      <th>11</th>
      <td>4.0</td>
      <td>30.0</td>
      <td>0.192374</td>
      <td>0.020265</td>
      <td>95.314777</td>
      <td>5.569713</td>
      <td>-0.718208</td>
      <td>0.002018</td>
      <td>0.000243</td>
      <td>0.0</td>
      <td>23.819987</td>
      <td>10.874634</td>
    </tr>
    <tr>
      <th>12</th>
      <td>4.0</td>
      <td>35.0</td>
      <td>0.282991</td>
      <td>0.029279</td>
      <td>100.647793</td>
      <td>11.576143</td>
      <td>0.608286</td>
      <td>0.002812</td>
      <td>0.000435</td>
      <td>0.0</td>
      <td>14.929497</td>
      <td>4.974473</td>
    </tr>
    <tr>
      <th>13</th>
      <td>4.0</td>
      <td>40.0</td>
      <td>0.225429</td>
      <td>0.012766</td>
      <td>97.966631</td>
      <td>1.677116</td>
      <td>0.168743</td>
      <td>0.002301</td>
      <td>0.000136</td>
      <td>0.0</td>
      <td>29.441552</td>
      <td>12.288914</td>
    </tr>
    <tr>
      <th>15</th>
      <td>4.5</td>
      <td>25.0</td>
      <td>0.368730</td>
      <td>0.025801</td>
      <td>110.902372</td>
      <td>0.830267</td>
      <td>-0.736267</td>
      <td>0.003325</td>
      <td>0.000234</td>
      <td>0.0</td>
      <td>22.536023</td>
      <td>6.247470</td>
    </tr>
    <tr>
      <th>16</th>
      <td>4.5</td>
      <td>30.0</td>
      <td>0.486402</td>
      <td>0.036084</td>
      <td>148.193267</td>
      <td>3.440830</td>
      <td>0.816213</td>
      <td>0.003282</td>
      <td>0.000255</td>
      <td>0.0</td>
      <td>19.046651</td>
      <td>4.109261</td>
    </tr>
    <tr>
      <th>18</th>
      <td>4.5</td>
      <td>40.0</td>
      <td>0.549232</td>
      <td>0.021351</td>
      <td>178.481651</td>
      <td>2.544896</td>
      <td>0.084106</td>
      <td>0.003077</td>
      <td>0.000127</td>
      <td>0.0</td>
      <td>34.946634</td>
      <td>11.931005</td>
    </tr>
    <tr>
      <th>19</th>
      <td>4.5</td>
      <td>45.0</td>
      <td>0.235716</td>
      <td>0.013338</td>
      <td>64.718239</td>
      <td>0.037679</td>
      <td>-0.113239</td>
      <td>0.003642</td>
      <td>0.000206</td>
      <td>0.0</td>
      <td>108.053681</td>
      <td>160.970528</td>
    </tr>
    <tr>
      <th>20</th>
      <td>5.0</td>
      <td>25.0</td>
      <td>0.008085</td>
      <td>0.006871</td>
      <td>15.067578</td>
      <td>20.447918</td>
      <td>0.276523</td>
      <td>0.000537</td>
      <td>0.000859</td>
      <td>0.0</td>
      <td>115.522134</td>
      <td>2593.086797</td>
    </tr>
    <tr>
      <th>21</th>
      <td>5.0</td>
      <td>30.0</td>
      <td>0.040665</td>
      <td>0.005659</td>
      <td>50.772187</td>
      <td>3.784251</td>
      <td>0.009319</td>
      <td>0.000801</td>
      <td>0.000126</td>
      <td>0.0</td>
      <td>115.512956</td>
      <td>453.016208</td>
    </tr>
    <tr>
      <th>22</th>
      <td>5.0</td>
      <td>35.0</td>
      <td>0.061677</td>
      <td>0.006158</td>
      <td>68.874565</td>
      <td>0.267258</td>
      <td>0.299400</td>
      <td>0.000895</td>
      <td>0.000089</td>
      <td>0.0</td>
      <td>115.519891</td>
      <td>286.705892</td>
    </tr>
    <tr>
      <th>23</th>
      <td>5.0</td>
      <td>40.0</td>
      <td>0.131196</td>
      <td>0.017131</td>
      <td>91.314407</td>
      <td>2.756719</td>
      <td>-0.426692</td>
      <td>0.001437</td>
      <td>0.000193</td>
      <td>0.0</td>
      <td>8.399863</td>
      <td>2.519856</td>
    </tr>
    <tr>
      <th>24</th>
      <td>5.0</td>
      <td>45.0</td>
      <td>0.129399</td>
      <td>0.014577</td>
      <td>96.735544</td>
      <td>2.562607</td>
      <td>-0.696719</td>
      <td>0.001338</td>
      <td>0.000155</td>
      <td>0.0</td>
      <td>11.438272</td>
      <td>2.986966</td>
    </tr>
    <tr>
      <th>26</th>
      <td>5.5</td>
      <td>30.0</td>
      <td>0.014374</td>
      <td>0.007402</td>
      <td>96.125085</td>
      <td>0.393705</td>
      <td>-0.200038</td>
      <td>0.000150</td>
      <td>0.000077</td>
      <td>0.0</td>
      <td>115.523248</td>
      <td>1471.972292</td>
    </tr>
    <tr>
      <th>27</th>
      <td>5.5</td>
      <td>35.0</td>
      <td>0.124370</td>
      <td>0.023968</td>
      <td>81.996103</td>
      <td>22.257271</td>
      <td>0.771045</td>
      <td>0.001517</td>
      <td>0.000505</td>
      <td>0.0</td>
      <td>5.841192</td>
      <td>1.400781</td>
    </tr>
    <tr>
      <th>28</th>
      <td>5.5</td>
      <td>40.0</td>
      <td>0.055892</td>
      <td>0.016036</td>
      <td>48.740460</td>
      <td>19.833207</td>
      <td>0.552798</td>
      <td>0.001147</td>
      <td>0.000571</td>
      <td>0.0</td>
      <td>6.132460</td>
      <td>2.957303</td>
    </tr>
    <tr>
      <th>29</th>
      <td>5.5</td>
      <td>45.0</td>
      <td>0.075249</td>
      <td>0.015052</td>
      <td>115.390172</td>
      <td>27.541849</td>
      <td>0.604743</td>
      <td>0.000652</td>
      <td>0.000203</td>
      <td>0.0</td>
      <td>10.834329</td>
      <td>5.343971</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
</section>
<section id="high-correlation-between-k-cat-and-k-m">
<h5>High correlation between <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span><a class="headerlink" href="#high-correlation-between-k-cat-and-k-m" title="Permalink to this headline">#</a></h5>
<p>The parameter estimates for <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> showed correlation. For experiments at pH 3 and pH 3.5, the parameters were correlated between <em>r<sup>2</sup></em> = 0.5 - 0.85. This indicates, that the highest initial substrate concentration, which was applied for parameter estimation was not sufficiently high. Ideally, the highest initial substrate concentration applied for a kinetic experiment should be 10-fold higher than <span class="math notranslate nohighlight">\(K_{m}\)</span> to allow independent estimates for <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> <span id="id2">[<a class="reference internal" href="welcome.html#id11" title="Keith F Tipton, Richard N Armstrong, Barbara M Bakker, Amos Bairoch, Athel Cornish-Bowden, Peter J Halling, Jan-Hendrik Hofmeyr, Thomas S Leyh, Carsten Kettner, Frank M Raushel, and others. Standards for reporting enzyme data: the strenda consortium: what it aims to do and why it should be helpful. Perspectives in Science, 1(1-6):131–137, 2014.">Tipton <em>et al.</em>, 2014</a>]</span>. In this scenario, only reactions with an initial substrate concentration up to 175 uM were used for parameter estimation, due to the limited photometric detection range of ABTS. Hence, ABTS was only applied 1.5-fold to 6-fold of the estimated <span class="math notranslate nohighlight">\(K_{m}\)</span>.<br />
For reactions at pH 4 and above, positive as well as negative correlations were observed. This might result from wrong parameter estimates, due to minimal enzyme activity at the given reaction conditions.</p>
<p>Catalytic efficiency for all experimental conditions is visualized in Fig. 12B, whereas the pH is color-coded. For reactions with identical pH values, <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> both increase with temperature. This might be attributed to the high correlation between the parameters. Thus, the true change of <span class="math notranslate nohighlight">\(k_{cat}\)</span> or <span class="math notranslate nohighlight">\(K_{m}\)</span> through pH or temperature cannot be assessed with certainty. Therefore, the catalytic efficiency <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span>, was a better measure to assess the enzyme activity under different reaction conditions.</p>
<p>For reactions at pH 5 and above, almost no catalytic activity was observed. Therefore, data of these experimental conditions is excluded from further analysis.</p>
</section>
<section id="parameter-estimates-for-k-cat-and-k-m">
<h5>Parameter estimates for <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span><a class="headerlink" href="#parameter-estimates-for-k-cat-and-k-m" title="Permalink to this headline">#</a></h5>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Visualize estimated parameters</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.8</span><span class="p">,</span> <span class="mf">4.8</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">ax</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()):</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
            <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">i</span><span class="o">==</span><span class="mi">1</span><span class="p">:</span>
        <span class="n">plot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">xdata</span><span class="o">=</span><span class="s2">&quot;Km [uM]&quot;</span><span class="p">,</span> <span class="n">ydata</span><span class="o">=</span><span class="s2">&quot;kcat [1/s]&quot;</span><span class="p">,</span> <span class="n">xerror</span><span class="o">=</span><span class="s2">&quot;Km stderr&quot;</span><span class="p">,</span> <span class="n">yerror</span><span class="o">=</span><span class="s2">&quot;kcat stderr&quot;</span><span class="p">,</span> <span class="n">colors</span><span class="o">=</span><span class="s2">&quot;pH&quot;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">plot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">xdata</span><span class="o">=</span><span class="s2">&quot;pH&quot;</span><span class="p">,</span> <span class="n">ydata</span><span class="o">=</span><span class="s2">&quot;kcat/Km [1/s * uM]&quot;</span><span class="p">,</span> <span class="n">xerror</span><span class="o">=</span><span class="s2">&quot;zeros&quot;</span><span class="p">,</span> <span class="n">yerror</span><span class="o">=</span><span class="s2">&quot;kcat/Km stderr&quot;</span><span class="p">,</span> <span class="n">colors</span><span class="o">=</span><span class="s2">&quot;temperature [C]&quot;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/SLAC_kinetic_characterization_19_0.png" src="_images/SLAC_kinetic_characterization_19_0.png" />
</div>
</div>
<p><em>Fig. 12: A: Catalytic efficiency of SLAC under different pH and temperature conditions. B: Kinetic parameters <span class="math notranslate nohighlight">\(k_{cat}\)</span> and {K</em>{m}} estimates for SLAC reactions at different pH values._</p>
<p><span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span> in relation to reaction pH is visualized in Fig. 12A. The highest catalytic efficiency was observed at pH 3 and 45°C. Increasing the pH by 0.5 reduced <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span> approximately by half. For higher pH values, <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span> is even further reduced. The catalytic efficiency is therefore highly sensitive to the pH. This might source from changes in the protonation state of either substrate or enzyme, hindering the efficient formation of the enzyme-substrate complex.
In terms of temperature, higher <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span> was achieved at higher temperatures. SLAC might even be more active above 45°C and under pH 3, since in previous studies SLAC was the most active at 95°C <span id="id3">[<a class="reference internal" href="welcome.html#id30" title="Alaric Prins, L Kleinsmidt, Nuraan Khan, Bronwyn Kirby, Tukayi Kudanga, Jannik Vollmer, Juergen Pleiss, Stephanie Burton, and M Le Roes-Hill. The effect of mutations near the t1 copper site on the biochemical characteristics of the small laccase from streptomyces coelicolor a3 (2). Enzyme and microbial technology, 68:23–32, 2015.">Prins <em>et al.</em>, 2015</a>]</span>.</p>
</section>
<section id="time-dependent-enzyme-inactivation">
<h5>Time-dependent enzyme inactivation<a class="headerlink" href="#time-dependent-enzyme-inactivation" title="Permalink to this headline">#</a></h5>
<p>The half life of SLAC was calculated with equation <a class="reference internal" href="#equation-enzyme-halflife">(16)</a> <span id="id4">[<a class="reference internal" href="welcome.html#id8" title="Engelbert Buxbaum. Enzyme Kinetics: Special Cases, pages 185–191. Springer International Publishing, Cham, 2015. URL: https://doi.org/10.1007/978-3-319-19920-7_8, doi:10.1007/978-3-319-19920-7_8.">Buxbaum, 2015</a>]</span>.</p>
<div class="math notranslate nohighlight" id="equation-enzyme-halflife">
<span class="eqno">(16)<a class="headerlink" href="#equation-enzyme-halflife" title="Permalink to this equation">#</a></span>\[t_{\frac{1}{2}} = \frac{ln(2)}{k_{inact}}\]</div>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Exclued inactive enzyme reactions above pH 4.5 and discard measurements with more than 100 % standard deviation on enzyme inactivation parameter</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">loc</span><span class="p">[:</span><span class="mi">19</span><span class="p">]</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span> <span class="c1"># more than 100 % standard deviation on enzyme inactivation parameter</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="mi">19</span><span class="p">)</span> <span class="c1"># more than 100 % standard deviation on enzyme inactivation parameter</span>

<span class="c1"># Plot half life over catalytic efficiency</span>
<span class="n">plot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">xdata</span><span class="o">=</span><span class="s2">&quot;kcat/Km [1/s * uM]&quot;</span><span class="p">,</span> <span class="n">ydata</span><span class="o">=</span><span class="s2">&quot;half life [min]&quot;</span><span class="p">,</span> <span class="n">xerror</span><span class="o">=</span><span class="s2">&quot;kcat/Km stderr&quot;</span><span class="p">,</span> <span class="n">yerror</span><span class="o">=</span><span class="s2">&quot;half life std&quot;</span><span class="p">,</span> <span class="n">colors</span><span class="o">=</span><span class="s2">&quot;pH&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/SLAC_kinetic_characterization_21_0.png" src="_images/SLAC_kinetic_characterization_21_0.png" />
</div>
</div>
<p>The enzyme’s calculated half life was between 8 - 18 min for reactions at pH 3 and pH 3.5, whereas enzyme reactions at higher pH values showed a half life between 15 - 35 min. Generally, reactions with higher catalytic efficiency showed a shorter half life compared to reactions with lower catalytic efficiency.
In order to check for corrections between <span class="math notranslate nohighlight">\(k_{inact}\)</span>, <span class="math notranslate nohighlight">\(k_{cat}\)</span>, <span class="math notranslate nohighlight">\(K_{m}\)</span>, <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span>, as well as the reaction temperature and pH, a correlation analysis was conducted across all datasets.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Calculate correlations</span>
<span class="n">df_corr</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;kcat stderr&quot;</span><span class="p">,</span> <span class="s2">&quot;Km stderr&quot;</span><span class="p">,</span> <span class="s2">&quot;correlation kcat/Km&quot;</span><span class="p">,</span> <span class="s2">&quot;kcat/Km stderr&quot;</span><span class="p">,</span> <span class="s2">&quot;zeros&quot;</span><span class="p">,</span> <span class="s2">&quot;half life std&quot;</span><span class="p">])</span>
<span class="n">rho</span> <span class="o">=</span> <span class="n">df_corr</span><span class="o">.</span><span class="n">corr</span><span class="p">()</span>

<span class="c1"># Visualize correlation heatmap</span>
<span class="c1">#sns.heatmap(rho, </span>
<span class="c1">#        xticklabels=rho.columns,</span>
<span class="c1">#        yticklabels=rho.columns,</span>
<span class="c1">#        cmap = sns.color_palette(&quot;vlag&quot;, as_cmap=True))</span>
<span class="c1">#plt.show()</span>

<span class="c1"># Calculate confidence intercal for correlations</span>
<span class="n">pval</span> <span class="o">=</span> <span class="n">df_corr</span><span class="o">.</span><span class="n">corr</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">:</span> <span class="n">pearsonr</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">)[</span><span class="mi">1</span><span class="p">])</span> <span class="o">-</span> <span class="n">np</span><span class="o">.</span><span class="n">eye</span><span class="p">(</span><span class="o">*</span><span class="n">rho</span><span class="o">.</span><span class="n">shape</span><span class="p">)</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">pval</span><span class="o">.</span><span class="n">applymap</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39;*&#39;</span> <span class="k">for</span> <span class="n">t</span> <span class="ow">in</span> <span class="p">[</span><span class="mf">.05</span><span class="p">]</span> <span class="k">if</span> <span class="n">x</span><span class="o">&lt;=</span><span class="n">t</span><span class="p">]))</span>
<span class="n">rho</span><span class="o">.</span><span class="n">round</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span> <span class="o">+</span> <span class="n">p</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>pH</th>
      <th>temperature [C]</th>
      <th>kcat [1/s]</th>
      <th>Km [uM]</th>
      <th>kcat/Km [1/s * uM]</th>
      <th>half life [min]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>pH</th>
      <td>1.0*</td>
      <td>-0.15</td>
      <td>-0.39</td>
      <td>0.88*</td>
      <td>-0.89*</td>
      <td>0.8*</td>
    </tr>
    <tr>
      <th>temperature [C]</th>
      <td>-0.15</td>
      <td>1.0*</td>
      <td>0.6*</td>
      <td>0.07</td>
      <td>0.27</td>
      <td>0.03</td>
    </tr>
    <tr>
      <th>kcat [1/s]</th>
      <td>-0.39</td>
      <td>0.6*</td>
      <td>1.0*</td>
      <td>-0.03</td>
      <td>0.58*</td>
      <td>-0.49</td>
    </tr>
    <tr>
      <th>Km [uM]</th>
      <td>0.88*</td>
      <td>0.07</td>
      <td>-0.03</td>
      <td>1.0*</td>
      <td>-0.75*</td>
      <td>0.68*</td>
    </tr>
    <tr>
      <th>kcat/Km [1/s * uM]</th>
      <td>-0.89*</td>
      <td>0.27</td>
      <td>0.58*</td>
      <td>-0.75*</td>
      <td>1.0*</td>
      <td>-0.71*</td>
    </tr>
    <tr>
      <th>half life [min]</th>
      <td>0.8*</td>
      <td>0.03</td>
      <td>-0.49</td>
      <td>0.68*</td>
      <td>-0.71*</td>
      <td>1.0*</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>The above table shows correlation between kinetic parameters and experimental conditions. Thereby, significant (p &lt; 0.05) correlations are labeled with ‘*’. The half life of SLAC is significantly correlated to <span class="math notranslate nohighlight">\(\frac{k_{cat}}{K_{m}}\)</span> as well as to pH. Due to correlation between <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span>, the enzyme’s half life is also correlated to <span class="math notranslate nohighlight">\(K_{m}\)</span>. Multiple reasons might be the reason for this observation. On the one hand, the observed correlation might be of technical origin. As such, the system of ordinary differential equations, describing the change in substrate and enzyme concentration, are not an accurate model for the reaction system. Therefore, high cross-correlation occurs, since individual observations cannot be attributed to individual parameters. On the other hand, enzyme activity and enzyme inactivation might be causally related. In this case, the formed ABTS radical might inactivate the enzyme by potential suicide inhibition, or the enzyme deteriorates through catalysis.</p>
</section>
</section>
<section id="conclusion">
<h4>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this headline">#</a></h4>
<section id="id5">
<h5>Data management<a class="headerlink" href="#id5" title="Permalink to this headline">#</a></h5>
<p>The established data pipeline allows scalable data preparation and analysis of enzyme kinetics experiments. In future experiments this workflow could be used to further expand the experimental parameter space. As a result, different enzymes with different substrates in different buffers at different temperatures and pH values could be analyzed.</p>
</section>
<section id="correlation-between-parameters">
<h5>Correlation between parameters<a class="headerlink" href="#correlation-between-parameters" title="Permalink to this headline">#</a></h5>
<p>The highest catalytic efficiency of SLAC was observed at pH 3 at 45°C and therefore on the edge of the investigated parameter space. Optimal reaction conditions might therefore be at an even lower pH and higher temperature. <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> were correlated, which is likely the result of a too low initial substrate concentration in relation to the true <span class="math notranslate nohighlight">\(K_{m}\)</span> of the enzyme at a given experimental condition. Hence, in adjoining experiments the product signal could be used for parameter estimation, since the concentration to absorbance ratio is lower compared to the one of the product. However, a reliable method for product quantification would need to be established first.</p>
</section>
<section id="enzyme-inactivation">
<h5>Enzyme inactivation<a class="headerlink" href="#enzyme-inactivation" title="Permalink to this headline">#</a></h5>
<p>SLAC showed a short half life of approximately 10 min at reaction conditions with the highest catalytic efficiency, which was correlated to the catalytic efficiency. Further experiments are required to investigate the observed inactivation. This might be done by comparing the half life of SLAC with different substrates as well as higher enzyme concentrations.</p>
</section>
</section>
<section id="save-modeling-results">
<h4>Save modeling results<a class="headerlink" href="#save-modeling-results" title="Permalink to this headline">#</a></h4>
<p>Lastly, the modeling results are written to the EnzymeML documents and the files are exported.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">result</span><span class="p">,</span> <span class="n">enzmldoc</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">results</span><span class="p">,</span> <span class="n">concentration_enzymemldocs</span><span class="p">):</span>

    <span class="c1"># Write modeling results to kinetic parameters</span>
    <span class="n">k_cat</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;k_cat&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;k_cat&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;1 / </span><span class="si">{</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">time_unit</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

    <span class="n">K_m</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;K_m&quot;</span><span class="p">,</span>
        <span class="n">value</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;Km&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
        <span class="n">unit</span><span class="o">=</span><span class="n">result</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">data_conc_unit</span><span class="p">)</span>

    <span class="c1"># Define kinetic model</span>
    <span class="n">model</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">KineticModel</span><span class="p">(</span>
        <span class="n">name</span><span class="o">=</span><span class="s2">&quot;irreversible Michaelis-Menten&quot;</span><span class="p">,</span>
        <span class="n">equation</span><span class="o">=</span><span class="s2">&quot;-k_cat * p0 * s0 / (K_m + s0)&quot;</span><span class="p">,</span>
        <span class="n">parameters</span><span class="o">=</span><span class="p">[</span><span class="n">k_cat</span><span class="p">,</span> <span class="n">K_m</span><span class="p">])</span>

    <span class="n">enzmldoc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">model</span> <span class="o">=</span> <span class="n">model</span>

    <span class="n">add_enzyme_inactivation_model</span><span class="p">(</span>
        <span class="n">result</span><span class="o">=</span><span class="n">result</span><span class="p">,</span>
        <span class="n">enzmldoc</span><span class="o">=</span><span class="n">enzmldoc</span><span class="p">)</span>
    


    <span class="c1"># Export EnzymeML documents</span>
    <span class="n">export</span> <span class="o">=</span> <span class="kc">False</span>
    <span class="k">if</span> <span class="n">export</span><span class="p">:</span>
        <span class="n">enzmldoc</span><span class="o">.</span><span class="n">toFile</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<span id="document-scenarios/enzyme_inactivation"></span><section class="tex2jax_ignore mathjax_ignore" id="scenario-d-br-time-dependent-enzyme-inactivation">
<h3><strong>Scenario D:</strong><br>Time-dependent enzyme inactivation<a class="headerlink" href="#scenario-d-br-time-dependent-enzyme-inactivation" title="Permalink to this headline">#</a></h3>
<p>Data provided by Paulo Durão (Microbial &amp; Enzyme Technology, Instituto de Tecnologia Química e Biológica, Oeiras, Portugal)</p>
<section id="project-background">
<h4>Project background<a class="headerlink" href="#project-background" title="Permalink to this headline">#</a></h4>
<p>All investigated enzyme reactions without inhibitor applied to the reaction showed progress curve behavior, which were not explainable by irreversible Michaelis-Menten kinetics. All experiments had in common, that enzyme reactions were carried out in 96-well polystyrene micro titer plates (MTP), whereas the change in substrate or product absorption was monitored photometrically.<br />
One hypothesis for the suspected enzyme inactivation was hydrophobic interaction between the enzyme and the MTP surface. Thereby, hydrophobic regions of the enzyme’s surface might interact with the polystyrene reaction vessel, potentially preventing substrate access to the active site of the enzyme. Another hypothesis for the observed gradual activity decrease was inactivation which is linked to the catalytic activity, as observed in the previous scenario.</p>
<p>In order to test the hypothesis, an experiment was designed in order to detach enzyme inactivation from enzyme activity. Therefore, enzyme was incubated in MTP wells prior to reaction start for different periods. Thereby, the enzyme activity should decrease depending on prior incubation time. If the hypothesis is correct, the calculated half life from the adsorption experiment
should match the half life of an enzyme kinetics experiment which was conducted in parallel and also considered an inactivation term.</p>
<section id="experimental-design">
<h5>Experimental design<a class="headerlink" href="#experimental-design" title="Permalink to this headline">#</a></h5>
<p><strong>Determination of enzyme inactivation through adsorption</strong><br />
The inactivation experiment was conducted by incubating CotA laccase from <em>Bacillus subtilis</em> in individual MTP wells up to 1 h prior to reaction start. Then, enzyme reactions were started in 10 min intervals by transferring 2 µL of incubated enzyme in individual MTP wells. Each proceeding enzyme reaction contained 256 nM CotA, 1 mM ABTS and was buffered in acetate buffer at pH 4. Product formation was followed photometrically at 420 nm and 25°C for 5 min, whereas concentrations were calculated assuming an extinction coefficient of ε = 36000 M<sup>-1</sup>cm<sup>-1</sup> for the ABTS radical product. Each experiment was performed as technical triplicates.</p>
<p><strong>Enzyme kinetics experiment</strong><br />
Enzymatic oxidation of ABTS to its radical form was followed photometrically at 420 nm at 25°C for 70 min.
Thereby, ABTS was applied in a range from 0.01 mM - 2 mM. Each proceeding enzyme reaction contained 256 nM CotA, and was buffered in acetate buffer at pH 4. Each experimental condition was repeated as technical quadruplicates. Concentrations were calculated via the provided extinction coefficient of the ABTS radical (ε = 36000 M<sup>-1</sup>cm<sup>-1</sup>).</p>
</section>
<section id="experimental-data">
<h5>Experimental data<a class="headerlink" href="#experimental-data" title="Permalink to this headline">#</a></h5>
<p>Measurement data was provided as an Excel file, containing time-course absorption data. Meta data was written into the EnzymeML Excel template. Then, the experimental data was written to an EnzymeML document by a parser function.</p>
</section>
</section>
<section id="adsorption-of-enzyme-to-micro-titer-plate-surface">
<h4>Adsorption of enzyme to micro titer plate surface<a class="headerlink" href="#adsorption-of-enzyme-to-micro-titer-plate-surface" title="Permalink to this headline">#</a></h4>
<section id="imports">
<h5>Imports<a class="headerlink" href="#imports" title="Permalink to this headline">#</a></h5>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">scipy.stats</span> <span class="kn">import</span> <span class="n">linregress</span>
<span class="kn">import</span> <span class="nn">pyenzyme</span> <span class="k">as</span> <span class="nn">pe</span>
<span class="kn">from</span> <span class="nn">EnzymePynetics.tools.parameterestimator</span> <span class="kn">import</span> <span class="n">ParameterEstimator</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">matplotlib.cm</span> <span class="k">as</span> <span class="nn">cm</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span>
<span class="kn">import</span> <span class="nn">string</span>
<span class="kn">from</span> <span class="nn">parser_functions</span> <span class="kn">import</span> <span class="n">add_enzyme_inactivation_model</span>

<span class="kn">import</span> <span class="nn">warnings</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s1">&#39;ignore&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="calculation-of-inactivation-rate">
<h5>Calculation of inactivation rate<a class="headerlink" href="#calculation-of-inactivation-rate" title="Permalink to this headline">#</a></h5>
<p>Experimental data was loaded into a Pandas DataFrame and the slopes of each incubation condition were calculated through linear regression. The resulting initial rates were then used to calculate the rate by which the enzyme activity decreases. Lastly the half life of the enzyme was calculated with equation <a class="reference internal" href="welcome.html#equation-enzyme-halflife">(16)</a>.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load excel</span>
<span class="n">path</span> <span class="o">=</span> <span class="s1">&#39;../../data/enzyme_inactivation/Slide 2 - Activity effect of incubating CotA in MTP.xlsx&#39;</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="n">sheet_name</span><span class="o">=</span><span class="s1">&#39;csv&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;time (min)&#39;</span><span class="p">)</span>

<span class="c1"># replace values of &#39;0**&#39; with nan-values, since the measurement is incorrect</span>
<span class="n">df</span><span class="p">[</span><span class="s1">&#39;0**&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>

<span class="c1"># Get data from Excel file</span>
<span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="nb">int</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">list</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">)</span> <span class="k">if</span> <span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;0&quot;</span><span class="p">)]</span>
<span class="n">time</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">values</span>
<span class="n">absorption</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="mi">7</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">22</span><span class="p">)</span>

<span class="c1"># Calculate concentrations</span>
<span class="n">extinction_coefficient</span> <span class="o">=</span> <span class="mi">36</span> <span class="c1"># (1/mM * 1/cm)</span>
<span class="n">optical_length</span> <span class="o">=</span> <span class="mf">0.65</span> <span class="c1"># cm</span>

<span class="k">def</span> <span class="nf">absorption_to_concentration</span><span class="p">(</span><span class="n">abso</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">abso</span> <span class="o">/</span> <span class="p">(</span><span class="n">extinction_coefficient</span><span class="o">*</span><span class="n">optical_length</span><span class="p">)</span>

<span class="n">concentration</span> <span class="o">=</span> <span class="n">absorption_to_concentration</span><span class="p">(</span><span class="n">absorption</span><span class="p">)</span>
<span class="n">concentration_mean</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanmean</span><span class="p">(</span><span class="n">concentration</span><span class="p">,</span> <span class="n">axis</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
<span class="n">concentration_std</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nanstd</span><span class="p">(</span><span class="n">concentration</span><span class="p">,</span> <span class="n">axis</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>

<span class="c1"># Linear regression</span>
<span class="n">slopes</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">intercepts</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">stderrs</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">time_set</span> <span class="ow">in</span> <span class="n">concentration</span><span class="p">:</span>
    <span class="n">mask</span> <span class="o">=</span> <span class="o">~</span><span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">time_set</span><span class="p">)</span>
    <span class="n">time_regression</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">tile</span><span class="p">(</span><span class="n">time</span><span class="p">,</span> <span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="n">mask</span><span class="o">.</span><span class="n">shape</span><span class="p">)[</span><span class="n">mask</span><span class="p">]</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>
    <span class="n">data_regression</span> <span class="o">=</span> <span class="n">time_set</span><span class="p">[</span><span class="n">mask</span><span class="p">]</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>
    <span class="n">slope</span><span class="p">,</span><span class="n">intercept</span><span class="p">,</span><span class="n">_</span><span class="p">,</span><span class="n">_</span><span class="p">,</span><span class="n">stderr</span> <span class="o">=</span> <span class="p">(</span><span class="n">linregress</span><span class="p">(</span><span class="n">time_regression</span><span class="p">,</span> <span class="n">data_regression</span><span class="p">))</span>
    <span class="n">slopes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">slope</span><span class="p">)</span>
    <span class="n">intercepts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">intercept</span><span class="p">)</span>
    <span class="n">stderrs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">stderr</span><span class="p">)</span>

<span class="c1"># Regression of the slopes between 10 and 60 min</span>
<span class="n">slope_of_slopes</span><span class="p">,</span> <span class="n">intercept_of_slopes</span><span class="p">,</span><span class="n">_</span><span class="p">,</span><span class="n">_</span><span class="p">,</span><span class="n">_</span> <span class="o">=</span> <span class="n">linregress</span><span class="p">(</span><span class="n">columns</span><span class="p">[</span><span class="mi">1</span><span class="p">:],</span> <span class="n">slopes</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span>

<span class="c1"># Calculate time at which enzyme activity is reduced by half</span>
<span class="n">half_activity</span> <span class="o">=</span> <span class="p">((</span><span class="n">slopes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">/</span><span class="mi">2</span> <span class="o">-</span> <span class="n">slopes</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span> <span class="o">/</span> <span class="n">slope_of_slopes</span><span class="p">)</span>

<span class="c1"># Plot results</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Calculated time after enzyme activity is halved based on linear regression: </span><span class="si">{</span><span class="n">half_activity</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> min&quot;</span><span class="p">)</span>

<span class="n">colors</span> <span class="o">=</span> <span class="n">cm</span><span class="o">.</span><span class="n">tab10</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">linspace</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">absorption</span><span class="p">)))</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.8</span><span class="p">,</span><span class="mf">4.8</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="k">for</span> <span class="n">mean</span><span class="p">,</span> <span class="n">std</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">slope</span><span class="p">,</span> <span class="n">intercept</span><span class="p">,</span> <span class="n">color</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">concentration_mean</span><span class="p">,</span> <span class="n">concentration_std</span><span class="p">,</span> <span class="n">columns</span><span class="p">,</span> <span class="n">slopes</span><span class="p">,</span> <span class="n">intercepts</span><span class="p">,</span> <span class="n">colors</span><span class="p">):</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">errorbar</span><span class="p">(</span><span class="n">time</span><span class="p">,</span> <span class="n">mean</span><span class="p">,</span> <span class="n">std</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="n">label</span><span class="p">,</span> <span class="n">alpha</span> <span class="o">=</span> <span class="mf">0.4</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">time</span><span class="p">),</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">time</span><span class="p">)</span><span class="o">*</span><span class="n">slope</span><span class="o">+</span><span class="n">intercept</span><span class="p">,</span> <span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;CotA incubation</span><span class="se">\n</span><span class="s2">in MTP-well [min]&quot;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;ABTS radical [mM]&#39;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;time [min]&#39;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Experimental data with calculated reaction rate by linear regression&#39;</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
            <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>


<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">errorbar</span><span class="p">(</span><span class="n">columns</span><span class="p">,</span> <span class="n">slopes</span><span class="p">,</span> <span class="n">stderrs</span><span class="p">,</span> <span class="n">fmt</span><span class="o">=</span><span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s2">&quot;initial rates within the first 5 min&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;CotA incubation time</span><span class="se">\n</span><span class="s2">in MTP-well [min]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;initial rate [mM min$^{-1}$]&#39;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">columns</span><span class="p">[</span><span class="mi">1</span><span class="p">:]),</span> <span class="n">slope_of_slopes</span><span class="o">*</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">columns</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span><span class="o">+</span><span class="n">intercept_of_slopes</span><span class="p">,</span> <span class="s2">&quot;--&quot;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">colors</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">label</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;regression&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">()</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Initial rates of CotA reaction&#39;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">1.1</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
        <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Calculated time after enzyme activity is halved based on linear regression: 105.09 min
</pre></div>
</div>
<img alt="_images/enzyme_inactivation_3_1.png" src="_images/enzyme_inactivation_3_1.png" />
</div>
</div>
<p><em>Fig. 13: Experimental data and regression results of CotA reaction with different enzyme incubation times.</em></p>
<p>Enzyme reactions without prior incubation showed the fastest reaction rate, whereas all reactions with prior incubation showed an approximately halved reaction rate (Fig. 13A). Thereby, reactions with longer incubation time showed gradually reduced reaction rates with increased incubation time. Enzyme reactions, especially reactions without incubation, showed large deviations across replicates. This presumably resulted from inhomogeneous mixing or deviating enzyme concentration due to the small volume of enzyme solution, which was used to start the reactions.<br />
Since the true reaction rate of reactions without prior incubation was highly uncertain and showed mixing issued, the respective data was not used to estimate the enzyme’s time-dependent inactivation rate.
Based on linear regression of the initial reaction rates, excluding the reaction without prior incubation, the enzyme activity is approximately halved after 105 min (Fig. 13B). This assumes validity of extrapolation.</p>
</section>
</section>
<section id="enzyme-kinetics-experiment">
<h4>Enzyme kinetics experiment<a class="headerlink" href="#enzyme-kinetics-experiment" title="Permalink to this headline">#</a></h4>
<section id="id1">
<h5>Experimental data<a class="headerlink" href="#id1" title="Permalink to this headline">#</a></h5>
<p>Experimental data was provided as an Excel file, whereas meta data of the experiment was filled in to an EnzymeML Excel spreadsheet. Measurement data was written to the EnzymeML document by a parser function.
Kinetic parameters were estimated with and without considering enzyme inactivation.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load experimental data from excel file</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="s2">&quot;../../data/enzyme_inactivation/Repetition CotA ABTS kinetics higher volumes 2nd time.xlsx&quot;</span><span class="p">,</span> <span class="n">sheet_name</span><span class="o">=</span><span class="s2">&quot;csv&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;Time(min)&quot;</span><span class="p">)</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">reshape</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">72</span><span class="p">)</span>
<span class="n">time</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">values</span>

<span class="c1"># Calculate concentrations</span>
<span class="n">extinction_coefficient</span> <span class="o">=</span> <span class="mi">36</span> <span class="c1"># (1/mM * 1/cm)</span>
<span class="n">optical_length</span> <span class="o">=</span> <span class="mf">0.65</span> <span class="c1"># cm</span>

<span class="k">def</span> <span class="nf">absorption_to_concentration</span><span class="p">(</span><span class="n">abso</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">abso</span> <span class="o">/</span> <span class="p">(</span><span class="n">extinction_coefficient</span><span class="o">*</span><span class="n">optical_length</span><span class="p">)</span>

<span class="n">concentration_data</span> <span class="o">=</span> <span class="n">absorption_to_concentration</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>

<span class="c1"># Parser function</span>
<span class="k">def</span> <span class="nf">data_to_EnzymeML</span><span class="p">(</span>
    <span class="n">template_path</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">measurement_data</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">ndarray</span><span class="p">,</span>
    <span class="n">species_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">data_unit</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">time_unit</span><span class="p">:</span> <span class="nb">str</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="p">:</span>

    <span class="n">enzmldoc</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">EnzymeMLDocument</span><span class="o">.</span><span class="n">fromTemplate</span><span class="p">(</span><span class="n">template_path</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">IDs</span><span class="p">,</span> <span class="n">concentration</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="o">.</span><span class="n">keys</span><span class="p">(),</span> <span class="n">measurement_data</span><span class="p">):</span>
        <span class="k">for</span> <span class="n">counter</span><span class="p">,</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">concentration</span><span class="p">):</span>
            <span class="n">rep</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">Replicate</span><span class="p">(</span>
                <span class="nb">id</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;Measurement</span><span class="si">{</span><span class="n">counter</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">,</span>
                <span class="n">species_id</span><span class="o">=</span><span class="n">species_id</span><span class="p">,</span>
                <span class="n">data</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">replicate</span><span class="p">),</span>
                <span class="n">data_unit</span><span class="o">=</span><span class="n">data_unit</span><span class="p">,</span>
                <span class="n">time</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">time</span><span class="p">),</span>
                <span class="n">time_unit</span><span class="o">=</span><span class="n">time_unit</span><span class="p">)</span>
            <span class="n">enzmldoc</span><span class="o">.</span><span class="n">getMeasurement</span><span class="p">(</span><span class="n">IDs</span><span class="p">)</span><span class="o">.</span><span class="n">addReplicates</span><span class="p">(</span><span class="n">rep</span><span class="p">,</span> <span class="n">enzmldoc</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">enzmldoc</span>

<span class="c1"># Write experimantal data to EnzymeML document vis parser function</span>
<span class="n">enzmldoc</span> <span class="o">=</span> <span class="n">data_to_EnzymeML</span><span class="p">(</span>
    <span class="n">template_path</span><span class="o">=</span><span class="s2">&quot;../../data/enzyme_inactivation/EnzymeML_CotA.xlsm&quot;</span><span class="p">,</span>
    <span class="n">measurement_data</span><span class="o">=</span><span class="n">concentration_data</span><span class="p">,</span>
    <span class="n">species_id</span><span class="o">=</span><span class="s2">&quot;s1&quot;</span><span class="p">,</span>
    <span class="n">data_unit</span><span class="o">=</span><span class="s2">&quot;mmole / l&quot;</span><span class="p">,</span>
    <span class="n">time_unit</span><span class="o">=</span><span class="s2">&quot;min&quot;</span><span class="p">)</span>

<span class="c1"># Visualize experimantal data</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.8</span><span class="p">,</span> <span class="mf">4.8</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="k">for</span> <span class="n">measurement</span> <span class="ow">in</span> <span class="n">enzmldoc</span><span class="o">.</span><span class="n">measurement_dict</span><span class="o">.</span><span class="n">values</span><span class="p">():</span>
    <span class="n">concentration</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">product</span> <span class="o">=</span> <span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s1&quot;</span><span class="p">)</span>
    <span class="n">init_substrate</span><span class="o">=</span><span class="n">measurement</span><span class="o">.</span><span class="n">getReactant</span><span class="p">(</span><span class="s2">&quot;s0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">init_conc</span>
    <span class="k">for</span> <span class="n">replicate</span> <span class="ow">in</span> <span class="n">product</span><span class="o">.</span><span class="n">replicates</span><span class="p">:</span>
        <span class="n">concentration</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">replicate</span><span class="o">.</span><span class="n">data</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">errorbar</span><span class="p">(</span><span class="n">time</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">concentration</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">),</span> <span class="n">np</span><span class="o">.</span><span class="n">std</span><span class="p">(</span><span class="n">concentration</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">),</span> <span class="n">label</span><span class="o">=</span><span class="n">init_substrate</span><span class="p">,</span>\
        <span class="n">fmt</span><span class="o">=</span><span class="s2">&quot;.&quot;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.5</span><span class="p">)</span>
    <span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">errorbar</span><span class="p">(</span><span class="n">time</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">concentration</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">),</span> <span class="n">np</span><span class="o">.</span><span class="n">std</span><span class="p">(</span><span class="n">concentration</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">),</span> <span class="n">label</span><span class="o">=</span><span class="n">init_substrate</span><span class="p">,</span>\
        <span class="n">fmt</span><span class="o">=</span><span class="s2">&quot;o&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Initial ABTS [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;Initial ABTS [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS radical [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;ABTS radical concentration over CotA reaction time-course &quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;ABTS radical concentration within the first 5 min&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS radical [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [min]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [min]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">([</span><span class="o">-</span><span class="mf">0.2</span><span class="p">,</span><span class="mf">5.2</span><span class="p">])</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">([</span><span class="mi">0</span><span class="p">,</span><span class="mf">0.01</span><span class="p">])</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
        <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">1.1</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
        <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/enzyme_inactivation_5_0.png" src="_images/enzyme_inactivation_5_0.png" />
</div>
</div>
<p><em>Fig 14. ABTS radical concentration over the time-course of CotA reaction.</em></p>
<p>The provided reaction data showed large deviations between individual repeats of reactions with identical initial ABTS concentrations. Especially for initial ABTS concentrations above 0.1 mM (Fig. 14A).
Furthermore, visual analysis of the initial measurement points indicated that reactions with an initial substrate concentration of 0.1 mM had an increased enzyme concentration compared to others. This is evident, because the mentioned reactions show the highest concentration at the initial measurement time (Fig. 14B). The order of initial measurement points is expected to reflect the initial substrate concentrations, since the substrate slightly absorbs at the product detection wavelength. The substrate of the mentioned reaction was likely applied correctly, since the final concentration at 70 min is between reactions with the next lower and next higher initial substrate concentration.</p>
</section>
</section>
<section id="kinetic-parameter-estimation">
<h4>Kinetic parameter estimation<a class="headerlink" href="#kinetic-parameter-estimation" title="Permalink to this headline">#</a></h4>
<section id="quality-control-through-modeling">
<h5>Quality control through modeling<a class="headerlink" href="#quality-control-through-modeling" title="Permalink to this headline">#</a></h5>
<p>An initial parameter estimation considering enzyme inactivation was conducted in order to check further systematic deviations. As a result, reactions with an initial ABTS concentration of 0.05 mM showed to deviate from all other reactions, since the respective product concentrations is lower than the model predicted. The mentioned measurements were the only ones where the measured concentration was distinctly and systematically below that of the model (Fig. 15A). Hence indicating that either less substrate or enzyme than defined in the protocol was applied to the respective reactions. In consequence, data from reactions with an initial substrate concentration of 0.05 and 0.1 mM were excluded from parameter estimation, because the identified systematic deviations in reaction conditions would distort the results.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Parameter estimation considering time-dependent enzyme inactivation</span>
<span class="n">CotA_kinetics_first_run</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="s2">&quot;product&quot;</span><span class="p">)</span>
<span class="n">CotA_kinetics_first_run</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">enzyme_inactivation</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">initial_substrate_concs</span><span class="o">=</span><span class="p">[</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">0.02</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.2</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">,</span> <span class="mf">2.0</span><span class="p">])</span>

<span class="c1"># Parameter estimation considering time-dependent enzyme inactivation</span>
<span class="n">CotA_kinetics_with_inactivation</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="s2">&quot;product&quot;</span><span class="p">)</span>
<span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">enzyme_inactivation</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">initial_substrate_concs</span><span class="o">=</span><span class="p">[</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">0.02</span><span class="p">,</span> <span class="mf">0.2</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">,</span> <span class="mf">2.0</span><span class="p">])</span>
<span class="n">df_inactivation</span> <span class="o">=</span> <span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">result_dict</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;kcat / Km [1/min * 1/mmole / l]&quot;</span><span class="p">])</span>
<span class="n">df_inactivation</span><span class="o">.</span><span class="n">insert</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="s2">&quot;Enzyme inactivation model&quot;</span><span class="p">,</span> <span class="s2">&quot;True&quot;</span><span class="p">)</span>

<span class="c1"># Visualize fit of the models</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.8</span><span class="p">,</span><span class="mf">4.8</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">CotA_kinetics_first_run</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="s2">&quot;irreversible Michaelis Menten&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">alpha</span> <span class="o">=</span><span class="mf">.2</span><span class="p">,</span>\
    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Parameter estimation based on full dataset&quot;</span><span class="p">)</span>
<span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="s2">&quot;irreversible Michaelis Menten&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">alpha</span> <span class="o">=</span><span class="mf">.2</span><span class="p">,</span>\
    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;Parameter estimation based on data subset&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS radical [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [min]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [min]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial ABTS [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial ABTS [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
        <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">1.1</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> 
        <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/enzyme_inactivation_7_0.png" src="_images/enzyme_inactivation_7_0.png" />
</div>
</div>
<p><em>Fig 15: Identification of systematic deviations through kinetic modeling.</em></p>
<p>By excluding the mentioned measurements with systematic issues, the model fit was improved (Fig 15B). Additionally, the correlation between parameter estimates for <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> was reduced from <em>r<sup>2</sup></em> = 0.518 to 0.204.</p>
</section>
<section id="parameter-estimation-with-and-without-considering-enzyme-inactivation">
<h5>Parameter estimation with and without considering enzyme inactivation<a class="headerlink" href="#parameter-estimation-with-and-without-considering-enzyme-inactivation" title="Permalink to this headline">#</a></h5>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Parameter estimation without time-dependent enzyme inactivation</span>
<span class="n">CotA_kinetics</span> <span class="o">=</span> <span class="n">ParameterEstimator</span><span class="o">.</span><span class="n">from_EnzymeML</span><span class="p">(</span><span class="n">enzmldoc</span><span class="p">,</span> <span class="s2">&quot;s1&quot;</span><span class="p">,</span> <span class="s2">&quot;product&quot;</span><span class="p">)</span>
<span class="n">CotA_kinetics</span><span class="o">.</span><span class="n">fit_models</span><span class="p">(</span><span class="n">enzyme_inactivation</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">display_output</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">initial_substrate_concs</span><span class="o">=</span><span class="p">[</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">0.02</span><span class="p">,</span> <span class="mf">0.2</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">,</span> <span class="mf">2.0</span><span class="p">])</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">CotA_kinetics</span><span class="o">.</span><span class="n">result_dict</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;kcat / Km [1/min * 1/mmole / l]&quot;</span><span class="p">])</span>
<span class="n">df</span><span class="o">.</span><span class="n">insert</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="s2">&quot;Enzyme inactivation model&quot;</span><span class="p">,</span> <span class="s2">&quot;False&quot;</span><span class="p">)</span>

<span class="n">results</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df_inactivation</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;AIC&quot;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">results</span><span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">set_table_attributes</span><span class="p">(</span><span class="s1">&#39;style=&quot;font-size: 12px&quot;&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style type="text/css">
</style>
<table id="T_43830" style="font-size: 12px">
  <thead>
    <tr>
      <th class="blank level0" >&nbsp;</th>
      <th id="T_43830_level0_col0" class="col_heading level0 col0" >AIC</th>
      <th id="T_43830_level0_col1" class="col_heading level0 col1" >Enzyme inactivation model</th>
      <th id="T_43830_level0_col2" class="col_heading level0 col2" >kcat [1/min]</th>
      <th id="T_43830_level0_col3" class="col_heading level0 col3" >Km [mmole / l]</th>
      <th id="T_43830_level0_col4" class="col_heading level0 col4" >Ki competitive [mmole / l]</th>
      <th id="T_43830_level0_col5" class="col_heading level0 col5" >Ki uncompetitive [mmole / l]</th>
      <th id="T_43830_level0_col6" class="col_heading level0 col6" >ki time-dep enzyme-inactiv. [1/min]</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th id="T_43830_level0_row0" class="row_heading level0 row0" >irreversible Michaelis Menten</th>
      <td id="T_43830_row0_col0" class="data row0 col0" >-22202</td>
      <td id="T_43830_row0_col1" class="data row0 col1" >True</td>
      <td id="T_43830_row0_col2" class="data row0 col2" >3.003 +/- 0.77%</td>
      <td id="T_43830_row0_col3" class="data row0 col3" >0.030 +/- 1.99%</td>
      <td id="T_43830_row0_col4" class="data row0 col4" >-</td>
      <td id="T_43830_row0_col5" class="data row0 col5" >-</td>
      <td id="T_43830_row0_col6" class="data row0 col6" >0.007 +/- 4.48%</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row1" class="row_heading level0 row1" >competitive product inhibition</th>
      <td id="T_43830_row1_col0" class="data row1 col0" >-22202</td>
      <td id="T_43830_row1_col1" class="data row1 col1" >True</td>
      <td id="T_43830_row1_col2" class="data row1 col2" >2.999 +/- 0.78%</td>
      <td id="T_43830_row1_col3" class="data row1 col3" >0.029 +/- 3.85%</td>
      <td id="T_43830_row1_col4" class="data row1 col4" >0.221 +/- 92.18%</td>
      <td id="T_43830_row1_col5" class="data row1 col5" >-</td>
      <td id="T_43830_row1_col6" class="data row1 col6" >0.007 +/- 5.06%</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row2" class="row_heading level0 row2" >substrate inhibition</th>
      <td id="T_43830_row2_col0" class="data row2 col0" >-22202</td>
      <td id="T_43830_row2_col1" class="data row2 col1" >True</td>
      <td id="T_43830_row2_col2" class="data row2 col2" >3.021 +/- 0.86%</td>
      <td id="T_43830_row2_col3" class="data row2 col3" >0.031 +/- 2.38%</td>
      <td id="T_43830_row2_col4" class="data row2 col4" >-</td>
      <td id="T_43830_row2_col5" class="data row2 col5" >208.577 +/- 67.12%</td>
      <td id="T_43830_row2_col6" class="data row2 col6" >0.007 +/- 4.48%</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row3" class="row_heading level0 row3" >uncompetitive product inhibition</th>
      <td id="T_43830_row3_col0" class="data row3 col0" >-22199</td>
      <td id="T_43830_row3_col1" class="data row3 col1" >True</td>
      <td id="T_43830_row3_col2" class="data row3 col2" >3.012 +/- 1.23%</td>
      <td id="T_43830_row3_col3" class="data row3 col3" >0.030 +/- 4.72%</td>
      <td id="T_43830_row3_col4" class="data row3 col4" >-</td>
      <td id="T_43830_row3_col5" class="data row3 col5" >2.096 +/- 361.49%</td>
      <td id="T_43830_row3_col6" class="data row3 col6" >0.006 +/- 15.30%</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row4" class="row_heading level0 row4" >non-competitive product inhibition</th>
      <td id="T_43830_row4_col0" class="data row4 col0" >-22198</td>
      <td id="T_43830_row4_col1" class="data row4 col1" >True</td>
      <td id="T_43830_row4_col2" class="data row4 col2" >3.012 +/- 2.44%</td>
      <td id="T_43830_row4_col3" class="data row4 col3" >0.029 +/- 6.56%</td>
      <td id="T_43830_row4_col4" class="data row4 col4" >0.167 +/- 132.63%</td>
      <td id="T_43830_row4_col5" class="data row4 col5" >1.113 +/- 466.95%</td>
      <td id="T_43830_row4_col6" class="data row4 col6" >0.006 +/- 40.69%</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row5" class="row_heading level0 row5" >non-competitive product inhibition</th>
      <td id="T_43830_row5_col0" class="data row5 col0" >-22185</td>
      <td id="T_43830_row5_col1" class="data row5 col1" >False</td>
      <td id="T_43830_row5_col2" class="data row5 col2" >3.241 +/- 1.40%</td>
      <td id="T_43830_row5_col3" class="data row5 col3" >0.033 +/- 5.05%</td>
      <td id="T_43830_row5_col4" class="data row5 col4" >0.038 +/- 22.41%</td>
      <td id="T_43830_row5_col5" class="data row5 col5" >0.073 +/- 6.86%</td>
      <td id="T_43830_row5_col6" class="data row5 col6" >nan</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row6" class="row_heading level0 row6" >uncompetitive product inhibition</th>
      <td id="T_43830_row6_col0" class="data row6 col0" >-22156</td>
      <td id="T_43830_row6_col1" class="data row6 col1" >False</td>
      <td id="T_43830_row6_col2" class="data row6 col2" >3.268 +/- 1.43%</td>
      <td id="T_43830_row6_col3" class="data row6 col3" >0.041 +/- 2.70%</td>
      <td id="T_43830_row6_col4" class="data row6 col4" >-</td>
      <td id="T_43830_row6_col5" class="data row6 col5" >0.066 +/- 6.35%</td>
      <td id="T_43830_row6_col6" class="data row6 col6" >nan</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row7" class="row_heading level0 row7" >competitive product inhibition</th>
      <td id="T_43830_row7_col0" class="data row7 col0" >-21840</td>
      <td id="T_43830_row7_col1" class="data row7 col1" >False</td>
      <td id="T_43830_row7_col2" class="data row7 col2" >2.586 +/- 0.35%</td>
      <td id="T_43830_row7_col3" class="data row7 col3" >0.020 +/- 6.10%</td>
      <td id="T_43830_row7_col4" class="data row7 col4" >0.016 +/- 15.88%</td>
      <td id="T_43830_row7_col5" class="data row7 col5" >-</td>
      <td id="T_43830_row7_col6" class="data row7 col6" >nan</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row8" class="row_heading level0 row8" >irreversible Michaelis Menten</th>
      <td id="T_43830_row8_col0" class="data row8 col0" >-21739</td>
      <td id="T_43830_row8_col1" class="data row8 col1" >False</td>
      <td id="T_43830_row8_col2" class="data row8 col2" >2.529 +/- 0.27%</td>
      <td id="T_43830_row8_col3" class="data row8 col3" >0.030 +/- 2.30%</td>
      <td id="T_43830_row8_col4" class="data row8 col4" >-</td>
      <td id="T_43830_row8_col5" class="data row8 col5" >-</td>
      <td id="T_43830_row8_col6" class="data row8 col6" >nan</td>
    </tr>
    <tr>
      <th id="T_43830_level0_row9" class="row_heading level0 row9" >substrate inhibition</th>
      <td id="T_43830_row9_col0" class="data row9 col0" >-21737</td>
      <td id="T_43830_row9_col1" class="data row9 col1" >False</td>
      <td id="T_43830_row9_col2" class="data row9 col2" >2.537 +/- 0.52%</td>
      <td id="T_43830_row9_col3" class="data row9 col3" >0.030 +/- 2.71%</td>
      <td id="T_43830_row9_col4" class="data row9 col4" >-</td>
      <td id="T_43830_row9_col5" class="data row9 col5" >352.069 +/- 134.63%</td>
      <td id="T_43830_row9_col6" class="data row9 col6" >nan</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Kinetic parameters were estimated with and without considering time-dependent enzyme inactivation.
Modeling results are listed in the table above.
All models with an additional parameter for enzyme inactivation resulted in lower AIC values, indicating a better fit of the experimental data to inactivation models. When comparing models, an additional parameter in a model is informative, if the difference in AIC between the two models is larger than 2 . Accordingly two additional parameters in a model compared to another model are informative, if the difference in AIC is larger than 4 <span id="id2">[<a class="reference internal" href="welcome.html#id16" title="Todd W Arnold. Uninformative parameters and model selection using akaike's information criterion. The Journal of Wildlife Management, 74(6):1175–1178, 2010.">Arnold, 2010</a>]</span>. Thus, the additional parameters in inhibition models considering enzyme activation were uninformative and did not enhance the fit quality. Additionally, the mentioned models showed high standard deviations above 80% for the respective <span class="math notranslate nohighlight">\(K_{i}\)</span> estimates.
Without considering enzyme inactivation, product inhibition models were favoured over the irreversible Michaelis-Menten model,
whereas irreversible Michaelis-Menten with enzyme inactivation showed the best fit statistic compared to all other models (Fig 16A). Thereby, <span class="math notranslate nohighlight">\(k_{cat}\)</span> was estimated to 3.003 min<sup>-1</sup> ± 0.77% whereas 30 µM ± 1.99%  was estimated for <span class="math notranslate nohighlight">\(K_{m}\)</span>. The respective half life of CotA was estimated at 95 min ± 4 min using equation <a class="reference internal" href="welcome.html#equation-enzyme-halflife">(16)</a>, whereas <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(k_{inact}\)</span> were highly correlated (corr &gt; 0.95). Irreversible Michelis-Menten model without enzyme concentration did not fit the measurement data, since the reaction rate is initially underestimated, and overestimated in later stages of the reaction (Fig 16B).</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">12.8</span><span class="p">,</span><span class="mf">9.2</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="s2">&quot;irreversible Michaelis Menten&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span> <span class="n">alpha</span> <span class="o">=</span><span class="mf">.2</span><span class="p">,</span>\
    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;irrev. Michaelis-Menten model with time-dependent enzyme inactivation&quot;</span><span class="p">)</span>
<span class="n">CotA_kinetics</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="s2">&quot;irreversible Michaelis Menten&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">1</span><span class="p">],</span> <span class="n">alpha</span> <span class="o">=</span><span class="mf">.2</span><span class="p">,</span>\
    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;irrev. Michaelis-Menten model&quot;</span><span class="p">)</span>
<span class="n">CotA_kinetics</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="s2">&quot;non-competitive product inhibition&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span> <span class="n">alpha</span> <span class="o">=</span><span class="mf">.2</span><span class="p">,</span>\
    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;non-competitive product inhibition model&quot;</span><span class="p">)</span>
<span class="n">CotA_kinetics</span><span class="o">.</span><span class="n">visualize</span><span class="p">(</span><span class="s2">&quot;uncompetitive product inhibition&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">1</span><span class="p">],</span> <span class="n">alpha</span> <span class="o">=</span><span class="mf">.2</span><span class="p">,</span>\
    <span class="n">title</span><span class="o">=</span><span class="s2">&quot;uncompetitive product inhibition model&quot;</span><span class="p">)</span>

<span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS radical [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;ABTS radical [mM]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [min]&quot;</span><span class="p">)</span>
<span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;time [min]&quot;</span><span class="p">)</span>
<span class="p">[</span><span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">title</span><span class="o">=</span><span class="s2">&quot;initial ABTS [mM]&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">ax</span> <span class="ow">in</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()]</span>

<span class="p">[</span><span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="n">string</span><span class="o">.</span><span class="n">ascii_uppercase</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">weight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">ax</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">())]</span>

<span class="c1"># axes[0]</span>
<span class="c1"># axes[1].text(1.1, 1.1, string.ascii_uppercase[1], transform=axes[0].transAxes, </span>
<span class="c1">#         size=20, weight=&#39;bold&#39;)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/enzyme_inactivation_11_0.png" src="_images/enzyme_inactivation_11_0.png" />
</div>
</div>
<p><em>Fig. 16: Different kinetic models fitted to CotA reaction data</em></p>
<p>Out of the models without enzyme inactivation, product inhibition models showed the lowest AIC. Qualitatively, the inhibition models  described the progress curve of the reaction (Fig. 16C and 16D) similarly to the best fitting model (Fig. 16A)</p>
</section>
</section>
<section id="project-discussion">
<h4>Project discussion<a class="headerlink" href="#project-discussion" title="Permalink to this headline">#</a></h4>
<section id="data-quality">
<h5>Data quality<a class="headerlink" href="#data-quality" title="Permalink to this headline">#</a></h5>
<p>Both datasets, for the absorption experiment and the enzyme kinetics experiment, showed quality issues. In the adsorption experiment the initial measurement showed large uncertainties as well as mixing issues, indicated by an apparent fluctuation in reaction rate. In the enzyme kinetics experiment, individual reactions were shown to have substrate and enzyme concentrations that deviated from the assay protocol. Furthermore, the measured product concentration showed large deviations among experimental repeats, indicating unequal enzyme concentrations. Hence, larger pipetting volume for the enzyme solution was suggested to reduce pipetting errors due to avoidable small pipetting volumes.
Furthermore, the product partners were advised to repeat the experiments to reduce technical issues.</p>
</section>
<section id="enzyme-inactivation">
<h5>Enzyme inactivation<a class="headerlink" href="#enzyme-inactivation" title="Permalink to this headline">#</a></h5>
<p>The observed time-dependent decrease in enzyme activity can be explained by enzyme inactivation and product inhibition. In the case of time-dependent enzyme inactivation, the catalytic activity decreases, since the catalyst becomes inactive. Thereby, the decrease in relative reaction rate is independent from product concentration.
In the case of product inhibition, the formed product reduces the reaction rate. Hence, the observed enzyme inactivation is dependent on product concentration.
Since both phenomena express themselves in an apparent decrease of reaction rate, they bare the potential to be confused, as the results of the enzyme kinetics experiment show. Visually, product inhibition models without enzyme inactivation and irreversible Michaelis-Menten model with enzyme inactivation are able to qualitatively describe the observed progress curve (Fig 16C, 16D).</p>
<p>Both independent and methodologically different experiments resulted in a comparable time period, in which the enzyme’s activity is halved. Whereas in the inactivation experiment the enzyme’s activity was halved after approximately 105 min, half the activity was left in the enzyme kinetics experiment after 95 min. Due to the design of the incubation experiment, the loss of enzyme activity showed to be independent from the catalysis itself. In consequence, product inhibition is not a conclusive explanation for the observed behavior. Product inhibition is further disfavored based on the modeling results, since the AIC of all inhibition models was higher compared the irreversible Michaelis-Menten model with time-dependent enzyme inactivation.</p>
<p>In conclusion, the observed inactivation in likely caused by hydrophobic interaction between the reaction vessel and the enzyme. Interactions between hydrophobic surfaces and enzymes leading to decreased enzymatic activity were already reported for ribonuclease A <span id="id3">[<a class="reference internal" href="welcome.html#id29" title="Yang Wei, Aby A Thyparambil, Yonnie Wu, and Robert A Latour. Adsorption-induced changes in ribonuclease a structure and enzymatic activity on solid surfaces. Langmuir, 30(49):14849–14858, 2014.">Wei <em>et al.</em>, 2014</a>]</span>. In this work, enzyme inactivation was observed across multiple enzyme classes, where all reactions were carried out in MTPs. Therefore, MTPs might cause enzyme inactivation for a broad range of different enzymes.</p>
<p>Misinterpretation of the apparent decrease in enzymatic activity might lead to wrong conclusions and therefore negatively impact the research process. As already mentioned the activity decrease might be wrongfully attributed to product inhibition. This attribution might lead to wrong design decisions if the enzyme in question should be applied to an enzymatic cascade reaction or make the enzyme entirely unfavorable for this particular application.<br />
Furthermore, screening applications might be impacted by enzyme inactivation, if yields of an enzymatic reactions should be determined over the time course of multiple hours. All determined yield might be estimated too low.
Therefore, enzyme inactivation in MTPs should be investigated in future experiments and the results should be reported on.</p>
</section>
</section>
<section id="save-modeling-results">
<h4>Save modeling results<a class="headerlink" href="#save-modeling-results" title="Permalink to this headline">#</a></h4>
<p>Lastly, the modeling results were written to the EnzymeML documents and the files were exported.</p>
<div class="cell tag_hide_input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Write modeling results to kinetic parameters</span>
<span class="n">k_cat</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
    <span class="n">name</span><span class="o">=</span><span class="s2">&quot;k_cat&quot;</span><span class="p">,</span>
    <span class="n">value</span><span class="o">=</span><span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;k_cat&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
    <span class="n">unit</span><span class="o">=</span><span class="sa">f</span><span class="s2">&quot;1 / </span><span class="si">{</span><span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">time_unit</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

<span class="n">K_m</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">enzymeml</span><span class="o">.</span><span class="n">models</span><span class="o">.</span><span class="n">KineticParameter</span><span class="p">(</span>
    <span class="n">name</span><span class="o">=</span><span class="s2">&quot;K_m&quot;</span><span class="p">,</span>
    <span class="n">value</span><span class="o">=</span><span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">get_model_results</span><span class="p">()[</span><span class="s2">&quot;Km&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
    <span class="n">unit</span><span class="o">=</span><span class="n">CotA_kinetics_with_inactivation</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">data_conc_unit</span><span class="p">)</span>

<span class="c1"># Define kinetic model</span>
<span class="n">model</span> <span class="o">=</span> <span class="n">pe</span><span class="o">.</span><span class="n">KineticModel</span><span class="p">(</span>
    <span class="n">name</span><span class="o">=</span><span class="s2">&quot;irreversible Michaelis-Menten&quot;</span><span class="p">,</span>
    <span class="n">equation</span><span class="o">=</span><span class="s2">&quot;-k_cat * p0 * s0 / (K_m + s0)&quot;</span><span class="p">,</span>
    <span class="n">parameters</span><span class="o">=</span><span class="p">[</span><span class="n">k_cat</span><span class="p">,</span> <span class="n">K_m</span><span class="p">])</span>

<span class="n">enzmldoc</span><span class="o">.</span><span class="n">getReaction</span><span class="p">(</span><span class="s2">&quot;r0&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">model</span> <span class="o">=</span> <span class="n">model</span>

<span class="n">add_enzyme_inactivation_model</span><span class="p">(</span>
    <span class="n">result</span><span class="o">=</span><span class="n">CotA_kinetics_with_inactivation</span><span class="p">,</span>
    <span class="n">enzmldoc</span><span class="o">=</span><span class="n">enzmldoc</span><span class="p">)</span>

<span class="c1"># Export EnzymeML documents</span>
<span class="n">export</span> <span class="o">=</span> <span class="kc">False</span>
<span class="k">if</span> <span class="n">export</span><span class="p">:</span>
    <span class="n">enzmldoc</span><span class="o">.</span><span class="n">toFile</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
</div>
</section>
<span id="document-discussion"></span><section class="tex2jax_ignore mathjax_ignore" id="discussion">
<h2>Discussion<a class="headerlink" href="#discussion" title="Permalink to this headline">#</a></h2>
<p>Ensuring data integrity throughout an analysis workflow is important to yield trustworthy results. Hence, the analysis workflow was conceptualized starting from photometric measurement data in order to control all data processing steps. For that purpose, discussing the output format of the analytical device with the experimental partners proved to be valuable to plan the data acquisition process. Thus, a project specific parser function was established, which transferred the measurement data to the respective EnzymeML containing information on the measurement conditions. Thereby, error-prone and tedious manual copying between files was avoided, ensuring raw data integrity from the very start of the workflow.</p>
<section id="data-integrity-of-the-analysis-workflow">
<h3>Data integrity of the analysis workflow<a class="headerlink" href="#data-integrity-of-the-analysis-workflow" title="Permalink to this headline">#</a></h3>
<p>Another aspect of data integrity is accurate concentration calculation based on the measured absorption signal. The relation between concentration to absorption signal of an analyte can be linear as well as non-linear. Therefore, wrong assumption of a linear relationship, albeit the relation is better described non-linearly, leads to avoidable inaccuracy in concentration calculation <span id="id1">[<a class="reference internal" href="welcome.html#id14" title="Kang-Hung Hsu and Chiachung Chen. The effect of calibration equations on the uncertainty of uv–vis spectrophotometric measurements. Measurement, 43(10):1525–1531, 2010.">Hsu and Chen, 2010</a>]</span> <span id="id2">[<a class="reference internal" href="welcome.html#id15" title="Julia Martin, Alberto Romero Gracia, Agustín G Asuero, and others. Fitting nonlinear calibration curves: no models perfect. Journal of Analytical Sciences, Methods and Instrumentation, 7(01):1, 2017.">Martin <em>et al.</em>, 2017</a>]</span> and thus imprecise kinetic parameter estimates.
Hence, concentration calculation based on extinction coefficients was discouraged, if the underlying relation between absorption and concentration of an analyte was not self-determined for the respective reaction system.
To enable precise concentration calculation of linear and non-linear absorption to concentration relationships, CaliPytion was applied with analyte standards provided by the experimental partners.
All of the analyzed standard curves, except for scenario C ABTS at pH 5, were best described by non-linear calibration equations based on their fit quality determined by AIC. Hence, uncertainties from concentration calculation were reduced and thus their impact on the kinetic parameters.</p>
<p>Besides data integrity, high quality data is required for accurate parameter estimates, since unnoticed experimental errors unintentionally distort the resulting kinetic parameters. Therefore, quality control of measurement data is an important aspect of every data analysis process. EnzymePynetics allows visualization of measurement data together with the fitted kinetic model. Ultimately, systematic deviations of single measurements were identified by simultaneously visualizing measurement data and the fitted model.
By this method, errors with pipetting, mixing, and temperature equilibration during experimental preparation as well as issues with the experimental design itself were identified in different scenarios.<br />
Systematic pipetting errors were disclosed by systematic deviations between the measurement data and the fitted model.
In scenario D individual enzyme reactions showed to have substrate concentrations which deviated from the assay protocol. Parameter estimation without the presumably erroneous measurements yielded a better fit between data and model based on lower standard deviation for the kinetic parameters. Hence the experimental partners were advised to repeat the deviating measurement.<br />
In two projects the progress-curve of the reactions indicate a lag phase after reaction start. In one case, enzyme reactions were started by adding enzyme with low volume. In consequence, inhomogeneous mixing led to increasing reaction rates until the enzyme was distributed evenly. As a measure, the project partners were advised to increase the pipetting volume of the enzyme solution.
In another case, temperature incubation effects in MTP caused by prolonged assay preparation time resulted in an initial lag phase. Due to small reaction volumes and low mass, MTPs have a low heat capacity and thus high susceptibility to temperature change. Hence, project partners were advised to pre-incubate the MTP within the photometer at reaction temperature for 5 min and then start the reaction by adding enzyme as quick as possible.
In another project, modeling results of the estimated <span class="math notranslate nohighlight">\(k_{cat}\)</span> and <span class="math notranslate nohighlight">\(K_{m}\)</span> were used to iteratively improve the design of the enzyme assay. In addition, the correlation between the aforementioned parameters was used as a measure to ensure that the highest substrate concentration applied was sufficiently high and thus the kinetic parameters could be determined independently.
Thereby, the appropriate enzyme concentration and substrate concentration range were identified through multiple round of lab experiments with subsequent kinetic modeling.</p>
<p>Assessing the data quality through modeling and enhancing assay designs proved to be a strength of the implemented progress curve method on which the parameter estimation of this workflow is based. In contrast to the predominantly applied initial rates method <span id="id3">[<a class="reference internal" href="welcome.html#id25" title="Qingxiu Tang and Thomas S Leyh. Precise, facile initial rate measurements. The Journal of Physical Chemistry B, 114(49):16131–16136, 2010.">Tang and Leyh, 2010</a>]</span>, the progress curve method offers intrinsic advantages with regard to methods reproducibility.
In initial rate kinetics no consensus on the linear reaction period on which the kinetic parameters are estimated exists. Hence, the linear period is manually determined or only assumed. Ultimately, the resulting parameters are influenced by subjective choice of the linear period. For progress curve analysis the entire dataset of a continuous enzyme assay is used. If the fit statistics reveal that the data is not in accordance with the model, either the assay should be repeated due to an underlying issue or the model should be questioned. Contrarily, initial rates method can mostly be applied by arbitrarily choosing any linear subset or only the initial two data points of a time course data set. In discussions with project partners this showed to be a common practice although, scientifically at least, questionable.</p>
<p>The choice of the modeling method is just one decision which impacts kinetic parameters in the long chain of actions from enzyme expression all the way to data modeling. Each individual treatment step needs to be documented and later reported in order to make the results of the experiment reproducible. The STRENDA guidelines offer a valuable orientation on the minimum reporting standards in order to make an enzyme kinetics experiment reproducible. However, the guidelines are mainly focused on laboratory aspects and do not specify reporting standards for data treatment steps. Hence methodological reproducibility is not given by solely adhering to STRENDA guidelines, if kinetic parameters should be reproduced. Therefore, the method for concentration calculation, as well as the method for parameter estimation should be added to the STRENDA guidelines. As a result, the importance of data analytic aspects for methodologically reproducible kinetic parameters would receive more attention.</p>
</section>
<section id="repeatability-reproducibility-and-fairness-of-the-analysis-workflow">
<h3>Repeatability, reproducibility and FAIRness of the analysis workflow<a class="headerlink" href="#repeatability-reproducibility-and-fairness-of-the-analysis-workflow" title="Permalink to this headline">#</a></h3>
<p>Availability of raw data is the prerequisite to repeat a data analysis process, which additionally reinforces trust in scientific findings and the process itself. The lack of raw data is therefore also described as one of the reasons for the reproducibility crisis, since reported findings cannot be verified <span id="id4">[<a class="reference internal" href="welcome.html#id21" title="Tsuyoshi Miyakawa. No raw data, no science: another possible source of the reproducibility crisis. 2020.">Miyakawa, 2020</a>]</span>.
A second necessity for reanalysis is documentation to provide information on data treatment steps that led to a particular result. Furthermore, detailed documentation makes the analysis comprehensive for others.<br />
The developed parameter estimation workflow unifies these two requirements. On the one hand, the workflow is based on raw data. On the other hand, the Jupyter Notebook facilitates the analysis, whilst containing documentation on all analysis steps. Thus the workflow is repeatable and reproducible on a methods and results level. Moreover, by unifying data with functional code the implementation of the analysis workflow satisfies the highest reproducibility standards in computer science <span id="id5">[<a class="reference internal" href="welcome.html#id22" title="Roger D Peng. Reproducible research in computational science. Science, 334(6060):1226–1227, 2011.">Peng, 2011</a>]</span>.
In conclusion, the Jupyter Notebook with the implemented workflow can be understood as a graph, describing how experimental data is transform to yield kinetic parameters. Thereby, the measurement data and modeling results are stored within an EnzymeML document and the information of the modeling process in the notebook file.</p>
<p>Starting from data, FAIR guiding principles were implemented on multiple levels of this thesis. On data level, all experimental data as well as the modeling results were stored as EnzymeML files, which are compliant with FAIR data principles <span id="id6">[<a class="reference internal" href="welcome.html#id4" title="Jürgen Pleiss. Standardized data, scalable documentation, sustainable storage–enzymeml as a basis for fair data management in biocatalysis. ChemCatChem, 13(18):3909–3913, 2021.">Pleiss, 2021</a>]</span> <span id="id7">[<a class="reference internal" href="welcome.html#id18" title="Jan Range, Colin Halupczok, Jens Lohmann, Neil Swainston, Carsten Kettner, Frank T Bergmann, Andreas Weidemann, Ulrike Wittig, Santiago Schnell, and Jürgen Pleiss. Enzymeml—a data exchange format for biocatalysis and enzymology. The FEBS Journal, 289(19):5864–5874, 2022.">Range <em>et al.</em>, 2022</a>]</span>. On a methods level, the workflow and all of its components were designed in FAIR fashion.<br />
The Python packages are findable and accessible on PyPI and GitHub, which represent the most important distribution platforms for Python code. Additionally, the software is interoperable with other software. This was achieved by segregating the data model from the functionality of the software. Therefore, other Python tools can utilize the functionalities of CaliPytion and EnzymePynetics by serving the underlying data models, and thus no modifications on the software functionalities are necessary. Furthermore, the data model and its vocabulary is described in the specifications of <a class="reference external" href="https://github.com/FAIRChemistry/CaliPytion/blob/main/specifications/CalibrationModel.md">CaliPytion</a> as well as <a class="reference external" href="https://github.com/haeussma/EnzymePynetics/blob/main/specifications/EnzymeKinetics.md">EnzymePynetics</a>, making the data model comprehensive for others. Furthermore, the software packages are reusable, since the software can be installed and applied by anyone due to the documentation.</p>
<p>Since data integrity as well as reproducibility of analysis were integral parts of this work, the thesis itself was conceptualized enabling reproducible data analysis. Therefore, the Jupyter Notebooks were integrated into the Jupyter Book, which constitutes this thesis. Thus, all data analysis is reproducible by launching the individual notebooks with Binder. As a result, a level of data integrity and transparency is achieved, which is not reachable with established word processing software. Furthermore, the Jupyter Book format allows to display computational content in its native form. Due to the toggle functionality of code cells, the text readability and code readability is equally maintained, which is not possible with analog formats. Hence, preserving readability as well as reproducibility by optimizing the thesis for the digital format was preferred over printed readability.
The Jupyter Book formate allows this thesis to be findable and accessible on GibHub, whereas interoperability between book and notebooks makes all code content of this work reusable and reanalyzable. In conclusion, the Jupyter Book format allows scientific reporting of computational content which is compliant with FAIR guiding principles.</p>
</section>
</section>
<span id="document-conclusion"></span><section class="tex2jax_ignore mathjax_ignore" id="conclusion-and-outlook">
<h2>Conclusion and Outlook<a class="headerlink" href="#conclusion-and-outlook" title="Permalink to this headline">#</a></h2>
<p>In this work, a reproducible and FAIR workflow for reproducible kinetic parameter estimation of enzyme reactions was established. Thereby, the output of an analytical device is directly used as the input of the modeling pipeline. By considering possible nonlinear relationships in the conversion of analytic data to concentration data, a source of inaccuracies is avoided. The parameter estimation is based on progress curve analysis. Since various kinetic models were implemented, models can be compared and insight into the kinetic mechanism can be obtained. Therefore, the implemented parameter estimator is suited for inhibition studies. After fitting of the models, visualization of the fitted model together with the underlying measurement data allows to identify individual systematic deviation between the model and the data. Hence, providing a measure for quality control.
The workflow is capable to process EnzymeML files by using the associated PyEnzyme application programming interface. The analysis workflow was implemented in a Jupyter Notebook, thus the workflow is conclusive, repeatable and reproducible.
The reproducibility of the workflow is mainly attributed to the implemented progress curve analysis method, together with the intrinsic characteristics of the Jupyter Notebook format. Due to the format and the associated Jupyter, Binder, and GitHub ecosystems,the analysis can be extensively documented as well as reanalyzed and shared among scientists.
The developed workflow supports the scientific community by making accurate and scalable progress curve analysis of enzyme reactions available, which is compliant with FAIR guiding principles.
Application of the workflow to four ongoing research projects demonstrated its capabilities to identify issues with the preparation of enzymatic assays, which might otherwise have remained unrecognized. Thus, the developments additionally support the field by potentially preventing the publication of incorrect kinetic parameters based on erroneous data.</p>
<p>In future developments, the workflow could be extended to enable parameter estimation of enzyme reactions with more than one substrate, by implementing more complex models. Furthermore, part of the quality control, which is currently carried out by visualizing, could be automated. Thereby, the user would be informed about systematic errors in the data set, which might have resulted from experimental errors. Moreover, tools could be implemented which assist scientist in assay design with focus on the design of minimal experiments. Hence, after an initial analysis round the user could be informed if the given information is sufficient for parameter estimation, or if the data set contains to little information. In result, initial substrate concentrations for the next round of lab experiments could be suggested.</p>
</section>
<span id="document-references"></span><section class="tex2jax_ignore mathjax_ignore" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">#</a></h2>
<div class="docutils container" id="id1">
<dl class="citation">
<dt class="label" id="id17"><span class="brackets">Aka98</span></dt>
<dd><p>Hirotogu Akaike. Information theory and an extension of the maximum likelihood principle. In <em>Selected papers of hirotugu akaike</em>, pages 199–213. Springer, 1998.</p>
</dd>
<dt class="label" id="id16"><span class="brackets">Arn10</span></dt>
<dd><p>Todd W Arnold. Uninformative parameters and model selection using akaike's information criterion. <em>The Journal of Wildlife Management</em>, 74(6):1175–1178, 2010.</p>
</dd>
<dt class="label" id="id32"><span class="brackets">Bak16</span></dt>
<dd><p>Monya Baker. Reproducibility crisis. <em>Nature</em>, 533(26):353–66, 2016.</p>
</dd>
<dt class="label" id="id33"><span class="brackets">BI15</span></dt>
<dd><p>C Glenn Begley and John PA Ioannidis. Reproducibility in science: improving the standard for basic and preclinical research. <em>Circulation research</em>, 116(1):116–126, 2015.</p>
</dd>
<dt class="label" id="id42"><span class="brackets">BFF+21</span></dt>
<dd><p>Elizabeth L. Bell, William Finnigan, Scott P. France, Anthony P. Green, Martin A. Hayes, Lorna J. Hepworth, Sarah L. Lovelock, Haruka Niikura, S\'ılvia Osuna, Elvira Romero, Katherine S. Ryan, Nicholas J. Turner, and Sabine L. Flitsch. Biocatalysis. <em>Nature Reviews Methods Primers</em>, June 2021. URL: <a class="reference external" href="https://doi.org/10.1038/s43586-021-00044-z">https://doi.org/10.1038/s43586-021-00044-z</a>, <a class="reference external" href="https://doi.org/10.1038/s43586-021-00044-z">doi:10.1038/s43586-021-00044-z</a>.</p>
</dd>
<dt class="label" id="id5"><span class="brackets">Bis17</span></dt>
<dd><p>Hans Bisswanger. <em>Enzyme kinetics: principles and methods</em>. John Wiley &amp; Sons, 2017.</p>
</dd>
<dt class="label" id="id9"><span class="brackets">BAC99</span></dt>
<dd><p>William F Busby, Joseph M Ackermann, and Charles L Crespi. Effect of methanol, ethanol, dimethyl sulfoxide, and acetonitrile on in vitro activities of cdna-expressed human cytochromes p-450. <em>Drug Metabolism and Disposition</em>, 27(2):246–249, 1999.</p>
</dd>
<dt class="label" id="id8"><span class="brackets">Bux15</span></dt>
<dd><p>Engelbert Buxbaum. <em>Enzyme Kinetics: Special Cases</em>, pages 185–191. Springer International Publishing, Cham, 2015. URL: <a class="reference external" href="https://doi.org/10.1007/978-3-319-19920-7_8">https://doi.org/10.1007/978-3-319-19920-7_8</a>, <a class="reference external" href="https://doi.org/10.1007/978-3-319-19920-7_8">doi:10.1007/978-3-319-19920-7_8</a>.</p>
</dd>
<dt class="label" id="id36"><span class="brackets">CDL+20</span></dt>
<dd><p>Thomas A Caswell, Michael Droettboom, Antony Lee, John Hunter, Eric Firing, Elliott Sales De Andrade, Tim Hoffmann, David Stansby, Jody Klymak, Nelle Varoquaux, and others. Matplotlib/matplotlib: rel: v3. 3.1. <em>Zenodo</em>, 2020.</p>
</dd>
<dt class="label" id="id38"><span class="brackets">Com20</span></dt>
<dd><p>Executable Books Community. Jupyter book. February 2020. URL: <a class="reference external" href="https://doi.org/10.5281/zenodo.4539666">https://doi.org/10.5281/zenodo.4539666</a>, <a class="reference external" href="https://doi.org/10.5281/zenodo.4539666">doi:10.5281/zenodo.4539666</a>.</p>
</dd>
<dt class="label" id="id6"><span class="brackets">DMMP20</span></dt>
<dd><p>Chantal Désirée Daub, Blessing Mabate, Samkelo Malgas, and Brett Ivan Pletschke. Fucoidan from ecklonia maxima is a powerful inhibitor of the diabetes-related enzyme, α-glucosidase. <em>International journal of biological macromolecules</em>, 151:412–420, 2020.</p>
</dd>
<dt class="label" id="id31"><span class="brackets">GFI16</span></dt>
<dd><p>Steven N Goodman, Daniele Fanelli, and John PA Ioannidis. What does research reproducibility mean? <em>Science translational medicine</em>, 8(341):341ps12–341ps12, 2016.</p>
</dd>
<dt class="label" id="id14"><span class="brackets">HC10</span></dt>
<dd><p>Kang-Hung Hsu and Chiachung Chen. The effect of calibration equations on the uncertainty of uv–vis spectrophotometric measurements. <em>Measurement</em>, 43(10):1525–1531, 2010.</p>
</dd>
<dt class="label" id="id37"><span class="brackets">KRKPerez+16</span></dt>
<dd><p>Thomas Kluyver, Benjamin Ragan-Kelley, Fernando Pérez, Brian E Granger, Matthias Bussonnier, Jonathan Frederic, Kyle Kelley, Jessica B Hamrick, Jason Grout, Sylvain Corlay, and others. <em>Jupyter Notebooks-a publishing format for reproducible computational workflows.</em> Volume 2016. IOS Press, 2016.</p>
</dd>
<dt class="label" id="id7"><span class="brackets">LLWZ08</span></dt>
<dd><p>Bo Li, Fei Lu, Xinjun Wei, and Ruixiang Zhao. Fucoidan: structure and bioactivity. <em>Molecules</em>, 13(8):1671–1695, 2008.</p>
</dd>
<dt class="label" id="id15"><span class="brackets">MGA+17</span></dt>
<dd><p>Julia Martin, Alberto Romero Gracia, Agustín G Asuero, and others. Fitting nonlinear calibration curves: no models perfect. <em>Journal of Analytical Sciences, Methods and Instrumentation</em>, 7(01):1, 2017.</p>
</dd>
<dt class="label" id="id21"><span class="brackets">Miy20</span></dt>
<dd><p>Tsuyoshi Miyakawa. No raw data, no science: another possible source of the reproducibility crisis. 2020.</p>
</dd>
<dt class="label" id="id34"><span class="brackets">MunafoNB+17</span></dt>
<dd><p>Marcus R Munafò, Brian A Nosek, Dorothy VM Bishop, Katherine S Button, Christopher D Chambers, Nathalie Percie du Sert, Uri Simonsohn, Eric-Jan Wagenmakers, Jennifer J Ware, and John Ioannidis. A manifesto for reproducible science. <em>Nature human behaviour</em>, 1(1):1–9, 2017.</p>
</dd>
<dt class="label" id="id35"><span class="brackets">NSA+16</span></dt>
<dd><p>Matthew Newville, Till Stensitzki, Daniel B Allen, Michal Rawlik, Antonino Ingargiola, and Andrew Nelson. Lmfit: non-linear least-square minimization and curve-fitting for python. <em>Astrophysics Source Code Library</em>, pages ascl–1606, 2016.</p>
</dd>
<dt class="label" id="id22"><span class="brackets">Pen11</span></dt>
<dd><p>Roger D Peng. Reproducible research in computational science. <em>Science</em>, 334(6060):1226–1227, 2011.</p>
</dd>
<dt class="label" id="id4"><span class="brackets">Ple21</span></dt>
<dd><p>Jürgen Pleiss. Standardized data, scalable documentation, sustainable storage–enzymeml as a basis for fair data management in biocatalysis. <em>ChemCatChem</em>, 13(18):3909–3913, 2021.</p>
</dd>
<dt class="label" id="id30"><span class="brackets">PKK+15</span></dt>
<dd><p>Alaric Prins, L Kleinsmidt, Nuraan Khan, Bronwyn Kirby, Tukayi Kudanga, Jannik Vollmer, Juergen Pleiss, Stephanie Burton, and M Le Roes-Hill. The effect of mutations near the t1 copper site on the biochemical characteristics of the small laccase from streptomyces coelicolor a3 (2). <em>Enzyme and microbial technology</em>, 68:23–32, 2015.</p>
</dd>
<dt class="label" id="id20"><span class="brackets">RKWA+18</span></dt>
<dd><p>Benjamin Ragan-Kelley, Carol Willing, F Akici, D Lippa, D Niederhut, and M Pacer. Binder 2.0-reproducible, interactive, sharable environments for science at scale. In <em>Proceedings of the 17th python in science conference</em>, 113–120. F. Akici, D. Lippa, D. Niederhut, and M. Pacer, eds., 2018.</p>
</dd>
<dt class="label" id="id12"><span class="brackets">RBS21</span></dt>
<dd><p>Jan Range, Frank Bergmann, and Neil Swainston. Pyenzyme 1.1.0-alpha. June 2021. URL: <a class="reference external" href="https://doi.org/10.5281/zenodo.5021263">https://doi.org/10.5281/zenodo.5021263</a>, <a class="reference external" href="https://doi.org/10.5281/zenodo.5021263">doi:10.5281/zenodo.5021263</a>.</p>
</dd>
<dt class="label" id="id18"><span class="brackets">RHL+22</span></dt>
<dd><p>Jan Range, Colin Halupczok, Jens Lohmann, Neil Swainston, Carsten Kettner, Frank T Bergmann, Andreas Weidemann, Ulrike Wittig, Santiago Schnell, and Jürgen Pleiss. Enzymeml—a data exchange format for biocatalysis and enzymology. <em>The FEBS Journal</em>, 289(19):5864–5874, 2022.</p>
</dd>
<dt class="label" id="id26"><span class="brackets">SLH+16</span></dt>
<dd><p>Xindi Shan, Xin Liu, Jiejie Hao, Chao Cai, Fei Fan, Yunlou Dun, Xiaoliang Zhao, Xiaoxiao Liu, Chunxia Li, and Guangli Yu. In vitro and in vivo hypoglycemic effects of brown algal fucoidans. <em>International Journal of Biological Macromolecules</em>, 82:249–255, 2016.</p>
</dd>
<dt class="label" id="id25"><span class="brackets">TL10</span></dt>
<dd><p>Qingxiu Tang and Thomas S Leyh. Precise, facile initial rate measurements. <em>The Journal of Physical Chemistry B</em>, 114(49):16131–16136, 2010.</p>
</dd>
<dt class="label" id="id11"><span class="brackets">TAB+14</span></dt>
<dd><p>Keith F Tipton, Richard N Armstrong, Barbara M Bakker, Amos Bairoch, Athel Cornish-Bowden, Peter J Halling, Jan-Hendrik Hofmeyr, Thomas S Leyh, Carsten Kettner, Frank M Raushel, and others. Standards for reporting enzyme data: the strenda consortium: what it aims to do and why it should be helpful. <em>Perspectives in Science</em>, 1(1-6):131–137, 2014.</p>
</dd>
<dt class="label" id="id29"><span class="brackets">WTWL14</span></dt>
<dd><p>Yang Wei, Aby A Thyparambil, Yonnie Wu, and Robert A Latour. Adsorption-induced changes in ribonuclease a structure and enzymatic activity on solid surfaces. <em>Langmuir</em>, 30(49):14849–14858, 2014.</p>
</dd>
<dt class="label" id="id13"><span class="brackets">WK08</span></dt>
<dd><p>Petri Widsten and Andreas Kandelbauer. Laccase applications in the forest products industry: a review. <em>Enzyme and microbial technology</em>, 42(4):293–307, 2008.</p>
</dd>
<dt class="label" id="id40"><span class="brackets">WDA+16</span></dt>
<dd><p>Mark D Wilkinson, Michel Dumontier, IJsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, Jan-Willem Boiten, Luiz Bonino da Silva Santos, Philip E Bourne, and others. The fair guiding principles for scientific data management and stewardship. <em>Scientific data</em>, 3(1):1–9, 2016.</p>
</dd>
</dl>
</div>
</section>
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Max Häußler<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>